# -*- coding: utf-8 -*-
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: ncbi/datasets/v2/datasets.proto
"""Generated protocol buffer code."""
from google.protobuf.internal import enum_type_wrapper
from google.protobuf import descriptor as _descriptor
from google.protobuf import message as _message
from google.protobuf import reflection as _reflection
from google.protobuf import symbol_database as _symbol_database
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()


from google.protobuf import timestamp_pb2 as google_dot_protobuf_dot_timestamp__pb2
from ncbi.datasets.options import openapi_pb2 as ncbi_dot_datasets_dot_options_dot_openapi__pb2
from ncbi.datasets.options import request_pb2 as ncbi_dot_datasets_dot_options_dot_request__pb2
from ncbi.datasets.v2.reports import assembly_revision_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_assembly__revision__pb2
from ncbi.datasets.v2.reports import assembly_sequence_info_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_assembly__sequence__info__pb2
from ncbi.datasets.v2.reports import common_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_common__pb2


DESCRIPTOR = _descriptor.FileDescriptor(
  name='ncbi/datasets/v2/datasets.proto',
  package='ncbi.datasets.v2',
  syntax='proto3',
  serialized_options=b'Z\020ncbi/datasets/v2\370\001\001',
  create_key=_descriptor._internal_create_key,
  serialized_pb=b'\n\x1fncbi/datasets/v2/datasets.proto\x12\x10ncbi.datasets.v2\x1a\x1fgoogle/protobuf/timestamp.proto\x1a#ncbi/datasets/options/openapi.proto\x1a#ncbi/datasets/options/request.proto\x1a\x30ncbi/datasets/v2/reports/assembly_revision.proto\x1a\x35ncbi/datasets/v2/reports/assembly_sequence_info.proto\x1a%ncbi/datasets/v2/reports/common.proto\"H\n\x14MethodPayloadRequest\x12\x16\n\x06method\x18\x01 \x01(\tR\x06method\x12\x18\n\x07payload\x18\x02 \x01(\tR\x07payload\"\xb2\x06\n\x1e\x41ssemblySequenceReportsRequest\x12X\n\taccession\x18\x01 \x01(\tB:\xc2\xf0\x19\x36\x12\x19\n\x10GCF_000001635.27\x12\x05Mouse\x12\x19\n\x10GCF_000001405.40\x12\x05HumanR\taccession\x12s\n\x0b\x63hromosomes\x18\x02 \x03(\tBQ\xc2\xf0\x19M\x12,\n\x01\x31\n\x01\x32\n\x01\x33\n\x01X\n\x01Y\n\x02MT\x12\x17Selected chromosomes #1\x12\x1d\n\x01X\n\x01Y\x12\x15Human sex chromosomesR\x0b\x63hromosomes\x12\xaf\x01\n\x0crole_filters\x18\x04 \x03(\tB\x8b\x01\xc2\xf0\x19\x86\x01\x12T\n\x12\x61ssembled-molecule\n\x14unlocalized-scaffold\n\x11unplaced-scaffold\x12\x15\x41ll types of sequence\x12.\n\x12\x61ssembled-molecule\x12\x18Only assembled moleculesR\x0broleFilters\x12\x82\x01\n\x0ctable_fields\x18\x03 \x03(\tB_\xc2\xf0\x19[\x12;\n\taccession\n\x08\x63hr-name\x12$Genome Accession and Chromosome Name\x12\x1c\n\x0fucsc-style-name\x12\tUCSC NameR\x0btableFields\x12\x43\n\x17\x63ount_assembly_unplaced\x18\x05 \x01(\x08\x42\x0b\xc2\xf0\x19\x07\n\x05\x66\x61lseR\x15\x63ountAssemblyUnplaced\x12%\n\tpage_size\x18\x08 \x01(\x05\x42\x08\xc2\xf0\x19\x04\n\x02\x32\x30R\x08pageSize\x12\x1d\n\npage_token\x18\t \x01(\tR\tpageToken\x12\\\n\x16include_tabular_header\x18\x0c \x01(\x0e\x32&.ncbi.datasets.v2.IncludeTabularHeaderR\x14includeTabularHeader\x12!\n\x0ctable_format\x18\r \x01(\tR\x0btableFormat\"\xb1\x02\n\x12SequenceReportPage\x12@\n\x07reports\x18\x01 \x03(\x0b\x32&.ncbi.datasets.v2.reports.SequenceInfoR\x07reports\x12\x1f\n\x0btotal_count\x18\x03 \x01(\rR\ntotalCount\x12&\n\x0fnext_page_token\x18\x04 \x01(\tR\rnextPageToken\x12!\n\x0c_report_type\x18\xf4\x03 \x01(\tR\nReportType\x12%\n\x0e_report_fields\x18\xf5\x03 \x03(\tR\x0cReportFields\x12\x1f\n\x0b_first_page\x18\xf6\x03 \x01(\x08R\tFirstPage\x12%\n\x0e_report_format\x18\xf7\x03 \x01(\tR\x0cReportFormat\"\xac\x06\n\tSortField\x12\xdf\x05\n\x05\x66ield\x18\x01 \x01(\tB\xc8\x05\xc2\xf0\x19\xc3\x05\x12\x0b\n\taccession\x12\x17\n\x15versionless-accession\x12\r\n\x0bsearch-text\x12\x08\n\x06tax-id\x12\x13\n\x11tax-ordinal-range\x12\x0f\n\rorganism-name\x12\x0b\n\tsubmitter\x12\x15\n\x13\x61ssembly-level-enum\x12\x11\n\x0f\x61ssembly-status\x12\x0e\n\x0crelease-date\x12\x0f\n\ris-ref-genome\x12\x0f\n\ris-rep-genome\x12\x0b\n\tis-refseq\x12\x0c\n\nis-primary\x12\x0c\n\nis-current\x12\x1c\n\x1apaired-assembly-is-current\x12!\n\x1fpaired-assembly-annotation-name\x12\x10\n\x0ehas-annotation\x12\r\n\x0bis-atypical\x12\x16\n\x14\x62ioproject-accession\x12\x17\n\x15total-sequence-length\x12\x11\n\x0fpair-is-current\x12\x1d\n\x1bpair-has-genbank-annotation\x12\x1c\n\x1apair-has-refseq-annotation\x12\x15\n\x13pair-has-annotation\x12\x0c\n\ncontig-n50\x12\x0e\n\x0cscaffold-n50\x12\x10\n\x0e\x62usco-complete\x12\x13\n\x11genbank-accession\x12\x12\n\x10refseq-accession\x12\x11\n\x0f\x61nnotation-name\x12\x0f\n\rwgs-accession\x12\x11\n\x0fsequencing-tech\x12\x15\n\x13\x62iosample-accession\x12\x0c\n\ngene-count\x12\x0f\n\rprotein-count\x12\n\n\x08modifierR\x05\x66ield\x12=\n\tdirection\x18\x02 \x01(\x0e\x32\x1f.ncbi.datasets.v2.SortDirectionR\tdirection\"\xbc\x08\n\x1d\x41ssemblyDatasetReportsRequest\x12\x8c\x01\n\x06taxons\x18\x06 \x03(\tBt\xc2\xf0\x19p\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy Identifier\x12+\n\nchimpanzee\n\x0f\x65\x61stern gorilla\x12\x0c\x43ommon Names\x12\x1f\n\x0cHomo sapiens\x12\x0fScientific NameR\x06taxons\x12u\n\x0b\x62ioprojects\x18\x08 \x03(\tBS\xc2\xf0\x19O\x12)\n\x0bPRJNA489243\x12\x1aVertebrate Genomes Project\x12\"\n\nPRJNA31257\x12\x14Human Genome ProjectR\x0b\x62ioprojects\x12v\n\naccessions\x18\x01 \x03(\tBV\xc2\xf0\x19R\x12\x35\n\x10GCF_000001405.40\n\x10GCF_000001635.27\x12\x0fHuman and Mouse\x12\x19\n\x10GCF_000001405.40\x12\x05HumanR\naccessions\x12L\n\x07\x66ilters\x18\x07 \x01(\x0b\x32\x32.ncbi.datasets.v2.AssemblyDatasetDescriptorsFilterR\x07\x66ilters\x12$\n\x0b\x63hromosomes\x18\x02 \x03(\tB\x02\x18\x01R\x0b\x63hromosomes\x12\x8d\x01\n\x0ctable_fields\x18\x03 \x03(\tBj\xc2\xf0\x19\x66\x12>\n\x12\x61ssminfo-accession\n\rassminfo-name\x12\x19Genome Accession and Name\x12$\n\x12\x61ssminfo-accession\x12\x0eJust accessionR\x0btableFields\x12\x66\n\x10returned_content\x18\n \x01(\x0e\x32;.ncbi.datasets.v2.AssemblyDatasetReportsRequest.ContentTypeR\x0freturnedContent\x12%\n\tpage_size\x18\x04 \x01(\x05\x42\x08\xc2\xf0\x19\x04\n\x02\x32\x30R\x08pageSize\x12\x1d\n\npage_token\x18\x05 \x01(\tR\tpageToken\x12/\n\x04sort\x18\x0b \x03(\x0b\x32\x1b.ncbi.datasets.v2.SortFieldR\x04sort\x12\\\n\x16include_tabular_header\x18\x0c \x01(\x0e\x32&.ncbi.datasets.v2.IncludeTabularHeaderR\x14includeTabularHeader\x12!\n\x0ctable_format\x18\r \x01(\tR\x0btableFormat\"9\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\x0c\n\x08\x41SSM_ACC\x10\x01\x12\x0e\n\nPAIRED_ACC\x10\x02\"\x95\x11\n AssemblyDatasetDescriptorsFilter\x12\x9d\x01\n\x0ereference_only\x18\x01 \x01(\x08\x42v\xc2\xf0\x19r\n\x05\x66\x61lse\x12\x33\n\x04true\x12+Limit to reference & representative genomes\x12\x34\n\x05\x66\x61lse\x12+Include reference and non-reference genomesR\rreferenceOnly\x12\xd1\x01\n\x0f\x61ssembly_source\x18\x03 \x01(\x0e\x32\x41.ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.AssemblySourceBe\xc2\xf0\x19\x61\n\x03\x61ll\x12!\n\x06refseq\x12\x17Limit to RefSeq genomes\x12#\n\x07genbank\x12\x18Limit to GenBank genomes\x12\x12\n\x03\x61ll\x12\x0b\x41ll genomesR\x0e\x61ssemblySource\x12y\n\x0ehas_annotation\x18\x07 \x01(\x08\x42R\xc2\xf0\x19N\n\x05\x66\x61lse\x12%\n\x04true\x12\x1dLimit to annotated assemblies\x12\x1e\n\x05\x66\x61lse\x12\x15Return all assembliesR\rhasAnnotation\x12\x96\x01\n\x16\x65xclude_paired_reports\x18\t \x01(\x08\x42`\xc2\xf0\x19\\\n\x05\x66\x61lse\x12\x1e\n\x05\x66\x61lse\x12\x15Return all assemblies\x12\x33\n\x04true\x12+Only return single record for GCA/GCF pairsR\x14\x65xcludePairedReports\x12\xc7\x01\n\x10\x65xclude_atypical\x18\x0b \x01(\x08\x42\x9b\x01\xc2\xf0\x19\x96\x01\n\x05\x66\x61lse\x12\x46\n\x05\x66\x61lse\x12=Return all selected genomes, including ones that are atypical\x12\x45\n\x04true\x12=Only return genomes that have not been classified as atypicalR\x0f\x65xcludeAtypical\x12\xf5\x01\n\x10\x61ssembly_version\x18\n \x01(\x0e\x32\x42.ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.AssemblyVersionB\x85\x01\xc2\xf0\x19\x80\x01\n\x07\x63urrent\x12)\n\x07\x63urrent\x12\x1eOnly return current assemblies\x12J\n\x0e\x61ll_assemblies\x12\x38Return all assemblies, including suppressed and replacedR\x0f\x61ssemblyVersion\x12\xd2\x01\n\x0e\x61ssembly_level\x18\x04 \x03(\x0e\x32\'.ncbi.datasets.v2.reports.AssemblyLevelB\x81\x01\xc2\xf0\x19}\x12=\n\nchromosome\n\x0f\x63omplete_genome\x12\x1e\x43hromosome and complete genome\x12<\n\x08scaffold\n\nchromosome\n\x0f\x63omplete_genome\x12\x13Scaffold and higherR\rassemblyLevel\x12j\n\x12\x66irst_release_date\x18\x05 \x01(\x0b\x32\x1a.google.protobuf.TimestampB \xc2\xf0\x19\x1c\x12\x1a\n\n01/10/2015\x12\x0cJan 10, 2015R\x10\x66irstReleaseDate\x12h\n\x11last_release_date\x18\x06 \x01(\x0b\x32\x1a.google.protobuf.TimestampB \xc2\xf0\x19\x1c\x12\x1a\n\n01/10/2021\x12\x0cJan 10, 2021R\x0flastReleaseDate\x12\x89\x01\n\x0bsearch_text\x18\x08 \x03(\tBh\xc2\xf0\x19\x64\x12:\n\x1bGenome Reference Consortium\x12\x1bGenome Reference Consortium\x12\x10\n\x06GRCh38\x12\x06GRCh38\x12\x14\n\x08\x43\x35\x37\x42L/6J\x12\x08\x43\x35\x37\x42L/6JR\nsearchText\x12\x91\x02\n\x15is_metagenome_derived\x18\x0c \x01(\x0e\x32J.ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.MetagenomeDerivedFilterB\x90\x01\xc2\xf0\x19\x8b\x01\x12\x44\n\x17metagenome_derived_only\x12)Return only metagenome derived assemblies\x12\x43\n\x1ametagenome_derived_exclude\x12%Exclude metagenome derived assembliesR\x13isMetagenomeDerived\"2\n\x0e\x41ssemblySource\x12\x07\n\x03\x61ll\x10\x00\x12\n\n\x06refseq\x10\x01\x12\x0b\n\x07genbank\x10\x02\"2\n\x0f\x41ssemblyVersion\x12\x0b\n\x07\x63urrent\x10\x00\x12\x12\n\x0e\x61ll_assemblies\x10\x01\"t\n\x17MetagenomeDerivedFilter\x12\x1c\n\x18METAGENOME_DERIVED_UNSET\x10\x00\x12\x1b\n\x17metagenome_derived_only\x10\x01\x12\x1e\n\x1ametagenome_derived_exclude\x10\x02\"\x84\x01\n\nAccessions\x12v\n\naccessions\x18\x01 \x03(\tBV\xc2\xf0\x19R\x12\x35\n\x10GCF_000001405.40\n\x10GCF_000001635.27\x12\x0fHuman and Mouse\x12\x19\n\x10GCF_000001405.40\x12\x05HumanR\naccessions\"\x95\x01\n\x17\x41ssemblyRevisionHistory\x12Y\n\x12\x61ssembly_revisions\x18\x01 \x03(\x0b\x32*.ncbi.datasets.v2.reports.AssemblyRevisionR\x11\x61ssemblyRevisions\x12\x1f\n\x0btotal_count\x18\x02 \x01(\rR\ntotalCount\"_\n\x1e\x41ssemblyRevisionHistoryRequest\x12=\n\taccession\x18\x01 \x01(\tB\x1f\xc2\xf0\x19\x1b\x12\x19\n\x10GCF_000001405.40\x12\x05HumanR\taccession\"l\n\x1e\x41ssemblyCheckMHistogramRequest\x12J\n\rspecies_taxon\x18\x01 \x01(\tB%\xc2\xf0\x19!\x12\x1f\n\x06\x32\x30\x32\x39\x35\x36\x12\x15\x41\x63inetobacter towneriR\x0cspeciesTaxon\"\xc0\x02\n\x1c\x41ssemblyCheckMHistogramReply\x12J\n\rspecies_taxid\x18\x01 \x01(\rB%\xc2\xf0\x19!\x12\x1f\n\x06\x32\x30\x32\x39\x35\x36\x12\x15\x41\x63inetobacter towneriR\x0cspeciesTaxid\x12q\n\x13histogram_intervals\x18\x02 \x03(\x0b\x32@.ncbi.datasets.v2.AssemblyCheckMHistogramReply.HistogramIntervalR\x12histogramIntervals\x1a\x61\n\x11HistogramInterval\x12\x1b\n\tstart_pos\x18\x01 \x01(\x02R\x08startPos\x12\x19\n\x08stop_pos\x18\x02 \x01(\x02R\x07stopPos\x12\x14\n\x05\x63ount\x18\x03 \x01(\x02R\x05\x63ount\"\x98\x01\n\nBioProject\x12\x1c\n\taccession\x18\x01 \x01(\tR\taccession\x12\x14\n\x05title\x18\x02 \x01(\tR\x05title\x12)\n\x10parent_accession\x18\x03 \x01(\tR\x0fparentAccession\x12+\n\x11parent_accessions\x18\x04 \x03(\tR\x10parentAccessions\"S\n\x11\x42ioProjectLineage\x12>\n\x0b\x62ioprojects\x18\x01 \x03(\x0b\x32\x1c.ncbi.datasets.v2.BioProjectR\x0b\x62ioprojects\"\x85\x02\n\tBuscoStat\x12#\n\rbusco_lineage\x18\x01 \x01(\tR\x0c\x62uscoLineage\x12\x1b\n\tbusco_ver\x18\x02 \x01(\tR\x08\x62uscoVer\x12\x1a\n\x08\x63omplete\x18\x03 \x01(\x02R\x08\x63omplete\x12\x1f\n\x0bsingle_copy\x18\x04 \x01(\x02R\nsingleCopy\x12\x1e\n\nduplicated\x18\x05 \x01(\x02R\nduplicated\x12\x1e\n\nfragmented\x18\x06 \x01(\x02R\nfragmented\x12\x18\n\x07missing\x18\x07 \x01(\x02R\x07missing\x12\x1f\n\x0btotal_count\x18\x08 \x01(\x04R\ntotalCount\"\x8c\x05\n\x15\x41nnotationForAssembly\x12\x12\n\x04name\x18\x01 \x01(\tR\x04name\x12\x16\n\x06source\x18\x02 \x01(\tR\x06source\x12%\n\x0erelease_number\x18\x03 \x01(\tR\rreleaseNumber\x12!\n\x0crelease_date\x18\x04 \x01(\tR\x0breleaseDate\x12\x1d\n\nreport_url\x18\x05 \x01(\tR\treportUrl\x12@\n\x04\x66ile\x18\x06 \x03(\x0b\x32,.ncbi.datasets.v2.AnnotationForAssembly.FileR\x04\x66ile\x12\x35\n\x05stats\x18\x07 \x01(\x0b\x32\x1f.ncbi.datasets.v2.FeatureCountsR\x05stats\x12\x31\n\x05\x62usco\x18\x08 \x01(\x0b\x32\x1b.ncbi.datasets.v2.BuscoStatR\x05\x62usco\x1ao\n\x04\x46ile\x12@\n\x04type\x18\x01 \x01(\x0e\x32,.ncbi.datasets.v2.AnnotationForAssembly.TypeR\x04type\x12%\n\x0e\x65stimated_size\x18\x02 \x01(\x04R\restimatedSize\"\xc0\x01\n\x04Type\x12\x0b\n\x07\x44\x45\x46\x41ULT\x10\x00\x12\x0e\n\nGENOME_GFF\x10\x01\x12\x0f\n\x0bGENOME_GBFF\x10\x02\x12\x11\n\tGENOME_GB\x10\x02\x1a\x02\x08\x01\x12\r\n\tRNA_FASTA\x10\x03\x12\x0e\n\nPROT_FASTA\x10\x04\x12\x0e\n\nGENOME_GTF\x10\x05\x12\r\n\tCDS_FASTA\x10\x06\x12\x0e\n\nASM_STRUCT\x10\x07\x12\x10\n\x0cGENOME_FASTA\x10\x08\x12\x13\n\x0fSEQUENCE_REPORT\x10\t\x1a\x02\x10\x01\"N\n\rFeatureCounts\x12=\n\x0bgene_counts\x18\x01 \x01(\x0b\x32\x1c.ncbi.datasets.v2.GeneCountsR\ngeneCounts\"I\n\nGeneCounts\x12\x14\n\x05total\x18\x01 \x01(\rR\x05total\x12%\n\x0eprotein_coding\x18\x02 \x01(\rR\rproteinCoding\"\x8b\x06\n\x16\x41ssemblyDatasetRequest\x12v\n\naccessions\x18\n \x03(\tBV\xc2\xf0\x19R\x12\x35\n\x10GCF_000001405.40\n\x10GCF_000001635.27\x12\x0fHuman and Mouse\x12\x19\n\x10GCF_000001405.40\x12\x05HumanR\naccessions\x12s\n\x0b\x63hromosomes\x18\x02 \x03(\tBQ\xc2\xf0\x19M\x12,\n\x01\x31\n\x01\x32\n\x01\x33\n\x01X\n\x01Y\n\x02MT\x12\x17Selected chromosomes #1\x12\x1d\n\x01X\n\x01Y\x12\x15Human sex chromosomesR\x0b\x63hromosomes\x12\xa0\x02\n\x17include_annotation_type\x18\x08 \x03(\x0e\x32,.ncbi.datasets.v2.AnnotationForAssembly.TypeB\xb9\x01\xda\xf0\x19\x02\x08\x02\xc2\xf0\x19\xae\x01\x12M\n\x0cGENOME_FASTA\n\nPROT_FASTA\n\tRNA_FASTA\x12&Genomic, Transcript & Protein Sequence\x12]\n\nGENOME_GFF\n\x0bGENOME_GBFF\n\nGENOME_GTF\n\nPROT_FASTA\n\tRNA_FASTA\n\x1a\x43\x44S_FASTA, SEQUENCE_REPORT\x12\x03\x41llR\x15includeAnnotationType\x12O\n\x08hydrated\x18\t \x01(\x0e\x32\x33.ncbi.datasets.v2.AssemblyDatasetRequest.ResolutionR\x08hydrated\x12\x32\n\x0binclude_tsv\x18\x0b \x01(\x08\x42\x11\xda\xf0\x19\x02\x08\x02\xc2\xf0\x19\x07\n\x05\x66\x61lseR\nincludeTsv\"6\n\nResolution\x12\x12\n\x0e\x46ULLY_HYDRATED\x10\x00\x12\x14\n\x10\x44\x41TA_REPORT_ONLY\x10\x01:$\xc2\xf0\x19 \n\x1e\x61\x63\x63\x65ssions:\n- GCF_000001405.40\"\x8f\x01\n\x1b\x41ssemblyDatasetAvailability\x12)\n\x10valid_assemblies\x18\x01 \x03(\tR\x0fvalidAssemblies\x12-\n\x12invalid_assemblies\x18\x02 \x03(\tR\x11invalidAssemblies\x12\x16\n\x06reason\x18\x03 \x01(\tR\x06reason\"\xac\r\n\x08Organism\x12\x15\n\x06tax_id\x18\x01 \x01(\x04R\x05taxId\x12\x19\n\x08sci_name\x18\x02 \x01(\tR\x07sciName\x12\x1f\n\x0b\x63ommon_name\x18\x03 \x01(\tR\ncommonName\x12\x1d\n\nblast_node\x18\x10 \x01(\x08R\tblastNode\x12\x14\n\x05\x62reed\x18\n \x01(\tR\x05\x62reed\x12\x1a\n\x08\x63ultivar\x18\x0b \x01(\tR\x08\x63ultivar\x12\x18\n\x07\x65\x63otype\x18\x0c \x01(\tR\x07\x65\x63otype\x12\x18\n\x07isolate\x18\r \x01(\tR\x07isolate\x12\x10\n\x03sex\x18\x0e \x01(\tR\x03sex\x12\x16\n\x06strain\x18\x0f \x01(\tR\x06strain\x12\x1f\n\x0bsearch_text\x18\x11 \x03(\tR\nsearchText\x12\x37\n\x04rank\x18\x05 \x01(\x0e\x32#.ncbi.datasets.v2.Organism.RankTypeR\x04rank\x12\"\n\rparent_tax_id\x18\x06 \x01(\x04R\x0bparentTaxId\x12)\n\x0e\x61ssembly_count\x18\x07 \x01(\x04\x42\x02\x18\x01R\rassemblyCount\x12J\n\x0f\x61ssembly_counts\x18\x13 \x01(\x0b\x32!.ncbi.datasets.v2.Organism.CountsR\x0e\x61ssemblyCounts\x12>\n\x06\x63ounts\x18\x16 \x03(\x0b\x32&.ncbi.datasets.v2.Organism.CountByTypeR\x06\x63ounts\x12\x36\n\x08\x63hildren\x18\x08 \x03(\x0b\x32\x1a.ncbi.datasets.v2.OrganismR\x08\x63hildren\x12\x32\n\x06merged\x18\t \x03(\x0b\x32\x1a.ncbi.datasets.v2.OrganismR\x06merged\x12$\n\x0emerged_tax_ids\x18\x12 \x03(\x04R\x0cmergedTaxIds\x12\x17\n\x07min_ord\x18\x14 \x01(\rR\x06minOrd\x12\x17\n\x07max_ord\x18\x15 \x01(\rR\x06maxOrd\x12\x16\n\x06weight\x18\x17 \x01(\rR\x06weight\x12\x10\n\x03key\x18\x64 \x01(\x04R\x03key\x12\x14\n\x05title\x18\x65 \x01(\tR\x05title\x12\x12\n\x04icon\x18\x66 \x01(\x08R\x04icon\x1a\x36\n\x06\x43ounts\x12\x12\n\x04node\x18\x01 \x01(\rR\x04node\x12\x18\n\x07subtree\x18\x02 \x01(\rR\x07subtree\x1ay\n\x0b\x43ountByType\x12/\n\x04type\x18\x01 \x01(\x0e\x32\x1b.ncbi.datasets.v2.CountTypeR\x04type\x12\x39\n\x06\x63ounts\x18\x02 \x01(\x0b\x32!.ncbi.datasets.v2.Organism.CountsR\x06\x63ounts\"\x9d\x05\n\x08RankType\x12\x0b\n\x07NO_RANK\x10\x00\x12\x10\n\x0cSUPERKINGDOM\x10\x01\x12\x0b\n\x07KINGDOM\x10\x02\x12\x0e\n\nSUBKINGDOM\x10\x03\x12\x0f\n\x0bSUPERPHYLUM\x10\x04\x12\r\n\tSUBPHYLUM\x10\x05\x12\n\n\x06PHYLUM\x10\x06\x12\t\n\x05\x43LADE\x10\x1f\x12\x0e\n\nSUPERCLASS\x10\x07\x12\t\n\x05\x43LASS\x10\x08\x12\x0c\n\x08SUBCLASS\x10\t\x12\x0e\n\nINFRACLASS\x10\n\x12\n\n\x06\x43OHORT\x10\x0b\x12\r\n\tSUBCOHORT\x10\x0c\x12\x0e\n\nSUPERORDER\x10\r\x12\t\n\x05ORDER\x10\x0e\x12\x0c\n\x08SUBORDER\x10\x0f\x12\x0e\n\nINFRAORDER\x10\x10\x12\r\n\tPARVORDER\x10\x11\x12\x0f\n\x0bSUPERFAMILY\x10\x12\x12\n\n\x06\x46\x41MILY\x10\x13\x12\r\n\tSUBFAMILY\x10\x14\x12\t\n\x05GENUS\x10\x15\x12\x0c\n\x08SUBGENUS\x10\x16\x12\x11\n\rSPECIES_GROUP\x10\x17\x12\x14\n\x10SPECIES_SUBGROUP\x10\x18\x12\x0b\n\x07SPECIES\x10\x19\x12\x0e\n\nSUBSPECIES\x10\x1a\x12\t\n\x05TRIBE\x10\x1b\x12\x0c\n\x08SUBTRIBE\x10\x1c\x12\t\n\x05\x46ORMA\x10\x1d\x12\x0c\n\x08VARIETAS\x10\x1e\x12\x0b\n\x06STRAIN\x10\xc0\x02\x12\x0c\n\x07SECTION\x10\xca\x02\x12\x0f\n\nSUBSECTION\x10\xd4\x02\x12\x0f\n\nPATHOGROUP\x10\xde\x02\x12\x0f\n\nSUBVARIETY\x10\xe8\x02\x12\r\n\x08GENOTYPE\x10\xf2\x02\x12\r\n\x08SEROTYPE\x10\xfc\x02\x12\x0c\n\x07ISOLATE\x10\x86\x03\x12\n\n\x05MORPH\x10\x90\x03\x12\x0b\n\x06SERIES\x10\x9a\x03\x12\x14\n\x0f\x46ORMA_SPECIALIS\x10\xa4\x03\x12\x0e\n\tSEROGROUP\x10\xae\x03\x12\x0c\n\x07\x42IOTYPE\x10\xb8\x03\"\x85\x02\n\x0eTaxTreeRequest\x12\x1b\n\x06tax_id\x18\x01 \x01(\x04\x42\x02\x18\x01H\x00R\x05taxId\x12!\n\ttax_token\x18\x02 \x01(\tB\x02\x18\x01H\x00R\x08taxToken\x12u\n\x05taxon\x18\x03 \x01(\tB]\xc2\xf0\x19Y\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy Identifier\x12\x14\n\x05human\x12\x0b\x43ommon Name\x12\x1f\n\x0cHomo sapiens\x12\x0fScientific NameH\x00R\x05taxon\x12\x30\n\rchildren_only\x18\x04 \x01(\x08\x42\x0b\xc2\xf0\x19\x07\n\x05\x66\x61lseR\x0c\x63hildrenOnlyB\n\n\x08taxonomy\"\xa5\x04\n\x14OrganismQueryRequest\x12+\n\x0eorganism_query\x18\x01 \x01(\tB\x02\x18\x01H\x00R\rorganismQuery\x12\x33\n\x0btaxon_query\x18\x02 \x01(\tB\x10\xc2\xf0\x19\x0c\x12\n\n\x03hum\x12\x03humH\x00R\ntaxonQuery\x12k\n\x0ftax_rank_filter\x18\x03 \x01(\x0e\x32\x34.ncbi.datasets.v2.OrganismQueryRequest.TaxRankFilterB\r\xc2\xf0\x19\t\n\x07speciesR\rtaxRankFilter\x12\x8f\x01\n\x15taxon_resource_filter\x18\x04 \x01(\x0e\x32:.ncbi.datasets.v2.OrganismQueryRequest.TaxonResourceFilterB\x1f\xc2\xf0\x19\x1b\n\x19TAXON_RESOURCE_FILTER_ALLR\x13taxonResourceFilter\".\n\rTaxRankFilter\x12\x0b\n\x07species\x10\x00\x12\x10\n\x0chigher_taxon\x10\x01\"v\n\x13TaxonResourceFilter\x12\x1d\n\x19TAXON_RESOURCE_FILTER_ALL\x10\x00\x12 \n\x1cTAXON_RESOURCE_FILTER_GENOME\x10\x01\x12\x1e\n\x1aTAXON_RESOURCE_FILTER_GENE\x10\x02\x42\x04\n\x02id\"\xed\x01\n\rSciNameAndIds\x12U\n\x10sci_name_and_ids\x18\x01 \x03(\x0b\x32,.ncbi.datasets.v2.SciNameAndIds.SciNameAndIdR\rsciNameAndIds\x1a\x84\x01\n\x0cSciNameAndId\x12\x19\n\x08sci_name\x18\x01 \x01(\tR\x07sciName\x12\x15\n\x06tax_id\x18\x02 \x01(\x04R\x05taxId\x12\x1f\n\x0b\x63ommon_name\x18\x03 \x01(\tR\ncommonName\x12!\n\x0cmatched_term\x18\x04 \x01(\tR\x0bmatchedTerm\"\xb8\x03\n\x0fOrthologRequest\x12\x35\n\x07gene_id\x18\x01 \x01(\rB\x1c\xc2\xf0\x19\x18\x12\x0c\n\x04\x32\x37\x37\x38\x12\x04GNAS\x12\x08\n\x01\x32\x12\x03\x41\x32MR\x06geneId\x12X\n\x10returned_content\x18\x02 \x01(\x0e\x32-.ncbi.datasets.v2.OrthologRequest.ContentTypeR\x0freturnedContent\x12\xa2\x01\n\x0ctaxon_filter\x18\x03 \x03(\tB\x7f\xc2\xf0\x19{\x12\'\n\x04\x39\x36\x30\x36\n\x05\x31\x30\x30\x39\x30\x12\x18NCBI Taxonomy Identifier\x12!\n\x05human\n\x0bhouse mouse\x12\x0b\x43ommon Name\x12-\n\x0chomo sapiens\n\x0cmus musculus\x12\x0fScientific NameR\x0btaxonFilter\x12%\n\tpage_size\x18\t \x01(\x05\x42\x08\xc2\xf0\x19\x04\n\x02\x32\x30R\x08pageSize\x12\x1d\n\npage_token\x18\n \x01(\tR\tpageToken\")\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\x0c\n\x08IDS_ONLY\x10\x01\"\x9e\x07\n\x12GeneDatasetRequest\x12[\n\x08gene_ids\x18\x01 \x03(\rB@\xc2\xf0\x19<\x12\x14\n\x05\x35\x39\x30\x36\x37\x12\x0bIL21 GeneID\x12$\n\x05\x35\x39\x30\x36\x37\n\x05\x35\x30\x36\x31\x35\x12\x14IL21 & IL21R GeneIDsR\x07geneIds\x12W\n\x17include_annotation_type\x18\x06 \x03(\x0e\x32\x17.ncbi.datasets.v2.FastaB\x06\xda\xf0\x19\x02\x08\x02R\x15includeAnnotationType\x12[\n\x10returned_content\x18\x07 \x01(\x0e\x32\x30.ncbi.datasets.v2.GeneDatasetRequest.ContentTypeR\x0freturnedContent\x12!\n\x0c\x66\x61sta_filter\x18\x0c \x03(\tR\x0b\x66\x61staFilter\x12\xd9\x01\n\naux_report\x18\r \x03(\x0e\x32:.ncbi.datasets.v2.GeneDatasetRequest.GeneDatasetReportTypeB~\xc2\xf0\x19z\x12x\n\x0ePRODUCT_REPORT\x12\x66Select additional report types to include with download. The base data report will always be included.R\tauxReport\x12\x80\x01\n\x0ctable_fields\x18\x08 \x03(\tB]\xc2\xf0\x19Y\x12?\n\x07gene-id\n\tgene-type\n\x0b\x64\x65scription\x12\x1cGeneID, type and description\x12\x16\n\x07gene-id\x12\x0bJust GeneIDR\x0btableFields\x12\x66\n\x11table_report_type\x18\x0e \x01(\x0e\x32:.ncbi.datasets.v2.GeneDatasetRequest.GeneDatasetReportTypeR\x0ftableReportType\")\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\x0c\n\x08IDS_ONLY\x10\x01\"?\n\x15GeneDatasetReportType\x12\x12\n\x0e\x44\x41TASET_REPORT\x10\x00\x12\x12\n\x0ePRODUCT_REPORT\x10\x01:\x1f\xc2\xf0\x19\x1b\n\x19gene_ids:\n- 59067\n- 50615\"\xe2\t\n\x19GeneDatasetReportsRequest\x12\x62\n\x10returned_content\x18\x01 \x01(\x0e\x32\x37.ncbi.datasets.v2.GeneDatasetReportsRequest.ContentTypeR\x0freturnedContent\x12[\n\x08gene_ids\x18\x02 \x03(\rB@\xc2\xf0\x19<\x12\x14\n\x05\x35\x39\x30\x36\x37\x12\x0bIL21 GeneID\x12$\n\x05\x35\x39\x30\x36\x37\n\x05\x35\x30\x36\x31\x35\x12\x14IL21 & IL21R GeneIDsR\x07geneIds\x12\x81\x01\n\naccessions\x18\x03 \x03(\tBa\xc2\xf0\x19]\x12.\n\x0bNM_021803.4\n\x0bNM_181078.3\x12\x12RefSeq transcripts\x12+\n\x0bNP_068575.1\n\x0bNP_851564.1\x12\x0fRefSeq proteinsR\naccessions\x12g\n\x11symbols_for_taxon\x18\x04 \x01(\x0b\x32;.ncbi.datasets.v2.GeneDatasetReportsRequest.SymbolsForTaxonR\x0fsymbolsForTaxon\x12s\n\x05taxon\x18\x05 \x01(\tB]\xc2\xf0\x19Y\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy Identifier\x12\x14\n\x05human\x12\x0b\x43ommon Name\x12\x1f\n\x0cHomo sapiens\x12\x0fScientific NameR\x05taxon\x12\x80\x01\n\x0ctable_fields\x18\x08 \x03(\tB]\xc2\xf0\x19Y\x12?\n\x07gene-id\n\tgene-type\n\x0b\x64\x65scription\x12\x1cGeneID, type and description\x12\x16\n\x07gene-id\x12\x0bJust GeneIDR\x0btableFields\x12!\n\x0ctable_format\x18\r \x01(\tR\x0btableFormat\x12%\n\tpage_size\x18\t \x01(\x05\x42\x08\xc2\xf0\x19\x04\n\x02\x32\x30R\x08pageSize\x12\x1d\n\npage_token\x18\n \x01(\tR\tpageToken\x12\x30\n\x05query\x18\x0b \x01(\tB\x1a\xc2\xf0\x19\x16\x12\x14\n\x12\x41\x32M immunoglobulinR\x05query\x12\x63\n\x05types\x18\x0c \x03(\x0e\x32\x1a.ncbi.datasets.v2.GeneTypeB1\xc2\xf0\x19-\x12+\n\x0ePROTEIN_CODING\x12\x19Only protein coding genesR\x05types\x1a\xb4\x01\n\x0fSymbolsForTaxon\x12,\n\x07symbols\x18\x01 \x03(\tB\x12\xc2\xf0\x19\x0e\x12\x0c\n\x04GNAS\x12\x04GNASR\x07symbols\x12s\n\x05taxon\x18\x02 \x01(\tB]\xc2\xf0\x19Y\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy Identifier\x12\x14\n\x05human\x12\x0b\x43ommon Name\x12\x1f\n\x0cHomo sapiens\x12\x0fScientific NameR\x05taxon\"G\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\x0c\n\x08IDS_ONLY\x10\x01\x12\x0b\n\x07PRODUCT\x10\x02\x12\x0f\n\x0b\x43OUNTS_ONLY\x10\x03:\x1f\xc2\xf0\x19\x1b\n\x19gene_ids:\n- 59067\n- 50615\"\x90\x01\n\x10GeneLinksRequest\x12[\n\x08gene_ids\x18\x01 \x03(\rB@\xc2\xf0\x19<\x12\x14\n\x05\x35\x39\x30\x36\x37\x12\x0bIL21 GeneID\x12$\n\x05\x35\x39\x30\x36\x37\n\x05\x35\x30\x36\x31\x35\x12\x14IL21 & IL21R GeneIDsR\x07geneIds:\x1f\xc2\xf0\x19\x1b\n\x19gene_ids:\n- 59067\n- 50615\"\xe8\x02\n\x0eGeneLinksReply\x12H\n\ngene_links\x18\x01 \x03(\x0b\x32).ncbi.datasets.v2.GeneLinksReply.GeneLinkR\tgeneLinks\x1a\xbe\x01\n\x08GeneLink\x12\x17\n\x07gene_id\x18\x01 \x01(\rR\x06geneId\x12S\n\x0egene_link_type\x18\x02 \x01(\x0e\x32-.ncbi.datasets.v2.GeneLinksReply.GeneLinkTypeR\x0cgeneLinkType\x12#\n\rresource_link\x18\x03 \x01(\tR\x0cresourceLink\x12\x1f\n\x0bresource_id\x18\x04 \x01(\tR\nresourceId\"K\n\x0cGeneLinkType\x12\x0b\n\x07\x44\x45\x46\x41ULT\x10\x00\x12\r\n\tGENE_LINK\x10\x01\x12\x0c\n\x08GDV_LINK\x10\x02\x12\x11\n\rORTHOLOG_LINK\x10\x03\"\xff\x01\n\x1cGeneChromosomeSummaryRequest\x12s\n\x05taxon\x18\x01 \x01(\tB]\xc2\xf0\x19Y\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy Identifier\x12\x14\n\x05human\x12\x0b\x43ommon Name\x12\x1f\n\x0cHomo sapiens\x12\x0fScientific NameR\x05taxon\x12j\n\x0f\x61nnotation_name\x18\x02 \x01(\tBA\xc2\xf0\x19=\x12;\n\x1f\x41nnotation Release 109.20211119\x12\x18Human annotation releaseR\x0e\x61nnotationName\"\xfd\x01\n\x1aGeneChromosomeSummaryReply\x12~\n\x19gene_chromosome_summaries\x18\x01 \x03(\x0b\x32\x42.ncbi.datasets.v2.GeneChromosomeSummaryReply.GeneChromosomeSummaryR\x17geneChromosomeSummaries\x1a_\n\x15GeneChromosomeSummary\x12\x12\n\x04name\x18\x01 \x01(\tR\x04name\x12\x14\n\x05\x63ount\x18\x02 \x01(\rR\x05\x63ount\x12\x1c\n\taccession\x18\x03 \x01(\tR\taccession\"\xc4\x01\n\x14\x41ssemblyLinksRequest\x12v\n\naccessions\x18\x01 \x03(\tBV\xc2\xf0\x19R\x12\x35\n\x10GCF_000001405.40\n\x10GCF_000001635.27\x12\x0fHuman and Mouse\x12\x19\n\x10GCF_000001405.40\x12\x05HumanR\naccessions:4\xc2\xf0\x19\x30\n.accessions:\n  accessions:\n  - GCF_000001405.40\"\xb8\x03\n\x12\x41ssemblyLinksReply\x12X\n\x0e\x61ssembly_links\x18\x01 \x03(\x0b\x32\x31.ncbi.datasets.v2.AssemblyLinksReply.AssemblyLinkR\rassemblyLinks\x1a\xe5\x01\n\x0c\x41ssemblyLink\x12\x1c\n\taccession\x18\x01 \x01(\tR\taccession\x12\x63\n\x12\x61ssembly_link_type\x18\x02 \x01(\x0e\x32\x35.ncbi.datasets.v2.AssemblyLinksReply.AssemblyLinkTypeR\x10\x61ssemblyLinkType\x12#\n\rresource_link\x18\x03 \x01(\tR\x0cresourceLink\x12-\n\x12linked_identifiers\x18\x04 \x03(\tR\x11linkedIdentifiers\"`\n\x10\x41ssemblyLinkType\x12\x0b\n\x07\x44\x45\x46\x41ULT\x10\x00\x12\x0c\n\x08GDV_LINK\x10\x01\x12\x0c\n\x08\x46TP_LINK\x10\x02\x12\x13\n\x0f\x41SSEMBLY_PUBMED\x10\x03\x12\x0e\n\nBLAST_LINK\x10\x04\"\x91\x0c\n\x13VirusDatasetRequest\x12\x63\n\naccessions\x18\x10 \x03(\tBC\xc2\xf0\x19?\x12=\n\x0bNC_045512.2\x12.SARS-COV-2 isolate Wuhan-Hu-1, complete genomeR\naccessions\x12Z\n\x05taxon\x18\x13 \x01(\tBD\xc2\xf0\x19@\x12\x15\n\x07\x32\x36\x39\x37\x30\x34\x39\x12\nSARS-COV-2\x12\x16\n\x05\x31\x31\x31\x31\x38\x12\rCoronaviridae\x12\x0f\n\x07\x31\x33\x33\x35\x36\x32\x36\x12\x04MERSR\x05taxon\x12u\n\x0brefseq_only\x18\x04 \x01(\x08\x42T\xc2\xf0\x19P\n\x05\x66\x61lse\x12\x1f\n\x04true\x12\x17Limit to RefSeq genomes\x12&\n\x05\x66\x61lse\x12\x1d\x42oth GenBank & RefSeq genomesR\nrefseqOnly\x12l\n\x0e\x61nnotated_only\x18\x05 \x01(\x08\x42\x45\xc2\xf0\x19\x41\n\x05\x66\x61lse\x12\"\n\x04true\x12\x1aLimit to annotated genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\rannotatedOnly\x12\x65\n\x0ereleased_since\x18\x06 \x01(\x0b\x32\x1a.google.protobuf.TimestampB\"\xc2\xf0\x19\x1e\x12\x1c\n\n2020-08-01\x12\x0e\x41ugust 1, 2020R\rreleasedSince\x12\x62\n\rupdated_since\x18\x11 \x01(\x0b\x32\x1a.google.protobuf.TimestampB!\xc2\xf0\x19\x1d\x12\x1b\n\n2021-07-18\x12\rJuly 18, 2021R\x0cupdatedSince\x12(\n\x04host\x18\x07 \x01(\tB\x14\xc2\xf0\x19\x10\x12\x0e\n\x05human\x12\x05humanR\x04host\x12\x37\n\x17pangolin_classification\x18\x14 \x01(\tR\x16pangolinClassification\x12@\n\x0cgeo_location\x18\x08 \x01(\tB\x1d\xc2\xf0\x19\x19\x12\n\n\x03USA\x12\x03USA\x12\x0b\n\x02TX\x12\x05TexasR\x0bgeoLocation\x12i\n\rcomplete_only\x18\t \x01(\x08\x42\x44\xc2\xf0\x19@\n\x05\x66\x61lse\x12!\n\x04true\x12\x19Limit to complete genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\x0c\x63ompleteOnly\x12\x98\x01\n\x0ctable_fields\x18\n \x03(\x0e\x32!.ncbi.datasets.v2.VirusTableFieldBR\xc2\xf0\x19N\x12L\n\x14nucleotide_accession\n\x11nucleotide_length\n\x10nuc_completeness\x12\x0f\x41ssembly FieldsR\x0btableFields\x12\x8f\x01\n\x10include_sequence\x18\x16 \x03(\x0e\x32#.ncbi.datasets.v2.ViralSequenceTypeB?\xc2\xf0\x19;\x12\x39\n\x06GENOME\n\x03\x43\x44S\n\x07PROTEIN\x12!Select viral sequences to includeR\x0fincludeSequence\x12\xc3\x01\n\naux_report\x18\x17 \x03(\x0e\x32(.ncbi.datasets.v2.VirusDatasetReportTypeBz\xc2\xf0\x19v\x12t\n\nANNOTATION\x12\x66Select additional report types to include with download. The base data report will always be included.R\tauxReport\x12m\n\x06\x66ormat\x18\x0e \x01(\x0e\x32\x1d.ncbi.datasets.v2.TableFormatB6\xc2\xf0\x19\x32\n\x03tsv\x12\n\n\x03tsv\x12\x03TSV\x12\n\n\x03\x63sv\x12\x03\x43SV\x12\x13\n\x05jsonl\x12\nJSON LinesR\x06\x66ormat\x12\x17\n\x07use_psg\x18\x63 \x01(\x08R\x06usePsg\"\xdf\x07\n\x12VirusDatasetFilter\x12\xac\x01\n\naccessions\x18\x02 \x03(\tB\x8b\x01\xc2\xf0\x19\x86\x01\x12\x83\x01\n\x17NC_045512.2,NC_038294.1\x12hSARS-COV-2 isolate Wuhan-Hu-1 and Middle East respiratory syndrome-related coronavirus, complete genomesR\naccessions\x12Z\n\x05taxon\x18\x03 \x01(\tBD\xc2\xf0\x19@\x12\x15\n\x07\x32\x36\x39\x37\x30\x34\x39\x12\nSARS-COV-2\x12\x16\n\x05\x31\x31\x31\x31\x38\x12\rCoronaviridae\x12\x0f\n\x07\x31\x33\x33\x35\x36\x32\x36\x12\x04MERSR\x05taxon\x12u\n\x0brefseq_only\x18\x04 \x01(\x08\x42T\xc2\xf0\x19P\n\x05\x66\x61lse\x12\x1f\n\x04true\x12\x17Limit to RefSeq genomes\x12&\n\x05\x66\x61lse\x12\x1d\x42oth GenBank & RefSeq genomesR\nrefseqOnly\x12l\n\x0e\x61nnotated_only\x18\x05 \x01(\x08\x42\x45\xc2\xf0\x19\x41\n\x05\x66\x61lse\x12\"\n\x04true\x12\x1aLimit to annotated genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\rannotatedOnly\x12\x65\n\x0ereleased_since\x18\x06 \x01(\x0b\x32\x1a.google.protobuf.TimestampB\"\xc2\xf0\x19\x1e\x12\x1c\n\n2020-08-01\x12\x0e\x41ugust 1, 2020R\rreleasedSince\x12\x62\n\rupdated_since\x18\x0b \x01(\x0b\x32\x1a.google.protobuf.TimestampB!\xc2\xf0\x19\x1d\x12\x1b\n\n2021-07-18\x12\rJuly 18, 2021R\x0cupdatedSince\x12(\n\x04host\x18\x07 \x01(\tB\x14\xc2\xf0\x19\x10\x12\x0e\n\x05human\x12\x05humanR\x04host\x12\x37\n\x17pangolin_classification\x18\x08 \x01(\tR\x16pangolinClassification\x12@\n\x0cgeo_location\x18\t \x01(\tB\x1d\xc2\xf0\x19\x19\x12\n\n\x03USA\x12\x03USA\x12\x0b\n\x02TX\x12\x05TexasR\x0bgeoLocation\x12i\n\rcomplete_only\x18\n \x01(\x08\x42\x44\xc2\xf0\x19@\n\x05\x66\x61lse\x12!\n\x04true\x12\x19Limit to complete genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\x0c\x63ompleteOnly\"\xe5\x03\n\x16VirusDataReportRequest\x12<\n\x06\x66ilter\x18\x01 \x01(\x0b\x32$.ncbi.datasets.v2.VirusDatasetFilterR\x06\x66ilter\x12_\n\x10returned_content\x18\x02 \x01(\x0e\x32\x34.ncbi.datasets.v2.VirusDataReportRequest.ContentTypeR\x0freturnedContent\x12i\n\x0ctable_fields\x18\x03 \x03(\tBF\xc2\xf0\x19\x42\x12@\n\taccession\n\x0bis-complete\n\x0cis-annotated\x12\x18Virus Data Report FieldsR\x0btableFields\x12!\n\x0ctable_format\x18\x06 \x01(\tR\x0btableFormat\x12%\n\tpage_size\x18\x04 \x01(\x05\x42\x08\xc2\xf0\x19\x04\n\x02\x32\x30R\x08pageSize\x12\x1d\n\npage_token\x18\x05 \x01(\tR\tpageToken\"0\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\x13\n\x0f\x41\x43\x43\x45SSIONS_ONLY\x10\x01:&\xc2\xf0\x19\"\n taxon: 2697049\nrefseq_only: true\"\xf4\x06\n\x15VirusAnnotationFilter\x12\xac\x01\n\naccessions\x18\x02 \x03(\tB\x8b\x01\xc2\xf0\x19\x86\x01\x12\x83\x01\n\x17NC_045512.2,NC_038294.1\x12hSARS-COV-2 isolate Wuhan-Hu-1 and Middle East respiratory syndrome-related coronavirus, complete genomesR\naccessions\x12Z\n\x05taxon\x18\x03 \x01(\tBD\xc2\xf0\x19@\x12\x15\n\x07\x32\x36\x39\x37\x30\x34\x39\x12\nSARS-COV-2\x12\x16\n\x05\x31\x31\x31\x31\x38\x12\rCoronaviridae\x12\x0f\n\x07\x31\x33\x33\x35\x36\x32\x36\x12\x04MERSR\x05taxon\x12u\n\x0brefseq_only\x18\x04 \x01(\x08\x42T\xc2\xf0\x19P\n\x05\x66\x61lse\x12\x1f\n\x04true\x12\x17Limit to RefSeq genomes\x12&\n\x05\x66\x61lse\x12\x1d\x42oth GenBank & RefSeq genomesR\nrefseqOnly\x12\x65\n\x0ereleased_since\x18\x06 \x01(\x0b\x32\x1a.google.protobuf.TimestampB\"\xc2\xf0\x19\x1e\x12\x1c\n\n2020-08-01\x12\x0e\x41ugust 1, 2020R\rreleasedSince\x12\x62\n\rupdated_since\x18\x0b \x01(\x0b\x32\x1a.google.protobuf.TimestampB!\xc2\xf0\x19\x1d\x12\x1b\n\n2021-07-18\x12\rJuly 18, 2021R\x0cupdatedSince\x12(\n\x04host\x18\x07 \x01(\tB\x14\xc2\xf0\x19\x10\x12\x0e\n\x05human\x12\x05humanR\x04host\x12\x37\n\x17pangolin_classification\x18\x08 \x01(\tR\x16pangolinClassification\x12@\n\x0cgeo_location\x18\t \x01(\tB\x1d\xc2\xf0\x19\x19\x12\n\n\x03USA\x12\x03USA\x12\x0b\n\x02TX\x12\x05TexasR\x0bgeoLocation\x12i\n\rcomplete_only\x18\n \x01(\x08\x42\x44\xc2\xf0\x19@\n\x05\x66\x61lse\x12!\n\x04true\x12\x19Limit to complete genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\x0c\x63ompleteOnly\"\xd4\x02\n\x1cVirusAnnotationReportRequest\x12?\n\x06\x66ilter\x18\x01 \x01(\x0b\x32\'.ncbi.datasets.v2.VirusAnnotationFilterR\x06\x66ilter\x12\x62\n\x0ctable_fields\x18\x03 \x03(\tB?\xc2\xf0\x19;\x12\x39\n\taccession\n\x0cisolate-name\x12\x1eVirus Annotation Report FieldsR\x0btableFields\x12!\n\x0ctable_format\x18\x06 \x01(\tR\x0btableFormat\x12%\n\tpage_size\x18\x04 \x01(\x05\x42\x08\xc2\xf0\x19\x04\n\x02\x32\x30R\x08pageSize\x12\x1d\n\npage_token\x18\x05 \x01(\tR\tpageToken:&\xc2\xf0\x19\"\n taxon: 2697049\nrefseq_only: true\"\xc9\x01\n\x18VirusAvailabilityRequest\x12\xac\x01\n\naccessions\x18\x01 \x03(\tB\x8b\x01\xc2\xf0\x19\x86\x01\x12\x83\x01\n\x17NC_045512.2,NC_038294.1\x12hSARS-COV-2 isolate Wuhan-Hu-1 and Middle East respiratory syndrome-related coronavirus, complete genomesR\naccessions\"\x87\x01\n\x11VirusAvailability\x12)\n\x10valid_accessions\x18\x01 \x03(\tR\x0fvalidAccessions\x12-\n\x12invalid_accessions\x18\x02 \x03(\tR\x11invalidAccessions\x12\x18\n\x07message\x18\x03 \x01(\tR\x07message\"\xfd\n\n\x1aSars2ProteinDatasetRequest\x12~\n\x08proteins\x18\x01 \x03(\tBb\xc2\xf0\x19^\x12\x1e\n\rspike protein\x12\rSpike Protein\x12<\n\rspike protein\n\x10\x65nvelope protein\n\x04RdRp\x12\x13SARS-CoV-2 proteinsR\x08proteins\x12u\n\x0brefseq_only\x18\x04 \x01(\x08\x42T\xc2\xf0\x19P\n\x05\x66\x61lse\x12\x1f\n\x04true\x12\x17Limit to RefSeq genomes\x12&\n\x05\x66\x61lse\x12\x1d\x42oth GenBank & RefSeq genomesR\nrefseqOnly\x12l\n\x0e\x61nnotated_only\x18\x05 \x01(\x08\x42\x45\xc2\xf0\x19\x41\n\x05\x66\x61lse\x12\"\n\x04true\x12\x1aLimit to annotated genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\rannotatedOnly\x12\x65\n\x0ereleased_since\x18\x06 \x01(\x0b\x32\x1a.google.protobuf.TimestampB\"\xc2\xf0\x19\x1e\x12\x1c\n\n2020-08-01\x12\x0e\x41ugust 1, 2020R\rreleasedSince\x12\x62\n\rupdated_since\x18\x0f \x01(\x0b\x32\x1a.google.protobuf.TimestampB!\xc2\xf0\x19\x1d\x12\x1b\n\n2021-07-18\x12\rJuly 18, 2021R\x0cupdatedSince\x12(\n\x04host\x18\x07 \x01(\tB\x14\xc2\xf0\x19\x10\x12\x0e\n\x05human\x12\x05humanR\x04host\x12@\n\x0cgeo_location\x18\x08 \x01(\tB\x1d\xc2\xf0\x19\x19\x12\n\n\x03USA\x12\x03USA\x12\x0b\n\x02TX\x12\x05TexasR\x0bgeoLocation\x12i\n\rcomplete_only\x18\t \x01(\x08\x42\x44\xc2\xf0\x19@\n\x05\x66\x61lse\x12!\n\x04true\x12\x19Limit to complete genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\x0c\x63ompleteOnly\x12\x98\x01\n\x0ctable_fields\x18\n \x03(\x0e\x32!.ncbi.datasets.v2.VirusTableFieldBR\xc2\xf0\x19N\x12L\n\x14nucleotide_accession\n\x11nucleotide_length\n\x10nuc_completeness\x12\x0f\x41ssembly FieldsR\x0btableFields\x12\x87\x01\n\x10include_sequence\x18\x16 \x03(\x0e\x32#.ncbi.datasets.v2.ViralSequenceTypeB7\xc2\xf0\x19\x33\x12\x31\n\x03\x43\x44S\n\x07PROTEIN\x12!Select viral sequences to includeR\x0fincludeSequence\x12\xc3\x01\n\naux_report\x18\x17 \x03(\x0e\x32(.ncbi.datasets.v2.VirusDatasetReportTypeBz\xc2\xf0\x19v\x12t\n\nANNOTATION\x12\x66Select additional report types to include with download. The base data report will always be included.R\tauxReport\x12m\n\x06\x66ormat\x18\x0e \x01(\x0e\x32\x1d.ncbi.datasets.v2.TableFormatB6\xc2\xf0\x19\x32\n\x03tsv\x12\n\n\x03tsv\x12\x03TSV\x12\n\n\x03\x63sv\x12\x03\x43SV\x12\x13\n\x05jsonl\x12\nJSON LinesR\x06\x66ormat\"\xb0\r\n\x0f\x44ownloadSummary\x12!\n\x0crecord_count\x18\x01 \x01(\rR\x0brecordCount\x12)\n\x0e\x61ssembly_count\x18\x02 \x01(\rB\x02\x18\x01R\rassemblyCount\x12J\n\x13resource_updated_on\x18\x04 \x01(\x0b\x32\x1a.google.protobuf.TimestampR\x11resourceUpdatedOn\x12\x46\n\x08hydrated\x18\x05 \x01(\x0b\x32*.ncbi.datasets.v2.DownloadSummary.HydratedR\x08hydrated\x12L\n\ndehydrated\x18\x06 \x01(\x0b\x32,.ncbi.datasets.v2.DownloadSummary.DehydratedR\ndehydrated\x12;\n\x06\x65rrors\x18\x07 \x03(\x0b\x32\x1f.ncbi.datasets.v2.reports.ErrorB\x02\x18\x01R\x06\x65rrors\x12=\n\x08messages\x18\x08 \x03(\x0b\x32!.ncbi.datasets.v2.reports.MessageR\x08messages\x12Y\n\x0f\x61vailable_files\x18\t \x01(\x0b\x32\x30.ncbi.datasets.v2.DownloadSummary.AvailableFilesR\x0e\x61vailableFiles\x1a\x8c\x01\n\x08Hydrated\x12\x33\n\x16\x65stimated_file_size_mb\x18\x01 \x01(\rR\x13\x65stimatedFileSizeMb\x12\x10\n\x03url\x18\x02 \x01(\tR\x03url\x12\x39\n\x19\x63li_download_command_line\x18\x03 \x01(\tR\x16\x63liDownloadCommandLine\x1a\xcb\x01\n\nDehydrated\x12\x33\n\x16\x65stimated_file_size_mb\x18\x01 \x01(\rR\x13\x65stimatedFileSizeMb\x12\x10\n\x03url\x18\x02 \x01(\tR\x03url\x12\x39\n\x19\x63li_download_command_line\x18\x03 \x01(\tR\x16\x63liDownloadCommandLine\x12;\n\x1a\x63li_rehydrate_command_line\x18\x04 \x01(\tR\x17\x63liRehydrateCommandLine\x1a\x45\n\x0b\x46ileSummary\x12\x1d\n\nfile_count\x18\x01 \x01(\rR\tfileCount\x12\x17\n\x07size_mb\x18\x02 \x01(\x02R\x06sizeMb\x1a\xf1\x05\n\x0e\x41vailableFiles\x12Y\n\x11\x61ll_genomic_fasta\x18\x01 \x01(\x0b\x32-.ncbi.datasets.v2.DownloadSummary.FileSummaryR\x0f\x61llGenomicFasta\x12L\n\ngenome_gff\x18\x02 \x01(\x0b\x32-.ncbi.datasets.v2.DownloadSummary.FileSummaryR\tgenomeGff\x12N\n\x0bgenome_gbff\x18\x03 \x01(\x0b\x32-.ncbi.datasets.v2.DownloadSummary.FileSummaryR\ngenomeGbff\x12J\n\trna_fasta\x18\x04 \x01(\x0b\x32-.ncbi.datasets.v2.DownloadSummary.FileSummaryR\x08rnaFasta\x12L\n\nprot_fasta\x18\x05 \x01(\x0b\x32-.ncbi.datasets.v2.DownloadSummary.FileSummaryR\tprotFasta\x12L\n\ngenome_gtf\x18\x06 \x01(\x0b\x32-.ncbi.datasets.v2.DownloadSummary.FileSummaryR\tgenomeGtf\x12J\n\tcds_fasta\x18\x07 \x01(\x0b\x32-.ncbi.datasets.v2.DownloadSummary.FileSummaryR\x08\x63\x64sFasta\x12V\n\x0fsequence_report\x18\x08 \x01(\x0b\x32-.ncbi.datasets.v2.DownloadSummary.FileSummaryR\x0esequenceReport\x12Z\n\x11\x61nnotation_report\x18\t \x01(\x0b\x32-.ncbi.datasets.v2.DownloadSummary.FileSummaryR\x10\x61nnotationReport\"#\n\rTabularOutput\x12\x12\n\x04\x64\x61ta\x18\x01 \x01(\tR\x04\x64\x61ta\"\xbb\x03\n\x18MicroBiggeDatasetRequest\x12,\n\x11opaque_solr_query\x18\x01 \x01(\tH\x00R\x0fopaqueSolrQuery\x12Q\n\x05\x66iles\x18\x02 \x03(\x0e\x32\x33.ncbi.datasets.v2.MicroBiggeDatasetRequest.FileTypeB\x06\xda\xf0\x19\x02\x08\x02R\x05\x66iles\x12o\n\x14\x65lement_flank_config\x18\x03 \x01(\x0b\x32=.ncbi.datasets.v2.MicroBiggeDatasetRequest.ElementFlankConfigR\x12\x65lementFlankConfig\x1a,\n\x12\x45lementFlankConfig\x12\x16\n\x06length\x18\x01 \x01(\rR\x06length\"v\n\x08\x46ileType\x12\x19\n\x15\x46ILE_TYPE_UNSPECIFIED\x10\x00\x12\x11\n\relement_fasta\x10\x01\x12\x17\n\x13\x65lement_flank_fasta\x10\x02\x12\x10\n\x0c\x63ontig_fasta\x10\x03\x12\x11\n\rprotein_fasta\x10\x04\x42\x07\n\x05query\"\x7f\n\x0cVersionReply\x12\x18\n\x07version\x18\x01 \x01(\tR\x07version\x12\x1b\n\tmajor_ver\x18\x02 \x01(\rR\x08majorVer\x12\x1b\n\tminor_ver\x18\x03 \x01(\rR\x08minorVer\x12\x1b\n\tpatch_ver\x18\x04 \x01(\rR\x08patchVer\"\x81\x05\n\x15ProkaryoteGeneRequest\x12\x93\x01\n\naccessions\x18\x01 \x03(\tBs\xc2\xf0\x19o\x12.\n\x0eWP_015878339.1\x12\x1c\x44NA-deoxyinosine glycosylase\x12=\n\x0eWP_001040085.1\x12+type II CRISPR RNA-guided endonuclease Cas9R\naccessions\x12\xf6\x01\n\x17include_annotation_type\x18\x02 \x03(\x0e\x32\x17.ncbi.datasets.v2.FastaB\xa4\x01\xda\xf0\x19\x02\x08\x02\xc2\xf0\x19\x99\x01\x12\x1e\n\nFASTA_GENE\x12\x10Gene range FASTA\x12?\n\tFASTA_RNA\n\rFASTA_PROTEIN\x12#Transcript & protein sequence FASTA\x12\x36\n\nFASTA_GENE\n\tFASTA_RNA\n\rFASTA_PROTEIN\x12\x0e\x41ll annotationR\x15includeAnnotationType\x12\x63\n\x11gene_flank_config\x18\x03 \x01(\x0b\x32\x37.ncbi.datasets.v2.ProkaryoteGeneRequest.GeneFlankConfigR\x0fgeneFlankConfig\x12\x14\n\x05taxon\x18\x04 \x01(\tR\x05taxon\x1a)\n\x0fGeneFlankConfig\x12\x16\n\x06length\x18\x01 \x01(\rR\x06length:3\xc2\xf0\x19/\n-accessions:\n- WP_000000001.1\n- WP_000000002.1\"L\n\x0cSleepRequest\x12\x1d\n\nsleep_msec\x18\x01 \x01(\x05R\tsleepMsec\x12\x1d\n\nerror_rate\x18\x02 \x01(\x02R\terrorRate\"\"\n\nSleepReply\x12\x14\n\x05reply\x18\x01 \x01(\tR\x05reply\"\xea\x04\n\x0cTaxonomyNode\x12\x15\n\x06tax_id\x18\x01 \x01(\rR\x05taxId\x12#\n\rorganism_name\x18\x02 \x01(\tR\x0corganismName\x12\x1f\n\x0b\x63ommon_name\x18\x03 \x01(\tR\ncommonName\x12\x18\n\x07lineage\x18\x04 \x03(\rR\x07lineage\x12\x1a\n\x08\x63hildren\x18\x05 \x03(\rR\x08\x63hildren\x12_\n-descendent_with_described_species_names_count\x18\x06 \x01(\rR(descendentWithDescribedSpeciesNamesCount\x12\x37\n\x04rank\x18\x07 \x01(\x0e\x32#.ncbi.datasets.v2.Organism.RankTypeR\x04rank\x12;\n\x1ahas_described_species_name\x18\x08 \x01(\x08R\x17hasDescribedSpeciesName\x12\x42\n\x06\x63ounts\x18\t \x03(\x0b\x32*.ncbi.datasets.v2.TaxonomyNode.CountByTypeR\x06\x63ounts\x12\x17\n\x07min_ord\x18\x14 \x01(\rR\x06minOrd\x12\x17\n\x07max_ord\x18\x15 \x01(\rR\x06maxOrd\x12$\n\x0e_indexed_names\x18\x65 \x03(\tR\x0cIndexedNames\x1aT\n\x0b\x43ountByType\x12/\n\x04type\x18\x01 \x01(\x0e\x32\x1b.ncbi.datasets.v2.CountTypeR\x04type\x12\x14\n\x05\x63ount\x18\x02 \x01(\rR\x05\x63ount\"\xd6\x02\n\x17TaxonomyMetadataRequest\x12\x91\x01\n\x06taxons\x18\x01 \x03(\tBy\xc2\xf0\x19u\x12\'\n\x04\x39\x36\x30\x36\n\x05\x31\x30\x30\x39\x30\x12\x18NCBI Taxonomy Identifier\x12\x1b\n\x05human\n\x05mouse\x12\x0b\x43ommon Name\x12-\n\x0cHomo sapiens\n\x0cMus musculus\x12\x0fScientific NameR\x06taxons\x12`\n\x10returned_content\x18\x02 \x01(\x0e\x32\x35.ncbi.datasets.v2.TaxonomyMetadataRequest.ContentTypeR\x0freturnedContent\"\'\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\n\n\x06TAXIDS\x10\x01:\x1c\xc2\xf0\x19\x18\n\x16taxons:\n- 9606\n- mouse\"\xd9\x01\n\rTaxonomyMatch\x12=\n\x08warnings\x18\x01 \x03(\x0b\x32!.ncbi.datasets.v2.reports.WarningR\x08warnings\x12\x37\n\x06\x65rrors\x18\x02 \x03(\x0b\x32\x1f.ncbi.datasets.v2.reports.ErrorR\x06\x65rrors\x12\x14\n\x05query\x18\x03 \x03(\tR\x05query\x12:\n\x08taxonomy\x18\x04 \x01(\x0b\x32\x1e.ncbi.datasets.v2.TaxonomyNodeR\x08taxonomy\"\xa1\x01\n\x18TaxonomyMetadataResponse\x12=\n\x08messages\x18\x01 \x03(\x0b\x32!.ncbi.datasets.v2.reports.MessageR\x08messages\x12\x46\n\x0etaxonomy_nodes\x18\x02 \x03(\x0b\x32\x1f.ncbi.datasets.v2.TaxonomyMatchR\rtaxonomyNodes\"\xa3\x02\n\x1eTaxonomyFilteredSubtreeRequest\x12\x91\x01\n\x06taxons\x18\x01 \x03(\tBy\xc2\xf0\x19u\x12\'\n\x04\x39\x36\x30\x36\n\x05\x31\x30\x30\x39\x30\x12\x18NCBI Taxonomy Identifier\x12\x1b\n\x05human\n\x05mouse\x12\x0b\x43ommon Name\x12-\n\x0cHomo sapiens\n\x0cMus musculus\x12\x0fScientific NameR\x06taxons\x12\'\n\x0fspecified_limit\x18\x02 \x01(\x08R\x0especifiedLimit\x12\x44\n\x0brank_limits\x18\x03 \x03(\x0e\x32#.ncbi.datasets.v2.Organism.RankTypeR\nrankLimits\"\xef\x04\n\x1fTaxonomyFilteredSubtreeResponse\x12\x1d\n\nroot_nodes\x18\x01 \x03(\rR\trootNodes\x12R\n\x05\x65\x64ges\x18\x02 \x03(\x0b\x32<.ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.EdgesEntryR\x05\x65\x64ges\x12=\n\x08warnings\x18\x03 \x03(\x0b\x32!.ncbi.datasets.v2.reports.WarningR\x08warnings\x12\x37\n\x06\x65rrors\x18\x04 \x03(\x0b\x32\x1f.ncbi.datasets.v2.reports.ErrorR\x06\x65rrors\x1a\xee\x01\n\x04\x45\x64ge\x12)\n\x10visible_children\x18\x01 \x03(\rR\x0fvisibleChildren\x12k\n\x0f\x63hildren_status\x18\x02 \x01(\x0e\x32\x42.ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.Edge.ChildStatusR\x0e\x63hildrenStatus\"N\n\x0b\x43hildStatus\x12\x0f\n\x0bUNSPECIFIED\x10\x00\x12\x15\n\x11HAS_MORE_CHILDREN\x10\x01\x12\x17\n\x13NO_VISIBLE_CHILDREN\x10\x02\x1ap\n\nEdgesEntry\x12\x10\n\x03key\x18\x01 \x01(\rR\x03key\x12L\n\x05value\x18\x02 \x01(\x0b\x32\x36.ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.EdgeR\x05value:\x02\x38\x01\"\xe9\x01\n\x0e\x44\x61tasetRequest\x12G\n\tgenome_v2\x18\x04 \x01(\x0b\x32(.ncbi.datasets.v2.AssemblyDatasetRequestH\x00R\x08genomeV2\x12?\n\x07gene_v2\x18\x05 \x01(\x0b\x32$.ncbi.datasets.v2.GeneDatasetRequestH\x00R\x06geneV2\x12\x42\n\x08virus_v2\x18\x06 \x01(\x0b\x32%.ncbi.datasets.v2.VirusDatasetRequestH\x00R\x07virusV2B\t\n\x07request\"\x8f\x01\n\x18GeneCountsByTaxonRequest\x12s\n\x05taxon\x18\x01 \x01(\tB]\xc2\xf0\x19Y\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy Identifier\x12\x14\n\x05human\x12\x0b\x43ommon Name\x12\x1f\n\x0cHomo sapiens\x12\x0fScientific NameR\x05taxon\"\xb2\x01\n\x16GeneCountsByTaxonReply\x12Q\n\x06report\x18\x01 \x03(\x0b\x32\x39.ncbi.datasets.v2.GeneCountsByTaxonReply.GeneTypeAndCountR\x06report\x1a\x45\n\x10GeneTypeAndCount\x12\x1b\n\tgene_type\x18\x01 \x01(\tR\x08geneType\x12\x14\n\x05\x63ount\x18\x02 \x01(\rR\x05\x63ount\"\xc1\x06\n\x17GenomeAnnotationRequest\x12\x1c\n\taccession\x18\x01 \x01(\tR\taccession\x12%\n\x0e\x61nnotation_ids\x18\r \x03(\tR\rannotationIds\x12\x18\n\x07symbols\x18\x02 \x03(\tR\x07symbols\x12\x1c\n\tlocations\x18\x0f \x03(\tR\tlocations\x12\x1d\n\ngene_types\x18\x0e \x03(\tR\tgeneTypes\x12\x1f\n\x0bsearch_text\x18\x05 \x03(\tR\nsearchText\x12/\n\x04sort\x18\x06 \x03(\x0b\x32\x1b.ncbi.datasets.v2.SortFieldR\x04sort\x12p\n\x17include_annotation_type\x18\x04 \x03(\x0e\x32\x38.ncbi.datasets.v2.GenomeAnnotationRequest.AnnotationTypeR\x15includeAnnotationType\x12%\n\tpage_size\x18\x08 \x01(\x05\x42\x08\xc2\xf0\x19\x04\n\x02\x32\x30R\x08pageSize\x12!\n\x0ctable_fields\x18\n \x03(\tR\x0btableFields\x12h\n\x0ctable_format\x18\x0b \x01(\x0e\x32\x45.ncbi.datasets.v2.GenomeAnnotationRequest.GenomeAnnotationTableFormatR\x0btableFormat\x12\\\n\x16include_tabular_header\x18\x0c \x01(\x0e\x32&.ncbi.datasets.v2.IncludeTabularHeaderR\x14includeTabularHeader\x12\x1d\n\npage_token\x18\t \x01(\tR\tpageToken\"N\n\x0e\x41nnotationType\x12\x0b\n\x07\x44\x45\x46\x41ULT\x10\x00\x12\x10\n\x0cGENOME_FASTA\x10\x01\x12\r\n\tRNA_FASTA\x10\x02\x12\x0e\n\nPROT_FASTA\x10\x03\"E\n\x1bGenomeAnnotationTableFormat\x12\x0c\n\x08NO_TABLE\x10\x00\x12\x0b\n\x07SUMMARY\x10\x01\x12\x0b\n\x07PRODUCT\x10\x02\"\xa7\x01\n!GenomeAnnotationTableSummaryReply\x12\x1c\n\taccession\x18\x01 \x01(\tR\taccession\x12 \n\x0b\x63hromosomes\x18\x02 \x03(\tR\x0b\x63hromosomes\x12\x1d\n\ngene_types\x18\x03 \x03(\tR\tgeneTypes\x12#\n\rempty_columns\x18\x04 \x03(\tR\x0c\x65mptyColumns*\x87\x01\n\x14IncludeTabularHeader\x12*\n&INCLUDE_TABULAR_HEADER_FIRST_PAGE_ONLY\x10\x00\x12!\n\x1dINCLUDE_TABULAR_HEADER_ALWAYS\x10\x01\x12 \n\x1cINCLUDE_TABULAR_HEADER_NEVER\x10\x02*l\n\rSortDirection\x12\x1e\n\x1aSORT_DIRECTION_UNSPECIFIED\x10\x00\x12\x1c\n\x18SORT_DIRECTION_ASCENDING\x10\x01\x12\x1d\n\x19SORT_DIRECTION_DESCENDING\x10\x02*\xf9\x02\n\tCountType\x12\x1a\n\x16\x43OUNT_TYPE_UNSPECIFIED\x10\x00\x12\x17\n\x13\x43OUNT_TYPE_ASSEMBLY\x10\x01\x12\x13\n\x0f\x43OUNT_TYPE_GENE\x10\x02\x12\x13\n\x0f\x43OUNT_TYPE_tRNA\x10\x03\x12\x13\n\x0f\x43OUNT_TYPE_rRNA\x10\x04\x12\x14\n\x10\x43OUNT_TYPE_snRNA\x10\x05\x12\x14\n\x10\x43OUNT_TYPE_scRNA\x10\x06\x12\x15\n\x11\x43OUNT_TYPE_snoRNA\x10\x07\x12\x1d\n\x19\x43OUNT_TYPE_PROTEIN_CODING\x10\x08\x12\x15\n\x11\x43OUNT_TYPE_PSEUDO\x10\t\x12\x19\n\x15\x43OUNT_TYPE_TRANSPOSON\x10\n\x12\x16\n\x12\x43OUNT_TYPE_miscRNA\x10\x0b\x12\x14\n\x10\x43OUNT_TYPE_ncRNA\x10\x0c\x12 \n\x1c\x43OUNT_TYPE_BIOLOGICAL_REGION\x10\r\x12\x14\n\x10\x43OUNT_TYPE_OTHER\x10\x0e*\x99\x01\n\x05\x46\x61sta\x12\x15\n\x11\x46\x41STA_UNSPECIFIED\x10\x00\x12\x0e\n\nFASTA_GENE\x10\x01\x12\r\n\tFASTA_RNA\x10\x02\x12\x11\n\rFASTA_PROTEIN\x10\x03\x12\x14\n\x10\x46\x41STA_GENE_FLANK\x10\x04\x12\r\n\tFASTA_CDS\x10\x05\x12\x10\n\x0c\x46\x41STA_5P_UTR\x10\x06\x12\x10\n\x0c\x46\x41STA_3P_UTR\x10\x07*\xb8\x01\n\x08GeneType\x12\x0b\n\x07UNKNOWN\x10\x00\x12\x08\n\x04tRNA\x10\x01\x12\x08\n\x04rRNA\x10\x02\x12\t\n\x05snRNA\x10\x03\x12\t\n\x05scRNA\x10\x04\x12\n\n\x06snoRNA\x10\x05\x12\x12\n\x0ePROTEIN_CODING\x10\x06\x12\n\n\x06PSEUDO\x10\x07\x12\x0e\n\nTRANSPOSON\x10\x08\x12\x0b\n\x07miscRNA\x10\t\x12\t\n\x05ncRNA\x10\n\x12\x15\n\x11\x42IOLOGICAL_REGION\x10\x0b\x12\n\n\x05OTHER\x10\xff\x01*\xa1\x03\n\x0fVirusTableField\x12\x0f\n\x0bunspecified\x10\x00\x12\x18\n\x14nucleotide_accession\x10\x01\x12\x12\n\x0especies_tax_id\x10\x02\x12\x10\n\x0cspecies_name\x10\x03\x12\t\n\x05genus\x10\x04\x12\n\n\x06\x66\x61mily\x10\x05\x12\x15\n\x11nucleotide_length\x10\x07\x12\x10\n\x0cisolate_name\x10\x08\x12\x11\n\rsequence_type\x10\t\x12\x14\n\x10nuc_completeness\x10\n\x12\x10\n\x0cgeo_location\x10\x0b\x12\x0c\n\x08us_state\x10\x0c\x12\r\n\thost_name\x10\r\x12\x0f\n\x0bhost_tax_id\x10\x0e\x12\x13\n\x0f\x63ollection_date\x10\x0f\x12\x0e\n\nbioproject\x10\x10\x12\r\n\tbiosample\x10\x11\x12\x14\n\x10polyprotein_name\x10\x13\x12\x10\n\x0cprotein_name\x10\x14\x12\x15\n\x11protein_accession\x10\x15\x12\x13\n\x0fprotein_synonym\x10\x16\x12\x0c\n\x08\x63\x64s_span\x10\x17**\n\x0bTableFormat\x12\x07\n\x03tsv\x10\x00\x12\x07\n\x03\x63sv\x10\x01\x12\t\n\x05jsonl\x10\x02*N\n\x11ViralSequenceType\x12\n\n\x06GENOME\x10\x00\x12\x07\n\x03\x43\x44S\x10\x01\x12\x0b\n\x07PROTEIN\x10\x02\x12\x08\n\x04NONE\x10\x03\x12\r\n\tBIOSAMPLE\x10\x04*R\n\x16VirusDatasetReportType\x12\x12\n\x0e\x44\x41TASET_REPORT\x10\x00\x12\x0e\n\nANNOTATION\x10\x01\x12\x14\n\x10\x42IOSAMPLE_REPORT\x10\x02\x42\x15Z\x10ncbi/datasets/v2\xf8\x01\x01\x62\x06proto3'
  ,
  dependencies=[google_dot_protobuf_dot_timestamp__pb2.DESCRIPTOR,ncbi_dot_datasets_dot_options_dot_openapi__pb2.DESCRIPTOR,ncbi_dot_datasets_dot_options_dot_request__pb2.DESCRIPTOR,ncbi_dot_datasets_dot_v2_dot_reports_dot_assembly__revision__pb2.DESCRIPTOR,ncbi_dot_datasets_dot_v2_dot_reports_dot_assembly__sequence__info__pb2.DESCRIPTOR,ncbi_dot_datasets_dot_v2_dot_reports_dot_common__pb2.DESCRIPTOR,])

_INCLUDETABULARHEADER = _descriptor.EnumDescriptor(
  name='IncludeTabularHeader',
  full_name='ncbi.datasets.v2.IncludeTabularHeader',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='INCLUDE_TABULAR_HEADER_FIRST_PAGE_ONLY', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='INCLUDE_TABULAR_HEADER_ALWAYS', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='INCLUDE_TABULAR_HEADER_NEVER', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=28669,
  serialized_end=28804,
)
_sym_db.RegisterEnumDescriptor(_INCLUDETABULARHEADER)

IncludeTabularHeader = enum_type_wrapper.EnumTypeWrapper(_INCLUDETABULARHEADER)
_SORTDIRECTION = _descriptor.EnumDescriptor(
  name='SortDirection',
  full_name='ncbi.datasets.v2.SortDirection',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='SORT_DIRECTION_UNSPECIFIED', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SORT_DIRECTION_ASCENDING', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SORT_DIRECTION_DESCENDING', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=28806,
  serialized_end=28914,
)
_sym_db.RegisterEnumDescriptor(_SORTDIRECTION)

SortDirection = enum_type_wrapper.EnumTypeWrapper(_SORTDIRECTION)
_COUNTTYPE = _descriptor.EnumDescriptor(
  name='CountType',
  full_name='ncbi.datasets.v2.CountType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_UNSPECIFIED', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_ASSEMBLY', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_GENE', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_tRNA', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_rRNA', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_snRNA', index=5, number=5,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_scRNA', index=6, number=6,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_snoRNA', index=7, number=7,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_PROTEIN_CODING', index=8, number=8,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_PSEUDO', index=9, number=9,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_TRANSPOSON', index=10, number=10,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_miscRNA', index=11, number=11,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_ncRNA', index=12, number=12,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_BIOLOGICAL_REGION', index=13, number=13,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_OTHER', index=14, number=14,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=28917,
  serialized_end=29294,
)
_sym_db.RegisterEnumDescriptor(_COUNTTYPE)

CountType = enum_type_wrapper.EnumTypeWrapper(_COUNTTYPE)
_FASTA = _descriptor.EnumDescriptor(
  name='Fasta',
  full_name='ncbi.datasets.v2.Fasta',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='FASTA_UNSPECIFIED', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FASTA_GENE', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FASTA_RNA', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FASTA_PROTEIN', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FASTA_GENE_FLANK', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FASTA_CDS', index=5, number=5,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FASTA_5P_UTR', index=6, number=6,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FASTA_3P_UTR', index=7, number=7,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=29297,
  serialized_end=29450,
)
_sym_db.RegisterEnumDescriptor(_FASTA)

Fasta = enum_type_wrapper.EnumTypeWrapper(_FASTA)
_GENETYPE = _descriptor.EnumDescriptor(
  name='GeneType',
  full_name='ncbi.datasets.v2.GeneType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='UNKNOWN', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='tRNA', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='rRNA', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='snRNA', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='scRNA', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='snoRNA', index=5, number=5,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PROTEIN_CODING', index=6, number=6,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PSEUDO', index=7, number=7,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='TRANSPOSON', index=8, number=8,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='miscRNA', index=9, number=9,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='ncRNA', index=10, number=10,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='BIOLOGICAL_REGION', index=11, number=11,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='OTHER', index=12, number=255,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=29453,
  serialized_end=29637,
)
_sym_db.RegisterEnumDescriptor(_GENETYPE)

GeneType = enum_type_wrapper.EnumTypeWrapper(_GENETYPE)
_VIRUSTABLEFIELD = _descriptor.EnumDescriptor(
  name='VirusTableField',
  full_name='ncbi.datasets.v2.VirusTableField',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='unspecified', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='nucleotide_accession', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='species_tax_id', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='species_name', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='genus', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='family', index=5, number=5,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='nucleotide_length', index=6, number=7,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='isolate_name', index=7, number=8,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='sequence_type', index=8, number=9,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='nuc_completeness', index=9, number=10,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='geo_location', index=10, number=11,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='us_state', index=11, number=12,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='host_name', index=12, number=13,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='host_tax_id', index=13, number=14,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='collection_date', index=14, number=15,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='bioproject', index=15, number=16,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='biosample', index=16, number=17,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='polyprotein_name', index=17, number=19,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='protein_name', index=18, number=20,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='protein_accession', index=19, number=21,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='protein_synonym', index=20, number=22,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='cds_span', index=21, number=23,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=29640,
  serialized_end=30057,
)
_sym_db.RegisterEnumDescriptor(_VIRUSTABLEFIELD)

VirusTableField = enum_type_wrapper.EnumTypeWrapper(_VIRUSTABLEFIELD)
_TABLEFORMAT = _descriptor.EnumDescriptor(
  name='TableFormat',
  full_name='ncbi.datasets.v2.TableFormat',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='tsv', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='csv', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='jsonl', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=30059,
  serialized_end=30101,
)
_sym_db.RegisterEnumDescriptor(_TABLEFORMAT)

TableFormat = enum_type_wrapper.EnumTypeWrapper(_TABLEFORMAT)
_VIRALSEQUENCETYPE = _descriptor.EnumDescriptor(
  name='ViralSequenceType',
  full_name='ncbi.datasets.v2.ViralSequenceType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='GENOME', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='CDS', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PROTEIN', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='NONE', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='BIOSAMPLE', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=30103,
  serialized_end=30181,
)
_sym_db.RegisterEnumDescriptor(_VIRALSEQUENCETYPE)

ViralSequenceType = enum_type_wrapper.EnumTypeWrapper(_VIRALSEQUENCETYPE)
_VIRUSDATASETREPORTTYPE = _descriptor.EnumDescriptor(
  name='VirusDatasetReportType',
  full_name='ncbi.datasets.v2.VirusDatasetReportType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='DATASET_REPORT', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='ANNOTATION', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='BIOSAMPLE_REPORT', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=30183,
  serialized_end=30265,
)
_sym_db.RegisterEnumDescriptor(_VIRUSDATASETREPORTTYPE)

VirusDatasetReportType = enum_type_wrapper.EnumTypeWrapper(_VIRUSDATASETREPORTTYPE)
INCLUDE_TABULAR_HEADER_FIRST_PAGE_ONLY = 0
INCLUDE_TABULAR_HEADER_ALWAYS = 1
INCLUDE_TABULAR_HEADER_NEVER = 2
SORT_DIRECTION_UNSPECIFIED = 0
SORT_DIRECTION_ASCENDING = 1
SORT_DIRECTION_DESCENDING = 2
COUNT_TYPE_UNSPECIFIED = 0
COUNT_TYPE_ASSEMBLY = 1
COUNT_TYPE_GENE = 2
COUNT_TYPE_tRNA = 3
COUNT_TYPE_rRNA = 4
COUNT_TYPE_snRNA = 5
COUNT_TYPE_scRNA = 6
COUNT_TYPE_snoRNA = 7
COUNT_TYPE_PROTEIN_CODING = 8
COUNT_TYPE_PSEUDO = 9
COUNT_TYPE_TRANSPOSON = 10
COUNT_TYPE_miscRNA = 11
COUNT_TYPE_ncRNA = 12
COUNT_TYPE_BIOLOGICAL_REGION = 13
COUNT_TYPE_OTHER = 14
FASTA_UNSPECIFIED = 0
FASTA_GENE = 1
FASTA_RNA = 2
FASTA_PROTEIN = 3
FASTA_GENE_FLANK = 4
FASTA_CDS = 5
FASTA_5P_UTR = 6
FASTA_3P_UTR = 7
UNKNOWN = 0
tRNA = 1
rRNA = 2
snRNA = 3
scRNA = 4
snoRNA = 5
PROTEIN_CODING = 6
PSEUDO = 7
TRANSPOSON = 8
miscRNA = 9
ncRNA = 10
BIOLOGICAL_REGION = 11
OTHER = 255
unspecified = 0
nucleotide_accession = 1
species_tax_id = 2
species_name = 3
genus = 4
family = 5
nucleotide_length = 7
isolate_name = 8
sequence_type = 9
nuc_completeness = 10
geo_location = 11
us_state = 12
host_name = 13
host_tax_id = 14
collection_date = 15
bioproject = 16
biosample = 17
polyprotein_name = 19
protein_name = 20
protein_accession = 21
protein_synonym = 22
cds_span = 23
tsv = 0
csv = 1
jsonl = 2
GENOME = 0
CDS = 1
PROTEIN = 2
NONE = 3
BIOSAMPLE = 4
DATASET_REPORT = 0
ANNOTATION = 1
BIOSAMPLE_REPORT = 2


_ASSEMBLYDATASETREPORTSREQUEST_CONTENTTYPE = _descriptor.EnumDescriptor(
  name='ContentType',
  full_name='ncbi.datasets.v2.AssemblyDatasetReportsRequest.ContentType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='COMPLETE', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='ASSM_ACC', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PAIRED_ACC', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=3350,
  serialized_end=3407,
)
_sym_db.RegisterEnumDescriptor(_ASSEMBLYDATASETREPORTSREQUEST_CONTENTTYPE)

_ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYSOURCE = _descriptor.EnumDescriptor(
  name='AssemblySource',
  full_name='ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.AssemblySource',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='all', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='refseq', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='genbank', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=5387,
  serialized_end=5437,
)
_sym_db.RegisterEnumDescriptor(_ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYSOURCE)

_ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYVERSION = _descriptor.EnumDescriptor(
  name='AssemblyVersion',
  full_name='ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.AssemblyVersion',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='current', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='all_assemblies', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=5439,
  serialized_end=5489,
)
_sym_db.RegisterEnumDescriptor(_ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYVERSION)

_ASSEMBLYDATASETDESCRIPTORSFILTER_METAGENOMEDERIVEDFILTER = _descriptor.EnumDescriptor(
  name='MetagenomeDerivedFilter',
  full_name='ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.MetagenomeDerivedFilter',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='METAGENOME_DERIVED_UNSET', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='metagenome_derived_only', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='metagenome_derived_exclude', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=5491,
  serialized_end=5607,
)
_sym_db.RegisterEnumDescriptor(_ASSEMBLYDATASETDESCRIPTORSFILTER_METAGENOMEDERIVEDFILTER)

_ANNOTATIONFORASSEMBLY_TYPE = _descriptor.EnumDescriptor(
  name='Type',
  full_name='ncbi.datasets.v2.AnnotationForAssembly.Type',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='DEFAULT', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GENOME_GFF', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GENOME_GBFF', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GENOME_GB', index=3, number=2,
      serialized_options=b'\010\001',
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='RNA_FASTA', index=4, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PROT_FASTA', index=5, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GENOME_GTF', index=6, number=5,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='CDS_FASTA', index=7, number=6,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='ASM_STRUCT', index=8, number=7,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GENOME_FASTA', index=9, number=8,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SEQUENCE_REPORT', index=10, number=9,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=b'\020\001',
  serialized_start=7391,
  serialized_end=7583,
)
_sym_db.RegisterEnumDescriptor(_ANNOTATIONFORASSEMBLY_TYPE)

_ASSEMBLYDATASETREQUEST_RESOLUTION = _descriptor.EnumDescriptor(
  name='Resolution',
  full_name='ncbi.datasets.v2.AssemblyDatasetRequest.Resolution',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='FULLY_HYDRATED', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='DATA_REPORT_ONLY', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=8428,
  serialized_end=8482,
)
_sym_db.RegisterEnumDescriptor(_ASSEMBLYDATASETREQUEST_RESOLUTION)

_ORGANISM_RANKTYPE = _descriptor.EnumDescriptor(
  name='RankType',
  full_name='ncbi.datasets.v2.Organism.RankType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='NO_RANK', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUPERKINGDOM', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='KINGDOM', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBKINGDOM', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUPERPHYLUM', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBPHYLUM', index=5, number=5,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PHYLUM', index=6, number=6,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='CLADE', index=7, number=31,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUPERCLASS', index=8, number=7,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='CLASS', index=9, number=8,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBCLASS', index=10, number=9,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='INFRACLASS', index=11, number=10,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COHORT', index=12, number=11,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBCOHORT', index=13, number=12,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUPERORDER', index=14, number=13,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='ORDER', index=15, number=14,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBORDER', index=16, number=15,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='INFRAORDER', index=17, number=16,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PARVORDER', index=18, number=17,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUPERFAMILY', index=19, number=18,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FAMILY', index=20, number=19,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBFAMILY', index=21, number=20,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GENUS', index=22, number=21,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBGENUS', index=23, number=22,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SPECIES_GROUP', index=24, number=23,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SPECIES_SUBGROUP', index=25, number=24,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SPECIES', index=26, number=25,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBSPECIES', index=27, number=26,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='TRIBE', index=28, number=27,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBTRIBE', index=29, number=28,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FORMA', index=30, number=29,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='VARIETAS', index=31, number=30,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='STRAIN', index=32, number=320,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SECTION', index=33, number=330,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBSECTION', index=34, number=340,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PATHOGROUP', index=35, number=350,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBVARIETY', index=36, number=360,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GENOTYPE', index=37, number=370,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SEROTYPE', index=38, number=380,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='ISOLATE', index=39, number=390,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='MORPH', index=40, number=400,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SERIES', index=41, number=410,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FORMA_SPECIALIS', index=42, number=420,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SEROGROUP', index=43, number=430,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='BIOTYPE', index=44, number=440,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=9708,
  serialized_end=10377,
)
_sym_db.RegisterEnumDescriptor(_ORGANISM_RANKTYPE)

_ORGANISMQUERYREQUEST_TAXRANKFILTER = _descriptor.EnumDescriptor(
  name='TaxRankFilter',
  full_name='ncbi.datasets.v2.OrganismQueryRequest.TaxRankFilter',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='species', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='higher_taxon', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=11021,
  serialized_end=11067,
)
_sym_db.RegisterEnumDescriptor(_ORGANISMQUERYREQUEST_TAXRANKFILTER)

_ORGANISMQUERYREQUEST_TAXONRESOURCEFILTER = _descriptor.EnumDescriptor(
  name='TaxonResourceFilter',
  full_name='ncbi.datasets.v2.OrganismQueryRequest.TaxonResourceFilter',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='TAXON_RESOURCE_FILTER_ALL', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='TAXON_RESOURCE_FILTER_GENOME', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='TAXON_RESOURCE_FILTER_GENE', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=11069,
  serialized_end=11187,
)
_sym_db.RegisterEnumDescriptor(_ORGANISMQUERYREQUEST_TAXONRESOURCEFILTER)

_ORTHOLOGREQUEST_CONTENTTYPE = _descriptor.EnumDescriptor(
  name='ContentType',
  full_name='ncbi.datasets.v2.OrthologRequest.ContentType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='COMPLETE', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='IDS_ONLY', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=11835,
  serialized_end=11876,
)
_sym_db.RegisterEnumDescriptor(_ORTHOLOGREQUEST_CONTENTTYPE)

_GENEDATASETREQUEST_CONTENTTYPE = _descriptor.EnumDescriptor(
  name='ContentType',
  full_name='ncbi.datasets.v2.GeneDatasetRequest.ContentType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='COMPLETE', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='IDS_ONLY', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=11835,
  serialized_end=11876,
)
_sym_db.RegisterEnumDescriptor(_GENEDATASETREQUEST_CONTENTTYPE)

_GENEDATASETREQUEST_GENEDATASETREPORTTYPE = _descriptor.EnumDescriptor(
  name='GeneDatasetReportType',
  full_name='ncbi.datasets.v2.GeneDatasetRequest.GeneDatasetReportType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='DATASET_REPORT', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PRODUCT_REPORT', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=12709,
  serialized_end=12772,
)
_sym_db.RegisterEnumDescriptor(_GENEDATASETREQUEST_GENEDATASETREPORTTYPE)

_GENEDATASETREPORTSREQUEST_CONTENTTYPE = _descriptor.EnumDescriptor(
  name='ContentType',
  full_name='ncbi.datasets.v2.GeneDatasetReportsRequest.ContentType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='COMPLETE', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='IDS_ONLY', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PRODUCT', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNTS_ONLY', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=13954,
  serialized_end=14025,
)
_sym_db.RegisterEnumDescriptor(_GENEDATASETREPORTSREQUEST_CONTENTTYPE)

_GENELINKSREPLY_GENELINKTYPE = _descriptor.EnumDescriptor(
  name='GeneLinkType',
  full_name='ncbi.datasets.v2.GeneLinksReply.GeneLinkType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='DEFAULT', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GENE_LINK', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GDV_LINK', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='ORTHOLOG_LINK', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=14493,
  serialized_end=14568,
)
_sym_db.RegisterEnumDescriptor(_GENELINKSREPLY_GENELINKTYPE)

_ASSEMBLYLINKSREPLY_ASSEMBLYLINKTYPE = _descriptor.EnumDescriptor(
  name='AssemblyLinkType',
  full_name='ncbi.datasets.v2.AssemblyLinksReply.AssemblyLinkType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='DEFAULT', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GDV_LINK', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FTP_LINK', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='ASSEMBLY_PUBMED', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='BLAST_LINK', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=15628,
  serialized_end=15724,
)
_sym_db.RegisterEnumDescriptor(_ASSEMBLYLINKSREPLY_ASSEMBLYLINKTYPE)

_VIRUSDATAREPORTREQUEST_CONTENTTYPE = _descriptor.EnumDescriptor(
  name='ContentType',
  full_name='ncbi.datasets.v2.VirusDataReportRequest.ContentType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='COMPLETE', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='ACCESSIONS_ONLY', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=18674,
  serialized_end=18722,
)
_sym_db.RegisterEnumDescriptor(_VIRUSDATAREPORTREQUEST_CONTENTTYPE)

_MICROBIGGEDATASETREQUEST_FILETYPE = _descriptor.EnumDescriptor(
  name='FileType',
  full_name='ncbi.datasets.v2.MicroBiggeDatasetRequest.FileType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='FILE_TYPE_UNSPECIFIED', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='element_fasta', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='element_flank_fasta', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='contig_fasta', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='protein_fasta', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=23813,
  serialized_end=23931,
)
_sym_db.RegisterEnumDescriptor(_MICROBIGGEDATASETREQUEST_FILETYPE)

_TAXONOMYMETADATAREQUEST_CONTENTTYPE = _descriptor.EnumDescriptor(
  name='ContentType',
  full_name='ncbi.datasets.v2.TaxonomyMetadataRequest.ContentType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='COMPLETE', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='TAXIDS', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=25724,
  serialized_end=25763,
)
_sym_db.RegisterEnumDescriptor(_TAXONOMYMETADATAREQUEST_CONTENTTYPE)

_TAXONOMYFILTEREDSUBTREERESPONSE_EDGE_CHILDSTATUS = _descriptor.EnumDescriptor(
  name='ChildStatus',
  full_name='ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.Edge.ChildStatus',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='UNSPECIFIED', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='HAS_MORE_CHILDREN', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='NO_VISIBLE_CHILDREN', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=26905,
  serialized_end=26983,
)
_sym_db.RegisterEnumDescriptor(_TAXONOMYFILTEREDSUBTREERESPONSE_EDGE_CHILDSTATUS)

_GENOMEANNOTATIONREQUEST_ANNOTATIONTYPE = _descriptor.EnumDescriptor(
  name='AnnotationType',
  full_name='ncbi.datasets.v2.GenomeAnnotationRequest.AnnotationType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='DEFAULT', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GENOME_FASTA', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='RNA_FASTA', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PROT_FASTA', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=28347,
  serialized_end=28425,
)
_sym_db.RegisterEnumDescriptor(_GENOMEANNOTATIONREQUEST_ANNOTATIONTYPE)

_GENOMEANNOTATIONREQUEST_GENOMEANNOTATIONTABLEFORMAT = _descriptor.EnumDescriptor(
  name='GenomeAnnotationTableFormat',
  full_name='ncbi.datasets.v2.GenomeAnnotationRequest.GenomeAnnotationTableFormat',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='NO_TABLE', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUMMARY', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PRODUCT', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=28427,
  serialized_end=28496,
)
_sym_db.RegisterEnumDescriptor(_GENOMEANNOTATIONREQUEST_GENOMEANNOTATIONTABLEFORMAT)


_METHODPAYLOADREQUEST = _descriptor.Descriptor(
  name='MethodPayloadRequest',
  full_name='ncbi.datasets.v2.MethodPayloadRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='method', full_name='ncbi.datasets.v2.MethodPayloadRequest.method', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='method', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='payload', full_name='ncbi.datasets.v2.MethodPayloadRequest.payload', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='payload', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=304,
  serialized_end=376,
)


_ASSEMBLYSEQUENCEREPORTSREQUEST = _descriptor.Descriptor(
  name='AssemblySequenceReportsRequest',
  full_name='ncbi.datasets.v2.AssemblySequenceReportsRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accession', full_name='ncbi.datasets.v2.AssemblySequenceReportsRequest.accession', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\0316\022\031\n\020GCF_000001635.27\022\005Mouse\022\031\n\020GCF_000001405.40\022\005Human', json_name='accession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='chromosomes', full_name='ncbi.datasets.v2.AssemblySequenceReportsRequest.chromosomes', index=1,
      number=2, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031M\022,\n\0011\n\0012\n\0013\n\001X\n\001Y\n\002MT\022\027Selected chromosomes #1\022\035\n\001X\n\001Y\022\025Human sex chromosomes', json_name='chromosomes', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='role_filters', full_name='ncbi.datasets.v2.AssemblySequenceReportsRequest.role_filters', index=2,
      number=4, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\206\001\022T\n\022assembled-molecule\n\024unlocalized-scaffold\n\021unplaced-scaffold\022\025All types of sequence\022.\n\022assembled-molecule\022\030Only assembled molecules', json_name='roleFilters', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_fields', full_name='ncbi.datasets.v2.AssemblySequenceReportsRequest.table_fields', index=3,
      number=3, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031[\022;\n\taccession\n\010chr-name\022$Genome Accession and Chromosome Name\022\034\n\017ucsc-style-name\022\tUCSC Name', json_name='tableFields', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='count_assembly_unplaced', full_name='ncbi.datasets.v2.AssemblySequenceReportsRequest.count_assembly_unplaced', index=4,
      number=5, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\007\n\005false', json_name='countAssemblyUnplaced', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_size', full_name='ncbi.datasets.v2.AssemblySequenceReportsRequest.page_size', index=5,
      number=8, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\004\n\00220', json_name='pageSize', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_token', full_name='ncbi.datasets.v2.AssemblySequenceReportsRequest.page_token', index=6,
      number=9, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='pageToken', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='include_tabular_header', full_name='ncbi.datasets.v2.AssemblySequenceReportsRequest.include_tabular_header', index=7,
      number=12, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='includeTabularHeader', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_format', full_name='ncbi.datasets.v2.AssemblySequenceReportsRequest.table_format', index=8,
      number=13, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='tableFormat', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=379,
  serialized_end=1197,
)


_SEQUENCEREPORTPAGE = _descriptor.Descriptor(
  name='SequenceReportPage',
  full_name='ncbi.datasets.v2.SequenceReportPage',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='reports', full_name='ncbi.datasets.v2.SequenceReportPage.reports', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='reports', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='total_count', full_name='ncbi.datasets.v2.SequenceReportPage.total_count', index=1,
      number=3, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='totalCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='next_page_token', full_name='ncbi.datasets.v2.SequenceReportPage.next_page_token', index=2,
      number=4, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='nextPageToken', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='_report_type', full_name='ncbi.datasets.v2.SequenceReportPage._report_type', index=3,
      number=500, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='ReportType', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='_report_fields', full_name='ncbi.datasets.v2.SequenceReportPage._report_fields', index=4,
      number=501, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='ReportFields', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='_first_page', full_name='ncbi.datasets.v2.SequenceReportPage._first_page', index=5,
      number=502, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='FirstPage', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='_report_format', full_name='ncbi.datasets.v2.SequenceReportPage._report_format', index=6,
      number=503, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='ReportFormat', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=1200,
  serialized_end=1505,
)


_SORTFIELD = _descriptor.Descriptor(
  name='SortField',
  full_name='ncbi.datasets.v2.SortField',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='field', full_name='ncbi.datasets.v2.SortField.field', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\303\005\022\013\n\taccession\022\027\n\025versionless-accession\022\r\n\013search-text\022\010\n\006tax-id\022\023\n\021tax-ordinal-range\022\017\n\rorganism-name\022\013\n\tsubmitter\022\025\n\023assembly-level-enum\022\021\n\017assembly-status\022\016\n\014release-date\022\017\n\ris-ref-genome\022\017\n\ris-rep-genome\022\013\n\tis-refseq\022\014\n\nis-primary\022\014\n\nis-current\022\034\n\032paired-assembly-is-current\022!\n\037paired-assembly-annotation-name\022\020\n\016has-annotation\022\r\n\013is-atypical\022\026\n\024bioproject-accession\022\027\n\025total-sequence-length\022\021\n\017pair-is-current\022\035\n\033pair-has-genbank-annotation\022\034\n\032pair-has-refseq-annotation\022\025\n\023pair-has-annotation\022\014\n\ncontig-n50\022\016\n\014scaffold-n50\022\020\n\016busco-complete\022\023\n\021genbank-accession\022\022\n\020refseq-accession\022\021\n\017annotation-name\022\017\n\rwgs-accession\022\021\n\017sequencing-tech\022\025\n\023biosample-accession\022\014\n\ngene-count\022\017\n\rprotein-count\022\n\n\010modifier', json_name='field', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='direction', full_name='ncbi.datasets.v2.SortField.direction', index=1,
      number=2, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='direction', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=1508,
  serialized_end=2320,
)


_ASSEMBLYDATASETREPORTSREQUEST = _descriptor.Descriptor(
  name='AssemblyDatasetReportsRequest',
  full_name='ncbi.datasets.v2.AssemblyDatasetReportsRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='taxons', full_name='ncbi.datasets.v2.AssemblyDatasetReportsRequest.taxons', index=0,
      number=6, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031p\022 \n\0049606\022\030NCBI Taxonomy Identifier\022+\n\nchimpanzee\n\017eastern gorilla\022\014Common Names\022\037\n\014Homo sapiens\022\017Scientific Name', json_name='taxons', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='bioprojects', full_name='ncbi.datasets.v2.AssemblyDatasetReportsRequest.bioprojects', index=1,
      number=8, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031O\022)\n\013PRJNA489243\022\032Vertebrate Genomes Project\022\"\n\nPRJNA31257\022\024Human Genome Project', json_name='bioprojects', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v2.AssemblyDatasetReportsRequest.accessions', index=2,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031R\0225\n\020GCF_000001405.40\n\020GCF_000001635.27\022\017Human and Mouse\022\031\n\020GCF_000001405.40\022\005Human', json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='filters', full_name='ncbi.datasets.v2.AssemblyDatasetReportsRequest.filters', index=3,
      number=7, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='filters', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='chromosomes', full_name='ncbi.datasets.v2.AssemblyDatasetReportsRequest.chromosomes', index=4,
      number=2, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='chromosomes', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_fields', full_name='ncbi.datasets.v2.AssemblyDatasetReportsRequest.table_fields', index=5,
      number=3, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031f\022>\n\022assminfo-accession\n\rassminfo-name\022\031Genome Accession and Name\022$\n\022assminfo-accession\022\016Just accession', json_name='tableFields', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='returned_content', full_name='ncbi.datasets.v2.AssemblyDatasetReportsRequest.returned_content', index=6,
      number=10, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='returnedContent', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_size', full_name='ncbi.datasets.v2.AssemblyDatasetReportsRequest.page_size', index=7,
      number=4, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\004\n\00220', json_name='pageSize', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_token', full_name='ncbi.datasets.v2.AssemblyDatasetReportsRequest.page_token', index=8,
      number=5, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='pageToken', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='sort', full_name='ncbi.datasets.v2.AssemblyDatasetReportsRequest.sort', index=9,
      number=11, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='sort', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='include_tabular_header', full_name='ncbi.datasets.v2.AssemblyDatasetReportsRequest.include_tabular_header', index=10,
      number=12, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='includeTabularHeader', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_format', full_name='ncbi.datasets.v2.AssemblyDatasetReportsRequest.table_format', index=11,
      number=13, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='tableFormat', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _ASSEMBLYDATASETREPORTSREQUEST_CONTENTTYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=2323,
  serialized_end=3407,
)


_ASSEMBLYDATASETDESCRIPTORSFILTER = _descriptor.Descriptor(
  name='AssemblyDatasetDescriptorsFilter',
  full_name='ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='reference_only', full_name='ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.reference_only', index=0,
      number=1, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031r\n\005false\0223\n\004true\022+Limit to reference & representative genomes\0224\n\005false\022+Include reference and non-reference genomes', json_name='referenceOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_source', full_name='ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.assembly_source', index=1,
      number=3, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031a\n\003all\022!\n\006refseq\022\027Limit to RefSeq genomes\022#\n\007genbank\022\030Limit to GenBank genomes\022\022\n\003all\022\013All genomes', json_name='assemblySource', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='has_annotation', full_name='ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.has_annotation', index=2,
      number=7, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031N\n\005false\022%\n\004true\022\035Limit to annotated assemblies\022\036\n\005false\022\025Return all assemblies', json_name='hasAnnotation', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='exclude_paired_reports', full_name='ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.exclude_paired_reports', index=3,
      number=9, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\\\n\005false\022\036\n\005false\022\025Return all assemblies\0223\n\004true\022+Only return single record for GCA/GCF pairs', json_name='excludePairedReports', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='exclude_atypical', full_name='ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.exclude_atypical', index=4,
      number=11, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\226\001\n\005false\022F\n\005false\022=Return all selected genomes, including ones that are atypical\022E\n\004true\022=Only return genomes that have not been classified as atypical', json_name='excludeAtypical', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_version', full_name='ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.assembly_version', index=5,
      number=10, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\200\001\n\007current\022)\n\007current\022\036Only return current assemblies\022J\n\016all_assemblies\0228Return all assemblies, including suppressed and replaced', json_name='assemblyVersion', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_level', full_name='ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.assembly_level', index=6,
      number=4, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031}\022=\n\nchromosome\n\017complete_genome\022\036Chromosome and complete genome\022<\n\010scaffold\n\nchromosome\n\017complete_genome\022\023Scaffold and higher', json_name='assemblyLevel', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='first_release_date', full_name='ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.first_release_date', index=7,
      number=5, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\034\022\032\n\n01/10/2015\022\014Jan 10, 2015', json_name='firstReleaseDate', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='last_release_date', full_name='ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.last_release_date', index=8,
      number=6, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\034\022\032\n\n01/10/2021\022\014Jan 10, 2021', json_name='lastReleaseDate', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='search_text', full_name='ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.search_text', index=9,
      number=8, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031d\022:\n\033Genome Reference Consortium\022\033Genome Reference Consortium\022\020\n\006GRCh38\022\006GRCh38\022\024\n\010C57BL/6J\022\010C57BL/6J', json_name='searchText', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='is_metagenome_derived', full_name='ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.is_metagenome_derived', index=10,
      number=12, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\213\001\022D\n\027metagenome_derived_only\022)Return only metagenome derived assemblies\022C\n\032metagenome_derived_exclude\022%Exclude metagenome derived assemblies', json_name='isMetagenomeDerived', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYSOURCE,
    _ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYVERSION,
    _ASSEMBLYDATASETDESCRIPTORSFILTER_METAGENOMEDERIVEDFILTER,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=3410,
  serialized_end=5607,
)


_ACCESSIONS = _descriptor.Descriptor(
  name='Accessions',
  full_name='ncbi.datasets.v2.Accessions',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v2.Accessions.accessions', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031R\0225\n\020GCF_000001405.40\n\020GCF_000001635.27\022\017Human and Mouse\022\031\n\020GCF_000001405.40\022\005Human', json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=5610,
  serialized_end=5742,
)


_ASSEMBLYREVISIONHISTORY = _descriptor.Descriptor(
  name='AssemblyRevisionHistory',
  full_name='ncbi.datasets.v2.AssemblyRevisionHistory',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='assembly_revisions', full_name='ncbi.datasets.v2.AssemblyRevisionHistory.assembly_revisions', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='assemblyRevisions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='total_count', full_name='ncbi.datasets.v2.AssemblyRevisionHistory.total_count', index=1,
      number=2, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='totalCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=5745,
  serialized_end=5894,
)


_ASSEMBLYREVISIONHISTORYREQUEST = _descriptor.Descriptor(
  name='AssemblyRevisionHistoryRequest',
  full_name='ncbi.datasets.v2.AssemblyRevisionHistoryRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accession', full_name='ncbi.datasets.v2.AssemblyRevisionHistoryRequest.accession', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\033\022\031\n\020GCF_000001405.40\022\005Human', json_name='accession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=5896,
  serialized_end=5991,
)


_ASSEMBLYCHECKMHISTOGRAMREQUEST = _descriptor.Descriptor(
  name='AssemblyCheckMHistogramRequest',
  full_name='ncbi.datasets.v2.AssemblyCheckMHistogramRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='species_taxon', full_name='ncbi.datasets.v2.AssemblyCheckMHistogramRequest.species_taxon', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031!\022\037\n\006202956\022\025Acinetobacter towneri', json_name='speciesTaxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=5993,
  serialized_end=6101,
)


_ASSEMBLYCHECKMHISTOGRAMREPLY_HISTOGRAMINTERVAL = _descriptor.Descriptor(
  name='HistogramInterval',
  full_name='ncbi.datasets.v2.AssemblyCheckMHistogramReply.HistogramInterval',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='start_pos', full_name='ncbi.datasets.v2.AssemblyCheckMHistogramReply.HistogramInterval.start_pos', index=0,
      number=1, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='startPos', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='stop_pos', full_name='ncbi.datasets.v2.AssemblyCheckMHistogramReply.HistogramInterval.stop_pos', index=1,
      number=2, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='stopPos', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='count', full_name='ncbi.datasets.v2.AssemblyCheckMHistogramReply.HistogramInterval.count', index=2,
      number=3, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='count', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=6327,
  serialized_end=6424,
)

_ASSEMBLYCHECKMHISTOGRAMREPLY = _descriptor.Descriptor(
  name='AssemblyCheckMHistogramReply',
  full_name='ncbi.datasets.v2.AssemblyCheckMHistogramReply',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='species_taxid', full_name='ncbi.datasets.v2.AssemblyCheckMHistogramReply.species_taxid', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031!\022\037\n\006202956\022\025Acinetobacter towneri', json_name='speciesTaxid', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='histogram_intervals', full_name='ncbi.datasets.v2.AssemblyCheckMHistogramReply.histogram_intervals', index=1,
      number=2, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='histogramIntervals', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_ASSEMBLYCHECKMHISTOGRAMREPLY_HISTOGRAMINTERVAL, ],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=6104,
  serialized_end=6424,
)


_BIOPROJECT = _descriptor.Descriptor(
  name='BioProject',
  full_name='ncbi.datasets.v2.BioProject',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accession', full_name='ncbi.datasets.v2.BioProject.accession', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='accession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='title', full_name='ncbi.datasets.v2.BioProject.title', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='title', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='parent_accession', full_name='ncbi.datasets.v2.BioProject.parent_accession', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='parentAccession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='parent_accessions', full_name='ncbi.datasets.v2.BioProject.parent_accessions', index=3,
      number=4, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='parentAccessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=6427,
  serialized_end=6579,
)


_BIOPROJECTLINEAGE = _descriptor.Descriptor(
  name='BioProjectLineage',
  full_name='ncbi.datasets.v2.BioProjectLineage',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='bioprojects', full_name='ncbi.datasets.v2.BioProjectLineage.bioprojects', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='bioprojects', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=6581,
  serialized_end=6664,
)


_BUSCOSTAT = _descriptor.Descriptor(
  name='BuscoStat',
  full_name='ncbi.datasets.v2.BuscoStat',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='busco_lineage', full_name='ncbi.datasets.v2.BuscoStat.busco_lineage', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='buscoLineage', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='busco_ver', full_name='ncbi.datasets.v2.BuscoStat.busco_ver', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='buscoVer', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='complete', full_name='ncbi.datasets.v2.BuscoStat.complete', index=2,
      number=3, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='complete', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='single_copy', full_name='ncbi.datasets.v2.BuscoStat.single_copy', index=3,
      number=4, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='singleCopy', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='duplicated', full_name='ncbi.datasets.v2.BuscoStat.duplicated', index=4,
      number=5, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='duplicated', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='fragmented', full_name='ncbi.datasets.v2.BuscoStat.fragmented', index=5,
      number=6, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='fragmented', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='missing', full_name='ncbi.datasets.v2.BuscoStat.missing', index=6,
      number=7, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='missing', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='total_count', full_name='ncbi.datasets.v2.BuscoStat.total_count', index=7,
      number=8, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='totalCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=6667,
  serialized_end=6928,
)


_ANNOTATIONFORASSEMBLY_FILE = _descriptor.Descriptor(
  name='File',
  full_name='ncbi.datasets.v2.AnnotationForAssembly.File',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='type', full_name='ncbi.datasets.v2.AnnotationForAssembly.File.type', index=0,
      number=1, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='type', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='estimated_size', full_name='ncbi.datasets.v2.AnnotationForAssembly.File.estimated_size', index=1,
      number=2, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='estimatedSize', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=7277,
  serialized_end=7388,
)

_ANNOTATIONFORASSEMBLY = _descriptor.Descriptor(
  name='AnnotationForAssembly',
  full_name='ncbi.datasets.v2.AnnotationForAssembly',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='name', full_name='ncbi.datasets.v2.AnnotationForAssembly.name', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='name', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='source', full_name='ncbi.datasets.v2.AnnotationForAssembly.source', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='source', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='release_number', full_name='ncbi.datasets.v2.AnnotationForAssembly.release_number', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='releaseNumber', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='release_date', full_name='ncbi.datasets.v2.AnnotationForAssembly.release_date', index=3,
      number=4, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='releaseDate', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='report_url', full_name='ncbi.datasets.v2.AnnotationForAssembly.report_url', index=4,
      number=5, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='reportUrl', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='file', full_name='ncbi.datasets.v2.AnnotationForAssembly.file', index=5,
      number=6, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='file', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='stats', full_name='ncbi.datasets.v2.AnnotationForAssembly.stats', index=6,
      number=7, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='stats', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='busco', full_name='ncbi.datasets.v2.AnnotationForAssembly.busco', index=7,
      number=8, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='busco', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_ANNOTATIONFORASSEMBLY_FILE, ],
  enum_types=[
    _ANNOTATIONFORASSEMBLY_TYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=6931,
  serialized_end=7583,
)


_FEATURECOUNTS = _descriptor.Descriptor(
  name='FeatureCounts',
  full_name='ncbi.datasets.v2.FeatureCounts',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='gene_counts', full_name='ncbi.datasets.v2.FeatureCounts.gene_counts', index=0,
      number=1, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='geneCounts', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=7585,
  serialized_end=7663,
)


_GENECOUNTS = _descriptor.Descriptor(
  name='GeneCounts',
  full_name='ncbi.datasets.v2.GeneCounts',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='total', full_name='ncbi.datasets.v2.GeneCounts.total', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='total', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='protein_coding', full_name='ncbi.datasets.v2.GeneCounts.protein_coding', index=1,
      number=2, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='proteinCoding', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=7665,
  serialized_end=7738,
)


_ASSEMBLYDATASETREQUEST = _descriptor.Descriptor(
  name='AssemblyDatasetRequest',
  full_name='ncbi.datasets.v2.AssemblyDatasetRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v2.AssemblyDatasetRequest.accessions', index=0,
      number=10, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031R\0225\n\020GCF_000001405.40\n\020GCF_000001635.27\022\017Human and Mouse\022\031\n\020GCF_000001405.40\022\005Human', json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='chromosomes', full_name='ncbi.datasets.v2.AssemblyDatasetRequest.chromosomes', index=1,
      number=2, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031M\022,\n\0011\n\0012\n\0013\n\001X\n\001Y\n\002MT\022\027Selected chromosomes #1\022\035\n\001X\n\001Y\022\025Human sex chromosomes', json_name='chromosomes', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='include_annotation_type', full_name='ncbi.datasets.v2.AssemblyDatasetRequest.include_annotation_type', index=2,
      number=8, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\332\360\031\002\010\002\302\360\031\256\001\022M\n\014GENOME_FASTA\n\nPROT_FASTA\n\tRNA_FASTA\022&Genomic, Transcript & Protein Sequence\022]\n\nGENOME_GFF\n\013GENOME_GBFF\n\nGENOME_GTF\n\nPROT_FASTA\n\tRNA_FASTA\n\032CDS_FASTA, SEQUENCE_REPORT\022\003All', json_name='includeAnnotationType', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='hydrated', full_name='ncbi.datasets.v2.AssemblyDatasetRequest.hydrated', index=3,
      number=9, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='hydrated', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='include_tsv', full_name='ncbi.datasets.v2.AssemblyDatasetRequest.include_tsv', index=4,
      number=11, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\332\360\031\002\010\002\302\360\031\007\n\005false', json_name='includeTsv', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _ASSEMBLYDATASETREQUEST_RESOLUTION,
  ],
  serialized_options=b'\302\360\031 \n\036accessions:\n- GCF_000001405.40',
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=7741,
  serialized_end=8520,
)


_ASSEMBLYDATASETAVAILABILITY = _descriptor.Descriptor(
  name='AssemblyDatasetAvailability',
  full_name='ncbi.datasets.v2.AssemblyDatasetAvailability',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='valid_assemblies', full_name='ncbi.datasets.v2.AssemblyDatasetAvailability.valid_assemblies', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='validAssemblies', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='invalid_assemblies', full_name='ncbi.datasets.v2.AssemblyDatasetAvailability.invalid_assemblies', index=1,
      number=2, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='invalidAssemblies', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='reason', full_name='ncbi.datasets.v2.AssemblyDatasetAvailability.reason', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='reason', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=8523,
  serialized_end=8666,
)


_ORGANISM_COUNTS = _descriptor.Descriptor(
  name='Counts',
  full_name='ncbi.datasets.v2.Organism.Counts',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='node', full_name='ncbi.datasets.v2.Organism.Counts.node', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='node', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='subtree', full_name='ncbi.datasets.v2.Organism.Counts.subtree', index=1,
      number=2, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='subtree', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=9528,
  serialized_end=9582,
)

_ORGANISM_COUNTBYTYPE = _descriptor.Descriptor(
  name='CountByType',
  full_name='ncbi.datasets.v2.Organism.CountByType',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='type', full_name='ncbi.datasets.v2.Organism.CountByType.type', index=0,
      number=1, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='type', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='counts', full_name='ncbi.datasets.v2.Organism.CountByType.counts', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='counts', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=9584,
  serialized_end=9705,
)

_ORGANISM = _descriptor.Descriptor(
  name='Organism',
  full_name='ncbi.datasets.v2.Organism',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='tax_id', full_name='ncbi.datasets.v2.Organism.tax_id', index=0,
      number=1, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='taxId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='sci_name', full_name='ncbi.datasets.v2.Organism.sci_name', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='sciName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='common_name', full_name='ncbi.datasets.v2.Organism.common_name', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='commonName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='blast_node', full_name='ncbi.datasets.v2.Organism.blast_node', index=3,
      number=16, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='blastNode', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='breed', full_name='ncbi.datasets.v2.Organism.breed', index=4,
      number=10, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='breed', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='cultivar', full_name='ncbi.datasets.v2.Organism.cultivar', index=5,
      number=11, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='cultivar', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='ecotype', full_name='ncbi.datasets.v2.Organism.ecotype', index=6,
      number=12, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='ecotype', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='isolate', full_name='ncbi.datasets.v2.Organism.isolate', index=7,
      number=13, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='isolate', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='sex', full_name='ncbi.datasets.v2.Organism.sex', index=8,
      number=14, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='sex', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='strain', full_name='ncbi.datasets.v2.Organism.strain', index=9,
      number=15, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='strain', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='search_text', full_name='ncbi.datasets.v2.Organism.search_text', index=10,
      number=17, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='searchText', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='rank', full_name='ncbi.datasets.v2.Organism.rank', index=11,
      number=5, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='rank', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='parent_tax_id', full_name='ncbi.datasets.v2.Organism.parent_tax_id', index=12,
      number=6, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='parentTaxId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_count', full_name='ncbi.datasets.v2.Organism.assembly_count', index=13,
      number=7, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='assemblyCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_counts', full_name='ncbi.datasets.v2.Organism.assembly_counts', index=14,
      number=19, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='assemblyCounts', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='counts', full_name='ncbi.datasets.v2.Organism.counts', index=15,
      number=22, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='counts', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='children', full_name='ncbi.datasets.v2.Organism.children', index=16,
      number=8, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='children', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='merged', full_name='ncbi.datasets.v2.Organism.merged', index=17,
      number=9, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='merged', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='merged_tax_ids', full_name='ncbi.datasets.v2.Organism.merged_tax_ids', index=18,
      number=18, type=4, cpp_type=4, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='mergedTaxIds', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='min_ord', full_name='ncbi.datasets.v2.Organism.min_ord', index=19,
      number=20, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='minOrd', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='max_ord', full_name='ncbi.datasets.v2.Organism.max_ord', index=20,
      number=21, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='maxOrd', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='weight', full_name='ncbi.datasets.v2.Organism.weight', index=21,
      number=23, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='weight', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='key', full_name='ncbi.datasets.v2.Organism.key', index=22,
      number=100, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='key', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='title', full_name='ncbi.datasets.v2.Organism.title', index=23,
      number=101, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='title', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='icon', full_name='ncbi.datasets.v2.Organism.icon', index=24,
      number=102, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='icon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_ORGANISM_COUNTS, _ORGANISM_COUNTBYTYPE, ],
  enum_types=[
    _ORGANISM_RANKTYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=8669,
  serialized_end=10377,
)


_TAXTREEREQUEST = _descriptor.Descriptor(
  name='TaxTreeRequest',
  full_name='ncbi.datasets.v2.TaxTreeRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='tax_id', full_name='ncbi.datasets.v2.TaxTreeRequest.tax_id', index=0,
      number=1, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='taxId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='tax_token', full_name='ncbi.datasets.v2.TaxTreeRequest.tax_token', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='taxToken', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxon', full_name='ncbi.datasets.v2.TaxTreeRequest.taxon', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031Y\022 \n\0049606\022\030NCBI Taxonomy Identifier\022\024\n\005human\022\013Common Name\022\037\n\014Homo sapiens\022\017Scientific Name', json_name='taxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='children_only', full_name='ncbi.datasets.v2.TaxTreeRequest.children_only', index=3,
      number=4, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\007\n\005false', json_name='childrenOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
    _descriptor.OneofDescriptor(
      name='taxonomy', full_name='ncbi.datasets.v2.TaxTreeRequest.taxonomy',
      index=0, containing_type=None,
      create_key=_descriptor._internal_create_key,
    fields=[]),
  ],
  serialized_start=10380,
  serialized_end=10641,
)


_ORGANISMQUERYREQUEST = _descriptor.Descriptor(
  name='OrganismQueryRequest',
  full_name='ncbi.datasets.v2.OrganismQueryRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='organism_query', full_name='ncbi.datasets.v2.OrganismQueryRequest.organism_query', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='organismQuery', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxon_query', full_name='ncbi.datasets.v2.OrganismQueryRequest.taxon_query', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\014\022\n\n\003hum\022\003hum', json_name='taxonQuery', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='tax_rank_filter', full_name='ncbi.datasets.v2.OrganismQueryRequest.tax_rank_filter', index=2,
      number=3, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\t\n\007species', json_name='taxRankFilter', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxon_resource_filter', full_name='ncbi.datasets.v2.OrganismQueryRequest.taxon_resource_filter', index=3,
      number=4, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\033\n\031TAXON_RESOURCE_FILTER_ALL', json_name='taxonResourceFilter', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _ORGANISMQUERYREQUEST_TAXRANKFILTER,
    _ORGANISMQUERYREQUEST_TAXONRESOURCEFILTER,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
    _descriptor.OneofDescriptor(
      name='id', full_name='ncbi.datasets.v2.OrganismQueryRequest.id',
      index=0, containing_type=None,
      create_key=_descriptor._internal_create_key,
    fields=[]),
  ],
  serialized_start=10644,
  serialized_end=11193,
)


_SCINAMEANDIDS_SCINAMEANDID = _descriptor.Descriptor(
  name='SciNameAndId',
  full_name='ncbi.datasets.v2.SciNameAndIds.SciNameAndId',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='sci_name', full_name='ncbi.datasets.v2.SciNameAndIds.SciNameAndId.sci_name', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='sciName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='tax_id', full_name='ncbi.datasets.v2.SciNameAndIds.SciNameAndId.tax_id', index=1,
      number=2, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='taxId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='common_name', full_name='ncbi.datasets.v2.SciNameAndIds.SciNameAndId.common_name', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='commonName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='matched_term', full_name='ncbi.datasets.v2.SciNameAndIds.SciNameAndId.matched_term', index=3,
      number=4, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='matchedTerm', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=11301,
  serialized_end=11433,
)

_SCINAMEANDIDS = _descriptor.Descriptor(
  name='SciNameAndIds',
  full_name='ncbi.datasets.v2.SciNameAndIds',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='sci_name_and_ids', full_name='ncbi.datasets.v2.SciNameAndIds.sci_name_and_ids', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='sciNameAndIds', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_SCINAMEANDIDS_SCINAMEANDID, ],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=11196,
  serialized_end=11433,
)


_ORTHOLOGREQUEST = _descriptor.Descriptor(
  name='OrthologRequest',
  full_name='ncbi.datasets.v2.OrthologRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='gene_id', full_name='ncbi.datasets.v2.OrthologRequest.gene_id', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\030\022\014\n\0042778\022\004GNAS\022\010\n\0012\022\003A2M', json_name='geneId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='returned_content', full_name='ncbi.datasets.v2.OrthologRequest.returned_content', index=1,
      number=2, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='returnedContent', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxon_filter', full_name='ncbi.datasets.v2.OrthologRequest.taxon_filter', index=2,
      number=3, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031{\022\'\n\0049606\n\00510090\022\030NCBI Taxonomy Identifier\022!\n\005human\n\013house mouse\022\013Common Name\022-\n\014homo sapiens\n\014mus musculus\022\017Scientific Name', json_name='taxonFilter', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_size', full_name='ncbi.datasets.v2.OrthologRequest.page_size', index=3,
      number=9, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\004\n\00220', json_name='pageSize', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_token', full_name='ncbi.datasets.v2.OrthologRequest.page_token', index=4,
      number=10, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='pageToken', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _ORTHOLOGREQUEST_CONTENTTYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=11436,
  serialized_end=11876,
)


_GENEDATASETREQUEST = _descriptor.Descriptor(
  name='GeneDatasetRequest',
  full_name='ncbi.datasets.v2.GeneDatasetRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='gene_ids', full_name='ncbi.datasets.v2.GeneDatasetRequest.gene_ids', index=0,
      number=1, type=13, cpp_type=3, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031<\022\024\n\00559067\022\013IL21 GeneID\022$\n\00559067\n\00550615\022\024IL21 & IL21R GeneIDs', json_name='geneIds', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='include_annotation_type', full_name='ncbi.datasets.v2.GeneDatasetRequest.include_annotation_type', index=1,
      number=6, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\332\360\031\002\010\002', json_name='includeAnnotationType', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='returned_content', full_name='ncbi.datasets.v2.GeneDatasetRequest.returned_content', index=2,
      number=7, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='returnedContent', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='fasta_filter', full_name='ncbi.datasets.v2.GeneDatasetRequest.fasta_filter', index=3,
      number=12, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='fastaFilter', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='aux_report', full_name='ncbi.datasets.v2.GeneDatasetRequest.aux_report', index=4,
      number=13, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031z\022x\n\016PRODUCT_REPORT\022fSelect additional report types to include with download. The base data report will always be included.', json_name='auxReport', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_fields', full_name='ncbi.datasets.v2.GeneDatasetRequest.table_fields', index=5,
      number=8, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031Y\022?\n\007gene-id\n\tgene-type\n\013description\022\034GeneID, type and description\022\026\n\007gene-id\022\013Just GeneID', json_name='tableFields', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_report_type', full_name='ncbi.datasets.v2.GeneDatasetRequest.table_report_type', index=6,
      number=14, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='tableReportType', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _GENEDATASETREQUEST_CONTENTTYPE,
    _GENEDATASETREQUEST_GENEDATASETREPORTTYPE,
  ],
  serialized_options=b'\302\360\031\033\n\031gene_ids:\n- 59067\n- 50615',
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=11879,
  serialized_end=12805,
)


_GENEDATASETREPORTSREQUEST_SYMBOLSFORTAXON = _descriptor.Descriptor(
  name='SymbolsForTaxon',
  full_name='ncbi.datasets.v2.GeneDatasetReportsRequest.SymbolsForTaxon',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='symbols', full_name='ncbi.datasets.v2.GeneDatasetReportsRequest.SymbolsForTaxon.symbols', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\016\022\014\n\004GNAS\022\004GNAS', json_name='symbols', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxon', full_name='ncbi.datasets.v2.GeneDatasetReportsRequest.SymbolsForTaxon.taxon', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031Y\022 \n\0049606\022\030NCBI Taxonomy Identifier\022\024\n\005human\022\013Common Name\022\037\n\014Homo sapiens\022\017Scientific Name', json_name='taxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=13772,
  serialized_end=13952,
)

_GENEDATASETREPORTSREQUEST = _descriptor.Descriptor(
  name='GeneDatasetReportsRequest',
  full_name='ncbi.datasets.v2.GeneDatasetReportsRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='returned_content', full_name='ncbi.datasets.v2.GeneDatasetReportsRequest.returned_content', index=0,
      number=1, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='returnedContent', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='gene_ids', full_name='ncbi.datasets.v2.GeneDatasetReportsRequest.gene_ids', index=1,
      number=2, type=13, cpp_type=3, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031<\022\024\n\00559067\022\013IL21 GeneID\022$\n\00559067\n\00550615\022\024IL21 & IL21R GeneIDs', json_name='geneIds', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v2.GeneDatasetReportsRequest.accessions', index=2,
      number=3, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031]\022.\n\013NM_021803.4\n\013NM_181078.3\022\022RefSeq transcripts\022+\n\013NP_068575.1\n\013NP_851564.1\022\017RefSeq proteins', json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='symbols_for_taxon', full_name='ncbi.datasets.v2.GeneDatasetReportsRequest.symbols_for_taxon', index=3,
      number=4, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='symbolsForTaxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxon', full_name='ncbi.datasets.v2.GeneDatasetReportsRequest.taxon', index=4,
      number=5, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031Y\022 \n\0049606\022\030NCBI Taxonomy Identifier\022\024\n\005human\022\013Common Name\022\037\n\014Homo sapiens\022\017Scientific Name', json_name='taxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_fields', full_name='ncbi.datasets.v2.GeneDatasetReportsRequest.table_fields', index=5,
      number=8, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031Y\022?\n\007gene-id\n\tgene-type\n\013description\022\034GeneID, type and description\022\026\n\007gene-id\022\013Just GeneID', json_name='tableFields', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_format', full_name='ncbi.datasets.v2.GeneDatasetReportsRequest.table_format', index=6,
      number=13, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='tableFormat', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_size', full_name='ncbi.datasets.v2.GeneDatasetReportsRequest.page_size', index=7,
      number=9, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\004\n\00220', json_name='pageSize', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_token', full_name='ncbi.datasets.v2.GeneDatasetReportsRequest.page_token', index=8,
      number=10, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='pageToken', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='query', full_name='ncbi.datasets.v2.GeneDatasetReportsRequest.query', index=9,
      number=11, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\026\022\024\n\022A2M immunoglobulin', json_name='query', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='types', full_name='ncbi.datasets.v2.GeneDatasetReportsRequest.types', index=10,
      number=12, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031-\022+\n\016PROTEIN_CODING\022\031Only protein coding genes', json_name='types', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_GENEDATASETREPORTSREQUEST_SYMBOLSFORTAXON, ],
  enum_types=[
    _GENEDATASETREPORTSREQUEST_CONTENTTYPE,
  ],
  serialized_options=b'\302\360\031\033\n\031gene_ids:\n- 59067\n- 50615',
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=12808,
  serialized_end=14058,
)


_GENELINKSREQUEST = _descriptor.Descriptor(
  name='GeneLinksRequest',
  full_name='ncbi.datasets.v2.GeneLinksRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='gene_ids', full_name='ncbi.datasets.v2.GeneLinksRequest.gene_ids', index=0,
      number=1, type=13, cpp_type=3, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031<\022\024\n\00559067\022\013IL21 GeneID\022$\n\00559067\n\00550615\022\024IL21 & IL21R GeneIDs', json_name='geneIds', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=b'\302\360\031\033\n\031gene_ids:\n- 59067\n- 50615',
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=14061,
  serialized_end=14205,
)


_GENELINKSREPLY_GENELINK = _descriptor.Descriptor(
  name='GeneLink',
  full_name='ncbi.datasets.v2.GeneLinksReply.GeneLink',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='gene_id', full_name='ncbi.datasets.v2.GeneLinksReply.GeneLink.gene_id', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='geneId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='gene_link_type', full_name='ncbi.datasets.v2.GeneLinksReply.GeneLink.gene_link_type', index=1,
      number=2, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='geneLinkType', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='resource_link', full_name='ncbi.datasets.v2.GeneLinksReply.GeneLink.resource_link', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='resourceLink', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='resource_id', full_name='ncbi.datasets.v2.GeneLinksReply.GeneLink.resource_id', index=3,
      number=4, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='resourceId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=14301,
  serialized_end=14491,
)

_GENELINKSREPLY = _descriptor.Descriptor(
  name='GeneLinksReply',
  full_name='ncbi.datasets.v2.GeneLinksReply',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='gene_links', full_name='ncbi.datasets.v2.GeneLinksReply.gene_links', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='geneLinks', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_GENELINKSREPLY_GENELINK, ],
  enum_types=[
    _GENELINKSREPLY_GENELINKTYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=14208,
  serialized_end=14568,
)


_GENECHROMOSOMESUMMARYREQUEST = _descriptor.Descriptor(
  name='GeneChromosomeSummaryRequest',
  full_name='ncbi.datasets.v2.GeneChromosomeSummaryRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='taxon', full_name='ncbi.datasets.v2.GeneChromosomeSummaryRequest.taxon', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031Y\022 \n\0049606\022\030NCBI Taxonomy Identifier\022\024\n\005human\022\013Common Name\022\037\n\014Homo sapiens\022\017Scientific Name', json_name='taxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='annotation_name', full_name='ncbi.datasets.v2.GeneChromosomeSummaryRequest.annotation_name', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031=\022;\n\037Annotation Release 109.20211119\022\030Human annotation release', json_name='annotationName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=14571,
  serialized_end=14826,
)


_GENECHROMOSOMESUMMARYREPLY_GENECHROMOSOMESUMMARY = _descriptor.Descriptor(
  name='GeneChromosomeSummary',
  full_name='ncbi.datasets.v2.GeneChromosomeSummaryReply.GeneChromosomeSummary',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='name', full_name='ncbi.datasets.v2.GeneChromosomeSummaryReply.GeneChromosomeSummary.name', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='name', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='count', full_name='ncbi.datasets.v2.GeneChromosomeSummaryReply.GeneChromosomeSummary.count', index=1,
      number=2, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='count', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='accession', full_name='ncbi.datasets.v2.GeneChromosomeSummaryReply.GeneChromosomeSummary.accession', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='accession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=14987,
  serialized_end=15082,
)

_GENECHROMOSOMESUMMARYREPLY = _descriptor.Descriptor(
  name='GeneChromosomeSummaryReply',
  full_name='ncbi.datasets.v2.GeneChromosomeSummaryReply',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='gene_chromosome_summaries', full_name='ncbi.datasets.v2.GeneChromosomeSummaryReply.gene_chromosome_summaries', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='geneChromosomeSummaries', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_GENECHROMOSOMESUMMARYREPLY_GENECHROMOSOMESUMMARY, ],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=14829,
  serialized_end=15082,
)


_ASSEMBLYLINKSREQUEST = _descriptor.Descriptor(
  name='AssemblyLinksRequest',
  full_name='ncbi.datasets.v2.AssemblyLinksRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v2.AssemblyLinksRequest.accessions', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031R\0225\n\020GCF_000001405.40\n\020GCF_000001635.27\022\017Human and Mouse\022\031\n\020GCF_000001405.40\022\005Human', json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=b'\302\360\0310\n.accessions:\n  accessions:\n  - GCF_000001405.40',
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=15085,
  serialized_end=15281,
)


_ASSEMBLYLINKSREPLY_ASSEMBLYLINK = _descriptor.Descriptor(
  name='AssemblyLink',
  full_name='ncbi.datasets.v2.AssemblyLinksReply.AssemblyLink',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accession', full_name='ncbi.datasets.v2.AssemblyLinksReply.AssemblyLink.accession', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='accession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_link_type', full_name='ncbi.datasets.v2.AssemblyLinksReply.AssemblyLink.assembly_link_type', index=1,
      number=2, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='assemblyLinkType', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='resource_link', full_name='ncbi.datasets.v2.AssemblyLinksReply.AssemblyLink.resource_link', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='resourceLink', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='linked_identifiers', full_name='ncbi.datasets.v2.AssemblyLinksReply.AssemblyLink.linked_identifiers', index=3,
      number=4, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='linkedIdentifiers', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=15397,
  serialized_end=15626,
)

_ASSEMBLYLINKSREPLY = _descriptor.Descriptor(
  name='AssemblyLinksReply',
  full_name='ncbi.datasets.v2.AssemblyLinksReply',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='assembly_links', full_name='ncbi.datasets.v2.AssemblyLinksReply.assembly_links', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='assemblyLinks', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_ASSEMBLYLINKSREPLY_ASSEMBLYLINK, ],
  enum_types=[
    _ASSEMBLYLINKSREPLY_ASSEMBLYLINKTYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=15284,
  serialized_end=15724,
)


_VIRUSDATASETREQUEST = _descriptor.Descriptor(
  name='VirusDatasetRequest',
  full_name='ncbi.datasets.v2.VirusDatasetRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v2.VirusDatasetRequest.accessions', index=0,
      number=16, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031?\022=\n\013NC_045512.2\022.SARS-COV-2 isolate Wuhan-Hu-1, complete genome', json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxon', full_name='ncbi.datasets.v2.VirusDatasetRequest.taxon', index=1,
      number=19, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031@\022\025\n\0072697049\022\nSARS-COV-2\022\026\n\00511118\022\rCoronaviridae\022\017\n\0071335626\022\004MERS', json_name='taxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='refseq_only', full_name='ncbi.datasets.v2.VirusDatasetRequest.refseq_only', index=2,
      number=4, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031P\n\005false\022\037\n\004true\022\027Limit to RefSeq genomes\022&\n\005false\022\035Both GenBank & RefSeq genomes', json_name='refseqOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='annotated_only', full_name='ncbi.datasets.v2.VirusDatasetRequest.annotated_only', index=3,
      number=5, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031A\n\005false\022\"\n\004true\022\032Limit to annotated genomes\022\024\n\005false\022\013All genomes', json_name='annotatedOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='released_since', full_name='ncbi.datasets.v2.VirusDatasetRequest.released_since', index=4,
      number=6, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\036\022\034\n\n2020-08-01\022\016August 1, 2020', json_name='releasedSince', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='updated_since', full_name='ncbi.datasets.v2.VirusDatasetRequest.updated_since', index=5,
      number=17, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\035\022\033\n\n2021-07-18\022\rJuly 18, 2021', json_name='updatedSince', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='host', full_name='ncbi.datasets.v2.VirusDatasetRequest.host', index=6,
      number=7, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\020\022\016\n\005human\022\005human', json_name='host', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='pangolin_classification', full_name='ncbi.datasets.v2.VirusDatasetRequest.pangolin_classification', index=7,
      number=20, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='pangolinClassification', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='geo_location', full_name='ncbi.datasets.v2.VirusDatasetRequest.geo_location', index=8,
      number=8, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\031\022\n\n\003USA\022\003USA\022\013\n\002TX\022\005Texas', json_name='geoLocation', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='complete_only', full_name='ncbi.datasets.v2.VirusDatasetRequest.complete_only', index=9,
      number=9, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031@\n\005false\022!\n\004true\022\031Limit to complete genomes\022\024\n\005false\022\013All genomes', json_name='completeOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_fields', full_name='ncbi.datasets.v2.VirusDatasetRequest.table_fields', index=10,
      number=10, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031N\022L\n\024nucleotide_accession\n\021nucleotide_length\n\020nuc_completeness\022\017Assembly Fields', json_name='tableFields', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='include_sequence', full_name='ncbi.datasets.v2.VirusDatasetRequest.include_sequence', index=11,
      number=22, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031;\0229\n\006GENOME\n\003CDS\n\007PROTEIN\022!Select viral sequences to include', json_name='includeSequence', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='aux_report', full_name='ncbi.datasets.v2.VirusDatasetRequest.aux_report', index=12,
      number=23, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031v\022t\n\nANNOTATION\022fSelect additional report types to include with download. The base data report will always be included.', json_name='auxReport', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='format', full_name='ncbi.datasets.v2.VirusDatasetRequest.format', index=13,
      number=14, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\0312\n\003tsv\022\n\n\003tsv\022\003TSV\022\n\n\003csv\022\003CSV\022\023\n\005jsonl\022\nJSON Lines', json_name='format', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='use_psg', full_name='ncbi.datasets.v2.VirusDatasetRequest.use_psg', index=14,
      number=99, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='usePsg', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=15727,
  serialized_end=17280,
)


_VIRUSDATASETFILTER = _descriptor.Descriptor(
  name='VirusDatasetFilter',
  full_name='ncbi.datasets.v2.VirusDatasetFilter',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v2.VirusDatasetFilter.accessions', index=0,
      number=2, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\206\001\022\203\001\n\027NC_045512.2,NC_038294.1\022hSARS-COV-2 isolate Wuhan-Hu-1 and Middle East respiratory syndrome-related coronavirus, complete genomes', json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxon', full_name='ncbi.datasets.v2.VirusDatasetFilter.taxon', index=1,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031@\022\025\n\0072697049\022\nSARS-COV-2\022\026\n\00511118\022\rCoronaviridae\022\017\n\0071335626\022\004MERS', json_name='taxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='refseq_only', full_name='ncbi.datasets.v2.VirusDatasetFilter.refseq_only', index=2,
      number=4, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031P\n\005false\022\037\n\004true\022\027Limit to RefSeq genomes\022&\n\005false\022\035Both GenBank & RefSeq genomes', json_name='refseqOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='annotated_only', full_name='ncbi.datasets.v2.VirusDatasetFilter.annotated_only', index=3,
      number=5, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031A\n\005false\022\"\n\004true\022\032Limit to annotated genomes\022\024\n\005false\022\013All genomes', json_name='annotatedOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='released_since', full_name='ncbi.datasets.v2.VirusDatasetFilter.released_since', index=4,
      number=6, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\036\022\034\n\n2020-08-01\022\016August 1, 2020', json_name='releasedSince', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='updated_since', full_name='ncbi.datasets.v2.VirusDatasetFilter.updated_since', index=5,
      number=11, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\035\022\033\n\n2021-07-18\022\rJuly 18, 2021', json_name='updatedSince', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='host', full_name='ncbi.datasets.v2.VirusDatasetFilter.host', index=6,
      number=7, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\020\022\016\n\005human\022\005human', json_name='host', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='pangolin_classification', full_name='ncbi.datasets.v2.VirusDatasetFilter.pangolin_classification', index=7,
      number=8, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='pangolinClassification', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='geo_location', full_name='ncbi.datasets.v2.VirusDatasetFilter.geo_location', index=8,
      number=9, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\031\022\n\n\003USA\022\003USA\022\013\n\002TX\022\005Texas', json_name='geoLocation', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='complete_only', full_name='ncbi.datasets.v2.VirusDatasetFilter.complete_only', index=9,
      number=10, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031@\n\005false\022!\n\004true\022\031Limit to complete genomes\022\024\n\005false\022\013All genomes', json_name='completeOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=17283,
  serialized_end=18274,
)


_VIRUSDATAREPORTREQUEST = _descriptor.Descriptor(
  name='VirusDataReportRequest',
  full_name='ncbi.datasets.v2.VirusDataReportRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='filter', full_name='ncbi.datasets.v2.VirusDataReportRequest.filter', index=0,
      number=1, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='filter', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='returned_content', full_name='ncbi.datasets.v2.VirusDataReportRequest.returned_content', index=1,
      number=2, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='returnedContent', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_fields', full_name='ncbi.datasets.v2.VirusDataReportRequest.table_fields', index=2,
      number=3, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031B\022@\n\taccession\n\013is-complete\n\014is-annotated\022\030Virus Data Report Fields', json_name='tableFields', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_format', full_name='ncbi.datasets.v2.VirusDataReportRequest.table_format', index=3,
      number=6, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='tableFormat', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_size', full_name='ncbi.datasets.v2.VirusDataReportRequest.page_size', index=4,
      number=4, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\004\n\00220', json_name='pageSize', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_token', full_name='ncbi.datasets.v2.VirusDataReportRequest.page_token', index=5,
      number=5, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='pageToken', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _VIRUSDATAREPORTREQUEST_CONTENTTYPE,
  ],
  serialized_options=b'\302\360\031\"\n taxon: 2697049\nrefseq_only: true',
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=18277,
  serialized_end=18762,
)


_VIRUSANNOTATIONFILTER = _descriptor.Descriptor(
  name='VirusAnnotationFilter',
  full_name='ncbi.datasets.v2.VirusAnnotationFilter',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v2.VirusAnnotationFilter.accessions', index=0,
      number=2, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\206\001\022\203\001\n\027NC_045512.2,NC_038294.1\022hSARS-COV-2 isolate Wuhan-Hu-1 and Middle East respiratory syndrome-related coronavirus, complete genomes', json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxon', full_name='ncbi.datasets.v2.VirusAnnotationFilter.taxon', index=1,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031@\022\025\n\0072697049\022\nSARS-COV-2\022\026\n\00511118\022\rCoronaviridae\022\017\n\0071335626\022\004MERS', json_name='taxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='refseq_only', full_name='ncbi.datasets.v2.VirusAnnotationFilter.refseq_only', index=2,
      number=4, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031P\n\005false\022\037\n\004true\022\027Limit to RefSeq genomes\022&\n\005false\022\035Both GenBank & RefSeq genomes', json_name='refseqOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='released_since', full_name='ncbi.datasets.v2.VirusAnnotationFilter.released_since', index=3,
      number=6, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\036\022\034\n\n2020-08-01\022\016August 1, 2020', json_name='releasedSince', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='updated_since', full_name='ncbi.datasets.v2.VirusAnnotationFilter.updated_since', index=4,
      number=11, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\035\022\033\n\n2021-07-18\022\rJuly 18, 2021', json_name='updatedSince', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='host', full_name='ncbi.datasets.v2.VirusAnnotationFilter.host', index=5,
      number=7, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\020\022\016\n\005human\022\005human', json_name='host', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='pangolin_classification', full_name='ncbi.datasets.v2.VirusAnnotationFilter.pangolin_classification', index=6,
      number=8, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='pangolinClassification', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='geo_location', full_name='ncbi.datasets.v2.VirusAnnotationFilter.geo_location', index=7,
      number=9, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\031\022\n\n\003USA\022\003USA\022\013\n\002TX\022\005Texas', json_name='geoLocation', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='complete_only', full_name='ncbi.datasets.v2.VirusAnnotationFilter.complete_only', index=8,
      number=10, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031@\n\005false\022!\n\004true\022\031Limit to complete genomes\022\024\n\005false\022\013All genomes', json_name='completeOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=18765,
  serialized_end=19649,
)


_VIRUSANNOTATIONREPORTREQUEST = _descriptor.Descriptor(
  name='VirusAnnotationReportRequest',
  full_name='ncbi.datasets.v2.VirusAnnotationReportRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='filter', full_name='ncbi.datasets.v2.VirusAnnotationReportRequest.filter', index=0,
      number=1, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='filter', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_fields', full_name='ncbi.datasets.v2.VirusAnnotationReportRequest.table_fields', index=1,
      number=3, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031;\0229\n\taccession\n\014isolate-name\022\036Virus Annotation Report Fields', json_name='tableFields', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_format', full_name='ncbi.datasets.v2.VirusAnnotationReportRequest.table_format', index=2,
      number=6, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='tableFormat', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_size', full_name='ncbi.datasets.v2.VirusAnnotationReportRequest.page_size', index=3,
      number=4, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\004\n\00220', json_name='pageSize', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_token', full_name='ncbi.datasets.v2.VirusAnnotationReportRequest.page_token', index=4,
      number=5, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='pageToken', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=b'\302\360\031\"\n taxon: 2697049\nrefseq_only: true',
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=19652,
  serialized_end=19992,
)


_VIRUSAVAILABILITYREQUEST = _descriptor.Descriptor(
  name='VirusAvailabilityRequest',
  full_name='ncbi.datasets.v2.VirusAvailabilityRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v2.VirusAvailabilityRequest.accessions', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\206\001\022\203\001\n\027NC_045512.2,NC_038294.1\022hSARS-COV-2 isolate Wuhan-Hu-1 and Middle East respiratory syndrome-related coronavirus, complete genomes', json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=19995,
  serialized_end=20196,
)


_VIRUSAVAILABILITY = _descriptor.Descriptor(
  name='VirusAvailability',
  full_name='ncbi.datasets.v2.VirusAvailability',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='valid_accessions', full_name='ncbi.datasets.v2.VirusAvailability.valid_accessions', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='validAccessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='invalid_accessions', full_name='ncbi.datasets.v2.VirusAvailability.invalid_accessions', index=1,
      number=2, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='invalidAccessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='message', full_name='ncbi.datasets.v2.VirusAvailability.message', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='message', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=20199,
  serialized_end=20334,
)


_SARS2PROTEINDATASETREQUEST = _descriptor.Descriptor(
  name='Sars2ProteinDatasetRequest',
  full_name='ncbi.datasets.v2.Sars2ProteinDatasetRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='proteins', full_name='ncbi.datasets.v2.Sars2ProteinDatasetRequest.proteins', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031^\022\036\n\rspike protein\022\rSpike Protein\022<\n\rspike protein\n\020envelope protein\n\004RdRp\022\023SARS-CoV-2 proteins', json_name='proteins', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='refseq_only', full_name='ncbi.datasets.v2.Sars2ProteinDatasetRequest.refseq_only', index=1,
      number=4, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031P\n\005false\022\037\n\004true\022\027Limit to RefSeq genomes\022&\n\005false\022\035Both GenBank & RefSeq genomes', json_name='refseqOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='annotated_only', full_name='ncbi.datasets.v2.Sars2ProteinDatasetRequest.annotated_only', index=2,
      number=5, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031A\n\005false\022\"\n\004true\022\032Limit to annotated genomes\022\024\n\005false\022\013All genomes', json_name='annotatedOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='released_since', full_name='ncbi.datasets.v2.Sars2ProteinDatasetRequest.released_since', index=3,
      number=6, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\036\022\034\n\n2020-08-01\022\016August 1, 2020', json_name='releasedSince', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='updated_since', full_name='ncbi.datasets.v2.Sars2ProteinDatasetRequest.updated_since', index=4,
      number=15, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\035\022\033\n\n2021-07-18\022\rJuly 18, 2021', json_name='updatedSince', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='host', full_name='ncbi.datasets.v2.Sars2ProteinDatasetRequest.host', index=5,
      number=7, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\020\022\016\n\005human\022\005human', json_name='host', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='geo_location', full_name='ncbi.datasets.v2.Sars2ProteinDatasetRequest.geo_location', index=6,
      number=8, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\031\022\n\n\003USA\022\003USA\022\013\n\002TX\022\005Texas', json_name='geoLocation', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='complete_only', full_name='ncbi.datasets.v2.Sars2ProteinDatasetRequest.complete_only', index=7,
      number=9, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031@\n\005false\022!\n\004true\022\031Limit to complete genomes\022\024\n\005false\022\013All genomes', json_name='completeOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_fields', full_name='ncbi.datasets.v2.Sars2ProteinDatasetRequest.table_fields', index=8,
      number=10, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031N\022L\n\024nucleotide_accession\n\021nucleotide_length\n\020nuc_completeness\022\017Assembly Fields', json_name='tableFields', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='include_sequence', full_name='ncbi.datasets.v2.Sars2ProteinDatasetRequest.include_sequence', index=9,
      number=22, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\0313\0221\n\003CDS\n\007PROTEIN\022!Select viral sequences to include', json_name='includeSequence', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='aux_report', full_name='ncbi.datasets.v2.Sars2ProteinDatasetRequest.aux_report', index=10,
      number=23, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031v\022t\n\nANNOTATION\022fSelect additional report types to include with download. The base data report will always be included.', json_name='auxReport', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='format', full_name='ncbi.datasets.v2.Sars2ProteinDatasetRequest.format', index=11,
      number=14, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\0312\n\003tsv\022\n\n\003tsv\022\003TSV\022\n\n\003csv\022\003CSV\022\023\n\005jsonl\022\nJSON Lines', json_name='format', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=20337,
  serialized_end=21742,
)


_DOWNLOADSUMMARY_HYDRATED = _descriptor.Descriptor(
  name='Hydrated',
  full_name='ncbi.datasets.v2.DownloadSummary.Hydrated',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='estimated_file_size_mb', full_name='ncbi.datasets.v2.DownloadSummary.Hydrated.estimated_file_size_mb', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='estimatedFileSizeMb', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='url', full_name='ncbi.datasets.v2.DownloadSummary.Hydrated.url', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='url', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='cli_download_command_line', full_name='ncbi.datasets.v2.DownloadSummary.Hydrated.cli_download_command_line', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='cliDownloadCommandLine', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=22284,
  serialized_end=22424,
)

_DOWNLOADSUMMARY_DEHYDRATED = _descriptor.Descriptor(
  name='Dehydrated',
  full_name='ncbi.datasets.v2.DownloadSummary.Dehydrated',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='estimated_file_size_mb', full_name='ncbi.datasets.v2.DownloadSummary.Dehydrated.estimated_file_size_mb', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='estimatedFileSizeMb', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='url', full_name='ncbi.datasets.v2.DownloadSummary.Dehydrated.url', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='url', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='cli_download_command_line', full_name='ncbi.datasets.v2.DownloadSummary.Dehydrated.cli_download_command_line', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='cliDownloadCommandLine', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='cli_rehydrate_command_line', full_name='ncbi.datasets.v2.DownloadSummary.Dehydrated.cli_rehydrate_command_line', index=3,
      number=4, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='cliRehydrateCommandLine', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=22427,
  serialized_end=22630,
)

_DOWNLOADSUMMARY_FILESUMMARY = _descriptor.Descriptor(
  name='FileSummary',
  full_name='ncbi.datasets.v2.DownloadSummary.FileSummary',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='file_count', full_name='ncbi.datasets.v2.DownloadSummary.FileSummary.file_count', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='fileCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='size_mb', full_name='ncbi.datasets.v2.DownloadSummary.FileSummary.size_mb', index=1,
      number=2, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='sizeMb', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=22632,
  serialized_end=22701,
)

_DOWNLOADSUMMARY_AVAILABLEFILES = _descriptor.Descriptor(
  name='AvailableFiles',
  full_name='ncbi.datasets.v2.DownloadSummary.AvailableFiles',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='all_genomic_fasta', full_name='ncbi.datasets.v2.DownloadSummary.AvailableFiles.all_genomic_fasta', index=0,
      number=1, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='allGenomicFasta', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='genome_gff', full_name='ncbi.datasets.v2.DownloadSummary.AvailableFiles.genome_gff', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='genomeGff', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='genome_gbff', full_name='ncbi.datasets.v2.DownloadSummary.AvailableFiles.genome_gbff', index=2,
      number=3, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='genomeGbff', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='rna_fasta', full_name='ncbi.datasets.v2.DownloadSummary.AvailableFiles.rna_fasta', index=3,
      number=4, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='rnaFasta', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='prot_fasta', full_name='ncbi.datasets.v2.DownloadSummary.AvailableFiles.prot_fasta', index=4,
      number=5, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='protFasta', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='genome_gtf', full_name='ncbi.datasets.v2.DownloadSummary.AvailableFiles.genome_gtf', index=5,
      number=6, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='genomeGtf', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='cds_fasta', full_name='ncbi.datasets.v2.DownloadSummary.AvailableFiles.cds_fasta', index=6,
      number=7, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='cdsFasta', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='sequence_report', full_name='ncbi.datasets.v2.DownloadSummary.AvailableFiles.sequence_report', index=7,
      number=8, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='sequenceReport', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='annotation_report', full_name='ncbi.datasets.v2.DownloadSummary.AvailableFiles.annotation_report', index=8,
      number=9, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='annotationReport', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=22704,
  serialized_end=23457,
)

_DOWNLOADSUMMARY = _descriptor.Descriptor(
  name='DownloadSummary',
  full_name='ncbi.datasets.v2.DownloadSummary',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='record_count', full_name='ncbi.datasets.v2.DownloadSummary.record_count', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='recordCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_count', full_name='ncbi.datasets.v2.DownloadSummary.assembly_count', index=1,
      number=2, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='assemblyCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='resource_updated_on', full_name='ncbi.datasets.v2.DownloadSummary.resource_updated_on', index=2,
      number=4, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='resourceUpdatedOn', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='hydrated', full_name='ncbi.datasets.v2.DownloadSummary.hydrated', index=3,
      number=5, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='hydrated', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='dehydrated', full_name='ncbi.datasets.v2.DownloadSummary.dehydrated', index=4,
      number=6, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='dehydrated', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='errors', full_name='ncbi.datasets.v2.DownloadSummary.errors', index=5,
      number=7, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='errors', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='messages', full_name='ncbi.datasets.v2.DownloadSummary.messages', index=6,
      number=8, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='messages', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='available_files', full_name='ncbi.datasets.v2.DownloadSummary.available_files', index=7,
      number=9, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='availableFiles', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_DOWNLOADSUMMARY_HYDRATED, _DOWNLOADSUMMARY_DEHYDRATED, _DOWNLOADSUMMARY_FILESUMMARY, _DOWNLOADSUMMARY_AVAILABLEFILES, ],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=21745,
  serialized_end=23457,
)


_TABULAROUTPUT = _descriptor.Descriptor(
  name='TabularOutput',
  full_name='ncbi.datasets.v2.TabularOutput',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='data', full_name='ncbi.datasets.v2.TabularOutput.data', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='data', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=23459,
  serialized_end=23494,
)


_MICROBIGGEDATASETREQUEST_ELEMENTFLANKCONFIG = _descriptor.Descriptor(
  name='ElementFlankConfig',
  full_name='ncbi.datasets.v2.MicroBiggeDatasetRequest.ElementFlankConfig',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='length', full_name='ncbi.datasets.v2.MicroBiggeDatasetRequest.ElementFlankConfig.length', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='length', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=23767,
  serialized_end=23811,
)

_MICROBIGGEDATASETREQUEST = _descriptor.Descriptor(
  name='MicroBiggeDatasetRequest',
  full_name='ncbi.datasets.v2.MicroBiggeDatasetRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='opaque_solr_query', full_name='ncbi.datasets.v2.MicroBiggeDatasetRequest.opaque_solr_query', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='opaqueSolrQuery', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='files', full_name='ncbi.datasets.v2.MicroBiggeDatasetRequest.files', index=1,
      number=2, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\332\360\031\002\010\002', json_name='files', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='element_flank_config', full_name='ncbi.datasets.v2.MicroBiggeDatasetRequest.element_flank_config', index=2,
      number=3, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='elementFlankConfig', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_MICROBIGGEDATASETREQUEST_ELEMENTFLANKCONFIG, ],
  enum_types=[
    _MICROBIGGEDATASETREQUEST_FILETYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
    _descriptor.OneofDescriptor(
      name='query', full_name='ncbi.datasets.v2.MicroBiggeDatasetRequest.query',
      index=0, containing_type=None,
      create_key=_descriptor._internal_create_key,
    fields=[]),
  ],
  serialized_start=23497,
  serialized_end=23940,
)


_VERSIONREPLY = _descriptor.Descriptor(
  name='VersionReply',
  full_name='ncbi.datasets.v2.VersionReply',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='version', full_name='ncbi.datasets.v2.VersionReply.version', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='version', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='major_ver', full_name='ncbi.datasets.v2.VersionReply.major_ver', index=1,
      number=2, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='majorVer', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='minor_ver', full_name='ncbi.datasets.v2.VersionReply.minor_ver', index=2,
      number=3, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='minorVer', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='patch_ver', full_name='ncbi.datasets.v2.VersionReply.patch_ver', index=3,
      number=4, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='patchVer', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=23942,
  serialized_end=24069,
)


_PROKARYOTEGENEREQUEST_GENEFLANKCONFIG = _descriptor.Descriptor(
  name='GeneFlankConfig',
  full_name='ncbi.datasets.v2.ProkaryoteGeneRequest.GeneFlankConfig',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='length', full_name='ncbi.datasets.v2.ProkaryoteGeneRequest.GeneFlankConfig.length', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='length', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=24619,
  serialized_end=24660,
)

_PROKARYOTEGENEREQUEST = _descriptor.Descriptor(
  name='ProkaryoteGeneRequest',
  full_name='ncbi.datasets.v2.ProkaryoteGeneRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v2.ProkaryoteGeneRequest.accessions', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031o\022.\n\016WP_015878339.1\022\034DNA-deoxyinosine glycosylase\022=\n\016WP_001040085.1\022+type II CRISPR RNA-guided endonuclease Cas9', json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='include_annotation_type', full_name='ncbi.datasets.v2.ProkaryoteGeneRequest.include_annotation_type', index=1,
      number=2, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\332\360\031\002\010\002\302\360\031\231\001\022\036\n\nFASTA_GENE\022\020Gene range FASTA\022?\n\tFASTA_RNA\n\rFASTA_PROTEIN\022#Transcript & protein sequence FASTA\0226\n\nFASTA_GENE\n\tFASTA_RNA\n\rFASTA_PROTEIN\022\016All annotation', json_name='includeAnnotationType', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='gene_flank_config', full_name='ncbi.datasets.v2.ProkaryoteGeneRequest.gene_flank_config', index=2,
      number=3, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='geneFlankConfig', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxon', full_name='ncbi.datasets.v2.ProkaryoteGeneRequest.taxon', index=3,
      number=4, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='taxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_PROKARYOTEGENEREQUEST_GENEFLANKCONFIG, ],
  enum_types=[
  ],
  serialized_options=b'\302\360\031/\n-accessions:\n- WP_000000001.1\n- WP_000000002.1',
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=24072,
  serialized_end=24713,
)


_SLEEPREQUEST = _descriptor.Descriptor(
  name='SleepRequest',
  full_name='ncbi.datasets.v2.SleepRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='sleep_msec', full_name='ncbi.datasets.v2.SleepRequest.sleep_msec', index=0,
      number=1, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='sleepMsec', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='error_rate', full_name='ncbi.datasets.v2.SleepRequest.error_rate', index=1,
      number=2, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='errorRate', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=24715,
  serialized_end=24791,
)


_SLEEPREPLY = _descriptor.Descriptor(
  name='SleepReply',
  full_name='ncbi.datasets.v2.SleepReply',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='reply', full_name='ncbi.datasets.v2.SleepReply.reply', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='reply', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=24793,
  serialized_end=24827,
)


_TAXONOMYNODE_COUNTBYTYPE = _descriptor.Descriptor(
  name='CountByType',
  full_name='ncbi.datasets.v2.TaxonomyNode.CountByType',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='type', full_name='ncbi.datasets.v2.TaxonomyNode.CountByType.type', index=0,
      number=1, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='type', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='count', full_name='ncbi.datasets.v2.TaxonomyNode.CountByType.count', index=1,
      number=2, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='count', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=25364,
  serialized_end=25448,
)

_TAXONOMYNODE = _descriptor.Descriptor(
  name='TaxonomyNode',
  full_name='ncbi.datasets.v2.TaxonomyNode',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='tax_id', full_name='ncbi.datasets.v2.TaxonomyNode.tax_id', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='taxId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='organism_name', full_name='ncbi.datasets.v2.TaxonomyNode.organism_name', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='organismName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='common_name', full_name='ncbi.datasets.v2.TaxonomyNode.common_name', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='commonName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='lineage', full_name='ncbi.datasets.v2.TaxonomyNode.lineage', index=3,
      number=4, type=13, cpp_type=3, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='lineage', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='children', full_name='ncbi.datasets.v2.TaxonomyNode.children', index=4,
      number=5, type=13, cpp_type=3, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='children', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='descendent_with_described_species_names_count', full_name='ncbi.datasets.v2.TaxonomyNode.descendent_with_described_species_names_count', index=5,
      number=6, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='descendentWithDescribedSpeciesNamesCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='rank', full_name='ncbi.datasets.v2.TaxonomyNode.rank', index=6,
      number=7, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='rank', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='has_described_species_name', full_name='ncbi.datasets.v2.TaxonomyNode.has_described_species_name', index=7,
      number=8, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='hasDescribedSpeciesName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='counts', full_name='ncbi.datasets.v2.TaxonomyNode.counts', index=8,
      number=9, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='counts', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='min_ord', full_name='ncbi.datasets.v2.TaxonomyNode.min_ord', index=9,
      number=20, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='minOrd', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='max_ord', full_name='ncbi.datasets.v2.TaxonomyNode.max_ord', index=10,
      number=21, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='maxOrd', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='_indexed_names', full_name='ncbi.datasets.v2.TaxonomyNode._indexed_names', index=11,
      number=101, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='IndexedNames', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_TAXONOMYNODE_COUNTBYTYPE, ],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=24830,
  serialized_end=25448,
)


_TAXONOMYMETADATAREQUEST = _descriptor.Descriptor(
  name='TaxonomyMetadataRequest',
  full_name='ncbi.datasets.v2.TaxonomyMetadataRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='taxons', full_name='ncbi.datasets.v2.TaxonomyMetadataRequest.taxons', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031u\022\'\n\0049606\n\00510090\022\030NCBI Taxonomy Identifier\022\033\n\005human\n\005mouse\022\013Common Name\022-\n\014Homo sapiens\n\014Mus musculus\022\017Scientific Name', json_name='taxons', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='returned_content', full_name='ncbi.datasets.v2.TaxonomyMetadataRequest.returned_content', index=1,
      number=2, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='returnedContent', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _TAXONOMYMETADATAREQUEST_CONTENTTYPE,
  ],
  serialized_options=b'\302\360\031\030\n\026taxons:\n- 9606\n- mouse',
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=25451,
  serialized_end=25793,
)


_TAXONOMYMATCH = _descriptor.Descriptor(
  name='TaxonomyMatch',
  full_name='ncbi.datasets.v2.TaxonomyMatch',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='warnings', full_name='ncbi.datasets.v2.TaxonomyMatch.warnings', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='warnings', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='errors', full_name='ncbi.datasets.v2.TaxonomyMatch.errors', index=1,
      number=2, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='errors', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='query', full_name='ncbi.datasets.v2.TaxonomyMatch.query', index=2,
      number=3, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='query', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxonomy', full_name='ncbi.datasets.v2.TaxonomyMatch.taxonomy', index=3,
      number=4, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='taxonomy', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=25796,
  serialized_end=26013,
)


_TAXONOMYMETADATARESPONSE = _descriptor.Descriptor(
  name='TaxonomyMetadataResponse',
  full_name='ncbi.datasets.v2.TaxonomyMetadataResponse',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='messages', full_name='ncbi.datasets.v2.TaxonomyMetadataResponse.messages', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='messages', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxonomy_nodes', full_name='ncbi.datasets.v2.TaxonomyMetadataResponse.taxonomy_nodes', index=1,
      number=2, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='taxonomyNodes', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=26016,
  serialized_end=26177,
)


_TAXONOMYFILTEREDSUBTREEREQUEST = _descriptor.Descriptor(
  name='TaxonomyFilteredSubtreeRequest',
  full_name='ncbi.datasets.v2.TaxonomyFilteredSubtreeRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='taxons', full_name='ncbi.datasets.v2.TaxonomyFilteredSubtreeRequest.taxons', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031u\022\'\n\0049606\n\00510090\022\030NCBI Taxonomy Identifier\022\033\n\005human\n\005mouse\022\013Common Name\022-\n\014Homo sapiens\n\014Mus musculus\022\017Scientific Name', json_name='taxons', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='specified_limit', full_name='ncbi.datasets.v2.TaxonomyFilteredSubtreeRequest.specified_limit', index=1,
      number=2, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='specifiedLimit', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='rank_limits', full_name='ncbi.datasets.v2.TaxonomyFilteredSubtreeRequest.rank_limits', index=2,
      number=3, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='rankLimits', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=26180,
  serialized_end=26471,
)


_TAXONOMYFILTEREDSUBTREERESPONSE_EDGE = _descriptor.Descriptor(
  name='Edge',
  full_name='ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.Edge',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='visible_children', full_name='ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.Edge.visible_children', index=0,
      number=1, type=13, cpp_type=3, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='visibleChildren', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='children_status', full_name='ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.Edge.children_status', index=1,
      number=2, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='childrenStatus', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _TAXONOMYFILTEREDSUBTREERESPONSE_EDGE_CHILDSTATUS,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=26745,
  serialized_end=26983,
)

_TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY = _descriptor.Descriptor(
  name='EdgesEntry',
  full_name='ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.EdgesEntry',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='key', full_name='ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.EdgesEntry.key', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='key', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='value', full_name='ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.EdgesEntry.value', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='value', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=b'8\001',
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=26985,
  serialized_end=27097,
)

_TAXONOMYFILTEREDSUBTREERESPONSE = _descriptor.Descriptor(
  name='TaxonomyFilteredSubtreeResponse',
  full_name='ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='root_nodes', full_name='ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.root_nodes', index=0,
      number=1, type=13, cpp_type=3, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='rootNodes', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='edges', full_name='ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.edges', index=1,
      number=2, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='edges', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='warnings', full_name='ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.warnings', index=2,
      number=3, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='warnings', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='errors', full_name='ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.errors', index=3,
      number=4, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='errors', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_TAXONOMYFILTEREDSUBTREERESPONSE_EDGE, _TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY, ],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=26474,
  serialized_end=27097,
)


_DATASETREQUEST = _descriptor.Descriptor(
  name='DatasetRequest',
  full_name='ncbi.datasets.v2.DatasetRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='genome_v2', full_name='ncbi.datasets.v2.DatasetRequest.genome_v2', index=0,
      number=4, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='genomeV2', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='gene_v2', full_name='ncbi.datasets.v2.DatasetRequest.gene_v2', index=1,
      number=5, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='geneV2', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='virus_v2', full_name='ncbi.datasets.v2.DatasetRequest.virus_v2', index=2,
      number=6, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='virusV2', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
    _descriptor.OneofDescriptor(
      name='request', full_name='ncbi.datasets.v2.DatasetRequest.request',
      index=0, containing_type=None,
      create_key=_descriptor._internal_create_key,
    fields=[]),
  ],
  serialized_start=27100,
  serialized_end=27333,
)


_GENECOUNTSBYTAXONREQUEST = _descriptor.Descriptor(
  name='GeneCountsByTaxonRequest',
  full_name='ncbi.datasets.v2.GeneCountsByTaxonRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='taxon', full_name='ncbi.datasets.v2.GeneCountsByTaxonRequest.taxon', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031Y\022 \n\0049606\022\030NCBI Taxonomy Identifier\022\024\n\005human\022\013Common Name\022\037\n\014Homo sapiens\022\017Scientific Name', json_name='taxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=27336,
  serialized_end=27479,
)


_GENECOUNTSBYTAXONREPLY_GENETYPEANDCOUNT = _descriptor.Descriptor(
  name='GeneTypeAndCount',
  full_name='ncbi.datasets.v2.GeneCountsByTaxonReply.GeneTypeAndCount',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='gene_type', full_name='ncbi.datasets.v2.GeneCountsByTaxonReply.GeneTypeAndCount.gene_type', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='geneType', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='count', full_name='ncbi.datasets.v2.GeneCountsByTaxonReply.GeneTypeAndCount.count', index=1,
      number=2, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='count', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=27591,
  serialized_end=27660,
)

_GENECOUNTSBYTAXONREPLY = _descriptor.Descriptor(
  name='GeneCountsByTaxonReply',
  full_name='ncbi.datasets.v2.GeneCountsByTaxonReply',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='report', full_name='ncbi.datasets.v2.GeneCountsByTaxonReply.report', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='report', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_GENECOUNTSBYTAXONREPLY_GENETYPEANDCOUNT, ],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=27482,
  serialized_end=27660,
)


_GENOMEANNOTATIONREQUEST = _descriptor.Descriptor(
  name='GenomeAnnotationRequest',
  full_name='ncbi.datasets.v2.GenomeAnnotationRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accession', full_name='ncbi.datasets.v2.GenomeAnnotationRequest.accession', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='accession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='annotation_ids', full_name='ncbi.datasets.v2.GenomeAnnotationRequest.annotation_ids', index=1,
      number=13, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='annotationIds', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='symbols', full_name='ncbi.datasets.v2.GenomeAnnotationRequest.symbols', index=2,
      number=2, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='symbols', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='locations', full_name='ncbi.datasets.v2.GenomeAnnotationRequest.locations', index=3,
      number=15, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='locations', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='gene_types', full_name='ncbi.datasets.v2.GenomeAnnotationRequest.gene_types', index=4,
      number=14, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='geneTypes', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='search_text', full_name='ncbi.datasets.v2.GenomeAnnotationRequest.search_text', index=5,
      number=5, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='searchText', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='sort', full_name='ncbi.datasets.v2.GenomeAnnotationRequest.sort', index=6,
      number=6, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='sort', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='include_annotation_type', full_name='ncbi.datasets.v2.GenomeAnnotationRequest.include_annotation_type', index=7,
      number=4, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='includeAnnotationType', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_size', full_name='ncbi.datasets.v2.GenomeAnnotationRequest.page_size', index=8,
      number=8, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\004\n\00220', json_name='pageSize', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_fields', full_name='ncbi.datasets.v2.GenomeAnnotationRequest.table_fields', index=9,
      number=10, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='tableFields', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_format', full_name='ncbi.datasets.v2.GenomeAnnotationRequest.table_format', index=10,
      number=11, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='tableFormat', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='include_tabular_header', full_name='ncbi.datasets.v2.GenomeAnnotationRequest.include_tabular_header', index=11,
      number=12, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='includeTabularHeader', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_token', full_name='ncbi.datasets.v2.GenomeAnnotationRequest.page_token', index=12,
      number=9, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='pageToken', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _GENOMEANNOTATIONREQUEST_ANNOTATIONTYPE,
    _GENOMEANNOTATIONREQUEST_GENOMEANNOTATIONTABLEFORMAT,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=27663,
  serialized_end=28496,
)


_GENOMEANNOTATIONTABLESUMMARYREPLY = _descriptor.Descriptor(
  name='GenomeAnnotationTableSummaryReply',
  full_name='ncbi.datasets.v2.GenomeAnnotationTableSummaryReply',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accession', full_name='ncbi.datasets.v2.GenomeAnnotationTableSummaryReply.accession', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='accession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='chromosomes', full_name='ncbi.datasets.v2.GenomeAnnotationTableSummaryReply.chromosomes', index=1,
      number=2, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='chromosomes', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='gene_types', full_name='ncbi.datasets.v2.GenomeAnnotationTableSummaryReply.gene_types', index=2,
      number=3, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='geneTypes', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='empty_columns', full_name='ncbi.datasets.v2.GenomeAnnotationTableSummaryReply.empty_columns', index=3,
      number=4, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='emptyColumns', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=28499,
  serialized_end=28666,
)

_ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['include_tabular_header'].enum_type = _INCLUDETABULARHEADER
_SEQUENCEREPORTPAGE.fields_by_name['reports'].message_type = ncbi_dot_datasets_dot_v2_dot_reports_dot_assembly__sequence__info__pb2._SEQUENCEINFO
_SORTFIELD.fields_by_name['direction'].enum_type = _SORTDIRECTION
_ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['filters'].message_type = _ASSEMBLYDATASETDESCRIPTORSFILTER
_ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['returned_content'].enum_type = _ASSEMBLYDATASETREPORTSREQUEST_CONTENTTYPE
_ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['sort'].message_type = _SORTFIELD
_ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['include_tabular_header'].enum_type = _INCLUDETABULARHEADER
_ASSEMBLYDATASETREPORTSREQUEST_CONTENTTYPE.containing_type = _ASSEMBLYDATASETREPORTSREQUEST
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['assembly_source'].enum_type = _ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYSOURCE
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['assembly_version'].enum_type = _ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYVERSION
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['assembly_level'].enum_type = ncbi_dot_datasets_dot_v2_dot_reports_dot_common__pb2._ASSEMBLYLEVEL
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['first_release_date'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['last_release_date'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['is_metagenome_derived'].enum_type = _ASSEMBLYDATASETDESCRIPTORSFILTER_METAGENOMEDERIVEDFILTER
_ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYSOURCE.containing_type = _ASSEMBLYDATASETDESCRIPTORSFILTER
_ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYVERSION.containing_type = _ASSEMBLYDATASETDESCRIPTORSFILTER
_ASSEMBLYDATASETDESCRIPTORSFILTER_METAGENOMEDERIVEDFILTER.containing_type = _ASSEMBLYDATASETDESCRIPTORSFILTER
_ASSEMBLYREVISIONHISTORY.fields_by_name['assembly_revisions'].message_type = ncbi_dot_datasets_dot_v2_dot_reports_dot_assembly__revision__pb2._ASSEMBLYREVISION
_ASSEMBLYCHECKMHISTOGRAMREPLY_HISTOGRAMINTERVAL.containing_type = _ASSEMBLYCHECKMHISTOGRAMREPLY
_ASSEMBLYCHECKMHISTOGRAMREPLY.fields_by_name['histogram_intervals'].message_type = _ASSEMBLYCHECKMHISTOGRAMREPLY_HISTOGRAMINTERVAL
_BIOPROJECTLINEAGE.fields_by_name['bioprojects'].message_type = _BIOPROJECT
_ANNOTATIONFORASSEMBLY_FILE.fields_by_name['type'].enum_type = _ANNOTATIONFORASSEMBLY_TYPE
_ANNOTATIONFORASSEMBLY_FILE.containing_type = _ANNOTATIONFORASSEMBLY
_ANNOTATIONFORASSEMBLY.fields_by_name['file'].message_type = _ANNOTATIONFORASSEMBLY_FILE
_ANNOTATIONFORASSEMBLY.fields_by_name['stats'].message_type = _FEATURECOUNTS
_ANNOTATIONFORASSEMBLY.fields_by_name['busco'].message_type = _BUSCOSTAT
_ANNOTATIONFORASSEMBLY_TYPE.containing_type = _ANNOTATIONFORASSEMBLY
_FEATURECOUNTS.fields_by_name['gene_counts'].message_type = _GENECOUNTS
_ASSEMBLYDATASETREQUEST.fields_by_name['include_annotation_type'].enum_type = _ANNOTATIONFORASSEMBLY_TYPE
_ASSEMBLYDATASETREQUEST.fields_by_name['hydrated'].enum_type = _ASSEMBLYDATASETREQUEST_RESOLUTION
_ASSEMBLYDATASETREQUEST_RESOLUTION.containing_type = _ASSEMBLYDATASETREQUEST
_ORGANISM_COUNTS.containing_type = _ORGANISM
_ORGANISM_COUNTBYTYPE.fields_by_name['type'].enum_type = _COUNTTYPE
_ORGANISM_COUNTBYTYPE.fields_by_name['counts'].message_type = _ORGANISM_COUNTS
_ORGANISM_COUNTBYTYPE.containing_type = _ORGANISM
_ORGANISM.fields_by_name['rank'].enum_type = _ORGANISM_RANKTYPE
_ORGANISM.fields_by_name['assembly_counts'].message_type = _ORGANISM_COUNTS
_ORGANISM.fields_by_name['counts'].message_type = _ORGANISM_COUNTBYTYPE
_ORGANISM.fields_by_name['children'].message_type = _ORGANISM
_ORGANISM.fields_by_name['merged'].message_type = _ORGANISM
_ORGANISM_RANKTYPE.containing_type = _ORGANISM
_TAXTREEREQUEST.oneofs_by_name['taxonomy'].fields.append(
  _TAXTREEREQUEST.fields_by_name['tax_id'])
_TAXTREEREQUEST.fields_by_name['tax_id'].containing_oneof = _TAXTREEREQUEST.oneofs_by_name['taxonomy']
_TAXTREEREQUEST.oneofs_by_name['taxonomy'].fields.append(
  _TAXTREEREQUEST.fields_by_name['tax_token'])
_TAXTREEREQUEST.fields_by_name['tax_token'].containing_oneof = _TAXTREEREQUEST.oneofs_by_name['taxonomy']
_TAXTREEREQUEST.oneofs_by_name['taxonomy'].fields.append(
  _TAXTREEREQUEST.fields_by_name['taxon'])
_TAXTREEREQUEST.fields_by_name['taxon'].containing_oneof = _TAXTREEREQUEST.oneofs_by_name['taxonomy']
_ORGANISMQUERYREQUEST.fields_by_name['tax_rank_filter'].enum_type = _ORGANISMQUERYREQUEST_TAXRANKFILTER
_ORGANISMQUERYREQUEST.fields_by_name['taxon_resource_filter'].enum_type = _ORGANISMQUERYREQUEST_TAXONRESOURCEFILTER
_ORGANISMQUERYREQUEST_TAXRANKFILTER.containing_type = _ORGANISMQUERYREQUEST
_ORGANISMQUERYREQUEST_TAXONRESOURCEFILTER.containing_type = _ORGANISMQUERYREQUEST
_ORGANISMQUERYREQUEST.oneofs_by_name['id'].fields.append(
  _ORGANISMQUERYREQUEST.fields_by_name['organism_query'])
_ORGANISMQUERYREQUEST.fields_by_name['organism_query'].containing_oneof = _ORGANISMQUERYREQUEST.oneofs_by_name['id']
_ORGANISMQUERYREQUEST.oneofs_by_name['id'].fields.append(
  _ORGANISMQUERYREQUEST.fields_by_name['taxon_query'])
_ORGANISMQUERYREQUEST.fields_by_name['taxon_query'].containing_oneof = _ORGANISMQUERYREQUEST.oneofs_by_name['id']
_SCINAMEANDIDS_SCINAMEANDID.containing_type = _SCINAMEANDIDS
_SCINAMEANDIDS.fields_by_name['sci_name_and_ids'].message_type = _SCINAMEANDIDS_SCINAMEANDID
_ORTHOLOGREQUEST.fields_by_name['returned_content'].enum_type = _ORTHOLOGREQUEST_CONTENTTYPE
_ORTHOLOGREQUEST_CONTENTTYPE.containing_type = _ORTHOLOGREQUEST
_GENEDATASETREQUEST.fields_by_name['include_annotation_type'].enum_type = _FASTA
_GENEDATASETREQUEST.fields_by_name['returned_content'].enum_type = _GENEDATASETREQUEST_CONTENTTYPE
_GENEDATASETREQUEST.fields_by_name['aux_report'].enum_type = _GENEDATASETREQUEST_GENEDATASETREPORTTYPE
_GENEDATASETREQUEST.fields_by_name['table_report_type'].enum_type = _GENEDATASETREQUEST_GENEDATASETREPORTTYPE
_GENEDATASETREQUEST_CONTENTTYPE.containing_type = _GENEDATASETREQUEST
_GENEDATASETREQUEST_GENEDATASETREPORTTYPE.containing_type = _GENEDATASETREQUEST
_GENEDATASETREPORTSREQUEST_SYMBOLSFORTAXON.containing_type = _GENEDATASETREPORTSREQUEST
_GENEDATASETREPORTSREQUEST.fields_by_name['returned_content'].enum_type = _GENEDATASETREPORTSREQUEST_CONTENTTYPE
_GENEDATASETREPORTSREQUEST.fields_by_name['symbols_for_taxon'].message_type = _GENEDATASETREPORTSREQUEST_SYMBOLSFORTAXON
_GENEDATASETREPORTSREQUEST.fields_by_name['types'].enum_type = _GENETYPE
_GENEDATASETREPORTSREQUEST_CONTENTTYPE.containing_type = _GENEDATASETREPORTSREQUEST
_GENELINKSREPLY_GENELINK.fields_by_name['gene_link_type'].enum_type = _GENELINKSREPLY_GENELINKTYPE
_GENELINKSREPLY_GENELINK.containing_type = _GENELINKSREPLY
_GENELINKSREPLY.fields_by_name['gene_links'].message_type = _GENELINKSREPLY_GENELINK
_GENELINKSREPLY_GENELINKTYPE.containing_type = _GENELINKSREPLY
_GENECHROMOSOMESUMMARYREPLY_GENECHROMOSOMESUMMARY.containing_type = _GENECHROMOSOMESUMMARYREPLY
_GENECHROMOSOMESUMMARYREPLY.fields_by_name['gene_chromosome_summaries'].message_type = _GENECHROMOSOMESUMMARYREPLY_GENECHROMOSOMESUMMARY
_ASSEMBLYLINKSREPLY_ASSEMBLYLINK.fields_by_name['assembly_link_type'].enum_type = _ASSEMBLYLINKSREPLY_ASSEMBLYLINKTYPE
_ASSEMBLYLINKSREPLY_ASSEMBLYLINK.containing_type = _ASSEMBLYLINKSREPLY
_ASSEMBLYLINKSREPLY.fields_by_name['assembly_links'].message_type = _ASSEMBLYLINKSREPLY_ASSEMBLYLINK
_ASSEMBLYLINKSREPLY_ASSEMBLYLINKTYPE.containing_type = _ASSEMBLYLINKSREPLY
_VIRUSDATASETREQUEST.fields_by_name['released_since'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_VIRUSDATASETREQUEST.fields_by_name['updated_since'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_VIRUSDATASETREQUEST.fields_by_name['table_fields'].enum_type = _VIRUSTABLEFIELD
_VIRUSDATASETREQUEST.fields_by_name['include_sequence'].enum_type = _VIRALSEQUENCETYPE
_VIRUSDATASETREQUEST.fields_by_name['aux_report'].enum_type = _VIRUSDATASETREPORTTYPE
_VIRUSDATASETREQUEST.fields_by_name['format'].enum_type = _TABLEFORMAT
_VIRUSDATASETFILTER.fields_by_name['released_since'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_VIRUSDATASETFILTER.fields_by_name['updated_since'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_VIRUSDATAREPORTREQUEST.fields_by_name['filter'].message_type = _VIRUSDATASETFILTER
_VIRUSDATAREPORTREQUEST.fields_by_name['returned_content'].enum_type = _VIRUSDATAREPORTREQUEST_CONTENTTYPE
_VIRUSDATAREPORTREQUEST_CONTENTTYPE.containing_type = _VIRUSDATAREPORTREQUEST
_VIRUSANNOTATIONFILTER.fields_by_name['released_since'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_VIRUSANNOTATIONFILTER.fields_by_name['updated_since'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_VIRUSANNOTATIONREPORTREQUEST.fields_by_name['filter'].message_type = _VIRUSANNOTATIONFILTER
_SARS2PROTEINDATASETREQUEST.fields_by_name['released_since'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_SARS2PROTEINDATASETREQUEST.fields_by_name['updated_since'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_SARS2PROTEINDATASETREQUEST.fields_by_name['table_fields'].enum_type = _VIRUSTABLEFIELD
_SARS2PROTEINDATASETREQUEST.fields_by_name['include_sequence'].enum_type = _VIRALSEQUENCETYPE
_SARS2PROTEINDATASETREQUEST.fields_by_name['aux_report'].enum_type = _VIRUSDATASETREPORTTYPE
_SARS2PROTEINDATASETREQUEST.fields_by_name['format'].enum_type = _TABLEFORMAT
_DOWNLOADSUMMARY_HYDRATED.containing_type = _DOWNLOADSUMMARY
_DOWNLOADSUMMARY_DEHYDRATED.containing_type = _DOWNLOADSUMMARY
_DOWNLOADSUMMARY_FILESUMMARY.containing_type = _DOWNLOADSUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.fields_by_name['all_genomic_fasta'].message_type = _DOWNLOADSUMMARY_FILESUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.fields_by_name['genome_gff'].message_type = _DOWNLOADSUMMARY_FILESUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.fields_by_name['genome_gbff'].message_type = _DOWNLOADSUMMARY_FILESUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.fields_by_name['rna_fasta'].message_type = _DOWNLOADSUMMARY_FILESUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.fields_by_name['prot_fasta'].message_type = _DOWNLOADSUMMARY_FILESUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.fields_by_name['genome_gtf'].message_type = _DOWNLOADSUMMARY_FILESUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.fields_by_name['cds_fasta'].message_type = _DOWNLOADSUMMARY_FILESUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.fields_by_name['sequence_report'].message_type = _DOWNLOADSUMMARY_FILESUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.fields_by_name['annotation_report'].message_type = _DOWNLOADSUMMARY_FILESUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.containing_type = _DOWNLOADSUMMARY
_DOWNLOADSUMMARY.fields_by_name['resource_updated_on'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_DOWNLOADSUMMARY.fields_by_name['hydrated'].message_type = _DOWNLOADSUMMARY_HYDRATED
_DOWNLOADSUMMARY.fields_by_name['dehydrated'].message_type = _DOWNLOADSUMMARY_DEHYDRATED
_DOWNLOADSUMMARY.fields_by_name['errors'].message_type = ncbi_dot_datasets_dot_v2_dot_reports_dot_common__pb2._ERROR
_DOWNLOADSUMMARY.fields_by_name['messages'].message_type = ncbi_dot_datasets_dot_v2_dot_reports_dot_common__pb2._MESSAGE
_DOWNLOADSUMMARY.fields_by_name['available_files'].message_type = _DOWNLOADSUMMARY_AVAILABLEFILES
_MICROBIGGEDATASETREQUEST_ELEMENTFLANKCONFIG.containing_type = _MICROBIGGEDATASETREQUEST
_MICROBIGGEDATASETREQUEST.fields_by_name['files'].enum_type = _MICROBIGGEDATASETREQUEST_FILETYPE
_MICROBIGGEDATASETREQUEST.fields_by_name['element_flank_config'].message_type = _MICROBIGGEDATASETREQUEST_ELEMENTFLANKCONFIG
_MICROBIGGEDATASETREQUEST_FILETYPE.containing_type = _MICROBIGGEDATASETREQUEST
_MICROBIGGEDATASETREQUEST.oneofs_by_name['query'].fields.append(
  _MICROBIGGEDATASETREQUEST.fields_by_name['opaque_solr_query'])
_MICROBIGGEDATASETREQUEST.fields_by_name['opaque_solr_query'].containing_oneof = _MICROBIGGEDATASETREQUEST.oneofs_by_name['query']
_PROKARYOTEGENEREQUEST_GENEFLANKCONFIG.containing_type = _PROKARYOTEGENEREQUEST
_PROKARYOTEGENEREQUEST.fields_by_name['include_annotation_type'].enum_type = _FASTA
_PROKARYOTEGENEREQUEST.fields_by_name['gene_flank_config'].message_type = _PROKARYOTEGENEREQUEST_GENEFLANKCONFIG
_TAXONOMYNODE_COUNTBYTYPE.fields_by_name['type'].enum_type = _COUNTTYPE
_TAXONOMYNODE_COUNTBYTYPE.containing_type = _TAXONOMYNODE
_TAXONOMYNODE.fields_by_name['rank'].enum_type = _ORGANISM_RANKTYPE
_TAXONOMYNODE.fields_by_name['counts'].message_type = _TAXONOMYNODE_COUNTBYTYPE
_TAXONOMYMETADATAREQUEST.fields_by_name['returned_content'].enum_type = _TAXONOMYMETADATAREQUEST_CONTENTTYPE
_TAXONOMYMETADATAREQUEST_CONTENTTYPE.containing_type = _TAXONOMYMETADATAREQUEST
_TAXONOMYMATCH.fields_by_name['warnings'].message_type = ncbi_dot_datasets_dot_v2_dot_reports_dot_common__pb2._WARNING
_TAXONOMYMATCH.fields_by_name['errors'].message_type = ncbi_dot_datasets_dot_v2_dot_reports_dot_common__pb2._ERROR
_TAXONOMYMATCH.fields_by_name['taxonomy'].message_type = _TAXONOMYNODE
_TAXONOMYMETADATARESPONSE.fields_by_name['messages'].message_type = ncbi_dot_datasets_dot_v2_dot_reports_dot_common__pb2._MESSAGE
_TAXONOMYMETADATARESPONSE.fields_by_name['taxonomy_nodes'].message_type = _TAXONOMYMATCH
_TAXONOMYFILTEREDSUBTREEREQUEST.fields_by_name['rank_limits'].enum_type = _ORGANISM_RANKTYPE
_TAXONOMYFILTEREDSUBTREERESPONSE_EDGE.fields_by_name['children_status'].enum_type = _TAXONOMYFILTEREDSUBTREERESPONSE_EDGE_CHILDSTATUS
_TAXONOMYFILTEREDSUBTREERESPONSE_EDGE.containing_type = _TAXONOMYFILTEREDSUBTREERESPONSE
_TAXONOMYFILTEREDSUBTREERESPONSE_EDGE_CHILDSTATUS.containing_type = _TAXONOMYFILTEREDSUBTREERESPONSE_EDGE
_TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY.fields_by_name['value'].message_type = _TAXONOMYFILTEREDSUBTREERESPONSE_EDGE
_TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY.containing_type = _TAXONOMYFILTEREDSUBTREERESPONSE
_TAXONOMYFILTEREDSUBTREERESPONSE.fields_by_name['edges'].message_type = _TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY
_TAXONOMYFILTEREDSUBTREERESPONSE.fields_by_name['warnings'].message_type = ncbi_dot_datasets_dot_v2_dot_reports_dot_common__pb2._WARNING
_TAXONOMYFILTEREDSUBTREERESPONSE.fields_by_name['errors'].message_type = ncbi_dot_datasets_dot_v2_dot_reports_dot_common__pb2._ERROR
_DATASETREQUEST.fields_by_name['genome_v2'].message_type = _ASSEMBLYDATASETREQUEST
_DATASETREQUEST.fields_by_name['gene_v2'].message_type = _GENEDATASETREQUEST
_DATASETREQUEST.fields_by_name['virus_v2'].message_type = _VIRUSDATASETREQUEST
_DATASETREQUEST.oneofs_by_name['request'].fields.append(
  _DATASETREQUEST.fields_by_name['genome_v2'])
_DATASETREQUEST.fields_by_name['genome_v2'].containing_oneof = _DATASETREQUEST.oneofs_by_name['request']
_DATASETREQUEST.oneofs_by_name['request'].fields.append(
  _DATASETREQUEST.fields_by_name['gene_v2'])
_DATASETREQUEST.fields_by_name['gene_v2'].containing_oneof = _DATASETREQUEST.oneofs_by_name['request']
_DATASETREQUEST.oneofs_by_name['request'].fields.append(
  _DATASETREQUEST.fields_by_name['virus_v2'])
_DATASETREQUEST.fields_by_name['virus_v2'].containing_oneof = _DATASETREQUEST.oneofs_by_name['request']
_GENECOUNTSBYTAXONREPLY_GENETYPEANDCOUNT.containing_type = _GENECOUNTSBYTAXONREPLY
_GENECOUNTSBYTAXONREPLY.fields_by_name['report'].message_type = _GENECOUNTSBYTAXONREPLY_GENETYPEANDCOUNT
_GENOMEANNOTATIONREQUEST.fields_by_name['sort'].message_type = _SORTFIELD
_GENOMEANNOTATIONREQUEST.fields_by_name['include_annotation_type'].enum_type = _GENOMEANNOTATIONREQUEST_ANNOTATIONTYPE
_GENOMEANNOTATIONREQUEST.fields_by_name['table_format'].enum_type = _GENOMEANNOTATIONREQUEST_GENOMEANNOTATIONTABLEFORMAT
_GENOMEANNOTATIONREQUEST.fields_by_name['include_tabular_header'].enum_type = _INCLUDETABULARHEADER
_GENOMEANNOTATIONREQUEST_ANNOTATIONTYPE.containing_type = _GENOMEANNOTATIONREQUEST
_GENOMEANNOTATIONREQUEST_GENOMEANNOTATIONTABLEFORMAT.containing_type = _GENOMEANNOTATIONREQUEST
DESCRIPTOR.message_types_by_name['MethodPayloadRequest'] = _METHODPAYLOADREQUEST
DESCRIPTOR.message_types_by_name['AssemblySequenceReportsRequest'] = _ASSEMBLYSEQUENCEREPORTSREQUEST
DESCRIPTOR.message_types_by_name['SequenceReportPage'] = _SEQUENCEREPORTPAGE
DESCRIPTOR.message_types_by_name['SortField'] = _SORTFIELD
DESCRIPTOR.message_types_by_name['AssemblyDatasetReportsRequest'] = _ASSEMBLYDATASETREPORTSREQUEST
DESCRIPTOR.message_types_by_name['AssemblyDatasetDescriptorsFilter'] = _ASSEMBLYDATASETDESCRIPTORSFILTER
DESCRIPTOR.message_types_by_name['Accessions'] = _ACCESSIONS
DESCRIPTOR.message_types_by_name['AssemblyRevisionHistory'] = _ASSEMBLYREVISIONHISTORY
DESCRIPTOR.message_types_by_name['AssemblyRevisionHistoryRequest'] = _ASSEMBLYREVISIONHISTORYREQUEST
DESCRIPTOR.message_types_by_name['AssemblyCheckMHistogramRequest'] = _ASSEMBLYCHECKMHISTOGRAMREQUEST
DESCRIPTOR.message_types_by_name['AssemblyCheckMHistogramReply'] = _ASSEMBLYCHECKMHISTOGRAMREPLY
DESCRIPTOR.message_types_by_name['BioProject'] = _BIOPROJECT
DESCRIPTOR.message_types_by_name['BioProjectLineage'] = _BIOPROJECTLINEAGE
DESCRIPTOR.message_types_by_name['BuscoStat'] = _BUSCOSTAT
DESCRIPTOR.message_types_by_name['AnnotationForAssembly'] = _ANNOTATIONFORASSEMBLY
DESCRIPTOR.message_types_by_name['FeatureCounts'] = _FEATURECOUNTS
DESCRIPTOR.message_types_by_name['GeneCounts'] = _GENECOUNTS
DESCRIPTOR.message_types_by_name['AssemblyDatasetRequest'] = _ASSEMBLYDATASETREQUEST
DESCRIPTOR.message_types_by_name['AssemblyDatasetAvailability'] = _ASSEMBLYDATASETAVAILABILITY
DESCRIPTOR.message_types_by_name['Organism'] = _ORGANISM
DESCRIPTOR.message_types_by_name['TaxTreeRequest'] = _TAXTREEREQUEST
DESCRIPTOR.message_types_by_name['OrganismQueryRequest'] = _ORGANISMQUERYREQUEST
DESCRIPTOR.message_types_by_name['SciNameAndIds'] = _SCINAMEANDIDS
DESCRIPTOR.message_types_by_name['OrthologRequest'] = _ORTHOLOGREQUEST
DESCRIPTOR.message_types_by_name['GeneDatasetRequest'] = _GENEDATASETREQUEST
DESCRIPTOR.message_types_by_name['GeneDatasetReportsRequest'] = _GENEDATASETREPORTSREQUEST
DESCRIPTOR.message_types_by_name['GeneLinksRequest'] = _GENELINKSREQUEST
DESCRIPTOR.message_types_by_name['GeneLinksReply'] = _GENELINKSREPLY
DESCRIPTOR.message_types_by_name['GeneChromosomeSummaryRequest'] = _GENECHROMOSOMESUMMARYREQUEST
DESCRIPTOR.message_types_by_name['GeneChromosomeSummaryReply'] = _GENECHROMOSOMESUMMARYREPLY
DESCRIPTOR.message_types_by_name['AssemblyLinksRequest'] = _ASSEMBLYLINKSREQUEST
DESCRIPTOR.message_types_by_name['AssemblyLinksReply'] = _ASSEMBLYLINKSREPLY
DESCRIPTOR.message_types_by_name['VirusDatasetRequest'] = _VIRUSDATASETREQUEST
DESCRIPTOR.message_types_by_name['VirusDatasetFilter'] = _VIRUSDATASETFILTER
DESCRIPTOR.message_types_by_name['VirusDataReportRequest'] = _VIRUSDATAREPORTREQUEST
DESCRIPTOR.message_types_by_name['VirusAnnotationFilter'] = _VIRUSANNOTATIONFILTER
DESCRIPTOR.message_types_by_name['VirusAnnotationReportRequest'] = _VIRUSANNOTATIONREPORTREQUEST
DESCRIPTOR.message_types_by_name['VirusAvailabilityRequest'] = _VIRUSAVAILABILITYREQUEST
DESCRIPTOR.message_types_by_name['VirusAvailability'] = _VIRUSAVAILABILITY
DESCRIPTOR.message_types_by_name['Sars2ProteinDatasetRequest'] = _SARS2PROTEINDATASETREQUEST
DESCRIPTOR.message_types_by_name['DownloadSummary'] = _DOWNLOADSUMMARY
DESCRIPTOR.message_types_by_name['TabularOutput'] = _TABULAROUTPUT
DESCRIPTOR.message_types_by_name['MicroBiggeDatasetRequest'] = _MICROBIGGEDATASETREQUEST
DESCRIPTOR.message_types_by_name['VersionReply'] = _VERSIONREPLY
DESCRIPTOR.message_types_by_name['ProkaryoteGeneRequest'] = _PROKARYOTEGENEREQUEST
DESCRIPTOR.message_types_by_name['SleepRequest'] = _SLEEPREQUEST
DESCRIPTOR.message_types_by_name['SleepReply'] = _SLEEPREPLY
DESCRIPTOR.message_types_by_name['TaxonomyNode'] = _TAXONOMYNODE
DESCRIPTOR.message_types_by_name['TaxonomyMetadataRequest'] = _TAXONOMYMETADATAREQUEST
DESCRIPTOR.message_types_by_name['TaxonomyMatch'] = _TAXONOMYMATCH
DESCRIPTOR.message_types_by_name['TaxonomyMetadataResponse'] = _TAXONOMYMETADATARESPONSE
DESCRIPTOR.message_types_by_name['TaxonomyFilteredSubtreeRequest'] = _TAXONOMYFILTEREDSUBTREEREQUEST
DESCRIPTOR.message_types_by_name['TaxonomyFilteredSubtreeResponse'] = _TAXONOMYFILTEREDSUBTREERESPONSE
DESCRIPTOR.message_types_by_name['DatasetRequest'] = _DATASETREQUEST
DESCRIPTOR.message_types_by_name['GeneCountsByTaxonRequest'] = _GENECOUNTSBYTAXONREQUEST
DESCRIPTOR.message_types_by_name['GeneCountsByTaxonReply'] = _GENECOUNTSBYTAXONREPLY
DESCRIPTOR.message_types_by_name['GenomeAnnotationRequest'] = _GENOMEANNOTATIONREQUEST
DESCRIPTOR.message_types_by_name['GenomeAnnotationTableSummaryReply'] = _GENOMEANNOTATIONTABLESUMMARYREPLY
DESCRIPTOR.enum_types_by_name['IncludeTabularHeader'] = _INCLUDETABULARHEADER
DESCRIPTOR.enum_types_by_name['SortDirection'] = _SORTDIRECTION
DESCRIPTOR.enum_types_by_name['CountType'] = _COUNTTYPE
DESCRIPTOR.enum_types_by_name['Fasta'] = _FASTA
DESCRIPTOR.enum_types_by_name['GeneType'] = _GENETYPE
DESCRIPTOR.enum_types_by_name['VirusTableField'] = _VIRUSTABLEFIELD
DESCRIPTOR.enum_types_by_name['TableFormat'] = _TABLEFORMAT
DESCRIPTOR.enum_types_by_name['ViralSequenceType'] = _VIRALSEQUENCETYPE
DESCRIPTOR.enum_types_by_name['VirusDatasetReportType'] = _VIRUSDATASETREPORTTYPE
_sym_db.RegisterFileDescriptor(DESCRIPTOR)

MethodPayloadRequest = _reflection.GeneratedProtocolMessageType('MethodPayloadRequest', (_message.Message,), {
  'DESCRIPTOR' : _METHODPAYLOADREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.MethodPayloadRequest)
  })
_sym_db.RegisterMessage(MethodPayloadRequest)

AssemblySequenceReportsRequest = _reflection.GeneratedProtocolMessageType('AssemblySequenceReportsRequest', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYSEQUENCEREPORTSREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblySequenceReportsRequest)
  })
_sym_db.RegisterMessage(AssemblySequenceReportsRequest)

SequenceReportPage = _reflection.GeneratedProtocolMessageType('SequenceReportPage', (_message.Message,), {
  'DESCRIPTOR' : _SEQUENCEREPORTPAGE,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.SequenceReportPage)
  })
_sym_db.RegisterMessage(SequenceReportPage)

SortField = _reflection.GeneratedProtocolMessageType('SortField', (_message.Message,), {
  'DESCRIPTOR' : _SORTFIELD,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.SortField)
  })
_sym_db.RegisterMessage(SortField)

AssemblyDatasetReportsRequest = _reflection.GeneratedProtocolMessageType('AssemblyDatasetReportsRequest', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYDATASETREPORTSREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyDatasetReportsRequest)
  })
_sym_db.RegisterMessage(AssemblyDatasetReportsRequest)

AssemblyDatasetDescriptorsFilter = _reflection.GeneratedProtocolMessageType('AssemblyDatasetDescriptorsFilter', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYDATASETDESCRIPTORSFILTER,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter)
  })
_sym_db.RegisterMessage(AssemblyDatasetDescriptorsFilter)

Accessions = _reflection.GeneratedProtocolMessageType('Accessions', (_message.Message,), {
  'DESCRIPTOR' : _ACCESSIONS,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.Accessions)
  })
_sym_db.RegisterMessage(Accessions)

AssemblyRevisionHistory = _reflection.GeneratedProtocolMessageType('AssemblyRevisionHistory', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYREVISIONHISTORY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyRevisionHistory)
  })
_sym_db.RegisterMessage(AssemblyRevisionHistory)

AssemblyRevisionHistoryRequest = _reflection.GeneratedProtocolMessageType('AssemblyRevisionHistoryRequest', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYREVISIONHISTORYREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyRevisionHistoryRequest)
  })
_sym_db.RegisterMessage(AssemblyRevisionHistoryRequest)

AssemblyCheckMHistogramRequest = _reflection.GeneratedProtocolMessageType('AssemblyCheckMHistogramRequest', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYCHECKMHISTOGRAMREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyCheckMHistogramRequest)
  })
_sym_db.RegisterMessage(AssemblyCheckMHistogramRequest)

AssemblyCheckMHistogramReply = _reflection.GeneratedProtocolMessageType('AssemblyCheckMHistogramReply', (_message.Message,), {

  'HistogramInterval' : _reflection.GeneratedProtocolMessageType('HistogramInterval', (_message.Message,), {
    'DESCRIPTOR' : _ASSEMBLYCHECKMHISTOGRAMREPLY_HISTOGRAMINTERVAL,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyCheckMHistogramReply.HistogramInterval)
    })
  ,
  'DESCRIPTOR' : _ASSEMBLYCHECKMHISTOGRAMREPLY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyCheckMHistogramReply)
  })
_sym_db.RegisterMessage(AssemblyCheckMHistogramReply)
_sym_db.RegisterMessage(AssemblyCheckMHistogramReply.HistogramInterval)

BioProject = _reflection.GeneratedProtocolMessageType('BioProject', (_message.Message,), {
  'DESCRIPTOR' : _BIOPROJECT,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.BioProject)
  })
_sym_db.RegisterMessage(BioProject)

BioProjectLineage = _reflection.GeneratedProtocolMessageType('BioProjectLineage', (_message.Message,), {
  'DESCRIPTOR' : _BIOPROJECTLINEAGE,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.BioProjectLineage)
  })
_sym_db.RegisterMessage(BioProjectLineage)

BuscoStat = _reflection.GeneratedProtocolMessageType('BuscoStat', (_message.Message,), {
  'DESCRIPTOR' : _BUSCOSTAT,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.BuscoStat)
  })
_sym_db.RegisterMessage(BuscoStat)

AnnotationForAssembly = _reflection.GeneratedProtocolMessageType('AnnotationForAssembly', (_message.Message,), {

  'File' : _reflection.GeneratedProtocolMessageType('File', (_message.Message,), {
    'DESCRIPTOR' : _ANNOTATIONFORASSEMBLY_FILE,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AnnotationForAssembly.File)
    })
  ,
  'DESCRIPTOR' : _ANNOTATIONFORASSEMBLY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AnnotationForAssembly)
  })
_sym_db.RegisterMessage(AnnotationForAssembly)
_sym_db.RegisterMessage(AnnotationForAssembly.File)

FeatureCounts = _reflection.GeneratedProtocolMessageType('FeatureCounts', (_message.Message,), {
  'DESCRIPTOR' : _FEATURECOUNTS,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.FeatureCounts)
  })
_sym_db.RegisterMessage(FeatureCounts)

GeneCounts = _reflection.GeneratedProtocolMessageType('GeneCounts', (_message.Message,), {
  'DESCRIPTOR' : _GENECOUNTS,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneCounts)
  })
_sym_db.RegisterMessage(GeneCounts)

AssemblyDatasetRequest = _reflection.GeneratedProtocolMessageType('AssemblyDatasetRequest', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYDATASETREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyDatasetRequest)
  })
_sym_db.RegisterMessage(AssemblyDatasetRequest)

AssemblyDatasetAvailability = _reflection.GeneratedProtocolMessageType('AssemblyDatasetAvailability', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYDATASETAVAILABILITY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyDatasetAvailability)
  })
_sym_db.RegisterMessage(AssemblyDatasetAvailability)

Organism = _reflection.GeneratedProtocolMessageType('Organism', (_message.Message,), {

  'Counts' : _reflection.GeneratedProtocolMessageType('Counts', (_message.Message,), {
    'DESCRIPTOR' : _ORGANISM_COUNTS,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.Organism.Counts)
    })
  ,

  'CountByType' : _reflection.GeneratedProtocolMessageType('CountByType', (_message.Message,), {
    'DESCRIPTOR' : _ORGANISM_COUNTBYTYPE,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.Organism.CountByType)
    })
  ,
  'DESCRIPTOR' : _ORGANISM,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.Organism)
  })
_sym_db.RegisterMessage(Organism)
_sym_db.RegisterMessage(Organism.Counts)
_sym_db.RegisterMessage(Organism.CountByType)

TaxTreeRequest = _reflection.GeneratedProtocolMessageType('TaxTreeRequest', (_message.Message,), {
  'DESCRIPTOR' : _TAXTREEREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxTreeRequest)
  })
_sym_db.RegisterMessage(TaxTreeRequest)

OrganismQueryRequest = _reflection.GeneratedProtocolMessageType('OrganismQueryRequest', (_message.Message,), {
  'DESCRIPTOR' : _ORGANISMQUERYREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.OrganismQueryRequest)
  })
_sym_db.RegisterMessage(OrganismQueryRequest)

SciNameAndIds = _reflection.GeneratedProtocolMessageType('SciNameAndIds', (_message.Message,), {

  'SciNameAndId' : _reflection.GeneratedProtocolMessageType('SciNameAndId', (_message.Message,), {
    'DESCRIPTOR' : _SCINAMEANDIDS_SCINAMEANDID,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.SciNameAndIds.SciNameAndId)
    })
  ,
  'DESCRIPTOR' : _SCINAMEANDIDS,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.SciNameAndIds)
  })
_sym_db.RegisterMessage(SciNameAndIds)
_sym_db.RegisterMessage(SciNameAndIds.SciNameAndId)

OrthologRequest = _reflection.GeneratedProtocolMessageType('OrthologRequest', (_message.Message,), {
  'DESCRIPTOR' : _ORTHOLOGREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.OrthologRequest)
  })
_sym_db.RegisterMessage(OrthologRequest)

GeneDatasetRequest = _reflection.GeneratedProtocolMessageType('GeneDatasetRequest', (_message.Message,), {
  'DESCRIPTOR' : _GENEDATASETREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneDatasetRequest)
  })
_sym_db.RegisterMessage(GeneDatasetRequest)

GeneDatasetReportsRequest = _reflection.GeneratedProtocolMessageType('GeneDatasetReportsRequest', (_message.Message,), {

  'SymbolsForTaxon' : _reflection.GeneratedProtocolMessageType('SymbolsForTaxon', (_message.Message,), {
    'DESCRIPTOR' : _GENEDATASETREPORTSREQUEST_SYMBOLSFORTAXON,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneDatasetReportsRequest.SymbolsForTaxon)
    })
  ,
  'DESCRIPTOR' : _GENEDATASETREPORTSREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneDatasetReportsRequest)
  })
_sym_db.RegisterMessage(GeneDatasetReportsRequest)
_sym_db.RegisterMessage(GeneDatasetReportsRequest.SymbolsForTaxon)

GeneLinksRequest = _reflection.GeneratedProtocolMessageType('GeneLinksRequest', (_message.Message,), {
  'DESCRIPTOR' : _GENELINKSREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneLinksRequest)
  })
_sym_db.RegisterMessage(GeneLinksRequest)

GeneLinksReply = _reflection.GeneratedProtocolMessageType('GeneLinksReply', (_message.Message,), {

  'GeneLink' : _reflection.GeneratedProtocolMessageType('GeneLink', (_message.Message,), {
    'DESCRIPTOR' : _GENELINKSREPLY_GENELINK,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneLinksReply.GeneLink)
    })
  ,
  'DESCRIPTOR' : _GENELINKSREPLY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneLinksReply)
  })
_sym_db.RegisterMessage(GeneLinksReply)
_sym_db.RegisterMessage(GeneLinksReply.GeneLink)

GeneChromosomeSummaryRequest = _reflection.GeneratedProtocolMessageType('GeneChromosomeSummaryRequest', (_message.Message,), {
  'DESCRIPTOR' : _GENECHROMOSOMESUMMARYREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneChromosomeSummaryRequest)
  })
_sym_db.RegisterMessage(GeneChromosomeSummaryRequest)

GeneChromosomeSummaryReply = _reflection.GeneratedProtocolMessageType('GeneChromosomeSummaryReply', (_message.Message,), {

  'GeneChromosomeSummary' : _reflection.GeneratedProtocolMessageType('GeneChromosomeSummary', (_message.Message,), {
    'DESCRIPTOR' : _GENECHROMOSOMESUMMARYREPLY_GENECHROMOSOMESUMMARY,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneChromosomeSummaryReply.GeneChromosomeSummary)
    })
  ,
  'DESCRIPTOR' : _GENECHROMOSOMESUMMARYREPLY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneChromosomeSummaryReply)
  })
_sym_db.RegisterMessage(GeneChromosomeSummaryReply)
_sym_db.RegisterMessage(GeneChromosomeSummaryReply.GeneChromosomeSummary)

AssemblyLinksRequest = _reflection.GeneratedProtocolMessageType('AssemblyLinksRequest', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYLINKSREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyLinksRequest)
  })
_sym_db.RegisterMessage(AssemblyLinksRequest)

AssemblyLinksReply = _reflection.GeneratedProtocolMessageType('AssemblyLinksReply', (_message.Message,), {

  'AssemblyLink' : _reflection.GeneratedProtocolMessageType('AssemblyLink', (_message.Message,), {
    'DESCRIPTOR' : _ASSEMBLYLINKSREPLY_ASSEMBLYLINK,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyLinksReply.AssemblyLink)
    })
  ,
  'DESCRIPTOR' : _ASSEMBLYLINKSREPLY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyLinksReply)
  })
_sym_db.RegisterMessage(AssemblyLinksReply)
_sym_db.RegisterMessage(AssemblyLinksReply.AssemblyLink)

VirusDatasetRequest = _reflection.GeneratedProtocolMessageType('VirusDatasetRequest', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSDATASETREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.VirusDatasetRequest)
  })
_sym_db.RegisterMessage(VirusDatasetRequest)

VirusDatasetFilter = _reflection.GeneratedProtocolMessageType('VirusDatasetFilter', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSDATASETFILTER,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.VirusDatasetFilter)
  })
_sym_db.RegisterMessage(VirusDatasetFilter)

VirusDataReportRequest = _reflection.GeneratedProtocolMessageType('VirusDataReportRequest', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSDATAREPORTREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.VirusDataReportRequest)
  })
_sym_db.RegisterMessage(VirusDataReportRequest)

VirusAnnotationFilter = _reflection.GeneratedProtocolMessageType('VirusAnnotationFilter', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSANNOTATIONFILTER,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.VirusAnnotationFilter)
  })
_sym_db.RegisterMessage(VirusAnnotationFilter)

VirusAnnotationReportRequest = _reflection.GeneratedProtocolMessageType('VirusAnnotationReportRequest', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSANNOTATIONREPORTREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.VirusAnnotationReportRequest)
  })
_sym_db.RegisterMessage(VirusAnnotationReportRequest)

VirusAvailabilityRequest = _reflection.GeneratedProtocolMessageType('VirusAvailabilityRequest', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSAVAILABILITYREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.VirusAvailabilityRequest)
  })
_sym_db.RegisterMessage(VirusAvailabilityRequest)

VirusAvailability = _reflection.GeneratedProtocolMessageType('VirusAvailability', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSAVAILABILITY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.VirusAvailability)
  })
_sym_db.RegisterMessage(VirusAvailability)

Sars2ProteinDatasetRequest = _reflection.GeneratedProtocolMessageType('Sars2ProteinDatasetRequest', (_message.Message,), {
  'DESCRIPTOR' : _SARS2PROTEINDATASETREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.Sars2ProteinDatasetRequest)
  })
_sym_db.RegisterMessage(Sars2ProteinDatasetRequest)

DownloadSummary = _reflection.GeneratedProtocolMessageType('DownloadSummary', (_message.Message,), {

  'Hydrated' : _reflection.GeneratedProtocolMessageType('Hydrated', (_message.Message,), {
    'DESCRIPTOR' : _DOWNLOADSUMMARY_HYDRATED,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.DownloadSummary.Hydrated)
    })
  ,

  'Dehydrated' : _reflection.GeneratedProtocolMessageType('Dehydrated', (_message.Message,), {
    'DESCRIPTOR' : _DOWNLOADSUMMARY_DEHYDRATED,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.DownloadSummary.Dehydrated)
    })
  ,

  'FileSummary' : _reflection.GeneratedProtocolMessageType('FileSummary', (_message.Message,), {
    'DESCRIPTOR' : _DOWNLOADSUMMARY_FILESUMMARY,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.DownloadSummary.FileSummary)
    })
  ,

  'AvailableFiles' : _reflection.GeneratedProtocolMessageType('AvailableFiles', (_message.Message,), {
    'DESCRIPTOR' : _DOWNLOADSUMMARY_AVAILABLEFILES,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.DownloadSummary.AvailableFiles)
    })
  ,
  'DESCRIPTOR' : _DOWNLOADSUMMARY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.DownloadSummary)
  })
_sym_db.RegisterMessage(DownloadSummary)
_sym_db.RegisterMessage(DownloadSummary.Hydrated)
_sym_db.RegisterMessage(DownloadSummary.Dehydrated)
_sym_db.RegisterMessage(DownloadSummary.FileSummary)
_sym_db.RegisterMessage(DownloadSummary.AvailableFiles)

TabularOutput = _reflection.GeneratedProtocolMessageType('TabularOutput', (_message.Message,), {
  'DESCRIPTOR' : _TABULAROUTPUT,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TabularOutput)
  })
_sym_db.RegisterMessage(TabularOutput)

MicroBiggeDatasetRequest = _reflection.GeneratedProtocolMessageType('MicroBiggeDatasetRequest', (_message.Message,), {

  'ElementFlankConfig' : _reflection.GeneratedProtocolMessageType('ElementFlankConfig', (_message.Message,), {
    'DESCRIPTOR' : _MICROBIGGEDATASETREQUEST_ELEMENTFLANKCONFIG,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.MicroBiggeDatasetRequest.ElementFlankConfig)
    })
  ,
  'DESCRIPTOR' : _MICROBIGGEDATASETREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.MicroBiggeDatasetRequest)
  })
_sym_db.RegisterMessage(MicroBiggeDatasetRequest)
_sym_db.RegisterMessage(MicroBiggeDatasetRequest.ElementFlankConfig)

VersionReply = _reflection.GeneratedProtocolMessageType('VersionReply', (_message.Message,), {
  'DESCRIPTOR' : _VERSIONREPLY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.VersionReply)
  })
_sym_db.RegisterMessage(VersionReply)

ProkaryoteGeneRequest = _reflection.GeneratedProtocolMessageType('ProkaryoteGeneRequest', (_message.Message,), {

  'GeneFlankConfig' : _reflection.GeneratedProtocolMessageType('GeneFlankConfig', (_message.Message,), {
    'DESCRIPTOR' : _PROKARYOTEGENEREQUEST_GENEFLANKCONFIG,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.ProkaryoteGeneRequest.GeneFlankConfig)
    })
  ,
  'DESCRIPTOR' : _PROKARYOTEGENEREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.ProkaryoteGeneRequest)
  })
_sym_db.RegisterMessage(ProkaryoteGeneRequest)
_sym_db.RegisterMessage(ProkaryoteGeneRequest.GeneFlankConfig)

SleepRequest = _reflection.GeneratedProtocolMessageType('SleepRequest', (_message.Message,), {
  'DESCRIPTOR' : _SLEEPREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.SleepRequest)
  })
_sym_db.RegisterMessage(SleepRequest)

SleepReply = _reflection.GeneratedProtocolMessageType('SleepReply', (_message.Message,), {
  'DESCRIPTOR' : _SLEEPREPLY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.SleepReply)
  })
_sym_db.RegisterMessage(SleepReply)

TaxonomyNode = _reflection.GeneratedProtocolMessageType('TaxonomyNode', (_message.Message,), {

  'CountByType' : _reflection.GeneratedProtocolMessageType('CountByType', (_message.Message,), {
    'DESCRIPTOR' : _TAXONOMYNODE_COUNTBYTYPE,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyNode.CountByType)
    })
  ,
  'DESCRIPTOR' : _TAXONOMYNODE,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyNode)
  })
_sym_db.RegisterMessage(TaxonomyNode)
_sym_db.RegisterMessage(TaxonomyNode.CountByType)

TaxonomyMetadataRequest = _reflection.GeneratedProtocolMessageType('TaxonomyMetadataRequest', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYMETADATAREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyMetadataRequest)
  })
_sym_db.RegisterMessage(TaxonomyMetadataRequest)

TaxonomyMatch = _reflection.GeneratedProtocolMessageType('TaxonomyMatch', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYMATCH,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyMatch)
  })
_sym_db.RegisterMessage(TaxonomyMatch)

TaxonomyMetadataResponse = _reflection.GeneratedProtocolMessageType('TaxonomyMetadataResponse', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYMETADATARESPONSE,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyMetadataResponse)
  })
_sym_db.RegisterMessage(TaxonomyMetadataResponse)

TaxonomyFilteredSubtreeRequest = _reflection.GeneratedProtocolMessageType('TaxonomyFilteredSubtreeRequest', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYFILTEREDSUBTREEREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyFilteredSubtreeRequest)
  })
_sym_db.RegisterMessage(TaxonomyFilteredSubtreeRequest)

TaxonomyFilteredSubtreeResponse = _reflection.GeneratedProtocolMessageType('TaxonomyFilteredSubtreeResponse', (_message.Message,), {

  'Edge' : _reflection.GeneratedProtocolMessageType('Edge', (_message.Message,), {
    'DESCRIPTOR' : _TAXONOMYFILTEREDSUBTREERESPONSE_EDGE,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.Edge)
    })
  ,

  'EdgesEntry' : _reflection.GeneratedProtocolMessageType('EdgesEntry', (_message.Message,), {
    'DESCRIPTOR' : _TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.EdgesEntry)
    })
  ,
  'DESCRIPTOR' : _TAXONOMYFILTEREDSUBTREERESPONSE,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse)
  })
_sym_db.RegisterMessage(TaxonomyFilteredSubtreeResponse)
_sym_db.RegisterMessage(TaxonomyFilteredSubtreeResponse.Edge)
_sym_db.RegisterMessage(TaxonomyFilteredSubtreeResponse.EdgesEntry)

DatasetRequest = _reflection.GeneratedProtocolMessageType('DatasetRequest', (_message.Message,), {
  'DESCRIPTOR' : _DATASETREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.DatasetRequest)
  })
_sym_db.RegisterMessage(DatasetRequest)

GeneCountsByTaxonRequest = _reflection.GeneratedProtocolMessageType('GeneCountsByTaxonRequest', (_message.Message,), {
  'DESCRIPTOR' : _GENECOUNTSBYTAXONREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneCountsByTaxonRequest)
  })
_sym_db.RegisterMessage(GeneCountsByTaxonRequest)

GeneCountsByTaxonReply = _reflection.GeneratedProtocolMessageType('GeneCountsByTaxonReply', (_message.Message,), {

  'GeneTypeAndCount' : _reflection.GeneratedProtocolMessageType('GeneTypeAndCount', (_message.Message,), {
    'DESCRIPTOR' : _GENECOUNTSBYTAXONREPLY_GENETYPEANDCOUNT,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneCountsByTaxonReply.GeneTypeAndCount)
    })
  ,
  'DESCRIPTOR' : _GENECOUNTSBYTAXONREPLY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneCountsByTaxonReply)
  })
_sym_db.RegisterMessage(GeneCountsByTaxonReply)
_sym_db.RegisterMessage(GeneCountsByTaxonReply.GeneTypeAndCount)

GenomeAnnotationRequest = _reflection.GeneratedProtocolMessageType('GenomeAnnotationRequest', (_message.Message,), {
  'DESCRIPTOR' : _GENOMEANNOTATIONREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GenomeAnnotationRequest)
  })
_sym_db.RegisterMessage(GenomeAnnotationRequest)

GenomeAnnotationTableSummaryReply = _reflection.GeneratedProtocolMessageType('GenomeAnnotationTableSummaryReply', (_message.Message,), {
  'DESCRIPTOR' : _GENOMEANNOTATIONTABLESUMMARYREPLY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GenomeAnnotationTableSummaryReply)
  })
_sym_db.RegisterMessage(GenomeAnnotationTableSummaryReply)


DESCRIPTOR._options = None
_ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['accession']._options = None
_ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['chromosomes']._options = None
_ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['role_filters']._options = None
_ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['table_fields']._options = None
_ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['count_assembly_unplaced']._options = None
_ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['page_size']._options = None
_SORTFIELD.fields_by_name['field']._options = None
_ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['taxons']._options = None
_ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['bioprojects']._options = None
_ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['accessions']._options = None
_ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['chromosomes']._options = None
_ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['table_fields']._options = None
_ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['page_size']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['reference_only']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['assembly_source']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['has_annotation']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['exclude_paired_reports']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['exclude_atypical']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['assembly_version']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['assembly_level']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['first_release_date']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['last_release_date']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['search_text']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['is_metagenome_derived']._options = None
_ACCESSIONS.fields_by_name['accessions']._options = None
_ASSEMBLYREVISIONHISTORYREQUEST.fields_by_name['accession']._options = None
_ASSEMBLYCHECKMHISTOGRAMREQUEST.fields_by_name['species_taxon']._options = None
_ASSEMBLYCHECKMHISTOGRAMREPLY.fields_by_name['species_taxid']._options = None
_ANNOTATIONFORASSEMBLY_TYPE._options = None
_ANNOTATIONFORASSEMBLY_TYPE.values_by_name["GENOME_GB"]._options = None
_ASSEMBLYDATASETREQUEST.fields_by_name['accessions']._options = None
_ASSEMBLYDATASETREQUEST.fields_by_name['chromosomes']._options = None
_ASSEMBLYDATASETREQUEST.fields_by_name['include_annotation_type']._options = None
_ASSEMBLYDATASETREQUEST.fields_by_name['include_tsv']._options = None
_ASSEMBLYDATASETREQUEST._options = None
_ORGANISM.fields_by_name['assembly_count']._options = None
_TAXTREEREQUEST.fields_by_name['tax_id']._options = None
_TAXTREEREQUEST.fields_by_name['tax_token']._options = None
_TAXTREEREQUEST.fields_by_name['taxon']._options = None
_TAXTREEREQUEST.fields_by_name['children_only']._options = None
_ORGANISMQUERYREQUEST.fields_by_name['organism_query']._options = None
_ORGANISMQUERYREQUEST.fields_by_name['taxon_query']._options = None
_ORGANISMQUERYREQUEST.fields_by_name['tax_rank_filter']._options = None
_ORGANISMQUERYREQUEST.fields_by_name['taxon_resource_filter']._options = None
_ORTHOLOGREQUEST.fields_by_name['gene_id']._options = None
_ORTHOLOGREQUEST.fields_by_name['taxon_filter']._options = None
_ORTHOLOGREQUEST.fields_by_name['page_size']._options = None
_GENEDATASETREQUEST.fields_by_name['gene_ids']._options = None
_GENEDATASETREQUEST.fields_by_name['include_annotation_type']._options = None
_GENEDATASETREQUEST.fields_by_name['aux_report']._options = None
_GENEDATASETREQUEST.fields_by_name['table_fields']._options = None
_GENEDATASETREQUEST._options = None
_GENEDATASETREPORTSREQUEST_SYMBOLSFORTAXON.fields_by_name['symbols']._options = None
_GENEDATASETREPORTSREQUEST_SYMBOLSFORTAXON.fields_by_name['taxon']._options = None
_GENEDATASETREPORTSREQUEST.fields_by_name['gene_ids']._options = None
_GENEDATASETREPORTSREQUEST.fields_by_name['accessions']._options = None
_GENEDATASETREPORTSREQUEST.fields_by_name['taxon']._options = None
_GENEDATASETREPORTSREQUEST.fields_by_name['table_fields']._options = None
_GENEDATASETREPORTSREQUEST.fields_by_name['page_size']._options = None
_GENEDATASETREPORTSREQUEST.fields_by_name['query']._options = None
_GENEDATASETREPORTSREQUEST.fields_by_name['types']._options = None
_GENEDATASETREPORTSREQUEST._options = None
_GENELINKSREQUEST.fields_by_name['gene_ids']._options = None
_GENELINKSREQUEST._options = None
_GENECHROMOSOMESUMMARYREQUEST.fields_by_name['taxon']._options = None
_GENECHROMOSOMESUMMARYREQUEST.fields_by_name['annotation_name']._options = None
_ASSEMBLYLINKSREQUEST.fields_by_name['accessions']._options = None
_ASSEMBLYLINKSREQUEST._options = None
_VIRUSDATASETREQUEST.fields_by_name['accessions']._options = None
_VIRUSDATASETREQUEST.fields_by_name['taxon']._options = None
_VIRUSDATASETREQUEST.fields_by_name['refseq_only']._options = None
_VIRUSDATASETREQUEST.fields_by_name['annotated_only']._options = None
_VIRUSDATASETREQUEST.fields_by_name['released_since']._options = None
_VIRUSDATASETREQUEST.fields_by_name['updated_since']._options = None
_VIRUSDATASETREQUEST.fields_by_name['host']._options = None
_VIRUSDATASETREQUEST.fields_by_name['geo_location']._options = None
_VIRUSDATASETREQUEST.fields_by_name['complete_only']._options = None
_VIRUSDATASETREQUEST.fields_by_name['table_fields']._options = None
_VIRUSDATASETREQUEST.fields_by_name['include_sequence']._options = None
_VIRUSDATASETREQUEST.fields_by_name['aux_report']._options = None
_VIRUSDATASETREQUEST.fields_by_name['format']._options = None
_VIRUSDATASETFILTER.fields_by_name['accessions']._options = None
_VIRUSDATASETFILTER.fields_by_name['taxon']._options = None
_VIRUSDATASETFILTER.fields_by_name['refseq_only']._options = None
_VIRUSDATASETFILTER.fields_by_name['annotated_only']._options = None
_VIRUSDATASETFILTER.fields_by_name['released_since']._options = None
_VIRUSDATASETFILTER.fields_by_name['updated_since']._options = None
_VIRUSDATASETFILTER.fields_by_name['host']._options = None
_VIRUSDATASETFILTER.fields_by_name['geo_location']._options = None
_VIRUSDATASETFILTER.fields_by_name['complete_only']._options = None
_VIRUSDATAREPORTREQUEST.fields_by_name['table_fields']._options = None
_VIRUSDATAREPORTREQUEST.fields_by_name['page_size']._options = None
_VIRUSDATAREPORTREQUEST._options = None
_VIRUSANNOTATIONFILTER.fields_by_name['accessions']._options = None
_VIRUSANNOTATIONFILTER.fields_by_name['taxon']._options = None
_VIRUSANNOTATIONFILTER.fields_by_name['refseq_only']._options = None
_VIRUSANNOTATIONFILTER.fields_by_name['released_since']._options = None
_VIRUSANNOTATIONFILTER.fields_by_name['updated_since']._options = None
_VIRUSANNOTATIONFILTER.fields_by_name['host']._options = None
_VIRUSANNOTATIONFILTER.fields_by_name['geo_location']._options = None
_VIRUSANNOTATIONFILTER.fields_by_name['complete_only']._options = None
_VIRUSANNOTATIONREPORTREQUEST.fields_by_name['table_fields']._options = None
_VIRUSANNOTATIONREPORTREQUEST.fields_by_name['page_size']._options = None
_VIRUSANNOTATIONREPORTREQUEST._options = None
_VIRUSAVAILABILITYREQUEST.fields_by_name['accessions']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['proteins']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['refseq_only']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['annotated_only']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['released_since']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['updated_since']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['host']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['geo_location']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['complete_only']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['table_fields']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['include_sequence']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['aux_report']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['format']._options = None
_DOWNLOADSUMMARY.fields_by_name['assembly_count']._options = None
_DOWNLOADSUMMARY.fields_by_name['errors']._options = None
_MICROBIGGEDATASETREQUEST.fields_by_name['files']._options = None
_PROKARYOTEGENEREQUEST.fields_by_name['accessions']._options = None
_PROKARYOTEGENEREQUEST.fields_by_name['include_annotation_type']._options = None
_PROKARYOTEGENEREQUEST._options = None
_TAXONOMYMETADATAREQUEST.fields_by_name['taxons']._options = None
_TAXONOMYMETADATAREQUEST._options = None
_TAXONOMYFILTEREDSUBTREEREQUEST.fields_by_name['taxons']._options = None
_TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY._options = None
_GENECOUNTSBYTAXONREQUEST.fields_by_name['taxon']._options = None
_GENOMEANNOTATIONREQUEST.fields_by_name['page_size']._options = None
# @@protoc_insertion_point(module_scope)
