# -*- coding: utf-8 -*-
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: ncbi/datasets/v1/reports/assembly.proto
"""Generated protocol buffer code."""
from google.protobuf.internal import enum_type_wrapper
from google.protobuf import descriptor as _descriptor
from google.protobuf import message as _message
from google.protobuf import reflection as _reflection
from google.protobuf import symbol_database as _symbol_database
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()


from ncbi.datasets.v1.reports import biosample_pb2 as ncbi_dot_datasets_dot_v1_dot_reports_dot_biosample__pb2
from ncbi.datasets.v1.reports import common_pb2 as ncbi_dot_datasets_dot_v1_dot_reports_dot_common__pb2
from ncbi.datasets.options import report_pb2 as ncbi_dot_datasets_dot_options_dot_report__pb2


DESCRIPTOR = _descriptor.FileDescriptor(
  name='ncbi/datasets/v1/reports/assembly.proto',
  package='ncbi.datasets.v1.reports',
  syntax='proto3',
  serialized_options=b'Z\030ncbi/datasets/v1/reports\370\001\001\302\363\030\327\n\n\033Genome Assembly Data Report\022\022AssemblyDataReport\032;<p>The downloaded genome package contains a genome assembly\032Idata report in <a href=\"https://jsonlines.readthedocs.io/\">JSON Lines</a>\032Pformat in the file: <br><br><em>ncbi_dataset/data/data_report.jsonl</em><br><br>\032xEach line of the genome assembly data report file is a hierarchical <a href=\"https://www.json.org/json-en.html\">JSON</a>\032yobject that represents a single genome assembly record. The schema of the genome assembly record is defined in the tables\032pbelow where each row describes a single field in the report or a sub-structure, which is a collection of fields.\032JThe outermost structure of the report is <em>AssemblyDataReport</em>.</p> \032Q<p>Table fields that include a <em>Table Field Mnemonic</em> can be used with the\032\220\001<a href=\"/datasets/v1/docs/reference-docs/command-line/dataformat/\">dataformat</a> command-line tool\'s <nobr><code>--fields</code></nobr> option\032\216\001Refer to the <a href=\"/datasets/v1/docs/reference-docs/command-line/dataformat/\">dataformat</a> CLI tool reference to see how you can use this\032Ptool to transform assembly data reports from JSON Lines to tabular formats.</p> \"\013Orientation\"\013SeqRangeSet\"\005Range\"\017LineageOrganism\"\010Organism*\221\001{{< readfile-code-block lang=\"json\" file=\"/docs_samples/datasets-cli/genome-report.cli.output\" begin=\"START data-report\" end=\"END data-report\">}}',
  create_key=_descriptor._internal_create_key,
  serialized_pb=b'\n\'ncbi/datasets/v1/reports/assembly.proto\x12\x18ncbi.datasets.v1.reports\x1a(ncbi/datasets/v1/reports/biosample.proto\x1a%ncbi/datasets/v1/reports/common.proto\x1a\"ncbi/datasets/options/report.proto\"\xbb\t\n\x12\x41ssemblyDataReport\x12g\n\x0b\x63ommon_name\x18\x01 \x01(\tBF\xc2\xf3\x18\x42\n\x0b\x63ommon-name\x12\x0b\x43ommon name2\x05Human2\tzebrafish2\x14pacific white shrimpR\ncommonName\x12\x83\x01\n\rorganism_name\x18\x02 \x01(\tB^\xc2\xf3\x18Z\n\rorganism-name\x12\rOrganism name2\x0cHomo sapiens2\x14\x41rabidopsis thaliana2\x16\x43\x61nis lupus familiarisR\x0corganismName\x12\x39\n\x05\x62reed\x18\x03 \x01(\tB#\xc2\xf3\x18\x1f\n\x05\x62reed\x12\x05\x42reed2\x08Hereford2\x05\x62oxerR\x05\x62reed\x12\x39\n\x08\x63ultivar\x18\x04 \x01(\tB\x1d\xc2\xf3\x18\x19\n\x08\x63ultivar\x12\x08\x43ultivar2\x03\x42\x37\x33R\x08\x63ultivar\x12\x38\n\x07\x65\x63otype\x18\x05 \x01(\tB\x1e\xc2\xf3\x18\x1a\n\x07\x65\x63otype\x12\x07\x45\x63otype2\x06\x41lpineR\x07\x65\x63otype\x12j\n\x07isolate\x18\x06 \x01(\tBP\xc2\xf3\x18L\n\x07isolate\x12\x07Isolate2/L1 Dominette 01449 registration number 421906802\x07Pmale09R\x07isolate\x12(\n\x03sex\x18\x07 \x01(\tB\x16\xc2\xf3\x18\x12\n\x03sex\x12\x03Sex2\x06\x66\x65maleR\x03sex\x12,\n\x06strain\x18\x08 \x01(\tB\x14\xc2\xf3\x18\x10\n\x06strain\x12\x06StrainR\x06strain\x12\x31\n\x06tax_id\x18\t \x01(\rB\x1a\xc2\xf3\x18\x16\n\x06tax-id\x12\x0cTaxonomic IDR\x05taxId\x12g\n\rassembly_info\x18\n \x01(\x0b\x32&.ncbi.datasets.v1.reports.AssemblyInfoB\x1a\xc2\xf3\x18\x16\n\tassminfo-\x12\tAssembly R\x0c\x61ssemblyInfo\x12q\n\x0e\x61ssembly_stats\x18\x0b \x01(\x0b\x32\'.ncbi.datasets.v1.reports.AssemblyStatsB!\xc2\xf3\x18\x1d\n\nassmstats-\x12\x0f\x41ssembly Stats R\rassemblyStats\x12l\n\x0eorganelle_info\x18\x0c \x03(\x0b\x32\'.ncbi.datasets.v1.reports.OrganelleInfoB\x1c\xc2\xf3\x18\x18\n\norganelle-\x12\nOrganelle R\rorganelleInfo\x12u\n\x0f\x61nnotation_info\x18\r \x01(\x0b\x32(.ncbi.datasets.v1.reports.AnnotationInfoB\"\xc2\xf3\x18\x1e\n\nannotinfo-\x12\x10\x41nnotation Info R\x0e\x61nnotationInfo\x12N\n\x08wgs_info\x18\x0e \x01(\x0b\x32!.ncbi.datasets.v1.reports.WGSInfoB\x10\xc2\xf3\x18\x0c\n\x04wgs-\x12\x04WGS R\x07wgsInfo\"\xfe\x03\n\x0eLinkedAssembly\x12T\n\x0flinked_assembly\x18\x01 \x01(\tB+\xc2\xf3\x18\'\n\taccession\x12\tAccession2\x0fGCA_000212995.1R\x0elinkedAssembly\x12r\n\rassembly_type\x18\x02 \x01(\x0e\x32;.ncbi.datasets.v1.reports.LinkedAssembly.LinkedAssemblyTypeB\x10\xc2\xf3\x18\x0c\n\x04type\x12\x04TypeR\x0c\x61ssemblyType\"\xa1\x02\n\x12LinkedAssemblyType\x12 \n\x1cLINKED_ASSEMBLY_TYPE_UNKNOWN\x10\x00\x12(\n$alternate_pseudohaplotype_of_diploid\x10\x01\x12(\n$principal_pseudohaplotype_of_diploid\x10\x02\x12!\n\x1dmaternal_haplotype_of_diploid\x10\x03\x12!\n\x1dpaternal_haplotype_of_diploid\x10\x04\x12\x0f\n\x0bhaplotype_1\x10\x06\x12\x0f\n\x0bhaplotype_2\x10\x07\x12\x0f\n\x0bhaplotype_3\x10\x08\x12\x0f\n\x0bhaplotype_4\x10\t\x12\x0b\n\x07haploid\x10\n\"\xf8\x01\n\x0ePairedAssembly\x12J\n\taccession\x18\x01 \x01(\tB,\xc2\xf3\x18(\n\taccession\x12\tAccession2\x10GCF_000001405.40R\taccession\x12_\n\x06status\x18\x02 \x01(\x0e\x32(.ncbi.datasets.v1.reports.AssemblyStatusB\x1d\xc2\xf3\x18\x19\n\x06status\x12\x06Status2\x07\x63urrentR\x06status\x12\x39\n\x0f\x61nnotation_name\x18\x03 \x01(\tB\x10\xc2\xf3\x18\x0c\n\x04name\x12\x04NameR\x0e\x61nnotationName\"\x85\x01\n\x0c\x41typicalInfo\x12?\n\x0bis_atypical\x18\x01 \x01(\x08\x42\x1e\xc2\xf3\x18\x1a\n\x0bis-atypical\x12\x0bIs AtypicalR\nisAtypical\x12\x34\n\x08warnings\x18\x03 \x03(\tB\x18\xc2\xf3\x18\x14\n\x08warnings\x12\x08WarningsR\x08warnings\"\xd8\x11\n\x0c\x41ssemblyInfo\x12[\n\x12\x61ssembly_accession\x18\x10 \x01(\tB,\xc2\xf3\x18(\n\taccession\x12\tAccession2\x10GCF_000001405.40R\x11\x61ssemblyAccession\x12z\n\x1a\x63urrent_assembly_accession\x18\x19 \x01(\tB<\xc2\xf3\x18\x38\n\x11\x63urrent-accession\x12\x11\x43urrent Accession2\x10GCF_000001405.40R\x18\x63urrentAssemblyAccession\x12p\n\x0f\x61ssembly_status\x18\x16 \x01(\x0e\x32(.ncbi.datasets.v1.reports.AssemblyStatusB\x1d\xc2\xf3\x18\x19\n\x06status\x12\x06Status2\x07\x63urrentR\x0e\x61ssemblyStatus\x12v\n\x19paired_assembly_accession\x18\x11 \x01(\tB:\xc2\xf3\x18\x36\n\x10paired_accession\x12\x10Paired Accession2\x10GCA_000001405.28R\x17pairedAssemblyAccession\x12v\n\x0fpaired_assembly\x18\x17 \x01(\x0b\x32(.ncbi.datasets.v1.reports.PairedAssemblyB#\xc2\xf3\x18\x1f\n\x0bpaired-assm\x12\x10Paired Assembly R\x0epairedAssembly\x12W\n\x0e\x61ssembly_level\x18\x01 \x01(\tB0\xc2\xf3\x18,\n\x05level\x12\x05Level2\nchromosome2\x08scaffold2\x06\x63ontigR\rassemblyLevel\x12N\n\rassembly_name\x18\x02 \x01(\tB)\xc2\xf3\x18%\n\x04name\x12\x04Name2\nGRCh38.p142\x0b\x41SM985889v3R\x0c\x61ssemblyName\x12U\n\rassembly_type\x18\x03 \x01(\tB0\xc2\xf3\x18,\n\x04type\x12\x04Type2\x15haploid-with-alt-loci2\x07haploidR\x0c\x61ssemblyType\x12z\n\x12\x62ioproject_lineage\x18\r \x03(\x0b\x32+.ncbi.datasets.v1.reports.BioProjectLineageB\x1e\xc2\xf3\x18\x1a\n\x0b\x62ioproject-\x12\x0b\x42ioProject R\x11\x62ioprojectLineage\x12O\n\x0fsubmission_date\x18\x05 \x01(\tB&\xc2\xf3\x18\"\n\x0fsubmission-date\x12\x0fSubmission DateR\x0esubmissionDate\x12@\n\x0b\x64\x65scription\x18\x06 \x01(\tB\x1e\xc2\xf3\x18\x1a\n\x0b\x64\x65scription\x12\x0b\x44\x65scriptionR\x0b\x64\x65scription\x12w\n\x16genbank_assm_accession\x18\x07 \x01(\tBA\xc2\xf3\x18=\n\x16genbank-assm-accession\x12\x11GenBank Accession2\x10GCA_000001405.28R\x14genbankAssmAccession\x12\x38\n\tsubmitter\x18\x08 \x01(\tB\x1a\xc2\xf3\x18\x16\n\tsubmitter\x12\tSubmitterR\tsubmitter\x12x\n\x0frefseq_category\x18\t \x01(\tBO\xc2\xf3\x18K\n\x0frefseq-category\x12\x0fRefseq Category2\x10reference genome2\x15representative genomeR\x0erefseqCategory\x12s\n\x15refseq_assm_accession\x18\n \x01(\tB?\xc2\xf3\x18;\n\x15refseq-assm-accession\x12\x10RefSeq Accession2\x10GCF_000001405.40R\x13refseqAssmAccession\x12T\n\x0eucsc_assm_name\x18\x0b \x01(\tB.\xc2\xf3\x18*\n\x0eucsc-assm-name\x12\x12UCSC Assembly Name2\x04hg38R\x0cucscAssmName\x12+\n\x0flinked_assembly\x18\x0c \x01(\tB\x02\x18\x01R\x0elinkedAssembly\x12z\n\x11linked_assemblies\x18\x15 \x03(\x0b\x32(.ncbi.datasets.v1.reports.LinkedAssemblyB#\xc2\xf3\x18\x1f\n\x0blinked-assm\x12\x10Linked Assembly R\x10linkedAssemblies\x12]\n\x08\x61typical\x18\x18 \x01(\x0b\x32&.ncbi.datasets.v1.reports.AtypicalInfoB\x19\xc2\xf3\x18\x15\n\x08\x61typical\x12\tAtypical R\x08\x61typical\x12O\n\x0fsequencing_tech\x18\x0e \x01(\tB&\xc2\xf3\x18\"\n\x0fsequencing-tech\x12\x0fSequencing TechR\x0esequencingTech\x12m\n\x13\x62iosample_accession\x18\x0f \x01(\tB<\xc2\xf3\x18\x38\n\x13\x62iosample-accession\x12\x13\x42ioSample Accession2\x0cSAMN03145444R\x12\x62iosampleAccession\x12i\n\tbiosample\x18\x13 \x01(\x0b\x32-.ncbi.datasets.v1.reports.BioSampleDescriptorB\x1c\xc2\xf3\x18\x18\n\nbiosample-\x12\nBioSample R\tbiosample\x12\x37\n\tblast_url\x18\x12 \x01(\tB\x1a\xc2\xf3\x18\x16\n\tblast-url\x12\tBlast URLR\x08\x62lastUrl\x12\x1a\n\x08\x63omments\x18\x14 \x01(\tR\x08\x63omments\"\xd5\x08\n\rAssemblyStats\x12}\n\x1btotal_number_of_chromosomes\x18\x01 \x01(\rB>\xc2\xf3\x18:\n\x1btotal-number-of-chromosomes\x12\x1bTotal Number of ChromosomesR\x18totalNumberOfChromosomes\x12\x63\n\x15total_sequence_length\x18\x02 \x01(\x04\x42/\xc2\xf3\x18+\n\x12total-sequence-len\x12\x15Total Sequence LengthR\x13totalSequenceLength\x12\x63\n\x15total_ungapped_length\x18\x03 \x01(\x04\x42/\xc2\xf3\x18+\n\x12total-ungapped-len\x12\x15Total Ungapped LengthR\x13totalUngappedLength\x12V\n\x11number_of_contigs\x18\x04 \x01(\rB*\xc2\xf3\x18&\n\x11number-of-contigs\x12\x11Number of ContigsR\x0fnumberOfContigs\x12;\n\ncontig_n50\x18\x05 \x01(\rB\x1c\xc2\xf3\x18\x18\n\ncontig-n50\x12\nContig N50R\tcontigN50\x12;\n\ncontig_l50\x18\x06 \x01(\rB\x1c\xc2\xf3\x18\x18\n\ncontig-l50\x12\nContig L50R\tcontigL50\x12^\n\x13number_of_scaffolds\x18\x07 \x01(\rB.\xc2\xf3\x18*\n\x13number-of-scaffolds\x12\x13Number of ScaffoldsR\x11numberOfScaffolds\x12\x43\n\x0cscaffold_n50\x18\x08 \x01(\rB \xc2\xf3\x18\x1c\n\x0cscaffold-n50\x12\x0cScaffold N50R\x0bscaffoldN50\x12\x43\n\x0cscaffold_l50\x18\t \x01(\rB \xc2\xf3\x18\x1c\n\x0cscaffold-l50\x12\x0cScaffold L50R\x0bscaffoldL50\x12\x81\x01\n\x1cgaps_between_scaffolds_count\x18\n \x01(\rB@\xc2\xf3\x18<\n\x1cgaps-between-scaffolds-count\x12\x1cGaps Between Scaffolds CountR\x19gapsBetweenScaffoldsCount\x12\x85\x01\n\x1dnumber_of_component_sequences\x18\x0b \x01(\rBB\xc2\xf3\x18>\n\x1dnumber-of-component-sequences\x12\x1dNumber of Component SequencesR\x1anumberOfComponentSequences\x12\x33\n\x08gc_count\x18\x0c \x01(\x04\x42\x18\xc2\xf3\x18\x14\n\x08gc-count\x12\x08GC CountR\x07gcCount\"\xd9\x03\n\rOrganelleInfo\x12G\n\rassembly_name\x18\x01 \x01(\tB\"\xc2\xf3\x18\x1e\n\rassembly-name\x12\rAssembly NameR\x0c\x61ssemblyName\x12[\n\x12infraspecific_name\x18\x02 \x01(\tB,\xc2\xf3\x18(\n\x12infraspecific-name\x12\x12Infraspecific NameR\x11infraspecificName\x12R\n\nbioproject\x18\x03 \x03(\tB2\xc2\xf3\x18.\n\x15\x62ioproject-accessions\x12\x15\x42ioProject AccessionsR\nbioproject\x12@\n\x0b\x64\x65scription\x18\x04 \x01(\tB\x1e\xc2\xf3\x18\x1a\n\x0b\x64\x65scription\x12\x0b\x44\x65scriptionR\x0b\x64\x65scription\x12R\n\x10total_seq_length\x18\x05 \x01(\x04\x42(\xc2\xf3\x18$\n\x10total-seq-length\x12\x10Total Seq LengthR\x0etotalSeqLength\x12\x38\n\tsubmitter\x18\x06 \x01(\tB\x1a\xc2\xf3\x18\x16\n\tsubmitter\x12\tSubmitterR\tsubmitter\"\x90\x03\n\x0e\x41nnotationInfo\x12$\n\x04name\x18\x01 \x01(\tB\x10\xc2\xf3\x18\x0c\n\x04name\x12\x04NameR\x04name\x12,\n\x06source\x18\x02 \x01(\tB\x14\xc2\xf3\x18\x10\n\x06source\x12\x06SourceR\x06source\x12\x43\n\x0crelease_date\x18\x03 \x01(\tB \xc2\xf3\x18\x1c\n\x0crelease-date\x12\x0cRelease DateR\x0breleaseDate\x12;\n\nreport_url\x18\x04 \x01(\tB\x1c\xc2\xf3\x18\x18\n\nreport-url\x12\nReport URLR\treportUrl\x12W\n\x05stats\x18\x05 \x01(\x0b\x32\'.ncbi.datasets.v1.reports.FeatureCountsB\x18\xc2\xf3\x18\x14\n\nfeatcount-\x12\x06\x43ount R\x05stats\x12O\n\x05\x62usco\x18\x06 \x01(\x0b\x32#.ncbi.datasets.v1.reports.BuscoStatB\x14\xc2\xf3\x18\x10\n\x06\x62usco-\x12\x06\x42USCO R\x05\x62usco\"j\n\rFeatureCounts\x12Y\n\x0bgene_counts\x18\x01 \x01(\x0b\x32$.ncbi.datasets.v1.reports.GeneCountsB\x12\xc2\xf3\x18\x0e\n\x05gene-\x12\x05Gene R\ngeneCounts\"\xdd\x03\n\tBuscoStat\x12;\n\rbusco_lineage\x18\x01 \x01(\tB\x16\xc2\xf3\x18\x12\n\x07lineage\x12\x07LineageR\x0c\x62uscoLineage\x12/\n\tbusco_ver\x18\x02 \x01(\tB\x12\xc2\xf3\x18\x0e\n\x03ver\x12\x07VersionR\x08\x62uscoVer\x12\x34\n\x08\x63omplete\x18\x03 \x01(\x02\x42\x18\xc2\xf3\x18\x14\n\x08\x63omplete\x12\x08\x43ompleteR\x08\x63omplete\x12>\n\x0bsingle_copy\x18\x04 \x01(\x02\x42\x1d\xc2\xf3\x18\x19\n\nsinglecopy\x12\x0bSingle CopyR\nsingleCopy\x12<\n\nduplicated\x18\x05 \x01(\x02\x42\x1c\xc2\xf3\x18\x18\n\nduplicated\x12\nDuplicatedR\nduplicated\x12<\n\nfragmented\x18\x06 \x01(\x02\x42\x1c\xc2\xf3\x18\x18\n\nfragmented\x12\nFragmentedR\nfragmented\x12\x30\n\x07missing\x18\x07 \x01(\x02\x42\x16\xc2\xf3\x18\x12\n\x07missing\x12\x07MissingR\x07missing\x12>\n\x0btotal_count\x18\x08 \x01(\x04\x42\x1d\xc2\xf3\x18\x19\n\ntotalcount\x12\x0bTotal CountR\ntotalCount\"\xa8\x02\n\nGeneCounts\x12(\n\x05total\x18\x01 \x01(\rB\x12\xc2\xf3\x18\x0e\n\x05total\x12\x05TotalR\x05total\x12K\n\x0eprotein_coding\x18\x02 \x01(\rB$\xc2\xf3\x18 \n\x0eprotein-coding\x12\x0eProtein-codingR\rproteinCoding\x12;\n\nnon_coding\x18\x03 \x01(\rB\x1c\xc2\xf3\x18\x18\n\nnon-coding\x12\nNon-codingR\tnonCoding\x12<\n\npseudogene\x18\x04 \x01(\rB\x1c\xc2\xf3\x18\x18\n\npseudogene\x12\nPseudogeneR\npseudogene\x12(\n\x05other\x18\x05 \x01(\rB\x12\xc2\xf3\x18\x0e\n\x05other\x12\x05OtherR\x05other\"\xde\x02\n\x07WGSInfo\x12p\n\x15wgs_project_accession\x18\x01 \x01(\tB<\xc2\xf3\x18\x38\n\x11project-accession\x12\x11project accession2\x06\x41\x41\x45X032\x08\x43\x41\x42HLF01R\x13wgsProjectAccession\x12i\n\x0emaster_wgs_url\x18\x02 \x01(\tBC\xc2\xf3\x18?\n\x03url\x12\x03URL23https://www.ncbi.nlm.nih.gov/nuccore/AAEX00000000.3R\x0cmasterWgsUrl\x12v\n\x0fwgs_contigs_url\x18\x03 \x01(\tBN\xc2\xf3\x18J\n\x0b\x63ontigs-url\x12\x0b\x63ontigs URL2.https://www.ncbi.nlm.nih.gov/Traces/wgs/AAEX03R\rwgsContigsUrl*e\n\x0e\x41ssemblyStatus\x12\x1b\n\x17\x41SSEMBLY_STATUS_UNKNOWN\x10\x00\x12\x0b\n\x07\x63urrent\x10\x01\x12\x0c\n\x08previous\x10\x02\x12\x0e\n\nsuppressed\x10\x03\x12\x0b\n\x07retired\x10\x04\x42\xf9\nZ\x18ncbi/datasets/v1/reports\xf8\x01\x01\xc2\xf3\x18\xd7\n\n\x1bGenome Assembly Data Report\x12\x12\x41ssemblyDataReport\x1a;<p>The downloaded genome package contains a genome assembly\x1aIdata report in <a href=\"https://jsonlines.readthedocs.io/\">JSON Lines</a>\x1aPformat in the file: <br><br><em>ncbi_dataset/data/data_report.jsonl</em><br><br>\x1axEach line of the genome assembly data report file is a hierarchical <a href=\"https://www.json.org/json-en.html\">JSON</a>\x1ayobject that represents a single genome assembly record. The schema of the genome assembly record is defined in the tables\x1apbelow where each row describes a single field in the report or a sub-structure, which is a collection of fields.\x1aJThe outermost structure of the report is <em>AssemblyDataReport</em>.</p> \x1aQ<p>Table fields that include a <em>Table Field Mnemonic</em> can be used with the\x1a\x90\x01<a href=\"/datasets/v1/docs/reference-docs/command-line/dataformat/\">dataformat</a> command-line tool\'s <nobr><code>--fields</code></nobr> option\x1a\x8e\x01Refer to the <a href=\"/datasets/v1/docs/reference-docs/command-line/dataformat/\">dataformat</a> CLI tool reference to see how you can use this\x1aPtool to transform assembly data reports from JSON Lines to tabular formats.</p> \"\x0bOrientation\"\x0bSeqRangeSet\"\x05Range\"\x0fLineageOrganism\"\x08Organism*\x91\x01{{< readfile-code-block lang=\"json\" file=\"/docs_samples/datasets-cli/genome-report.cli.output\" begin=\"START data-report\" end=\"END data-report\">}}b\x06proto3'
  ,
  dependencies=[ncbi_dot_datasets_dot_v1_dot_reports_dot_biosample__pb2.DESCRIPTOR,ncbi_dot_datasets_dot_v1_dot_reports_dot_common__pb2.DESCRIPTOR,ncbi_dot_datasets_dot_options_dot_report__pb2.DESCRIPTOR,])

_ASSEMBLYSTATUS = _descriptor.EnumDescriptor(
  name='AssemblyStatus',
  full_name='ncbi.datasets.v1.reports.AssemblyStatus',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='ASSEMBLY_STATUS_UNKNOWN', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='current', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='previous', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='suppressed', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='retired', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=7798,
  serialized_end=7899,
)
_sym_db.RegisterEnumDescriptor(_ASSEMBLYSTATUS)

AssemblyStatus = enum_type_wrapper.EnumTypeWrapper(_ASSEMBLYSTATUS)
ASSEMBLY_STATUS_UNKNOWN = 0
current = 1
previous = 2
suppressed = 3
retired = 4


_LINKEDASSEMBLY_LINKEDASSEMBLYTYPE = _descriptor.EnumDescriptor(
  name='LinkedAssemblyType',
  full_name='ncbi.datasets.v1.reports.LinkedAssembly.LinkedAssemblyType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='LINKED_ASSEMBLY_TYPE_UNKNOWN', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='alternate_pseudohaplotype_of_diploid', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='principal_pseudohaplotype_of_diploid', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='maternal_haplotype_of_diploid', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='paternal_haplotype_of_diploid', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='haplotype_1', index=5, number=6,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='haplotype_2', index=6, number=7,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='haplotype_3', index=7, number=8,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='haplotype_4', index=8, number=9,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='haploid', index=9, number=10,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=1622,
  serialized_end=1911,
)
_sym_db.RegisterEnumDescriptor(_LINKEDASSEMBLY_LINKEDASSEMBLYTYPE)


_ASSEMBLYDATAREPORT = _descriptor.Descriptor(
  name='AssemblyDataReport',
  full_name='ncbi.datasets.v1.reports.AssemblyDataReport',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='common_name', full_name='ncbi.datasets.v1.reports.AssemblyDataReport.common_name', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030B\n\013common-name\022\013Common name2\005Human2\tzebrafish2\024pacific white shrimp', json_name='commonName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='organism_name', full_name='ncbi.datasets.v1.reports.AssemblyDataReport.organism_name', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030Z\n\rorganism-name\022\rOrganism name2\014Homo sapiens2\024Arabidopsis thaliana2\026Canis lupus familiaris', json_name='organismName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='breed', full_name='ncbi.datasets.v1.reports.AssemblyDataReport.breed', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\037\n\005breed\022\005Breed2\010Hereford2\005boxer', json_name='breed', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='cultivar', full_name='ncbi.datasets.v1.reports.AssemblyDataReport.cultivar', index=3,
      number=4, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\031\n\010cultivar\022\010Cultivar2\003B73', json_name='cultivar', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='ecotype', full_name='ncbi.datasets.v1.reports.AssemblyDataReport.ecotype', index=4,
      number=5, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\032\n\007ecotype\022\007Ecotype2\006Alpine', json_name='ecotype', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='isolate', full_name='ncbi.datasets.v1.reports.AssemblyDataReport.isolate', index=5,
      number=6, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030L\n\007isolate\022\007Isolate2/L1 Dominette 01449 registration number 421906802\007Pmale09', json_name='isolate', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='sex', full_name='ncbi.datasets.v1.reports.AssemblyDataReport.sex', index=6,
      number=7, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\022\n\003sex\022\003Sex2\006female', json_name='sex', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='strain', full_name='ncbi.datasets.v1.reports.AssemblyDataReport.strain', index=7,
      number=8, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\020\n\006strain\022\006Strain', json_name='strain', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='tax_id', full_name='ncbi.datasets.v1.reports.AssemblyDataReport.tax_id', index=8,
      number=9, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\026\n\006tax-id\022\014Taxonomic ID', json_name='taxId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_info', full_name='ncbi.datasets.v1.reports.AssemblyDataReport.assembly_info', index=9,
      number=10, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\026\n\tassminfo-\022\tAssembly ', json_name='assemblyInfo', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_stats', full_name='ncbi.datasets.v1.reports.AssemblyDataReport.assembly_stats', index=10,
      number=11, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\035\n\nassmstats-\022\017Assembly Stats ', json_name='assemblyStats', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='organelle_info', full_name='ncbi.datasets.v1.reports.AssemblyDataReport.organelle_info', index=11,
      number=12, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\030\n\norganelle-\022\nOrganelle ', json_name='organelleInfo', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='annotation_info', full_name='ncbi.datasets.v1.reports.AssemblyDataReport.annotation_info', index=12,
      number=13, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\036\n\nannotinfo-\022\020Annotation Info ', json_name='annotationInfo', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='wgs_info', full_name='ncbi.datasets.v1.reports.AssemblyDataReport.wgs_info', index=13,
      number=14, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\014\n\004wgs-\022\004WGS ', json_name='wgsInfo', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=187,
  serialized_end=1398,
)


_LINKEDASSEMBLY = _descriptor.Descriptor(
  name='LinkedAssembly',
  full_name='ncbi.datasets.v1.reports.LinkedAssembly',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='linked_assembly', full_name='ncbi.datasets.v1.reports.LinkedAssembly.linked_assembly', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\'\n\taccession\022\tAccession2\017GCA_000212995.1', json_name='linkedAssembly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_type', full_name='ncbi.datasets.v1.reports.LinkedAssembly.assembly_type', index=1,
      number=2, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\014\n\004type\022\004Type', json_name='assemblyType', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _LINKEDASSEMBLY_LINKEDASSEMBLYTYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=1401,
  serialized_end=1911,
)


_PAIREDASSEMBLY = _descriptor.Descriptor(
  name='PairedAssembly',
  full_name='ncbi.datasets.v1.reports.PairedAssembly',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accession', full_name='ncbi.datasets.v1.reports.PairedAssembly.accession', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030(\n\taccession\022\tAccession2\020GCF_000001405.40', json_name='accession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='status', full_name='ncbi.datasets.v1.reports.PairedAssembly.status', index=1,
      number=2, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\031\n\006status\022\006Status2\007current', json_name='status', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='annotation_name', full_name='ncbi.datasets.v1.reports.PairedAssembly.annotation_name', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\014\n\004name\022\004Name', json_name='annotationName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=1914,
  serialized_end=2162,
)


_ATYPICALINFO = _descriptor.Descriptor(
  name='AtypicalInfo',
  full_name='ncbi.datasets.v1.reports.AtypicalInfo',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='is_atypical', full_name='ncbi.datasets.v1.reports.AtypicalInfo.is_atypical', index=0,
      number=1, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\032\n\013is-atypical\022\013Is Atypical', json_name='isAtypical', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='warnings', full_name='ncbi.datasets.v1.reports.AtypicalInfo.warnings', index=1,
      number=3, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\024\n\010warnings\022\010Warnings', json_name='warnings', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=2165,
  serialized_end=2298,
)


_ASSEMBLYINFO = _descriptor.Descriptor(
  name='AssemblyInfo',
  full_name='ncbi.datasets.v1.reports.AssemblyInfo',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='assembly_accession', full_name='ncbi.datasets.v1.reports.AssemblyInfo.assembly_accession', index=0,
      number=16, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030(\n\taccession\022\tAccession2\020GCF_000001405.40', json_name='assemblyAccession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='current_assembly_accession', full_name='ncbi.datasets.v1.reports.AssemblyInfo.current_assembly_accession', index=1,
      number=25, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\0308\n\021current-accession\022\021Current Accession2\020GCF_000001405.40', json_name='currentAssemblyAccession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_status', full_name='ncbi.datasets.v1.reports.AssemblyInfo.assembly_status', index=2,
      number=22, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\031\n\006status\022\006Status2\007current', json_name='assemblyStatus', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='paired_assembly_accession', full_name='ncbi.datasets.v1.reports.AssemblyInfo.paired_assembly_accession', index=3,
      number=17, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\0306\n\020paired_accession\022\020Paired Accession2\020GCA_000001405.28', json_name='pairedAssemblyAccession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='paired_assembly', full_name='ncbi.datasets.v1.reports.AssemblyInfo.paired_assembly', index=4,
      number=23, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\037\n\013paired-assm\022\020Paired Assembly ', json_name='pairedAssembly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_level', full_name='ncbi.datasets.v1.reports.AssemblyInfo.assembly_level', index=5,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030,\n\005level\022\005Level2\nchromosome2\010scaffold2\006contig', json_name='assemblyLevel', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_name', full_name='ncbi.datasets.v1.reports.AssemblyInfo.assembly_name', index=6,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030%\n\004name\022\004Name2\nGRCh38.p142\013ASM985889v3', json_name='assemblyName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_type', full_name='ncbi.datasets.v1.reports.AssemblyInfo.assembly_type', index=7,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030,\n\004type\022\004Type2\025haploid-with-alt-loci2\007haploid', json_name='assemblyType', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='bioproject_lineage', full_name='ncbi.datasets.v1.reports.AssemblyInfo.bioproject_lineage', index=8,
      number=13, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\032\n\013bioproject-\022\013BioProject ', json_name='bioprojectLineage', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='submission_date', full_name='ncbi.datasets.v1.reports.AssemblyInfo.submission_date', index=9,
      number=5, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\"\n\017submission-date\022\017Submission Date', json_name='submissionDate', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='description', full_name='ncbi.datasets.v1.reports.AssemblyInfo.description', index=10,
      number=6, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\032\n\013description\022\013Description', json_name='description', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='genbank_assm_accession', full_name='ncbi.datasets.v1.reports.AssemblyInfo.genbank_assm_accession', index=11,
      number=7, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030=\n\026genbank-assm-accession\022\021GenBank Accession2\020GCA_000001405.28', json_name='genbankAssmAccession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='submitter', full_name='ncbi.datasets.v1.reports.AssemblyInfo.submitter', index=12,
      number=8, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\026\n\tsubmitter\022\tSubmitter', json_name='submitter', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='refseq_category', full_name='ncbi.datasets.v1.reports.AssemblyInfo.refseq_category', index=13,
      number=9, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030K\n\017refseq-category\022\017Refseq Category2\020reference genome2\025representative genome', json_name='refseqCategory', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='refseq_assm_accession', full_name='ncbi.datasets.v1.reports.AssemblyInfo.refseq_assm_accession', index=14,
      number=10, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030;\n\025refseq-assm-accession\022\020RefSeq Accession2\020GCF_000001405.40', json_name='refseqAssmAccession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='ucsc_assm_name', full_name='ncbi.datasets.v1.reports.AssemblyInfo.ucsc_assm_name', index=15,
      number=11, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030*\n\016ucsc-assm-name\022\022UCSC Assembly Name2\004hg38', json_name='ucscAssmName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='linked_assembly', full_name='ncbi.datasets.v1.reports.AssemblyInfo.linked_assembly', index=16,
      number=12, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='linkedAssembly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='linked_assemblies', full_name='ncbi.datasets.v1.reports.AssemblyInfo.linked_assemblies', index=17,
      number=21, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\037\n\013linked-assm\022\020Linked Assembly ', json_name='linkedAssemblies', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='atypical', full_name='ncbi.datasets.v1.reports.AssemblyInfo.atypical', index=18,
      number=24, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\025\n\010atypical\022\tAtypical ', json_name='atypical', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='sequencing_tech', full_name='ncbi.datasets.v1.reports.AssemblyInfo.sequencing_tech', index=19,
      number=14, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\"\n\017sequencing-tech\022\017Sequencing Tech', json_name='sequencingTech', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='biosample_accession', full_name='ncbi.datasets.v1.reports.AssemblyInfo.biosample_accession', index=20,
      number=15, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\0308\n\023biosample-accession\022\023BioSample Accession2\014SAMN03145444', json_name='biosampleAccession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='biosample', full_name='ncbi.datasets.v1.reports.AssemblyInfo.biosample', index=21,
      number=19, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\030\n\nbiosample-\022\nBioSample ', json_name='biosample', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='blast_url', full_name='ncbi.datasets.v1.reports.AssemblyInfo.blast_url', index=22,
      number=18, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\026\n\tblast-url\022\tBlast URL', json_name='blastUrl', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='comments', full_name='ncbi.datasets.v1.reports.AssemblyInfo.comments', index=23,
      number=20, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='comments', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=2301,
  serialized_end=4565,
)


_ASSEMBLYSTATS = _descriptor.Descriptor(
  name='AssemblyStats',
  full_name='ncbi.datasets.v1.reports.AssemblyStats',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='total_number_of_chromosomes', full_name='ncbi.datasets.v1.reports.AssemblyStats.total_number_of_chromosomes', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030:\n\033total-number-of-chromosomes\022\033Total Number of Chromosomes', json_name='totalNumberOfChromosomes', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='total_sequence_length', full_name='ncbi.datasets.v1.reports.AssemblyStats.total_sequence_length', index=1,
      number=2, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030+\n\022total-sequence-len\022\025Total Sequence Length', json_name='totalSequenceLength', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='total_ungapped_length', full_name='ncbi.datasets.v1.reports.AssemblyStats.total_ungapped_length', index=2,
      number=3, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030+\n\022total-ungapped-len\022\025Total Ungapped Length', json_name='totalUngappedLength', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='number_of_contigs', full_name='ncbi.datasets.v1.reports.AssemblyStats.number_of_contigs', index=3,
      number=4, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030&\n\021number-of-contigs\022\021Number of Contigs', json_name='numberOfContigs', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='contig_n50', full_name='ncbi.datasets.v1.reports.AssemblyStats.contig_n50', index=4,
      number=5, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\030\n\ncontig-n50\022\nContig N50', json_name='contigN50', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='contig_l50', full_name='ncbi.datasets.v1.reports.AssemblyStats.contig_l50', index=5,
      number=6, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\030\n\ncontig-l50\022\nContig L50', json_name='contigL50', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='number_of_scaffolds', full_name='ncbi.datasets.v1.reports.AssemblyStats.number_of_scaffolds', index=6,
      number=7, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030*\n\023number-of-scaffolds\022\023Number of Scaffolds', json_name='numberOfScaffolds', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='scaffold_n50', full_name='ncbi.datasets.v1.reports.AssemblyStats.scaffold_n50', index=7,
      number=8, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\034\n\014scaffold-n50\022\014Scaffold N50', json_name='scaffoldN50', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='scaffold_l50', full_name='ncbi.datasets.v1.reports.AssemblyStats.scaffold_l50', index=8,
      number=9, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\034\n\014scaffold-l50\022\014Scaffold L50', json_name='scaffoldL50', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='gaps_between_scaffolds_count', full_name='ncbi.datasets.v1.reports.AssemblyStats.gaps_between_scaffolds_count', index=9,
      number=10, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030<\n\034gaps-between-scaffolds-count\022\034Gaps Between Scaffolds Count', json_name='gapsBetweenScaffoldsCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='number_of_component_sequences', full_name='ncbi.datasets.v1.reports.AssemblyStats.number_of_component_sequences', index=10,
      number=11, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030>\n\035number-of-component-sequences\022\035Number of Component Sequences', json_name='numberOfComponentSequences', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='gc_count', full_name='ncbi.datasets.v1.reports.AssemblyStats.gc_count', index=11,
      number=12, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\024\n\010gc-count\022\010GC Count', json_name='gcCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=4568,
  serialized_end=5677,
)


_ORGANELLEINFO = _descriptor.Descriptor(
  name='OrganelleInfo',
  full_name='ncbi.datasets.v1.reports.OrganelleInfo',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='assembly_name', full_name='ncbi.datasets.v1.reports.OrganelleInfo.assembly_name', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\036\n\rassembly-name\022\rAssembly Name', json_name='assemblyName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='infraspecific_name', full_name='ncbi.datasets.v1.reports.OrganelleInfo.infraspecific_name', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030(\n\022infraspecific-name\022\022Infraspecific Name', json_name='infraspecificName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='bioproject', full_name='ncbi.datasets.v1.reports.OrganelleInfo.bioproject', index=2,
      number=3, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030.\n\025bioproject-accessions\022\025BioProject Accessions', json_name='bioproject', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='description', full_name='ncbi.datasets.v1.reports.OrganelleInfo.description', index=3,
      number=4, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\032\n\013description\022\013Description', json_name='description', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='total_seq_length', full_name='ncbi.datasets.v1.reports.OrganelleInfo.total_seq_length', index=4,
      number=5, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030$\n\020total-seq-length\022\020Total Seq Length', json_name='totalSeqLength', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='submitter', full_name='ncbi.datasets.v1.reports.OrganelleInfo.submitter', index=5,
      number=6, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\026\n\tsubmitter\022\tSubmitter', json_name='submitter', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=5680,
  serialized_end=6153,
)


_ANNOTATIONINFO = _descriptor.Descriptor(
  name='AnnotationInfo',
  full_name='ncbi.datasets.v1.reports.AnnotationInfo',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='name', full_name='ncbi.datasets.v1.reports.AnnotationInfo.name', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\014\n\004name\022\004Name', json_name='name', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='source', full_name='ncbi.datasets.v1.reports.AnnotationInfo.source', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\020\n\006source\022\006Source', json_name='source', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='release_date', full_name='ncbi.datasets.v1.reports.AnnotationInfo.release_date', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\034\n\014release-date\022\014Release Date', json_name='releaseDate', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='report_url', full_name='ncbi.datasets.v1.reports.AnnotationInfo.report_url', index=3,
      number=4, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\030\n\nreport-url\022\nReport URL', json_name='reportUrl', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='stats', full_name='ncbi.datasets.v1.reports.AnnotationInfo.stats', index=4,
      number=5, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\024\n\nfeatcount-\022\006Count ', json_name='stats', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='busco', full_name='ncbi.datasets.v1.reports.AnnotationInfo.busco', index=5,
      number=6, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\020\n\006busco-\022\006BUSCO ', json_name='busco', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=6156,
  serialized_end=6556,
)


_FEATURECOUNTS = _descriptor.Descriptor(
  name='FeatureCounts',
  full_name='ncbi.datasets.v1.reports.FeatureCounts',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='gene_counts', full_name='ncbi.datasets.v1.reports.FeatureCounts.gene_counts', index=0,
      number=1, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\016\n\005gene-\022\005Gene ', json_name='geneCounts', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=6558,
  serialized_end=6664,
)


_BUSCOSTAT = _descriptor.Descriptor(
  name='BuscoStat',
  full_name='ncbi.datasets.v1.reports.BuscoStat',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='busco_lineage', full_name='ncbi.datasets.v1.reports.BuscoStat.busco_lineage', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\022\n\007lineage\022\007Lineage', json_name='buscoLineage', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='busco_ver', full_name='ncbi.datasets.v1.reports.BuscoStat.busco_ver', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\016\n\003ver\022\007Version', json_name='buscoVer', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='complete', full_name='ncbi.datasets.v1.reports.BuscoStat.complete', index=2,
      number=3, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\024\n\010complete\022\010Complete', json_name='complete', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='single_copy', full_name='ncbi.datasets.v1.reports.BuscoStat.single_copy', index=3,
      number=4, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\031\n\nsinglecopy\022\013Single Copy', json_name='singleCopy', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='duplicated', full_name='ncbi.datasets.v1.reports.BuscoStat.duplicated', index=4,
      number=5, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\030\n\nduplicated\022\nDuplicated', json_name='duplicated', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='fragmented', full_name='ncbi.datasets.v1.reports.BuscoStat.fragmented', index=5,
      number=6, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\030\n\nfragmented\022\nFragmented', json_name='fragmented', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='missing', full_name='ncbi.datasets.v1.reports.BuscoStat.missing', index=6,
      number=7, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\022\n\007missing\022\007Missing', json_name='missing', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='total_count', full_name='ncbi.datasets.v1.reports.BuscoStat.total_count', index=7,
      number=8, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\031\n\ntotalcount\022\013Total Count', json_name='totalCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=6667,
  serialized_end=7144,
)


_GENECOUNTS = _descriptor.Descriptor(
  name='GeneCounts',
  full_name='ncbi.datasets.v1.reports.GeneCounts',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='total', full_name='ncbi.datasets.v1.reports.GeneCounts.total', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\016\n\005total\022\005Total', json_name='total', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='protein_coding', full_name='ncbi.datasets.v1.reports.GeneCounts.protein_coding', index=1,
      number=2, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030 \n\016protein-coding\022\016Protein-coding', json_name='proteinCoding', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='non_coding', full_name='ncbi.datasets.v1.reports.GeneCounts.non_coding', index=2,
      number=3, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\030\n\nnon-coding\022\nNon-coding', json_name='nonCoding', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='pseudogene', full_name='ncbi.datasets.v1.reports.GeneCounts.pseudogene', index=3,
      number=4, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\030\n\npseudogene\022\nPseudogene', json_name='pseudogene', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='other', full_name='ncbi.datasets.v1.reports.GeneCounts.other', index=4,
      number=5, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\016\n\005other\022\005Other', json_name='other', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=7147,
  serialized_end=7443,
)


_WGSINFO = _descriptor.Descriptor(
  name='WGSInfo',
  full_name='ncbi.datasets.v1.reports.WGSInfo',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='wgs_project_accession', full_name='ncbi.datasets.v1.reports.WGSInfo.wgs_project_accession', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\0308\n\021project-accession\022\021project accession2\006AAEX032\010CABHLF01', json_name='wgsProjectAccession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='master_wgs_url', full_name='ncbi.datasets.v1.reports.WGSInfo.master_wgs_url', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030?\n\003url\022\003URL23https://www.ncbi.nlm.nih.gov/nuccore/AAEX00000000.3', json_name='masterWgsUrl', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='wgs_contigs_url', full_name='ncbi.datasets.v1.reports.WGSInfo.wgs_contigs_url', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030J\n\013contigs-url\022\013contigs URL2.https://www.ncbi.nlm.nih.gov/Traces/wgs/AAEX03', json_name='wgsContigsUrl', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=7446,
  serialized_end=7796,
)

_ASSEMBLYDATAREPORT.fields_by_name['assembly_info'].message_type = _ASSEMBLYINFO
_ASSEMBLYDATAREPORT.fields_by_name['assembly_stats'].message_type = _ASSEMBLYSTATS
_ASSEMBLYDATAREPORT.fields_by_name['organelle_info'].message_type = _ORGANELLEINFO
_ASSEMBLYDATAREPORT.fields_by_name['annotation_info'].message_type = _ANNOTATIONINFO
_ASSEMBLYDATAREPORT.fields_by_name['wgs_info'].message_type = _WGSINFO
_LINKEDASSEMBLY.fields_by_name['assembly_type'].enum_type = _LINKEDASSEMBLY_LINKEDASSEMBLYTYPE
_LINKEDASSEMBLY_LINKEDASSEMBLYTYPE.containing_type = _LINKEDASSEMBLY
_PAIREDASSEMBLY.fields_by_name['status'].enum_type = _ASSEMBLYSTATUS
_ASSEMBLYINFO.fields_by_name['assembly_status'].enum_type = _ASSEMBLYSTATUS
_ASSEMBLYINFO.fields_by_name['paired_assembly'].message_type = _PAIREDASSEMBLY
_ASSEMBLYINFO.fields_by_name['bioproject_lineage'].message_type = ncbi_dot_datasets_dot_v1_dot_reports_dot_common__pb2._BIOPROJECTLINEAGE
_ASSEMBLYINFO.fields_by_name['linked_assemblies'].message_type = _LINKEDASSEMBLY
_ASSEMBLYINFO.fields_by_name['atypical'].message_type = _ATYPICALINFO
_ASSEMBLYINFO.fields_by_name['biosample'].message_type = ncbi_dot_datasets_dot_v1_dot_reports_dot_biosample__pb2._BIOSAMPLEDESCRIPTOR
_ANNOTATIONINFO.fields_by_name['stats'].message_type = _FEATURECOUNTS
_ANNOTATIONINFO.fields_by_name['busco'].message_type = _BUSCOSTAT
_FEATURECOUNTS.fields_by_name['gene_counts'].message_type = _GENECOUNTS
DESCRIPTOR.message_types_by_name['AssemblyDataReport'] = _ASSEMBLYDATAREPORT
DESCRIPTOR.message_types_by_name['LinkedAssembly'] = _LINKEDASSEMBLY
DESCRIPTOR.message_types_by_name['PairedAssembly'] = _PAIREDASSEMBLY
DESCRIPTOR.message_types_by_name['AtypicalInfo'] = _ATYPICALINFO
DESCRIPTOR.message_types_by_name['AssemblyInfo'] = _ASSEMBLYINFO
DESCRIPTOR.message_types_by_name['AssemblyStats'] = _ASSEMBLYSTATS
DESCRIPTOR.message_types_by_name['OrganelleInfo'] = _ORGANELLEINFO
DESCRIPTOR.message_types_by_name['AnnotationInfo'] = _ANNOTATIONINFO
DESCRIPTOR.message_types_by_name['FeatureCounts'] = _FEATURECOUNTS
DESCRIPTOR.message_types_by_name['BuscoStat'] = _BUSCOSTAT
DESCRIPTOR.message_types_by_name['GeneCounts'] = _GENECOUNTS
DESCRIPTOR.message_types_by_name['WGSInfo'] = _WGSINFO
DESCRIPTOR.enum_types_by_name['AssemblyStatus'] = _ASSEMBLYSTATUS
_sym_db.RegisterFileDescriptor(DESCRIPTOR)

AssemblyDataReport = _reflection.GeneratedProtocolMessageType('AssemblyDataReport', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYDATAREPORT,
  '__module__' : 'ncbi.datasets.v1.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.AssemblyDataReport)
  })
_sym_db.RegisterMessage(AssemblyDataReport)

LinkedAssembly = _reflection.GeneratedProtocolMessageType('LinkedAssembly', (_message.Message,), {
  'DESCRIPTOR' : _LINKEDASSEMBLY,
  '__module__' : 'ncbi.datasets.v1.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.LinkedAssembly)
  })
_sym_db.RegisterMessage(LinkedAssembly)

PairedAssembly = _reflection.GeneratedProtocolMessageType('PairedAssembly', (_message.Message,), {
  'DESCRIPTOR' : _PAIREDASSEMBLY,
  '__module__' : 'ncbi.datasets.v1.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.PairedAssembly)
  })
_sym_db.RegisterMessage(PairedAssembly)

AtypicalInfo = _reflection.GeneratedProtocolMessageType('AtypicalInfo', (_message.Message,), {
  'DESCRIPTOR' : _ATYPICALINFO,
  '__module__' : 'ncbi.datasets.v1.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.AtypicalInfo)
  })
_sym_db.RegisterMessage(AtypicalInfo)

AssemblyInfo = _reflection.GeneratedProtocolMessageType('AssemblyInfo', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYINFO,
  '__module__' : 'ncbi.datasets.v1.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.AssemblyInfo)
  })
_sym_db.RegisterMessage(AssemblyInfo)

AssemblyStats = _reflection.GeneratedProtocolMessageType('AssemblyStats', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYSTATS,
  '__module__' : 'ncbi.datasets.v1.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.AssemblyStats)
  })
_sym_db.RegisterMessage(AssemblyStats)

OrganelleInfo = _reflection.GeneratedProtocolMessageType('OrganelleInfo', (_message.Message,), {
  'DESCRIPTOR' : _ORGANELLEINFO,
  '__module__' : 'ncbi.datasets.v1.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.OrganelleInfo)
  })
_sym_db.RegisterMessage(OrganelleInfo)

AnnotationInfo = _reflection.GeneratedProtocolMessageType('AnnotationInfo', (_message.Message,), {
  'DESCRIPTOR' : _ANNOTATIONINFO,
  '__module__' : 'ncbi.datasets.v1.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.AnnotationInfo)
  })
_sym_db.RegisterMessage(AnnotationInfo)

FeatureCounts = _reflection.GeneratedProtocolMessageType('FeatureCounts', (_message.Message,), {
  'DESCRIPTOR' : _FEATURECOUNTS,
  '__module__' : 'ncbi.datasets.v1.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.FeatureCounts)
  })
_sym_db.RegisterMessage(FeatureCounts)

BuscoStat = _reflection.GeneratedProtocolMessageType('BuscoStat', (_message.Message,), {
  'DESCRIPTOR' : _BUSCOSTAT,
  '__module__' : 'ncbi.datasets.v1.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.BuscoStat)
  })
_sym_db.RegisterMessage(BuscoStat)

GeneCounts = _reflection.GeneratedProtocolMessageType('GeneCounts', (_message.Message,), {
  'DESCRIPTOR' : _GENECOUNTS,
  '__module__' : 'ncbi.datasets.v1.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.GeneCounts)
  })
_sym_db.RegisterMessage(GeneCounts)

WGSInfo = _reflection.GeneratedProtocolMessageType('WGSInfo', (_message.Message,), {
  'DESCRIPTOR' : _WGSINFO,
  '__module__' : 'ncbi.datasets.v1.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.WGSInfo)
  })
_sym_db.RegisterMessage(WGSInfo)


DESCRIPTOR._options = None
_ASSEMBLYDATAREPORT.fields_by_name['common_name']._options = None
_ASSEMBLYDATAREPORT.fields_by_name['organism_name']._options = None
_ASSEMBLYDATAREPORT.fields_by_name['breed']._options = None
_ASSEMBLYDATAREPORT.fields_by_name['cultivar']._options = None
_ASSEMBLYDATAREPORT.fields_by_name['ecotype']._options = None
_ASSEMBLYDATAREPORT.fields_by_name['isolate']._options = None
_ASSEMBLYDATAREPORT.fields_by_name['sex']._options = None
_ASSEMBLYDATAREPORT.fields_by_name['strain']._options = None
_ASSEMBLYDATAREPORT.fields_by_name['tax_id']._options = None
_ASSEMBLYDATAREPORT.fields_by_name['assembly_info']._options = None
_ASSEMBLYDATAREPORT.fields_by_name['assembly_stats']._options = None
_ASSEMBLYDATAREPORT.fields_by_name['organelle_info']._options = None
_ASSEMBLYDATAREPORT.fields_by_name['annotation_info']._options = None
_ASSEMBLYDATAREPORT.fields_by_name['wgs_info']._options = None
_LINKEDASSEMBLY.fields_by_name['linked_assembly']._options = None
_LINKEDASSEMBLY.fields_by_name['assembly_type']._options = None
_PAIREDASSEMBLY.fields_by_name['accession']._options = None
_PAIREDASSEMBLY.fields_by_name['status']._options = None
_PAIREDASSEMBLY.fields_by_name['annotation_name']._options = None
_ATYPICALINFO.fields_by_name['is_atypical']._options = None
_ATYPICALINFO.fields_by_name['warnings']._options = None
_ASSEMBLYINFO.fields_by_name['assembly_accession']._options = None
_ASSEMBLYINFO.fields_by_name['current_assembly_accession']._options = None
_ASSEMBLYINFO.fields_by_name['assembly_status']._options = None
_ASSEMBLYINFO.fields_by_name['paired_assembly_accession']._options = None
_ASSEMBLYINFO.fields_by_name['paired_assembly']._options = None
_ASSEMBLYINFO.fields_by_name['assembly_level']._options = None
_ASSEMBLYINFO.fields_by_name['assembly_name']._options = None
_ASSEMBLYINFO.fields_by_name['assembly_type']._options = None
_ASSEMBLYINFO.fields_by_name['bioproject_lineage']._options = None
_ASSEMBLYINFO.fields_by_name['submission_date']._options = None
_ASSEMBLYINFO.fields_by_name['description']._options = None
_ASSEMBLYINFO.fields_by_name['genbank_assm_accession']._options = None
_ASSEMBLYINFO.fields_by_name['submitter']._options = None
_ASSEMBLYINFO.fields_by_name['refseq_category']._options = None
_ASSEMBLYINFO.fields_by_name['refseq_assm_accession']._options = None
_ASSEMBLYINFO.fields_by_name['ucsc_assm_name']._options = None
_ASSEMBLYINFO.fields_by_name['linked_assembly']._options = None
_ASSEMBLYINFO.fields_by_name['linked_assemblies']._options = None
_ASSEMBLYINFO.fields_by_name['atypical']._options = None
_ASSEMBLYINFO.fields_by_name['sequencing_tech']._options = None
_ASSEMBLYINFO.fields_by_name['biosample_accession']._options = None
_ASSEMBLYINFO.fields_by_name['biosample']._options = None
_ASSEMBLYINFO.fields_by_name['blast_url']._options = None
_ASSEMBLYSTATS.fields_by_name['total_number_of_chromosomes']._options = None
_ASSEMBLYSTATS.fields_by_name['total_sequence_length']._options = None
_ASSEMBLYSTATS.fields_by_name['total_ungapped_length']._options = None
_ASSEMBLYSTATS.fields_by_name['number_of_contigs']._options = None
_ASSEMBLYSTATS.fields_by_name['contig_n50']._options = None
_ASSEMBLYSTATS.fields_by_name['contig_l50']._options = None
_ASSEMBLYSTATS.fields_by_name['number_of_scaffolds']._options = None
_ASSEMBLYSTATS.fields_by_name['scaffold_n50']._options = None
_ASSEMBLYSTATS.fields_by_name['scaffold_l50']._options = None
_ASSEMBLYSTATS.fields_by_name['gaps_between_scaffolds_count']._options = None
_ASSEMBLYSTATS.fields_by_name['number_of_component_sequences']._options = None
_ASSEMBLYSTATS.fields_by_name['gc_count']._options = None
_ORGANELLEINFO.fields_by_name['assembly_name']._options = None
_ORGANELLEINFO.fields_by_name['infraspecific_name']._options = None
_ORGANELLEINFO.fields_by_name['bioproject']._options = None
_ORGANELLEINFO.fields_by_name['description']._options = None
_ORGANELLEINFO.fields_by_name['total_seq_length']._options = None
_ORGANELLEINFO.fields_by_name['submitter']._options = None
_ANNOTATIONINFO.fields_by_name['name']._options = None
_ANNOTATIONINFO.fields_by_name['source']._options = None
_ANNOTATIONINFO.fields_by_name['release_date']._options = None
_ANNOTATIONINFO.fields_by_name['report_url']._options = None
_ANNOTATIONINFO.fields_by_name['stats']._options = None
_ANNOTATIONINFO.fields_by_name['busco']._options = None
_FEATURECOUNTS.fields_by_name['gene_counts']._options = None
_BUSCOSTAT.fields_by_name['busco_lineage']._options = None
_BUSCOSTAT.fields_by_name['busco_ver']._options = None
_BUSCOSTAT.fields_by_name['complete']._options = None
_BUSCOSTAT.fields_by_name['single_copy']._options = None
_BUSCOSTAT.fields_by_name['duplicated']._options = None
_BUSCOSTAT.fields_by_name['fragmented']._options = None
_BUSCOSTAT.fields_by_name['missing']._options = None
_BUSCOSTAT.fields_by_name['total_count']._options = None
_GENECOUNTS.fields_by_name['total']._options = None
_GENECOUNTS.fields_by_name['protein_coding']._options = None
_GENECOUNTS.fields_by_name['non_coding']._options = None
_GENECOUNTS.fields_by_name['pseudogene']._options = None
_GENECOUNTS.fields_by_name['other']._options = None
_WGSINFO.fields_by_name['wgs_project_accession']._options = None
_WGSINFO.fields_by_name['master_wgs_url']._options = None
_WGSINFO.fields_by_name['wgs_contigs_url']._options = None
# @@protoc_insertion_point(module_scope)
