# -*- coding: utf-8 -*-
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: ncbi/datasets/v1/reports/gene.proto
"""Generated protocol buffer code."""
from google.protobuf import descriptor as _descriptor
from google.protobuf import message as _message
from google.protobuf import reflection as _reflection
from google.protobuf import symbol_database as _symbol_database
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()


from ncbi.datasets.v1.reports import common_pb2 as ncbi_dot_datasets_dot_v1_dot_reports_dot_common__pb2
from ncbi.datasets.options import report_pb2 as ncbi_dot_datasets_dot_options_dot_report__pb2


DESCRIPTOR = _descriptor.FileDescriptor(
  name='ncbi/datasets/v1/reports/gene.proto',
  package='ncbi.datasets.v1.reports',
  syntax='proto3',
  serialized_options=b'Z\030ncbi/datasets/v1/reports\370\001\001\302\363\030\373\t\n\013Gene Report\022\016GeneDescriptor\032=<p>The downloaded gene package contains a gene data report in\032:<a href=\"https://jsonlines.readthedocs.io/\">JSON Lines</a>\032Oformat in the file:<br><br><em>ncbi_dataset/data/data_report.jsonl</em><br><br>\032mEach line of the gene data report file is a hierarchical <a href=\"https://www.json.org/json-en.html\">JSON</a>\032iobject that represents a single gene record. The schema of the gene record is defined in the tables below\032jwhere each row describes a single field in the report or a sub-structure, which is a collection of fields.\032EThe outermost structure of the report is <em>GeneDescriptor</em>.</p>\032Q<p>Table fields that include a <em>Table Field Mnemonic</em> can be used with the\032\221\001<a href=\"/datasets/docs/v1/reference-docs/command-line/dataformat/\">dataformat</a> command-line tool\'s <nobr><code>--fields</code></nobr> option.\032\223\001Refer to the <a href=\"/datasets/docs/v1/reference-docs/command-line/dataformat/\">dataformat</a> CLI tool reference to see how you can use this tool\032Fto transform gene data reports from JSON Lines to tabular formats.</p>\"\017LineageOrganism\"\010Organism\"\017GeneDescriptors*\221\001{{< readfile-code-block lang=\"json\" file=\"/docs_samples/dataformat-cli/gene-to-tsv.cli.output\" begin=\"START data-report\" end=\"END data-report\">}}',
  create_key=_descriptor._internal_create_key,
  serialized_pb=b'\n#ncbi/datasets/v1/reports/gene.proto\x12\x18ncbi.datasets.v1.reports\x1a%ncbi/datasets/v1/reports/common.proto\x1a\"ncbi/datasets/options/report.proto\"\xac\x01\n\rMaturePeptide\x12G\n\x11\x61\x63\x63\x65ssion_version\x18\x01 \x01(\tB\x1a\xc2\xf3\x18\x16\n\taccession\x12\tAccessionR\x10\x61\x63\x63\x65ssionVersion\x12$\n\x04name\x18\x02 \x01(\tB\x10\xc2\xf3\x18\x0c\n\x04name\x12\x04NameR\x04name\x12,\n\x06length\x18\x03 \x01(\rB\x14\xc2\xf3\x18\x10\n\x06length\x12\x06LengthR\x06length\"\xed\x03\n\x07Protein\x12W\n\x11\x61\x63\x63\x65ssion_version\x18\x01 \x01(\tB*\xc2\xf3\x18&\n\taccession\x12\tAccession2\x0eNP_001296812.1R\x10\x61\x63\x63\x65ssionVersion\x12\x32\n\x04name\x18\x02 \x01(\tB\x1e\xc2\xf3\x18\x1a\n\x04name\x12\x04Name2\x0cprotein ALEXR\x04name\x12\x31\n\x06length\x18\x03 \x01(\rB\x19\xc2\xf3\x18\x15\n\x06length\x12\x06Length2\x03\x36\x32\x36R\x06length\x12G\n\x0cisoform_name\x18\x05 \x01(\tB$\xc2\xf3\x18 \n\x07isoform\x12\x07Isoform2\x0cisoform AlexR\x0bisoformName\x12\x62\n\x0f\x65nsembl_protein\x18\x06 \x01(\tB9\xc2\xf3\x18\x35\n\x0f\x65nsembl-protein\x12\x0f\x45nsembl Protein2\x11\x45NSP00000302237.3R\x0e\x65nsemblProtein\x12u\n\x0fmature_peptides\x18\x04 \x03(\x0b\x32\'.ncbi.datasets.v1.reports.MaturePeptideB#\xc2\xf3\x18\x1f\n\x0cmat-peptide-\x12\x0fMature Peptide R\x0ematurePeptides\"\xc9\x02\n\x0fGenomicLocation\x12W\n\x19genomic_accession_version\x18\x01 \x01(\tB\x1b\xc2\xf3\x18\x17\n\taccession\x12\nAccession R\x17genomicAccessionVersion\x12=\n\rsequence_name\x18\x02 \x01(\tB\x18\xc2\xf3\x18\x14\n\x08seq-name\x12\x08Seq NameR\x0csequenceName\x12R\n\rgenomic_range\x18\x03 \x01(\x0b\x32\x1f.ncbi.datasets.v1.reports.RangeB\x0c\xc2\xf3\x18\x08\n\x06range-R\x0cgenomicRange\x12J\n\x05\x65xons\x18\x04 \x03(\x0b\x32\x1f.ncbi.datasets.v1.reports.RangeB\x13\xc2\xf3\x18\x0f\n\x05\x65xon-\x12\x06\x45xons R\x05\x65xons\"\xd3\x07\n\nTranscript\x12G\n\x11\x61\x63\x63\x65ssion_version\x18\x01 \x01(\tB\x1a\xc2\xf3\x18\x16\n\taccession\x12\tAccessionR\x10\x61\x63\x63\x65ssionVersion\x12\x46\n\x04name\x18\x02 \x01(\tB2\xc2\xf3\x18.\n\x04name\x12\x0fTranscript Name2\x15transcript variant 12R\x04name\x12=\n\x06length\x18\x03 \x01(\rB%\xc2\xf3\x18!\n\x06length\x12\x11Transcript Length2\x04\x33\x31\x38\x30R\x06length\x12N\n\rgenomic_range\x18\x04 \x01(\x0b\x32%.ncbi.datasets.v1.reports.SeqRangeSetB\x02\x18\x01R\x0cgenomicRange\x12?\n\x05\x65xons\x18\x05 \x01(\x0b\x32%.ncbi.datasets.v1.reports.SeqRangeSetB\x02\x18\x01R\x05\x65xons\x12I\n\x03\x63\x64s\x18\x06 \x01(\x0b\x32%.ncbi.datasets.v1.reports.SeqRangeSetB\x10\xc2\xf3\x18\x0c\n\x04\x63\x64s-\x12\x04\x43\x44S R\x03\x63\x64s\x12y\n\x11genomic_locations\x18\n \x03(\x0b\x32).ncbi.datasets.v1.reports.GenomicLocationB!\xc2\xf3\x18\x1d\n\x11genomic-location-\x12\x08Genomic R\x10genomicLocations\x12n\n\x12\x65nsembl_transcript\x18\x08 \x01(\tB?\xc2\xf3\x18;\n\x12\x65nsembl-transcript\x12\x12\x45nsembl Transcript2\x11\x45NST00000306120.3R\x11\x65nsemblTranscript\x12U\n\x07protein\x18\x07 \x01(\x0b\x32!.ncbi.datasets.v1.reports.ProteinB\x18\xc2\xf3\x18\x14\n\x08protein-\x12\x08Protein R\x07protein\x12\x64\n\x04type\x18\t \x01(\x0e\x32\x33.ncbi.datasets.v1.reports.Transcript.TranscriptTypeB\x1b\xc2\xf3\x18\x17\n\x0ftranscript-type\x12\x04TypeR\x04type\"q\n\x0eTranscriptType\x12\x0b\n\x07UNKNOWN\x10\x00\x12\x12\n\x0ePROTEIN_CODING\x10\x01\x12\x0e\n\nNON_CODING\x10\x02\x12\x18\n\x14PROTEIN_CODING_MODEL\x10\x03\x12\x14\n\x10NON_CODING_MODEL\x10\x04\"\xd9\x02\n\rGenomicRegion\x12\x64\n\ngene_range\x18\x01 \x01(\x0b\x32%.ncbi.datasets.v1.reports.SeqRangeSetB\x1e\xc2\xf3\x18\x1a\n\x0bgene-range-\x12\x0bGene Range R\tgeneRange\x12}\n\x04type\x18\x02 \x01(\x0e\x32\x39.ncbi.datasets.v1.reports.GenomicRegion.GenomicRegionTypeB.\xc2\xf3\x18*\n\x13genomic-region-type\x12\x13Genomic Region TypeR\x04type\"c\n\x11GenomicRegionType\x12\x0b\n\x07UNKNOWN\x10\x00\x12\x0f\n\x0bREFSEQ_GENE\x10\x01\x12\x0e\n\nPSEUDOGENE\x10\x02\x12\x15\n\x11\x42IOLOGICAL_REGION\x10\x03\x12\t\n\x05OTHER\x10\x04\"\xa8\x02\n\nAnnotation\x12\x43\n\x0crelease_name\x18\x01 \x01(\tB \xc2\xf3\x18\x1c\n\x0crelease-name\x12\x0cRelease NameR\x0breleaseName\x12\x43\n\x0crelease_date\x18\x02 \x01(\tB \xc2\xf3\x18\x1c\n\x0crelease-date\x12\x0cRelease DateR\x0breleaseDate\x12\x8f\x01\n\x13\x61ssemblies_in_scope\x18\x03 \x03(\x0b\x32-.ncbi.datasets.v1.reports.AnnotatedAssembliesB0\xc2\xf3\x18,\n\x14\x61ssemblies-in-scope-\x12\x14\x41ssemblies in Scope R\x11\x61ssembliesInScope\"u\n\x13\x41nnotatedAssemblies\x12\x38\n\taccession\x18\x01 \x01(\tB\x1a\xc2\xf3\x18\x16\n\taccession\x12\tAccessionR\taccession\x12$\n\x04name\x18\x02 \x01(\tB\x10\xc2\xf3\x18\x0c\n\x04name\x12\x04NameR\x04name\"\xc9\x11\n\x0eGeneDescriptor\x12\x39\n\x07gene_id\x18\x01 \x01(\x04\x42 \xc2\xf3\x18\x1c\n\x07gene-id\x12\x0bNCBI GeneID2\x04\x32\x37\x37\x38R\x06geneId\x12\x32\n\x06symbol\x18\x02 \x01(\tB\x1a\xc2\xf3\x18\x16\n\x06symbol\x12\x06Symbol2\x04GNASR\x06symbol\x12T\n\x0b\x64\x65scription\x18\x03 \x01(\tB2\xc2\xf3\x18.\n\x0b\x64\x65scription\x12\x0b\x44\x65scription2\x12GNAS complex locusR\x0b\x64\x65scription\x12\x37\n\x06tax_id\x18\x04 \x01(\x04\x42 \xc2\xf3\x18\x1c\n\x06tax-id\x12\x0cTaxonomic ID2\x04\x39\x36\x30\x36R\x05taxId\x12\x46\n\x07taxname\x18\x05 \x01(\tB,\xc2\xf3\x18(\n\x08tax-name\x12\x0eTaxonomic Name2\x0cHomo sapiensR\x07taxname\x12\x46\n\x0b\x63ommon_name\x18\x0c \x01(\tB%\xc2\xf3\x18!\n\x0b\x63ommon-name\x12\x0b\x43ommon Name2\x05humanR\ncommonName\x12\x61\n\x04type\x18\x06 \x01(\x0e\x32\x31.ncbi.datasets.v1.reports.GeneDescriptor.GeneTypeB\x1a\xc2\xf3\x18\x16\n\tgene-type\x12\tGene TypeR\x04type\x12\x65\n\x08rna_type\x18\x0b \x01(\x0e\x32\x30.ncbi.datasets.v1.reports.GeneDescriptor.RnaTypeB\x18\xc2\xf3\x18\x14\n\x08rna-type\x12\x08RNA TypeR\x07rnaType\x12g\n\x0borientation\x18\x07 \x01(\x0e\x32%.ncbi.datasets.v1.reports.OrientationB\x1e\xc2\xf3\x18\x1a\n\x0borientation\x12\x0bOrientationR\x0borientation\x12r\n\x0egenomic_ranges\x18\x08 \x03(\x0b\x32%.ncbi.datasets.v1.reports.SeqRangeSetB$\xc2\xf3\x18 \n\x0egenomic-range-\x12\x0eGenomic Range R\rgenomicRanges\x12\x82\x01\n\x13reference_standards\x18\x15 \x03(\x0b\x32\'.ncbi.datasets.v1.reports.GenomicRegionB(\xc2\xf3\x18$\n\rref-standard-\x12\x13Reference Standard R\x12referenceStandards\x12x\n\x0fgenomic_regions\x18\x16 \x03(\x0b\x32\'.ncbi.datasets.v1.reports.GenomicRegionB&\xc2\xf3\x18\"\n\x0fgenomic-region-\x12\x0fGenomic Region R\x0egenomicRegions\x12\x66\n\x0btranscripts\x18\t \x03(\x0b\x32$.ncbi.datasets.v1.reports.TranscriptB\x1e\xc2\xf3\x18\x1a\n\x0btranscript-\x12\x0bTranscript R\x0btranscripts\x12W\n\x08proteins\x18\n \x03(\x0b\x32!.ncbi.datasets.v1.reports.ProteinB\x18\xc2\xf3\x18\x14\n\x08protein-\x12\x08Protein R\x08proteins\x12\"\n\nchromosome\x18\r \x01(\tB\x02\x18\x01R\nchromosome\x12H\n\x0b\x63hromosomes\x18\x0e \x03(\tB&\xc2\xf3\x18\"\n\x0b\x63hromosomes\x12\x0b\x43hromosomes2\x01\x31\x32\x03X,YR\x0b\x63hromosomes\x12\x82\x01\n\x16nomenclature_authority\x18\x0f \x01(\x0b\x32/.ncbi.datasets.v1.reports.NomenclatureAuthorityB\x1a\xc2\xf3\x18\x16\n\x05name-\x12\rNomenclature R\x15nomenclatureAuthority\x12\x64\n\x15swiss_prot_accessions\x18\x10 \x03(\tB0\xc2\xf3\x18,\n\x14swissprot-accessions\x12\x14SwissProt AccessionsR\x13swissProtAccessions\x12P\n\x10\x65nsembl_gene_ids\x18\x11 \x03(\tB&\xc2\xf3\x18\"\n\x0f\x65nsembl-geneids\x12\x0f\x45nsembl GeneIDsR\x0e\x65nsemblGeneIds\x12\x33\n\x08omim_ids\x18\x12 \x03(\tB\x18\xc2\xf3\x18\x14\n\x08omim-ids\x12\x08OMIM IDsR\x07omimIds\x12\x34\n\x08synonyms\x18\x13 \x03(\tB\x18\xc2\xf3\x18\x14\n\x08synonyms\x12\x08SynonymsR\x08synonyms\x12V\n\x10replaced_gene_id\x18\x14 \x01(\x04\x42,\xc2\xf3\x18(\n\x10replaced-gene-id\x12\x14Replaced NCBI GeneIDR\x0ereplacedGeneId\x12\x66\n\x0b\x61nnotations\x18\x17 \x03(\x0b\x32$.ncbi.datasets.v1.reports.AnnotationB\x1e\xc2\xf3\x18\x1a\n\x0b\x61nnotation-\x12\x0b\x41nnotation R\x0b\x61nnotations\"\xb8\x01\n\x08GeneType\x12\x0b\n\x07UNKNOWN\x10\x00\x12\x08\n\x04tRNA\x10\x01\x12\x08\n\x04rRNA\x10\x02\x12\t\n\x05snRNA\x10\x03\x12\t\n\x05scRNA\x10\x04\x12\n\n\x06snoRNA\x10\x05\x12\x12\n\x0ePROTEIN_CODING\x10\x06\x12\n\n\x06PSEUDO\x10\x07\x12\x0e\n\nTRANSPOSON\x10\x08\x12\x0b\n\x07miscRNA\x10\t\x12\t\n\x05ncRNA\x10\n\x12\x15\n\x11\x42IOLOGICAL_REGION\x10\x0b\x12\n\n\x05OTHER\x10\xff\x01\"1\n\x07RnaType\x12\x0f\n\x0brna_UNKNOWN\x10\x00\x12\n\n\x06premsg\x10\x01\x12\t\n\x05tmRna\x10\x02\"\x90\x01\n\x15NomenclatureAuthority\x12>\n\tauthority\x18\x01 \x01(\tB \xc2\xf3\x18\x1c\n\tauthority\x12\tAuthority2\x04HGNCR\tauthority\x12\x37\n\nidentifier\x18\x02 \x01(\tB\x17\xc2\xf3\x18\x13\n\x02id\x12\x02ID2\tHGNC:4392R\nidentifier\"Q\n\x0fGeneDescriptors\x12>\n\x05genes\x18\x01 \x03(\x0b\x32(.ncbi.datasets.v1.reports.GeneDescriptorR\x05genesB\x9d\nZ\x18ncbi/datasets/v1/reports\xf8\x01\x01\xc2\xf3\x18\xfb\t\n\x0bGene Report\x12\x0eGeneDescriptor\x1a=<p>The downloaded gene package contains a gene data report in\x1a:<a href=\"https://jsonlines.readthedocs.io/\">JSON Lines</a>\x1aOformat in the file:<br><br><em>ncbi_dataset/data/data_report.jsonl</em><br><br>\x1amEach line of the gene data report file is a hierarchical <a href=\"https://www.json.org/json-en.html\">JSON</a>\x1aiobject that represents a single gene record. The schema of the gene record is defined in the tables below\x1ajwhere each row describes a single field in the report or a sub-structure, which is a collection of fields.\x1a\x45The outermost structure of the report is <em>GeneDescriptor</em>.</p>\x1aQ<p>Table fields that include a <em>Table Field Mnemonic</em> can be used with the\x1a\x91\x01<a href=\"/datasets/docs/v1/reference-docs/command-line/dataformat/\">dataformat</a> command-line tool\'s <nobr><code>--fields</code></nobr> option.\x1a\x93\x01Refer to the <a href=\"/datasets/docs/v1/reference-docs/command-line/dataformat/\">dataformat</a> CLI tool reference to see how you can use this tool\x1a\x46to transform gene data reports from JSON Lines to tabular formats.</p>\"\x0fLineageOrganism\"\x08Organism\"\x0fGeneDescriptors*\x91\x01{{< readfile-code-block lang=\"json\" file=\"/docs_samples/dataformat-cli/gene-to-tsv.cli.output\" begin=\"START data-report\" end=\"END data-report\">}}b\x06proto3'
  ,
  dependencies=[ncbi_dot_datasets_dot_v1_dot_reports_dot_common__pb2.DESCRIPTOR,ncbi_dot_datasets_dot_options_dot_report__pb2.DESCRIPTOR,])



_TRANSCRIPT_TRANSCRIPTTYPE = _descriptor.EnumDescriptor(
  name='TranscriptType',
  full_name='ncbi.datasets.v1.reports.Transcript.TranscriptType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='UNKNOWN', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PROTEIN_CODING', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='NON_CODING', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PROTEIN_CODING_MODEL', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='NON_CODING_MODEL', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=2010,
  serialized_end=2123,
)
_sym_db.RegisterEnumDescriptor(_TRANSCRIPT_TRANSCRIPTTYPE)

_GENOMICREGION_GENOMICREGIONTYPE = _descriptor.EnumDescriptor(
  name='GenomicRegionType',
  full_name='ncbi.datasets.v1.reports.GenomicRegion.GenomicRegionType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='UNKNOWN', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='REFSEQ_GENE', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PSEUDOGENE', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='BIOLOGICAL_REGION', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='OTHER', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=2372,
  serialized_end=2471,
)
_sym_db.RegisterEnumDescriptor(_GENOMICREGION_GENOMICREGIONTYPE)

_GENEDESCRIPTOR_GENETYPE = _descriptor.EnumDescriptor(
  name='GeneType',
  full_name='ncbi.datasets.v1.reports.GeneDescriptor.GeneType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='UNKNOWN', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='tRNA', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='rRNA', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='snRNA', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='scRNA', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='snoRNA', index=5, number=5,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PROTEIN_CODING', index=6, number=6,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PSEUDO', index=7, number=7,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='TRANSPOSON', index=8, number=8,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='miscRNA', index=9, number=9,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='ncRNA', index=10, number=10,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='BIOLOGICAL_REGION', index=11, number=11,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='OTHER', index=12, number=255,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=4906,
  serialized_end=5090,
)
_sym_db.RegisterEnumDescriptor(_GENEDESCRIPTOR_GENETYPE)

_GENEDESCRIPTOR_RNATYPE = _descriptor.EnumDescriptor(
  name='RnaType',
  full_name='ncbi.datasets.v1.reports.GeneDescriptor.RnaType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='rna_UNKNOWN', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='premsg', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='tmRna', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=5092,
  serialized_end=5141,
)
_sym_db.RegisterEnumDescriptor(_GENEDESCRIPTOR_RNATYPE)


_MATUREPEPTIDE = _descriptor.Descriptor(
  name='MaturePeptide',
  full_name='ncbi.datasets.v1.reports.MaturePeptide',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accession_version', full_name='ncbi.datasets.v1.reports.MaturePeptide.accession_version', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\026\n\taccession\022\tAccession', json_name='accessionVersion', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='name', full_name='ncbi.datasets.v1.reports.MaturePeptide.name', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\014\n\004name\022\004Name', json_name='name', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='length', full_name='ncbi.datasets.v1.reports.MaturePeptide.length', index=2,
      number=3, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\020\n\006length\022\006Length', json_name='length', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=141,
  serialized_end=313,
)


_PROTEIN = _descriptor.Descriptor(
  name='Protein',
  full_name='ncbi.datasets.v1.reports.Protein',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accession_version', full_name='ncbi.datasets.v1.reports.Protein.accession_version', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030&\n\taccession\022\tAccession2\016NP_001296812.1', json_name='accessionVersion', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='name', full_name='ncbi.datasets.v1.reports.Protein.name', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\032\n\004name\022\004Name2\014protein ALEX', json_name='name', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='length', full_name='ncbi.datasets.v1.reports.Protein.length', index=2,
      number=3, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\025\n\006length\022\006Length2\003626', json_name='length', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='isoform_name', full_name='ncbi.datasets.v1.reports.Protein.isoform_name', index=3,
      number=5, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030 \n\007isoform\022\007Isoform2\014isoform Alex', json_name='isoformName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='ensembl_protein', full_name='ncbi.datasets.v1.reports.Protein.ensembl_protein', index=4,
      number=6, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\0305\n\017ensembl-protein\022\017Ensembl Protein2\021ENSP00000302237.3', json_name='ensemblProtein', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='mature_peptides', full_name='ncbi.datasets.v1.reports.Protein.mature_peptides', index=5,
      number=4, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\037\n\014mat-peptide-\022\017Mature Peptide ', json_name='maturePeptides', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=316,
  serialized_end=809,
)


_GENOMICLOCATION = _descriptor.Descriptor(
  name='GenomicLocation',
  full_name='ncbi.datasets.v1.reports.GenomicLocation',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='genomic_accession_version', full_name='ncbi.datasets.v1.reports.GenomicLocation.genomic_accession_version', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\027\n\taccession\022\nAccession ', json_name='genomicAccessionVersion', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='sequence_name', full_name='ncbi.datasets.v1.reports.GenomicLocation.sequence_name', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\024\n\010seq-name\022\010Seq Name', json_name='sequenceName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='genomic_range', full_name='ncbi.datasets.v1.reports.GenomicLocation.genomic_range', index=2,
      number=3, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\010\n\006range-', json_name='genomicRange', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='exons', full_name='ncbi.datasets.v1.reports.GenomicLocation.exons', index=3,
      number=4, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\017\n\005exon-\022\006Exons ', json_name='exons', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=812,
  serialized_end=1141,
)


_TRANSCRIPT = _descriptor.Descriptor(
  name='Transcript',
  full_name='ncbi.datasets.v1.reports.Transcript',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accession_version', full_name='ncbi.datasets.v1.reports.Transcript.accession_version', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\026\n\taccession\022\tAccession', json_name='accessionVersion', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='name', full_name='ncbi.datasets.v1.reports.Transcript.name', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030.\n\004name\022\017Transcript Name2\025transcript variant 12', json_name='name', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='length', full_name='ncbi.datasets.v1.reports.Transcript.length', index=2,
      number=3, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030!\n\006length\022\021Transcript Length2\0043180', json_name='length', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='genomic_range', full_name='ncbi.datasets.v1.reports.Transcript.genomic_range', index=3,
      number=4, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='genomicRange', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='exons', full_name='ncbi.datasets.v1.reports.Transcript.exons', index=4,
      number=5, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='exons', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='cds', full_name='ncbi.datasets.v1.reports.Transcript.cds', index=5,
      number=6, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\014\n\004cds-\022\004CDS ', json_name='cds', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='genomic_locations', full_name='ncbi.datasets.v1.reports.Transcript.genomic_locations', index=6,
      number=10, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\035\n\021genomic-location-\022\010Genomic ', json_name='genomicLocations', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='ensembl_transcript', full_name='ncbi.datasets.v1.reports.Transcript.ensembl_transcript', index=7,
      number=8, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030;\n\022ensembl-transcript\022\022Ensembl Transcript2\021ENST00000306120.3', json_name='ensemblTranscript', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='protein', full_name='ncbi.datasets.v1.reports.Transcript.protein', index=8,
      number=7, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\024\n\010protein-\022\010Protein ', json_name='protein', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='type', full_name='ncbi.datasets.v1.reports.Transcript.type', index=9,
      number=9, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\027\n\017transcript-type\022\004Type', json_name='type', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _TRANSCRIPT_TRANSCRIPTTYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=1144,
  serialized_end=2123,
)


_GENOMICREGION = _descriptor.Descriptor(
  name='GenomicRegion',
  full_name='ncbi.datasets.v1.reports.GenomicRegion',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='gene_range', full_name='ncbi.datasets.v1.reports.GenomicRegion.gene_range', index=0,
      number=1, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\032\n\013gene-range-\022\013Gene Range ', json_name='geneRange', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='type', full_name='ncbi.datasets.v1.reports.GenomicRegion.type', index=1,
      number=2, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030*\n\023genomic-region-type\022\023Genomic Region Type', json_name='type', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _GENOMICREGION_GENOMICREGIONTYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=2126,
  serialized_end=2471,
)


_ANNOTATION = _descriptor.Descriptor(
  name='Annotation',
  full_name='ncbi.datasets.v1.reports.Annotation',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='release_name', full_name='ncbi.datasets.v1.reports.Annotation.release_name', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\034\n\014release-name\022\014Release Name', json_name='releaseName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='release_date', full_name='ncbi.datasets.v1.reports.Annotation.release_date', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\034\n\014release-date\022\014Release Date', json_name='releaseDate', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assemblies_in_scope', full_name='ncbi.datasets.v1.reports.Annotation.assemblies_in_scope', index=2,
      number=3, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030,\n\024assemblies-in-scope-\022\024Assemblies in Scope ', json_name='assembliesInScope', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=2474,
  serialized_end=2770,
)


_ANNOTATEDASSEMBLIES = _descriptor.Descriptor(
  name='AnnotatedAssemblies',
  full_name='ncbi.datasets.v1.reports.AnnotatedAssemblies',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accession', full_name='ncbi.datasets.v1.reports.AnnotatedAssemblies.accession', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\026\n\taccession\022\tAccession', json_name='accession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='name', full_name='ncbi.datasets.v1.reports.AnnotatedAssemblies.name', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\014\n\004name\022\004Name', json_name='name', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=2772,
  serialized_end=2889,
)


_GENEDESCRIPTOR = _descriptor.Descriptor(
  name='GeneDescriptor',
  full_name='ncbi.datasets.v1.reports.GeneDescriptor',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='gene_id', full_name='ncbi.datasets.v1.reports.GeneDescriptor.gene_id', index=0,
      number=1, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\034\n\007gene-id\022\013NCBI GeneID2\0042778', json_name='geneId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='symbol', full_name='ncbi.datasets.v1.reports.GeneDescriptor.symbol', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\026\n\006symbol\022\006Symbol2\004GNAS', json_name='symbol', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='description', full_name='ncbi.datasets.v1.reports.GeneDescriptor.description', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030.\n\013description\022\013Description2\022GNAS complex locus', json_name='description', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='tax_id', full_name='ncbi.datasets.v1.reports.GeneDescriptor.tax_id', index=3,
      number=4, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\034\n\006tax-id\022\014Taxonomic ID2\0049606', json_name='taxId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxname', full_name='ncbi.datasets.v1.reports.GeneDescriptor.taxname', index=4,
      number=5, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030(\n\010tax-name\022\016Taxonomic Name2\014Homo sapiens', json_name='taxname', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='common_name', full_name='ncbi.datasets.v1.reports.GeneDescriptor.common_name', index=5,
      number=12, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030!\n\013common-name\022\013Common Name2\005human', json_name='commonName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='type', full_name='ncbi.datasets.v1.reports.GeneDescriptor.type', index=6,
      number=6, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\026\n\tgene-type\022\tGene Type', json_name='type', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='rna_type', full_name='ncbi.datasets.v1.reports.GeneDescriptor.rna_type', index=7,
      number=11, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\024\n\010rna-type\022\010RNA Type', json_name='rnaType', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='orientation', full_name='ncbi.datasets.v1.reports.GeneDescriptor.orientation', index=8,
      number=7, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\032\n\013orientation\022\013Orientation', json_name='orientation', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='genomic_ranges', full_name='ncbi.datasets.v1.reports.GeneDescriptor.genomic_ranges', index=9,
      number=8, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030 \n\016genomic-range-\022\016Genomic Range ', json_name='genomicRanges', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='reference_standards', full_name='ncbi.datasets.v1.reports.GeneDescriptor.reference_standards', index=10,
      number=21, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030$\n\rref-standard-\022\023Reference Standard ', json_name='referenceStandards', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='genomic_regions', full_name='ncbi.datasets.v1.reports.GeneDescriptor.genomic_regions', index=11,
      number=22, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\"\n\017genomic-region-\022\017Genomic Region ', json_name='genomicRegions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='transcripts', full_name='ncbi.datasets.v1.reports.GeneDescriptor.transcripts', index=12,
      number=9, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\032\n\013transcript-\022\013Transcript ', json_name='transcripts', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='proteins', full_name='ncbi.datasets.v1.reports.GeneDescriptor.proteins', index=13,
      number=10, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\024\n\010protein-\022\010Protein ', json_name='proteins', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='chromosome', full_name='ncbi.datasets.v1.reports.GeneDescriptor.chromosome', index=14,
      number=13, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='chromosome', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='chromosomes', full_name='ncbi.datasets.v1.reports.GeneDescriptor.chromosomes', index=15,
      number=14, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\"\n\013chromosomes\022\013Chromosomes2\00112\003X,Y', json_name='chromosomes', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='nomenclature_authority', full_name='ncbi.datasets.v1.reports.GeneDescriptor.nomenclature_authority', index=16,
      number=15, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\026\n\005name-\022\rNomenclature ', json_name='nomenclatureAuthority', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='swiss_prot_accessions', full_name='ncbi.datasets.v1.reports.GeneDescriptor.swiss_prot_accessions', index=17,
      number=16, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030,\n\024swissprot-accessions\022\024SwissProt Accessions', json_name='swissProtAccessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='ensembl_gene_ids', full_name='ncbi.datasets.v1.reports.GeneDescriptor.ensembl_gene_ids', index=18,
      number=17, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\"\n\017ensembl-geneids\022\017Ensembl GeneIDs', json_name='ensemblGeneIds', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='omim_ids', full_name='ncbi.datasets.v1.reports.GeneDescriptor.omim_ids', index=19,
      number=18, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\024\n\010omim-ids\022\010OMIM IDs', json_name='omimIds', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='synonyms', full_name='ncbi.datasets.v1.reports.GeneDescriptor.synonyms', index=20,
      number=19, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\024\n\010synonyms\022\010Synonyms', json_name='synonyms', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='replaced_gene_id', full_name='ncbi.datasets.v1.reports.GeneDescriptor.replaced_gene_id', index=21,
      number=20, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030(\n\020replaced-gene-id\022\024Replaced NCBI GeneID', json_name='replacedGeneId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='annotations', full_name='ncbi.datasets.v1.reports.GeneDescriptor.annotations', index=22,
      number=23, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\032\n\013annotation-\022\013Annotation ', json_name='annotations', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _GENEDESCRIPTOR_GENETYPE,
    _GENEDESCRIPTOR_RNATYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=2892,
  serialized_end=5141,
)


_NOMENCLATUREAUTHORITY = _descriptor.Descriptor(
  name='NomenclatureAuthority',
  full_name='ncbi.datasets.v1.reports.NomenclatureAuthority',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='authority', full_name='ncbi.datasets.v1.reports.NomenclatureAuthority.authority', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\034\n\tauthority\022\tAuthority2\004HGNC', json_name='authority', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='identifier', full_name='ncbi.datasets.v1.reports.NomenclatureAuthority.identifier', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\363\030\023\n\002id\022\002ID2\tHGNC:4392', json_name='identifier', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=5144,
  serialized_end=5288,
)


_GENEDESCRIPTORS = _descriptor.Descriptor(
  name='GeneDescriptors',
  full_name='ncbi.datasets.v1.reports.GeneDescriptors',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='genes', full_name='ncbi.datasets.v1.reports.GeneDescriptors.genes', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='genes', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=5290,
  serialized_end=5371,
)

_PROTEIN.fields_by_name['mature_peptides'].message_type = _MATUREPEPTIDE
_GENOMICLOCATION.fields_by_name['genomic_range'].message_type = ncbi_dot_datasets_dot_v1_dot_reports_dot_common__pb2._RANGE
_GENOMICLOCATION.fields_by_name['exons'].message_type = ncbi_dot_datasets_dot_v1_dot_reports_dot_common__pb2._RANGE
_TRANSCRIPT.fields_by_name['genomic_range'].message_type = ncbi_dot_datasets_dot_v1_dot_reports_dot_common__pb2._SEQRANGESET
_TRANSCRIPT.fields_by_name['exons'].message_type = ncbi_dot_datasets_dot_v1_dot_reports_dot_common__pb2._SEQRANGESET
_TRANSCRIPT.fields_by_name['cds'].message_type = ncbi_dot_datasets_dot_v1_dot_reports_dot_common__pb2._SEQRANGESET
_TRANSCRIPT.fields_by_name['genomic_locations'].message_type = _GENOMICLOCATION
_TRANSCRIPT.fields_by_name['protein'].message_type = _PROTEIN
_TRANSCRIPT.fields_by_name['type'].enum_type = _TRANSCRIPT_TRANSCRIPTTYPE
_TRANSCRIPT_TRANSCRIPTTYPE.containing_type = _TRANSCRIPT
_GENOMICREGION.fields_by_name['gene_range'].message_type = ncbi_dot_datasets_dot_v1_dot_reports_dot_common__pb2._SEQRANGESET
_GENOMICREGION.fields_by_name['type'].enum_type = _GENOMICREGION_GENOMICREGIONTYPE
_GENOMICREGION_GENOMICREGIONTYPE.containing_type = _GENOMICREGION
_ANNOTATION.fields_by_name['assemblies_in_scope'].message_type = _ANNOTATEDASSEMBLIES
_GENEDESCRIPTOR.fields_by_name['type'].enum_type = _GENEDESCRIPTOR_GENETYPE
_GENEDESCRIPTOR.fields_by_name['rna_type'].enum_type = _GENEDESCRIPTOR_RNATYPE
_GENEDESCRIPTOR.fields_by_name['orientation'].enum_type = ncbi_dot_datasets_dot_v1_dot_reports_dot_common__pb2._ORIENTATION
_GENEDESCRIPTOR.fields_by_name['genomic_ranges'].message_type = ncbi_dot_datasets_dot_v1_dot_reports_dot_common__pb2._SEQRANGESET
_GENEDESCRIPTOR.fields_by_name['reference_standards'].message_type = _GENOMICREGION
_GENEDESCRIPTOR.fields_by_name['genomic_regions'].message_type = _GENOMICREGION
_GENEDESCRIPTOR.fields_by_name['transcripts'].message_type = _TRANSCRIPT
_GENEDESCRIPTOR.fields_by_name['proteins'].message_type = _PROTEIN
_GENEDESCRIPTOR.fields_by_name['nomenclature_authority'].message_type = _NOMENCLATUREAUTHORITY
_GENEDESCRIPTOR.fields_by_name['annotations'].message_type = _ANNOTATION
_GENEDESCRIPTOR_GENETYPE.containing_type = _GENEDESCRIPTOR
_GENEDESCRIPTOR_RNATYPE.containing_type = _GENEDESCRIPTOR
_GENEDESCRIPTORS.fields_by_name['genes'].message_type = _GENEDESCRIPTOR
DESCRIPTOR.message_types_by_name['MaturePeptide'] = _MATUREPEPTIDE
DESCRIPTOR.message_types_by_name['Protein'] = _PROTEIN
DESCRIPTOR.message_types_by_name['GenomicLocation'] = _GENOMICLOCATION
DESCRIPTOR.message_types_by_name['Transcript'] = _TRANSCRIPT
DESCRIPTOR.message_types_by_name['GenomicRegion'] = _GENOMICREGION
DESCRIPTOR.message_types_by_name['Annotation'] = _ANNOTATION
DESCRIPTOR.message_types_by_name['AnnotatedAssemblies'] = _ANNOTATEDASSEMBLIES
DESCRIPTOR.message_types_by_name['GeneDescriptor'] = _GENEDESCRIPTOR
DESCRIPTOR.message_types_by_name['NomenclatureAuthority'] = _NOMENCLATUREAUTHORITY
DESCRIPTOR.message_types_by_name['GeneDescriptors'] = _GENEDESCRIPTORS
_sym_db.RegisterFileDescriptor(DESCRIPTOR)

MaturePeptide = _reflection.GeneratedProtocolMessageType('MaturePeptide', (_message.Message,), {
  'DESCRIPTOR' : _MATUREPEPTIDE,
  '__module__' : 'ncbi.datasets.v1.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.MaturePeptide)
  })
_sym_db.RegisterMessage(MaturePeptide)

Protein = _reflection.GeneratedProtocolMessageType('Protein', (_message.Message,), {
  'DESCRIPTOR' : _PROTEIN,
  '__module__' : 'ncbi.datasets.v1.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.Protein)
  })
_sym_db.RegisterMessage(Protein)

GenomicLocation = _reflection.GeneratedProtocolMessageType('GenomicLocation', (_message.Message,), {
  'DESCRIPTOR' : _GENOMICLOCATION,
  '__module__' : 'ncbi.datasets.v1.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.GenomicLocation)
  })
_sym_db.RegisterMessage(GenomicLocation)

Transcript = _reflection.GeneratedProtocolMessageType('Transcript', (_message.Message,), {
  'DESCRIPTOR' : _TRANSCRIPT,
  '__module__' : 'ncbi.datasets.v1.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.Transcript)
  })
_sym_db.RegisterMessage(Transcript)

GenomicRegion = _reflection.GeneratedProtocolMessageType('GenomicRegion', (_message.Message,), {
  'DESCRIPTOR' : _GENOMICREGION,
  '__module__' : 'ncbi.datasets.v1.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.GenomicRegion)
  })
_sym_db.RegisterMessage(GenomicRegion)

Annotation = _reflection.GeneratedProtocolMessageType('Annotation', (_message.Message,), {
  'DESCRIPTOR' : _ANNOTATION,
  '__module__' : 'ncbi.datasets.v1.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.Annotation)
  })
_sym_db.RegisterMessage(Annotation)

AnnotatedAssemblies = _reflection.GeneratedProtocolMessageType('AnnotatedAssemblies', (_message.Message,), {
  'DESCRIPTOR' : _ANNOTATEDASSEMBLIES,
  '__module__' : 'ncbi.datasets.v1.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.AnnotatedAssemblies)
  })
_sym_db.RegisterMessage(AnnotatedAssemblies)

GeneDescriptor = _reflection.GeneratedProtocolMessageType('GeneDescriptor', (_message.Message,), {
  'DESCRIPTOR' : _GENEDESCRIPTOR,
  '__module__' : 'ncbi.datasets.v1.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.GeneDescriptor)
  })
_sym_db.RegisterMessage(GeneDescriptor)

NomenclatureAuthority = _reflection.GeneratedProtocolMessageType('NomenclatureAuthority', (_message.Message,), {
  'DESCRIPTOR' : _NOMENCLATUREAUTHORITY,
  '__module__' : 'ncbi.datasets.v1.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.NomenclatureAuthority)
  })
_sym_db.RegisterMessage(NomenclatureAuthority)

GeneDescriptors = _reflection.GeneratedProtocolMessageType('GeneDescriptors', (_message.Message,), {
  'DESCRIPTOR' : _GENEDESCRIPTORS,
  '__module__' : 'ncbi.datasets.v1.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.reports.GeneDescriptors)
  })
_sym_db.RegisterMessage(GeneDescriptors)


DESCRIPTOR._options = None
_MATUREPEPTIDE.fields_by_name['accession_version']._options = None
_MATUREPEPTIDE.fields_by_name['name']._options = None
_MATUREPEPTIDE.fields_by_name['length']._options = None
_PROTEIN.fields_by_name['accession_version']._options = None
_PROTEIN.fields_by_name['name']._options = None
_PROTEIN.fields_by_name['length']._options = None
_PROTEIN.fields_by_name['isoform_name']._options = None
_PROTEIN.fields_by_name['ensembl_protein']._options = None
_PROTEIN.fields_by_name['mature_peptides']._options = None
_GENOMICLOCATION.fields_by_name['genomic_accession_version']._options = None
_GENOMICLOCATION.fields_by_name['sequence_name']._options = None
_GENOMICLOCATION.fields_by_name['genomic_range']._options = None
_GENOMICLOCATION.fields_by_name['exons']._options = None
_TRANSCRIPT.fields_by_name['accession_version']._options = None
_TRANSCRIPT.fields_by_name['name']._options = None
_TRANSCRIPT.fields_by_name['length']._options = None
_TRANSCRIPT.fields_by_name['genomic_range']._options = None
_TRANSCRIPT.fields_by_name['exons']._options = None
_TRANSCRIPT.fields_by_name['cds']._options = None
_TRANSCRIPT.fields_by_name['genomic_locations']._options = None
_TRANSCRIPT.fields_by_name['ensembl_transcript']._options = None
_TRANSCRIPT.fields_by_name['protein']._options = None
_TRANSCRIPT.fields_by_name['type']._options = None
_GENOMICREGION.fields_by_name['gene_range']._options = None
_GENOMICREGION.fields_by_name['type']._options = None
_ANNOTATION.fields_by_name['release_name']._options = None
_ANNOTATION.fields_by_name['release_date']._options = None
_ANNOTATION.fields_by_name['assemblies_in_scope']._options = None
_ANNOTATEDASSEMBLIES.fields_by_name['accession']._options = None
_ANNOTATEDASSEMBLIES.fields_by_name['name']._options = None
_GENEDESCRIPTOR.fields_by_name['gene_id']._options = None
_GENEDESCRIPTOR.fields_by_name['symbol']._options = None
_GENEDESCRIPTOR.fields_by_name['description']._options = None
_GENEDESCRIPTOR.fields_by_name['tax_id']._options = None
_GENEDESCRIPTOR.fields_by_name['taxname']._options = None
_GENEDESCRIPTOR.fields_by_name['common_name']._options = None
_GENEDESCRIPTOR.fields_by_name['type']._options = None
_GENEDESCRIPTOR.fields_by_name['rna_type']._options = None
_GENEDESCRIPTOR.fields_by_name['orientation']._options = None
_GENEDESCRIPTOR.fields_by_name['genomic_ranges']._options = None
_GENEDESCRIPTOR.fields_by_name['reference_standards']._options = None
_GENEDESCRIPTOR.fields_by_name['genomic_regions']._options = None
_GENEDESCRIPTOR.fields_by_name['transcripts']._options = None
_GENEDESCRIPTOR.fields_by_name['proteins']._options = None
_GENEDESCRIPTOR.fields_by_name['chromosome']._options = None
_GENEDESCRIPTOR.fields_by_name['chromosomes']._options = None
_GENEDESCRIPTOR.fields_by_name['nomenclature_authority']._options = None
_GENEDESCRIPTOR.fields_by_name['swiss_prot_accessions']._options = None
_GENEDESCRIPTOR.fields_by_name['ensembl_gene_ids']._options = None
_GENEDESCRIPTOR.fields_by_name['omim_ids']._options = None
_GENEDESCRIPTOR.fields_by_name['synonyms']._options = None
_GENEDESCRIPTOR.fields_by_name['replaced_gene_id']._options = None
_GENEDESCRIPTOR.fields_by_name['annotations']._options = None
_NOMENCLATUREAUTHORITY.fields_by_name['authority']._options = None
_NOMENCLATUREAUTHORITY.fields_by_name['identifier']._options = None
# @@protoc_insertion_point(module_scope)
