# -*- coding: utf-8 -*-
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: ncbi/datasets/v1/datasets.proto
"""Generated protocol buffer code."""
from google.protobuf.internal import enum_type_wrapper
from google.protobuf import descriptor as _descriptor
from google.protobuf import message as _message
from google.protobuf import reflection as _reflection
from google.protobuf import symbol_database as _symbol_database
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()


from google.protobuf import timestamp_pb2 as google_dot_protobuf_dot_timestamp__pb2
from ncbi.datasets.options import openapi_pb2 as ncbi_dot_datasets_dot_options_dot_openapi__pb2
from ncbi.datasets.options import request_pb2 as ncbi_dot_datasets_dot_options_dot_request__pb2
from ncbi.datasets.v1 import gene_descriptor_pb2 as ncbi_dot_datasets_dot_v1_dot_gene__descriptor__pb2
from ncbi.datasets.v1.reports import biosample_pb2 as ncbi_dot_datasets_dot_v1_dot_reports_dot_biosample__pb2


DESCRIPTOR = _descriptor.FileDescriptor(
  name='ncbi/datasets/v1/datasets.proto',
  package='ncbi.datasets.v1',
  syntax='proto3',
  serialized_options=b'Z\020ncbi/datasets/v1\370\001\001',
  create_key=_descriptor._internal_create_key,
  serialized_pb=b'\n\x1fncbi/datasets/v1/datasets.proto\x12\x10ncbi.datasets.v1\x1a\x1fgoogle/protobuf/timestamp.proto\x1a#ncbi/datasets/options/openapi.proto\x1a#ncbi/datasets/options/request.proto\x1a&ncbi/datasets/v1/gene_descriptor.proto\x1a(ncbi/datasets/v1/reports/biosample.proto\"H\n\x14MethodPayloadRequest\x12\x16\n\x06method\x18\x01 \x01(\tR\x06method\x12\x18\n\x07payload\x18\x02 \x01(\tR\x07payload\"\x99\x0f\n AssemblyDatasetDescriptorsFilter\x12\x9d\x01\n\x0ereference_only\x18\x01 \x01(\x08\x42v\xc2\xf0\x19r\n\x05\x66\x61lse\x12\x33\n\x04true\x12+Limit to reference & representative genomes\x12\x34\n\x05\x66\x61lse\x12+Include reference and non-reference genomesR\rreferenceOnly\x12#\n\x0brefseq_only\x18\x02 \x01(\x08\x42\x02\x18\x01R\nrefseqOnly\x12\xd1\x01\n\x0f\x61ssembly_source\x18\x03 \x01(\x0e\x32\x41.ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter.AssemblySourceBe\xc2\xf0\x19\x61\n\x03\x61ll\x12!\n\x06refseq\x12\x17Limit to RefSeq genomes\x12#\n\x07genbank\x12\x18Limit to GenBank genomes\x12\x12\n\x03\x61ll\x12\x0b\x41ll genomesR\x0e\x61ssemblySource\x12y\n\x0ehas_annotation\x18\x07 \x01(\x08\x42R\xc2\xf0\x19N\n\x05\x66\x61lse\x12%\n\x04true\x12\x1dLimit to annotated assemblies\x12\x1e\n\x05\x66\x61lse\x12\x15Return all assembliesR\rhasAnnotation\x12\x96\x01\n\x16\x65xclude_paired_reports\x18\t \x01(\x08\x42`\xc2\xf0\x19\\\n\x05\x66\x61lse\x12\x1e\n\x05\x66\x61lse\x12\x15Return all assemblies\x12\x33\n\x04true\x12+Only return single record for GCA/GCF pairsR\x14\x65xcludePairedReports\x12\xc7\x01\n\x10\x65xclude_atypical\x18\x0b \x01(\x08\x42\x9b\x01\xc2\xf0\x19\x96\x01\n\x05\x66\x61lse\x12\x46\n\x05\x66\x61lse\x12=Return all selected genomes, including ones that are atypical\x12\x45\n\x04true\x12=Only return genomes that have not been classified as atypicalR\x0f\x65xcludeAtypical\x12\xf5\x01\n\x10\x61ssembly_version\x18\n \x01(\x0e\x32\x42.ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter.AssemblyVersionB\x85\x01\xc2\xf0\x19\x80\x01\n\x07\x63urrent\x12)\n\x07\x63urrent\x12\x1eOnly return current assemblies\x12J\n\x0e\x61ll_assemblies\x12\x38Return all assemblies, including suppressed and replacedR\x0f\x61ssemblyVersion\x12\xeb\x01\n\x0e\x61ssembly_level\x18\x04 \x03(\x0e\x32@.ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter.AssemblyLevelB\x81\x01\xc2\xf0\x19}\x12=\n\nchromosome\n\x0f\x63omplete_genome\x12\x1e\x43hromosome and complete genome\x12<\n\x08scaffold\n\nchromosome\n\x0f\x63omplete_genome\x12\x13Scaffold and higherR\rassemblyLevel\x12j\n\x12\x66irst_release_date\x18\x05 \x01(\x0b\x32\x1a.google.protobuf.TimestampB \xc2\xf0\x19\x1c\x12\x1a\n\n01/10/2015\x12\x0cJan 10, 2015R\x10\x66irstReleaseDate\x12h\n\x11last_release_date\x18\x06 \x01(\x0b\x32\x1a.google.protobuf.TimestampB \xc2\xf0\x19\x1c\x12\x1a\n\n01/10/2021\x12\x0cJan 10, 2021R\x0flastReleaseDate\x12\x89\x01\n\x0bsearch_text\x18\x08 \x03(\tBh\xc2\xf0\x19\x64\x12:\n\x1bGenome Reference Consortium\x12\x1bGenome Reference Consortium\x12\x10\n\x06GRCh38\x12\x06GRCh38\x12\x14\n\x08\x43\x35\x37\x42L/6J\x12\x08\x43\x35\x37\x42L/6JR\nsearchText\"2\n\x0e\x41ssemblySource\x12\x07\n\x03\x61ll\x10\x00\x12\n\n\x06refseq\x10\x01\x12\x0b\n\x07genbank\x10\x02\"2\n\x0f\x41ssemblyVersion\x12\x0b\n\x07\x63urrent\x10\x00\x12\x12\n\x0e\x61ll_assemblies\x10\x01\"N\n\rAssemblyLevel\x12\x0e\n\nchromosome\x10\x00\x12\x0c\n\x08scaffold\x10\x01\x12\n\n\x06\x63ontig\x10\x02\x12\x13\n\x0f\x63omplete_genome\x10\x03\"\x84\x01\n\nAccessions\x12v\n\naccessions\x18\x01 \x03(\tBV\xc2\xf0\x19R\x12\x35\n\x10GCF_000001405.40\n\x10GCF_000001635.27\x12\x0fHuman and Mouse\x12\x19\n\x10GCF_000001405.40\x12\x05HumanR\naccessions\"\xdf\x06\n\x17\x41ssemblyMetadataRequest\x12s\n\x05taxon\x18\x01 \x01(\tB]\xc2\xf0\x19Y\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy Identifier\x12\x14\n\x05human\x12\x0b\x43ommon Name\x12\x1f\n\x0cHomo sapiens\x12\x0fScientific NameR\x05taxon\x12<\n\naccessions\x18\x02 \x01(\x0b\x32\x1c.ncbi.datasets.v1.AccessionsR\naccessions\x12W\n\x0b\x62ioprojects\x18\x07 \x01(\x0b\x32\x35.ncbi.datasets.v1.AssemblyMetadataRequest.BioprojectsR\x0b\x62ioprojects\x12L\n\x07\x66ilters\x18\x04 \x01(\x0b\x32\x32.ncbi.datasets.v1.AssemblyDatasetDescriptorsFilterR\x07\x66ilters\x12\x33\n\x0ftax_exact_match\x18\x05 \x01(\x08\x42\x0b\xc2\xf0\x19\x07\n\x05\x66\x61lseR\rtaxExactMatch\x12`\n\x10returned_content\x18\x06 \x01(\x0e\x32\x35.ncbi.datasets.v1.AssemblyMetadataRequest.ContentTypeR\x0freturnedContent\x12\x1a\n\x05limit\x18\x03 \x01(\tB\x02\x18\x01H\x00R\x05limit\x12\'\n\tpage_size\x18\x08 \x01(\x05\x42\x08\xc2\xf0\x19\x04\n\x02\x32\x30H\x00R\x08pageSize\x12\x1d\n\npage_token\x18\t \x01(\tR\tpageToken\x1a\x82\x01\n\x0b\x42ioprojects\x12s\n\naccessions\x18\x01 \x03(\tBS\xc2\xf0\x19O\x12)\n\x0bPRJNA489243\x12\x1aVertebrate Genomes Project\x12\"\n\nPRJNA31257\x12\x14Human Genome ProjectR\naccessions\")\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\x0c\n\x08\x41SSM_ACC\x10\x01:4\xc2\xf0\x19\x30\n.accessions:\n  accessions:\n  - GCF_000001405.40B\t\n\x07records\"\x8b\x04\n!AssemblyDatasetDescriptorsRequest\x12\x17\n\x06tax_id\x18\x01 \x01(\rH\x00R\x05taxId\x12\x1f\n\x08org_name\x18\x05 \x01(\tB\x02\x18\x01H\x00R\x07orgName\x12\x1b\n\x08tax_name\x18\x07 \x01(\tH\x00R\x07taxName\x12/\n\x12\x61ssembly_accession\x18\x04 \x01(\tH\x00R\x11\x61ssemblyAccession\x12\x1c\n\x06\x63utoff\x18\x02 \x01(\rB\x02\x18\x01H\x01R\x06\x63utoff\x12\x16\n\x05limit\x18\x08 \x01(\tH\x01R\x05limit\x12L\n\x07\x66ilters\x18\x03 \x01(\x0b\x32\x32.ncbi.datasets.v1.AssemblyDatasetDescriptorsFilterR\x07\x66ilters\x12&\n\x0ftax_exact_match\x18\x06 \x01(\x08R\rtaxExactMatch\x12j\n\x10returned_content\x18\t \x01(\x0e\x32?.ncbi.datasets.v1.AssemblyDatasetDescriptorsRequest.ContentTypeR\x0freturnedContent\"5\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\x0c\n\x08\x41SSM_ACC\x10\x01\x12\n\n\x06\x43OUNTS\x10\x02\x42\x04\n\x02idB\t\n\x07records\"\x86\x01\n\x1a\x41ssemblyDatasetDescriptors\x12G\n\x08\x64\x61tasets\x18\x01 \x03(\x0b\x32+.ncbi.datasets.v1.AssemblyDatasetDescriptorR\x08\x64\x61tasets\x12\x1f\n\x0btotal_count\x18\x02 \x01(\rR\ntotalCount\"\x98\x01\n\nBioProject\x12\x1c\n\taccession\x18\x01 \x01(\tR\taccession\x12\x14\n\x05title\x18\x02 \x01(\tR\x05title\x12)\n\x10parent_accession\x18\x03 \x01(\tR\x0fparentAccession\x12+\n\x11parent_accessions\x18\x04 \x03(\tR\x10parentAccessions\"S\n\x11\x42ioProjectLineage\x12>\n\x0b\x62ioprojects\x18\x01 \x03(\x0b\x32\x1c.ncbi.datasets.v1.BioProjectR\x0b\x62ioprojects\"\x9a\x08\n\x19\x41ssemblyDatasetDescriptor\x12-\n\x12\x61ssembly_accession\x18\x01 \x01(\tR\x11\x61ssemblyAccession\x12:\n\x19paired_assembly_accession\x18\x11 \x01(\tR\x17pairedAssemblyAccession\x12T\n\x13\x62ioproject_lineages\x18\x0e \x03(\x0b\x32#.ncbi.datasets.v1.BioProjectLineageR\x12\x62ioprojectLineages\x12!\n\x0c\x64isplay_name\x18\x02 \x01(\tR\x0b\x64isplayName\x12,\n\x03org\x18\n \x01(\x0b\x32\x1a.ncbi.datasets.v1.OrganismR\x03org\x12X\n\x0b\x63hromosomes\x18\x04 \x03(\x0b\x32\x36.ncbi.datasets.v1.AssemblyDatasetDescriptor.ChromosomeR\x0b\x63hromosomes\x12+\n\x11\x61ssembly_category\x18\x05 \x01(\tR\x10\x61ssemblyCategory\x12X\n\x13\x61nnotation_metadata\x18\x0b \x01(\x0b\x32\'.ncbi.datasets.v1.AnnotationForAssemblyR\x12\x61nnotationMetadata\x12%\n\x0e\x61ssembly_level\x18\x07 \x01(\tR\rassemblyLevel\x12\x1c\n\tsubmitter\x18\x0f \x01(\tR\tsubmitter\x12\'\n\x0fsubmission_date\x18\x08 \x01(\tR\x0esubmissionDate\x12\x1d\n\ncontig_n50\x18\t \x01(\rR\tcontigN50\x12%\n\x0e\x65stimated_size\x18\x0c \x01(\x04R\restimatedSize\x12\x1d\n\nseq_length\x18\r \x01(\x04R\tseqLength\x12/\n\x13\x62iosample_accession\x18\x10 \x01(\tR\x12\x62iosampleAccession\x12K\n\tbiosample\x18\x14 \x01(\x0b\x32-.ncbi.datasets.v1.reports.BioSampleDescriptorR\tbiosample\x12\x1b\n\tblast_url\x18\x12 \x01(\tR\x08\x62lastUrl\x12\x19\n\x08gc_count\x18\x13 \x01(\x04R\x07gcCount\x1a\x80\x01\n\nChromosome\x12\x12\n\x04name\x18\x01 \x01(\tR\x04name\x12\x16\n\x06length\x18\x02 \x01(\x04R\x06length\x12+\n\x11\x61\x63\x63\x65ssion_version\x18\x03 \x01(\tR\x10\x61\x63\x63\x65ssionVersion\x12\x19\n\x08gc_count\x18\x04 \x01(\x04R\x07gcCount\"\x85\x02\n\tBuscoStat\x12#\n\rbusco_lineage\x18\x01 \x01(\tR\x0c\x62uscoLineage\x12\x1b\n\tbusco_ver\x18\x02 \x01(\tR\x08\x62uscoVer\x12\x1a\n\x08\x63omplete\x18\x03 \x01(\x02R\x08\x63omplete\x12\x1f\n\x0bsingle_copy\x18\x04 \x01(\x02R\nsingleCopy\x12\x1e\n\nduplicated\x18\x05 \x01(\x02R\nduplicated\x12\x1e\n\nfragmented\x18\x06 \x01(\x02R\nfragmented\x12\x18\n\x07missing\x18\x07 \x01(\x02R\x07missing\x12\x1f\n\x0btotal_count\x18\x08 \x01(\x04R\ntotalCount\"\xf7\x04\n\x15\x41nnotationForAssembly\x12\x12\n\x04name\x18\x01 \x01(\tR\x04name\x12\x16\n\x06source\x18\x02 \x01(\tR\x06source\x12%\n\x0erelease_number\x18\x03 \x01(\tR\rreleaseNumber\x12!\n\x0crelease_date\x18\x04 \x01(\tR\x0breleaseDate\x12\x1d\n\nreport_url\x18\x05 \x01(\tR\treportUrl\x12@\n\x04\x66ile\x18\x06 \x03(\x0b\x32,.ncbi.datasets.v1.AnnotationForAssembly.FileR\x04\x66ile\x12\x35\n\x05stats\x18\x07 \x01(\x0b\x32\x1f.ncbi.datasets.v1.FeatureCountsR\x05stats\x12\x31\n\x05\x62usco\x18\x08 \x01(\x0b\x32\x1b.ncbi.datasets.v1.BuscoStatR\x05\x62usco\x1ao\n\x04\x46ile\x12@\n\x04type\x18\x01 \x01(\x0e\x32,.ncbi.datasets.v1.AnnotationForAssembly.TypeR\x04type\x12%\n\x0e\x65stimated_size\x18\x02 \x01(\x04R\restimatedSize\"\xab\x01\n\x04Type\x12\x0b\n\x07\x44\x45\x46\x41ULT\x10\x00\x12\x0e\n\nGENOME_GFF\x10\x01\x12\x0f\n\x0bGENOME_GBFF\x10\x02\x12\x11\n\tGENOME_GB\x10\x02\x1a\x02\x08\x01\x12\r\n\tRNA_FASTA\x10\x03\x12\x0e\n\nPROT_FASTA\x10\x04\x12\x0e\n\nGENOME_GTF\x10\x05\x12\r\n\tCDS_FASTA\x10\x06\x12\x0e\n\nASM_STRUCT\x10\x07\x12\x10\n\x0cGENOME_FASTA\x10\x08\x1a\x02\x10\x01\"N\n\rFeatureCounts\x12=\n\x0bgene_counts\x18\x01 \x01(\x0b\x32\x1c.ncbi.datasets.v1.GeneCountsR\ngeneCounts\"I\n\nGeneCounts\x12\x14\n\x05total\x18\x01 \x01(\rR\x05total\x12%\n\x0eprotein_coding\x18\x02 \x01(\rR\rproteinCoding\"\xd5\x07\n\x16\x41ssemblyDatasetRequest\x12\x33\n\x13\x61ssembly_accessions\x18\x01 \x03(\tB\x02\x18\x01R\x12\x61ssemblyAccessions\x12v\n\naccessions\x18\n \x03(\tBV\xc2\xf0\x19R\x12\x35\n\x10GCF_000001405.40\n\x10GCF_000001635.27\x12\x0fHuman and Mouse\x12\x19\n\x10GCF_000001405.40\x12\x05HumanR\naccessions\x12s\n\x0b\x63hromosomes\x18\x02 \x03(\tBQ\xc2\xf0\x19M\x12,\n\x01\x31\n\x01\x32\n\x01\x33\n\x01X\n\x01Y\n\x02MT\x12\x17Selected chromosomes #1\x12\x1d\n\x01X\n\x01Y\x12\x15Human sex chromosomesR\x0b\x63hromosomes\x12\x37\n\x12include_annotation\x18\x03 \x01(\x08\x42\x08\x18\x01\xda\xf0\x19\x02\x08\x02R\x11includeAnnotation\x12\x81\x01\n\x10\x65xclude_sequence\x18\x07 \x01(\x08\x42V\xda\xf0\x19\x02\x08\x02\xc2\xf0\x19L\n\x05\x66\x61lse\x12 \n\x04true\x12\x18\x45xclude genomic sequence\x12!\n\x05\x66\x61lse\x12\x18Include genomic sequenceR\x0f\x65xcludeSequence\x12\xf8\x01\n\x17include_annotation_type\x18\x08 \x03(\x0e\x32,.ncbi.datasets.v1.AnnotationForAssembly.TypeB\x91\x01\xda\xf0\x19\x02\x08\x02\xc2\xf0\x19\x86\x01\x12\x36\n\nPROT_FASTA\n\tRNA_FASTA\x12\x1dTranscript & Protein Sequence\x12L\n\nGENOME_GFF\n\x0bGENOME_GBFF\n\nGENOME_GTF\n\nPROT_FASTA\n\tRNA_FASTA\n\tCDS_FASTA\x12\x03\x41llR\x15includeAnnotationType\x12O\n\x08hydrated\x18\t \x01(\x0e\x32\x33.ncbi.datasets.v1.AssemblyDatasetRequest.ResolutionR\x08hydrated\x12\x32\n\x0binclude_tsv\x18\x0b \x01(\x08\x42\x11\xda\xf0\x19\x02\x08\x02\xc2\xf0\x19\x07\n\x05\x66\x61lseR\nincludeTsv\"6\n\nResolution\x12\x12\n\x0e\x46ULLY_HYDRATED\x10\x00\x12\x14\n\x10\x44\x41TA_REPORT_ONLY\x10\x01:$\xc2\xf0\x19 \n\x1e\x61\x63\x63\x65ssions:\n- GCF_000001405.40\"\x8f\x01\n\x1b\x41ssemblyDatasetAvailability\x12)\n\x10valid_assemblies\x18\x01 \x03(\tR\x0fvalidAssemblies\x12-\n\x12invalid_assemblies\x18\x02 \x03(\tR\x11invalidAssemblies\x12\x16\n\x06reason\x18\x03 \x01(\tR\x06reason\")\n\x0cTaxidRequest\x12\x19\n\x08org_name\x18\x01 \x01(\tR\x07orgName\"#\n\nTaxidReply\x12\x15\n\x06tax_id\x18\x01 \x01(\rR\x05taxId\"\xac\r\n\x08Organism\x12\x15\n\x06tax_id\x18\x01 \x01(\x04R\x05taxId\x12\x19\n\x08sci_name\x18\x02 \x01(\tR\x07sciName\x12\x1f\n\x0b\x63ommon_name\x18\x03 \x01(\tR\ncommonName\x12\x1d\n\nblast_node\x18\x10 \x01(\x08R\tblastNode\x12\x14\n\x05\x62reed\x18\n \x01(\tR\x05\x62reed\x12\x1a\n\x08\x63ultivar\x18\x0b \x01(\tR\x08\x63ultivar\x12\x18\n\x07\x65\x63otype\x18\x0c \x01(\tR\x07\x65\x63otype\x12\x18\n\x07isolate\x18\r \x01(\tR\x07isolate\x12\x10\n\x03sex\x18\x0e \x01(\tR\x03sex\x12\x16\n\x06strain\x18\x0f \x01(\tR\x06strain\x12\x1f\n\x0bsearch_text\x18\x11 \x03(\tR\nsearchText\x12\x37\n\x04rank\x18\x05 \x01(\x0e\x32#.ncbi.datasets.v1.Organism.RankTypeR\x04rank\x12\"\n\rparent_tax_id\x18\x06 \x01(\x04R\x0bparentTaxId\x12)\n\x0e\x61ssembly_count\x18\x07 \x01(\x04\x42\x02\x18\x01R\rassemblyCount\x12J\n\x0f\x61ssembly_counts\x18\x13 \x01(\x0b\x32!.ncbi.datasets.v1.Organism.CountsR\x0e\x61ssemblyCounts\x12>\n\x06\x63ounts\x18\x16 \x03(\x0b\x32&.ncbi.datasets.v1.Organism.CountByTypeR\x06\x63ounts\x12\x36\n\x08\x63hildren\x18\x08 \x03(\x0b\x32\x1a.ncbi.datasets.v1.OrganismR\x08\x63hildren\x12\x32\n\x06merged\x18\t \x03(\x0b\x32\x1a.ncbi.datasets.v1.OrganismR\x06merged\x12$\n\x0emerged_tax_ids\x18\x12 \x03(\x04R\x0cmergedTaxIds\x12\x17\n\x07min_ord\x18\x14 \x01(\rR\x06minOrd\x12\x17\n\x07max_ord\x18\x15 \x01(\rR\x06maxOrd\x12\x16\n\x06weight\x18\x17 \x01(\rR\x06weight\x12\x10\n\x03key\x18\x64 \x01(\x04R\x03key\x12\x14\n\x05title\x18\x65 \x01(\tR\x05title\x12\x12\n\x04icon\x18\x66 \x01(\x08R\x04icon\x1a\x36\n\x06\x43ounts\x12\x12\n\x04node\x18\x01 \x01(\rR\x04node\x12\x18\n\x07subtree\x18\x02 \x01(\rR\x07subtree\x1ay\n\x0b\x43ountByType\x12/\n\x04type\x18\x01 \x01(\x0e\x32\x1b.ncbi.datasets.v1.CountTypeR\x04type\x12\x39\n\x06\x63ounts\x18\x02 \x01(\x0b\x32!.ncbi.datasets.v1.Organism.CountsR\x06\x63ounts\"\x9d\x05\n\x08RankType\x12\x0b\n\x07NO_RANK\x10\x00\x12\x10\n\x0cSUPERKINGDOM\x10\x01\x12\x0b\n\x07KINGDOM\x10\x02\x12\x0e\n\nSUBKINGDOM\x10\x03\x12\x0f\n\x0bSUPERPHYLUM\x10\x04\x12\r\n\tSUBPHYLUM\x10\x05\x12\n\n\x06PHYLUM\x10\x06\x12\t\n\x05\x43LADE\x10\x1f\x12\x0e\n\nSUPERCLASS\x10\x07\x12\t\n\x05\x43LASS\x10\x08\x12\x0c\n\x08SUBCLASS\x10\t\x12\x0e\n\nINFRACLASS\x10\n\x12\n\n\x06\x43OHORT\x10\x0b\x12\r\n\tSUBCOHORT\x10\x0c\x12\x0e\n\nSUPERORDER\x10\r\x12\t\n\x05ORDER\x10\x0e\x12\x0c\n\x08SUBORDER\x10\x0f\x12\x0e\n\nINFRAORDER\x10\x10\x12\r\n\tPARVORDER\x10\x11\x12\x0f\n\x0bSUPERFAMILY\x10\x12\x12\n\n\x06\x46\x41MILY\x10\x13\x12\r\n\tSUBFAMILY\x10\x14\x12\t\n\x05GENUS\x10\x15\x12\x0c\n\x08SUBGENUS\x10\x16\x12\x11\n\rSPECIES_GROUP\x10\x17\x12\x14\n\x10SPECIES_SUBGROUP\x10\x18\x12\x0b\n\x07SPECIES\x10\x19\x12\x0e\n\nSUBSPECIES\x10\x1a\x12\t\n\x05TRIBE\x10\x1b\x12\x0c\n\x08SUBTRIBE\x10\x1c\x12\t\n\x05\x46ORMA\x10\x1d\x12\x0c\n\x08VARIETAS\x10\x1e\x12\x0b\n\x06STRAIN\x10\xc0\x02\x12\x0c\n\x07SECTION\x10\xca\x02\x12\x0f\n\nSUBSECTION\x10\xd4\x02\x12\x0f\n\nPATHOGROUP\x10\xde\x02\x12\x0f\n\nSUBVARIETY\x10\xe8\x02\x12\r\n\x08GENOTYPE\x10\xf2\x02\x12\r\n\x08SEROTYPE\x10\xfc\x02\x12\x0c\n\x07ISOLATE\x10\x86\x03\x12\n\n\x05MORPH\x10\x90\x03\x12\x0b\n\x06SERIES\x10\x9a\x03\x12\x14\n\x0f\x46ORMA_SPECIALIS\x10\xa4\x03\x12\x0e\n\tSEROGROUP\x10\xae\x03\x12\x0c\n\x07\x42IOTYPE\x10\xb8\x03\"\x85\x02\n\x0eTaxTreeRequest\x12\x1b\n\x06tax_id\x18\x01 \x01(\x04\x42\x02\x18\x01H\x00R\x05taxId\x12!\n\ttax_token\x18\x02 \x01(\tB\x02\x18\x01H\x00R\x08taxToken\x12u\n\x05taxon\x18\x03 \x01(\tB]\xc2\xf0\x19Y\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy Identifier\x12\x14\n\x05human\x12\x0b\x43ommon Name\x12\x1f\n\x0cHomo sapiens\x12\x0fScientific NameH\x00R\x05taxon\x12\x30\n\rchildren_only\x18\x04 \x01(\x08\x42\x0b\xc2\xf0\x19\x07\n\x05\x66\x61lseR\x0c\x63hildrenOnlyB\n\n\x08taxonomy\"\xc8\x03\n\x14OrganismQueryRequest\x12+\n\x0eorganism_query\x18\x01 \x01(\tB\x02\x18\x01H\x00R\rorganismQuery\x12\x33\n\x0btaxon_query\x18\x02 \x01(\tB\x10\xc2\xf0\x19\x0c\x12\n\n\x03hum\x12\x03humH\x00R\ntaxonQuery\x12k\n\x0ftax_rank_filter\x18\x03 \x01(\x0e\x32\x34.ncbi.datasets.v1.OrganismQueryRequest.TaxRankFilterB\r\xc2\xf0\x19\t\n\x07speciesR\rtaxRankFilter\x12u\n\x13tax_resource_filter\x18\x04 \x01(\x0e\x32:.ncbi.datasets.v1.OrganismQueryRequest.TaxonResourceFilterB\t\xc2\xf0\x19\x05\n\x03\x61llR\x11taxResourceFilter\".\n\rTaxRankFilter\x12\x0b\n\x07species\x10\x00\x12\x10\n\x0chigher_taxon\x10\x01\"4\n\x13TaxonResourceFilter\x12\x07\n\x03\x61ll\x10\x00\x12\n\n\x06genome\x10\x01\x12\x08\n\x04gene\x10\x02\x42\x04\n\x02id\"\xed\x01\n\rSciNameAndIds\x12U\n\x10sci_name_and_ids\x18\x01 \x03(\x0b\x32,.ncbi.datasets.v1.SciNameAndIds.SciNameAndIdR\rsciNameAndIds\x1a\x84\x01\n\x0cSciNameAndId\x12\x19\n\x08sci_name\x18\x01 \x01(\tR\x07sciName\x12\x15\n\x06tax_id\x18\x02 \x01(\x04R\x05taxId\x12\x1f\n\x0b\x63ommon_name\x18\x03 \x01(\tR\ncommonName\x12!\n\x0cmatched_term\x18\x04 \x01(\tR\x0bmatchedTerm\"\xf2\x02\n\x0fOrthologRequest\x12\x35\n\x07gene_id\x18\x01 \x01(\rB\x1c\xc2\xf0\x19\x18\x12\x0c\n\x04\x32\x37\x37\x38\x12\x04GNAS\x12\x08\n\x01\x32\x12\x03\x41\x32MR\x06geneId\x12X\n\x10returned_content\x18\x02 \x01(\x0e\x32-.ncbi.datasets.v1.OrthologRequest.ContentTypeR\x0freturnedContent\x12\xa2\x01\n\x0ctaxon_filter\x18\x03 \x03(\tB\x7f\xc2\xf0\x19{\x12\'\n\x04\x39\x36\x30\x36\n\x05\x31\x30\x30\x39\x30\x12\x18NCBI Taxonomy Identifier\x12!\n\x05human\n\x0bhouse mouse\x12\x0b\x43ommon Name\x12-\n\x0chomo sapiens\n\x0cmus musculus\x12\x0fScientific NameR\x0btaxonFilter\")\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\x0c\n\x08IDS_ONLY\x10\x01\"\x83\x01\n\x0bOrthologSet\x12&\n\x0fortholog_set_id\x18\x01 \x01(\rR\rorthologSetId\x12\x16\n\x06method\x18\x02 \x01(\tR\x06method\x12\x34\n\x05genes\x18\x03 \x01(\x0b\x32\x1e.ncbi.datasets.v1.GeneMetadataR\x05genes\"\xb5\n\n\x12GeneDatasetRequest\x12[\n\x08gene_ids\x18\x01 \x03(\rB@\xc2\xf0\x19<\x12\x14\n\x05\x35\x39\x30\x36\x37\x12\x0bIL21 GeneID\x12$\n\x05\x35\x39\x30\x36\x37\n\x05\x35\x30\x36\x31\x35\x12\x14IL21 & IL21R GeneIDsR\x07geneIds\x12\x81\x01\n\naccessions\x18\n \x03(\tBa\xc2\xf0\x19]\x12.\n\x0bNM_021803.4\n\x0bNM_181078.3\x12\x12RefSeq transcripts\x12+\n\x0bNP_068575.1\n\x0bNP_851564.1\x12\x0fRefSeq proteinsR\naccessions\x12`\n\x11symbols_for_taxon\x18\t \x01(\x0b\x32\x34.ncbi.datasets.v1.GeneDatasetRequest.SymbolsForTaxonR\x0fsymbolsForTaxon\x12s\n\x05taxon\x18\x03 \x01(\tB]\xc2\xf0\x19Y\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy Identifier\x12\x14\n\x05human\x12\x0b\x43ommon Name\x12\x1f\n\x0cHomo sapiens\x12\x0fScientific NameR\x05taxon\x12W\n\x17include_annotation_type\x18\x06 \x03(\x0e\x32\x17.ncbi.datasets.v1.FastaB\x06\xda\xf0\x19\x02\x08\x02R\x15includeAnnotationType\x12[\n\x10returned_content\x18\x07 \x01(\x0e\x32\x30.ncbi.datasets.v1.GeneDatasetRequest.ContentTypeR\x0freturnedContent\x12J\n\x0bsort_schema\x18\x08 \x01(\x0b\x32).ncbi.datasets.v1.GeneDatasetRequest.SortR\nsortSchema\x12\x14\n\x05limit\x18\x0b \x01(\tR\x05limit\x12!\n\x0c\x66\x61sta_filter\x18\x0c \x03(\tR\x0b\x66\x61staFilter\x1a\xb4\x01\n\x0fSymbolsForTaxon\x12,\n\x07symbols\x18\x01 \x03(\tB\x12\xc2\xf0\x19\x0e\x12\x0c\n\x04GNAS\x12\x04GNASR\x07symbols\x12s\n\x05taxon\x18\x02 \x01(\tB]\xc2\xf0\x19Y\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy Identifier\x12\x14\n\x05human\x12\x0b\x43ommon Name\x12\x1f\n\x0cHomo sapiens\x12\x0fScientific NameR\x05taxon\x1a$\n\x07GeneIds\x12\x19\n\x08gene_ids\x18\x01 \x03(\rR\x07geneIds\x1a\x8b\x01\n\x04Sort\x12\x44\n\x05\x66ield\x18\x01 \x01(\x0e\x32..ncbi.datasets.v1.GeneDatasetRequest.SortFieldR\x05\x66ield\x12=\n\tdirection\x18\x02 \x01(\x0e\x32\x1f.ncbi.datasets.v1.SortDirectionR\tdirection\")\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\x0c\n\x08IDS_ONLY\x10\x01\"u\n\tSortField\x12\x1a\n\x16SORT_FIELD_UNSPECIFIED\x10\x00\x12\x16\n\x12SORT_FIELD_GENE_ID\x10\x01\x12\x18\n\x14SORT_FIELD_GENE_TYPE\x10\x02\x12\x1a\n\x16SORT_FIELD_GENE_SYMBOL\x10\x03:\x1f\xc2\xf0\x19\x1b\n\x19gene_ids:\n- 59067\n- 50615\"\xc4\x05\n\x05\x45rror\x12[\n\x13\x61ssembly_error_code\x18\x06 \x01(\x0e\x32).ncbi.datasets.v1.Error.AssemblyErrorCodeH\x00R\x11\x61ssemblyErrorCode\x12O\n\x0fgene_error_code\x18\x07 \x01(\x0e\x32%.ncbi.datasets.v1.Error.GeneErrorCodeH\x00R\rgeneErrorCode\x12R\n\x10virus_error_code\x18\x08 \x01(\x0e\x32&.ncbi.datasets.v1.Error.VirusErrorCodeH\x00R\x0evirusErrorCode\x12\x16\n\x06reason\x18\x05 \x01(\tR\x06reason\x12\x18\n\x07message\x18\x02 \x01(\tR\x07message\x12/\n\x11valid_identifiers\x18\x03 \x03(\tB\x02\x18\x01R\x10validIdentifiers\x12/\n\x13invalid_identifiers\x18\x04 \x03(\tR\x12invalidIdentifiers\"\xa0\x01\n\x11\x41ssemblyErrorCode\x12\x1f\n\x1bUNKNOWN_ASSEMBLY_ERROR_CODE\x10\x00\x12\x1a\n\x16INVALID_BIOPROJECT_IDS\x10\x01\x12!\n\x1dNO_ASSEMBLIES_FOR_BIOPROJECTS\x10\x02\x12\x11\n\rINVALID_TAXON\x10\x03\x12\x18\n\x14MISSING_SEARCH_FIELD\x10\x04\"J\n\rGeneErrorCode\x12\x1b\n\x17UNKNOWN_GENE_ERROR_CODE\x10\x00\x12\x1c\n\x18INCOMPLETE_LOOKUP_SYMBOL\x10\x05\".\n\x0eVirusErrorCode\x12\x1c\n\x18UNKNOWN_VIRUS_ERROR_CODE\x10\x00\x42\x06\n\x04\x63ode\"\xd7\x04\n\x07Warning\x12W\n\x11gene_warning_code\x18\x07 \x01(\x0e\x32).ncbi.datasets.v1.Warning.GeneWarningCodeH\x00R\x0fgeneWarningCode\x12\x16\n\x06reason\x18\x02 \x01(\tR\x06reason\x12\x18\n\x07message\x18\x03 \x01(\tR\x07message\x12\x45\n\x0breplaced_id\x18\x04 \x01(\x0b\x32$.ncbi.datasets.v1.Warning.ReplacedIdR\nreplacedId\x12\x37\n\x17unrecognized_identifier\x18\x05 \x01(\tR\x16unrecognizedIdentifier\x1a\x46\n\nReplacedId\x12\x1c\n\trequested\x18\x01 \x01(\tR\trequested\x12\x1a\n\x08returned\x18\x02 \x01(\tR\x08returned\"\xf0\x01\n\x0fGeneWarningCode\x12\x1d\n\x19UNKNOWN_GENE_WARNING_CODE\x10\x00\x12\x1e\n\x1a\x41\x43\x43\x45SSION_VERSION_MISMATCH\x10\x01\x12\x14\n\x10REPLACED_GENE_ID\x10\x02\x12\x18\n\x14\x44ISCONTINUED_GENE_ID\x10\x03\x12\x18\n\x14UNRECOGNIZED_GENE_ID\x10\x04\x12\x1c\n\x18UNRECOGNIZED_GENE_SYMBOL\x10\x05\x12\x1a\n\x16UNRECOGNIZED_ACCESSION\x10\x06\x12\x1a\n\x16UNRECOGNIZED_TAX_TOKEN\x10\x07\x42\x06\n\x04\x63ode\"|\n\x07Message\x12/\n\x05\x65rror\x18\x01 \x01(\x0b\x32\x17.ncbi.datasets.v1.ErrorH\x00R\x05\x65rror\x12\x35\n\x07warning\x18\x02 \x01(\x0b\x32\x19.ncbi.datasets.v1.WarningH\x00R\x07warningB\t\n\x07message\"\xfa\x01\n\tGeneMatch\x12\x39\n\x08messages\x18\x01 \x03(\x0b\x32\x19.ncbi.datasets.v1.MessageB\x02\x18\x01R\x08messages\x12\x35\n\x08warnings\x18\x03 \x03(\x0b\x32\x19.ncbi.datasets.v1.WarningR\x08warnings\x12\x14\n\x05query\x18\x04 \x03(\tR\x05query\x12\x34\n\x04gene\x18\x02 \x01(\x0b\x32 .ncbi.datasets.v1.GeneDescriptorR\x04gene\x12/\n\x06\x65rrors\x18\x05 \x03(\x0b\x32\x17.ncbi.datasets.v1.ErrorR\x06\x65rrors\"\x99\x01\n\x0cGeneMetadata\x12\x35\n\x08messages\x18\x01 \x03(\x0b\x32\x19.ncbi.datasets.v1.MessageR\x08messages\x12\x31\n\x05genes\x18\x02 \x03(\x0b\x32\x1b.ncbi.datasets.v1.GeneMatchR\x05genes\x12\x1f\n\x0btotal_count\x18\x03 \x01(\rR\ntotalCount\"\x8f\x01\n\rAssemblyMatch\x12\x35\n\x08messages\x18\x01 \x03(\x0b\x32\x19.ncbi.datasets.v1.MessageR\x08messages\x12G\n\x08\x61ssembly\x18\x02 \x01(\x0b\x32+.ncbi.datasets.v1.AssemblyDatasetDescriptorR\x08\x61ssembly\"\xd3\x01\n\x10\x41ssemblyMetadata\x12\x35\n\x08messages\x18\x01 \x03(\x0b\x32\x19.ncbi.datasets.v1.MessageR\x08messages\x12?\n\nassemblies\x18\x02 \x03(\x0b\x32\x1f.ncbi.datasets.v1.AssemblyMatchR\nassemblies\x12\x1f\n\x0btotal_count\x18\x03 \x01(\rR\ntotalCount\x12&\n\x0fnext_page_token\x18\x04 \x01(\tR\rnextPageToken\"z\n\x0fGeneDescriptors\x12\x36\n\x05genes\x18\x01 \x03(\x0b\x32 .ncbi.datasets.v1.GeneDescriptorR\x05genes\x12/\n\x06\x65rrors\x18\x02 \x03(\x0b\x32\x17.ncbi.datasets.v1.ErrorR\x06\x65rrors\"\x88\x01\n\x12\x41nnotationForVirus\"r\n\x04Type\x12\x0b\n\x07\x44\x45\x46\x41ULT\x10\x00\x12\r\n\tCDS_FASTA\x10\x01\x12\x0e\n\nPROT_FASTA\x10\x02\x12\x0f\n\x0bGENOME_GBFF\x10\x03\x12\x0f\n\x0bGENOME_GPFF\x10\x04\x12\r\n\tPDB_FILES\x10\x05\x12\r\n\tBIOSAMPLE\x10\x06\"\xb7\x0c\n\x13VirusDatasetRequest\x12\x19\n\x06tax_id\x18\x01 \x01(\rB\x02\x18\x01R\x05taxId\x12\x1d\n\x08tax_name\x18\x02 \x01(\tB\x02\x18\x01R\x07taxName\x12\x1c\n\taccession\x18\x12 \x01(\tR\taccession\x12\x63\n\naccessions\x18\x10 \x03(\tBC\xc2\xf0\x19?\x12=\n\x0bNC_045512.2\x12.SARS-COV-2 isolate Wuhan-Hu-1, complete genomeR\naccessions\x12Z\n\x05taxon\x18\x13 \x01(\tBD\xc2\xf0\x19@\x12\x15\n\x07\x32\x36\x39\x37\x30\x34\x39\x12\nSARS-COV-2\x12\x16\n\x05\x31\x31\x31\x31\x38\x12\rCoronaviridae\x12\x0f\n\x07\x31\x33\x33\x35\x36\x32\x36\x12\x04MERSR\x05taxon\x12u\n\x0brefseq_only\x18\x04 \x01(\x08\x42T\xc2\xf0\x19P\n\x05\x66\x61lse\x12\x1f\n\x04true\x12\x17Limit to RefSeq genomes\x12&\n\x05\x66\x61lse\x12\x1d\x42oth GenBank & RefSeq genomesR\nrefseqOnly\x12l\n\x0e\x61nnotated_only\x18\x05 \x01(\x08\x42\x45\xc2\xf0\x19\x41\n\x05\x66\x61lse\x12\"\n\x04true\x12\x1aLimit to annotated genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\rannotatedOnly\x12\x65\n\x0ereleased_since\x18\x06 \x01(\x0b\x32\x1a.google.protobuf.TimestampB\"\xc2\xf0\x19\x1e\x12\x1c\n\n2020-08-01\x12\x0e\x41ugust 1, 2020R\rreleasedSince\x12\x62\n\rupdated_since\x18\x11 \x01(\x0b\x32\x1a.google.protobuf.TimestampB!\xc2\xf0\x19\x1d\x12\x1b\n\n2021-07-18\x12\rJuly 18, 2021R\x0cupdatedSince\x12(\n\x04host\x18\x07 \x01(\tB\x14\xc2\xf0\x19\x10\x12\x0e\n\x05human\x12\x05humanR\x04host\x12\x37\n\x17pangolin_classification\x18\x14 \x01(\tR\x16pangolinClassification\x12@\n\x0cgeo_location\x18\x08 \x01(\tB\x1d\xc2\xf0\x19\x19\x12\n\n\x03USA\x12\x03USA\x12\x0b\n\x02TX\x12\x05TexasR\x0bgeoLocation\x12i\n\rcomplete_only\x18\t \x01(\x08\x42\x44\xc2\xf0\x19@\n\x05\x66\x61lse\x12!\n\x04true\x12\x19Limit to complete genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\x0c\x63ompleteOnly\x12\x98\x01\n\x0ctable_fields\x18\n \x03(\x0e\x32!.ncbi.datasets.v1.VirusTableFieldBR\xc2\xf0\x19N\x12L\n\x14nucleotide_accession\n\x11nucleotide_length\n\x10nuc_completeness\x12\x0f\x41ssembly FieldsR\x0btableFields\x12\x81\x01\n\x10\x65xclude_sequence\x18\x0c \x01(\x08\x42V\xda\xf0\x19\x02\x08\x02\xc2\xf0\x19L\n\x05\x66\x61lse\x12 \n\x04true\x12\x18\x45xclude genomic sequence\x12!\n\x05\x66\x61lse\x12\x18Include genomic sequenceR\x0f\x65xcludeSequence\x12\xb8\x01\n\x17include_annotation_type\x18\r \x03(\x0e\x32).ncbi.datasets.v1.AnnotationForVirus.TypeBU\xda\xf0\x19\x02\x08\x02\xc2\xf0\x19K\x12 \n\tCDS_FASTA\x12\x13Transcript sequence\x12\'\n\tCDS_FASTA\n\nPROT_FASTA\x12\x0e\x41ll annotationR\x15includeAnnotationType\x12m\n\x06\x66ormat\x18\x0e \x01(\x0e\x32\x1d.ncbi.datasets.v1.TableFormatB6\xc2\xf0\x19\x32\n\x03tsv\x12\n\n\x03tsv\x12\x03TSV\x12\n\n\x03\x63sv\x12\x03\x43SV\x12\x13\n\x05jsonl\x12\nJSON LinesR\x06\x66ormat\"\xdf\x07\n\x12VirusDatasetFilter\x12\xac\x01\n\naccessions\x18\x02 \x03(\tB\x8b\x01\xc2\xf0\x19\x86\x01\x12\x83\x01\n\x17NC_045512.2,NC_038294.1\x12hSARS-COV-2 isolate Wuhan-Hu-1 and Middle East respiratory syndrome-related coronavirus, complete genomesR\naccessions\x12Z\n\x05taxon\x18\x03 \x01(\tBD\xc2\xf0\x19@\x12\x15\n\x07\x32\x36\x39\x37\x30\x34\x39\x12\nSARS-COV-2\x12\x16\n\x05\x31\x31\x31\x31\x38\x12\rCoronaviridae\x12\x0f\n\x07\x31\x33\x33\x35\x36\x32\x36\x12\x04MERSR\x05taxon\x12u\n\x0brefseq_only\x18\x04 \x01(\x08\x42T\xc2\xf0\x19P\n\x05\x66\x61lse\x12\x1f\n\x04true\x12\x17Limit to RefSeq genomes\x12&\n\x05\x66\x61lse\x12\x1d\x42oth GenBank & RefSeq genomesR\nrefseqOnly\x12l\n\x0e\x61nnotated_only\x18\x05 \x01(\x08\x42\x45\xc2\xf0\x19\x41\n\x05\x66\x61lse\x12\"\n\x04true\x12\x1aLimit to annotated genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\rannotatedOnly\x12\x65\n\x0ereleased_since\x18\x06 \x01(\x0b\x32\x1a.google.protobuf.TimestampB\"\xc2\xf0\x19\x1e\x12\x1c\n\n2020-08-01\x12\x0e\x41ugust 1, 2020R\rreleasedSince\x12\x62\n\rupdated_since\x18\x0b \x01(\x0b\x32\x1a.google.protobuf.TimestampB!\xc2\xf0\x19\x1d\x12\x1b\n\n2021-07-18\x12\rJuly 18, 2021R\x0cupdatedSince\x12(\n\x04host\x18\x07 \x01(\tB\x14\xc2\xf0\x19\x10\x12\x0e\n\x05human\x12\x05humanR\x04host\x12\x37\n\x17pangolin_classification\x18\x08 \x01(\tR\x16pangolinClassification\x12@\n\x0cgeo_location\x18\t \x01(\tB\x1d\xc2\xf0\x19\x19\x12\n\n\x03USA\x12\x03USA\x12\x0b\n\x02TX\x12\x05TexasR\x0bgeoLocation\x12i\n\rcomplete_only\x18\n \x01(\x08\x42\x44\xc2\xf0\x19@\n\x05\x66\x61lse\x12!\n\x04true\x12\x19Limit to complete genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\x0c\x63ompleteOnly\"\xe0\x03\n\x16VirusDataReportRequest\x12<\n\x06\x66ilter\x18\x01 \x01(\x0b\x32$.ncbi.datasets.v1.VirusDatasetFilterR\x06\x66ilter\x12_\n\x10returned_content\x18\x02 \x01(\x0e\x32\x34.ncbi.datasets.v1.VirusDataReportRequest.ContentTypeR\x0freturnedContent\x12\x64\n\x0ctable_fields\x18\x03 \x03(\tBA\xc2\xf0\x19=\x12;\n\taccession\n\x0bis-complete\n\x0cis-annotated\x12\x13Virus Genome FieldsR\x0btableFields\x12!\n\x0ctable_format\x18\x06 \x01(\tR\x0btableFormat\x12%\n\tpage_size\x18\x04 \x01(\x05\x42\x08\xc2\xf0\x19\x04\n\x02\x32\x30R\x08pageSize\x12\x1d\n\npage_token\x18\x05 \x01(\tR\tpageToken\"0\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\x13\n\x0f\x41\x43\x43\x45SSIONS_ONLY\x10\x01:&\xc2\xf0\x19\"\n taxon: 2697049\nrefseq_only: true\"\xec\n\n\x1aSars2ProteinDatasetRequest\x12~\n\x08proteins\x18\x01 \x03(\tBb\xc2\xf0\x19^\x12\x1e\n\rspike protein\x12\rSpike Protein\x12<\n\rspike protein\n\x10\x65nvelope protein\n\x04RdRp\x12\x13SARS-CoV-2 proteinsR\x08proteins\x12u\n\x0brefseq_only\x18\x04 \x01(\x08\x42T\xc2\xf0\x19P\n\x05\x66\x61lse\x12\x1f\n\x04true\x12\x17Limit to RefSeq genomes\x12&\n\x05\x66\x61lse\x12\x1d\x42oth GenBank & RefSeq genomesR\nrefseqOnly\x12l\n\x0e\x61nnotated_only\x18\x05 \x01(\x08\x42\x45\xc2\xf0\x19\x41\n\x05\x66\x61lse\x12\"\n\x04true\x12\x1aLimit to annotated genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\rannotatedOnly\x12\x65\n\x0ereleased_since\x18\x06 \x01(\x0b\x32\x1a.google.protobuf.TimestampB\"\xc2\xf0\x19\x1e\x12\x1c\n\n2020-08-01\x12\x0e\x41ugust 1, 2020R\rreleasedSince\x12\x62\n\rupdated_since\x18\x0f \x01(\x0b\x32\x1a.google.protobuf.TimestampB!\xc2\xf0\x19\x1d\x12\x1b\n\n2021-07-18\x12\rJuly 18, 2021R\x0cupdatedSince\x12(\n\x04host\x18\x07 \x01(\tB\x14\xc2\xf0\x19\x10\x12\x0e\n\x05human\x12\x05humanR\x04host\x12@\n\x0cgeo_location\x18\x08 \x01(\tB\x1d\xc2\xf0\x19\x19\x12\n\n\x03USA\x12\x03USA\x12\x0b\n\x02TX\x12\x05TexasR\x0bgeoLocation\x12i\n\rcomplete_only\x18\t \x01(\x08\x42\x44\xc2\xf0\x19@\n\x05\x66\x61lse\x12!\n\x04true\x12\x19Limit to complete genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\x0c\x63ompleteOnly\x12\x98\x01\n\x0ctable_fields\x18\n \x03(\x0e\x32!.ncbi.datasets.v1.VirusTableFieldBR\xc2\xf0\x19N\x12L\n\x14nucleotide_accession\n\x11nucleotide_length\n\x10nuc_completeness\x12\x0f\x41ssembly FieldsR\x0btableFields\x12\x81\x01\n\x10\x65xclude_sequence\x18\x0c \x01(\x08\x42V\xda\xf0\x19\x02\x08\x02\xc2\xf0\x19L\n\x05\x66\x61lse\x12 \n\x04true\x12\x18\x45xclude genomic sequence\x12!\n\x05\x66\x61lse\x12\x18Include genomic sequenceR\x0f\x65xcludeSequence\x12\xb8\x01\n\x17include_annotation_type\x18\r \x03(\x0e\x32).ncbi.datasets.v1.AnnotationForVirus.TypeBU\xda\xf0\x19\x02\x08\x02\xc2\xf0\x19K\x12 \n\tCDS_FASTA\x12\x13Transcript sequence\x12\'\n\tCDS_FASTA\n\nPROT_FASTA\x12\x0e\x41ll annotationR\x15includeAnnotationType\x12m\n\x06\x66ormat\x18\x0e \x01(\x0e\x32\x1d.ncbi.datasets.v1.TableFormatB6\xc2\xf0\x19\x32\n\x03tsv\x12\n\n\x03tsv\x12\x03TSV\x12\n\n\x03\x63sv\x12\x03\x43SV\x12\x13\n\x05jsonl\x12\nJSON LinesR\x06\x66ormat\"\xc9\x01\n\x18VirusAvailabilityRequest\x12\xac\x01\n\naccessions\x18\x01 \x03(\tB\x8b\x01\xc2\xf0\x19\x86\x01\x12\x83\x01\n\x17NC_045512.2,NC_038294.1\x12hSARS-COV-2 isolate Wuhan-Hu-1 and Middle East respiratory syndrome-related coronavirus, complete genomesR\naccessions\"\x87\x01\n\x11VirusAvailability\x12)\n\x10valid_accessions\x18\x01 \x03(\tR\x0fvalidAccessions\x12-\n\x12invalid_accessions\x18\x02 \x03(\tR\x11invalidAccessions\x12\x18\n\x07message\x18\x03 \x01(\tR\x07message\"\xc4\x0c\n\x0f\x44ownloadSummary\x12!\n\x0crecord_count\x18\x01 \x01(\rR\x0brecordCount\x12)\n\x0e\x61ssembly_count\x18\x02 \x01(\rB\x02\x18\x01R\rassemblyCount\x12J\n\x13resource_updated_on\x18\x04 \x01(\x0b\x32\x1a.google.protobuf.TimestampR\x11resourceUpdatedOn\x12\x46\n\x08hydrated\x18\x05 \x01(\x0b\x32*.ncbi.datasets.v1.DownloadSummary.HydratedR\x08hydrated\x12L\n\ndehydrated\x18\x06 \x01(\x0b\x32,.ncbi.datasets.v1.DownloadSummary.DehydratedR\ndehydrated\x12\x33\n\x06\x65rrors\x18\x07 \x03(\x0b\x32\x17.ncbi.datasets.v1.ErrorB\x02\x18\x01R\x06\x65rrors\x12\x35\n\x08messages\x18\x08 \x03(\x0b\x32\x19.ncbi.datasets.v1.MessageR\x08messages\x12Y\n\x0f\x61vailable_files\x18\t \x01(\x0b\x32\x30.ncbi.datasets.v1.DownloadSummary.AvailableFilesR\x0e\x61vailableFiles\x1a\x8c\x01\n\x08Hydrated\x12\x33\n\x16\x65stimated_file_size_mb\x18\x01 \x01(\rR\x13\x65stimatedFileSizeMb\x12\x10\n\x03url\x18\x02 \x01(\tR\x03url\x12\x39\n\x19\x63li_download_command_line\x18\x03 \x01(\tR\x16\x63liDownloadCommandLine\x1a\xcb\x01\n\nDehydrated\x12\x33\n\x16\x65stimated_file_size_mb\x18\x01 \x01(\rR\x13\x65stimatedFileSizeMb\x12\x10\n\x03url\x18\x02 \x01(\tR\x03url\x12\x39\n\x19\x63li_download_command_line\x18\x03 \x01(\tR\x16\x63liDownloadCommandLine\x12;\n\x1a\x63li_rehydrate_command_line\x18\x04 \x01(\tR\x17\x63liRehydrateCommandLine\x1a\x45\n\x0b\x46ileSummary\x12\x1d\n\nfile_count\x18\x01 \x01(\rR\tfileCount\x12\x17\n\x07size_mb\x18\x02 \x01(\x02R\x06sizeMb\x1a\x95\x05\n\x0e\x41vailableFiles\x12Y\n\x11\x61ll_genomic_fasta\x18\x01 \x01(\x0b\x32-.ncbi.datasets.v1.DownloadSummary.FileSummaryR\x0f\x61llGenomicFasta\x12L\n\ngenome_gff\x18\x02 \x01(\x0b\x32-.ncbi.datasets.v1.DownloadSummary.FileSummaryR\tgenomeGff\x12N\n\x0bgenome_gbff\x18\x03 \x01(\x0b\x32-.ncbi.datasets.v1.DownloadSummary.FileSummaryR\ngenomeGbff\x12J\n\trna_fasta\x18\x04 \x01(\x0b\x32-.ncbi.datasets.v1.DownloadSummary.FileSummaryR\x08rnaFasta\x12L\n\nprot_fasta\x18\x05 \x01(\x0b\x32-.ncbi.datasets.v1.DownloadSummary.FileSummaryR\tprotFasta\x12L\n\ngenome_gtf\x18\x06 \x01(\x0b\x32-.ncbi.datasets.v1.DownloadSummary.FileSummaryR\tgenomeGtf\x12J\n\tcds_fasta\x18\x07 \x01(\x0b\x32-.ncbi.datasets.v1.DownloadSummary.FileSummaryR\x08\x63\x64sFasta\x12V\n\x0fsequence_report\x18\x08 \x01(\x0b\x32-.ncbi.datasets.v1.DownloadSummary.FileSummaryR\x0esequenceReport\"#\n\rTabularOutput\x12\x12\n\x04\x64\x61ta\x18\x01 \x01(\tR\x04\x64\x61ta\",\n\x12\x45lementFlankConfig\x12\x16\n\x06length\x18\x01 \x01(\rR\x06length\"\x81\x03\n\x1cRefGeneCatalogDatasetRequest\x12,\n\x11opaque_solr_query\x18\x01 \x01(\tH\x00R\x0fopaqueSolrQuery\x12U\n\x05\x66iles\x18\x02 \x03(\x0e\x32\x37.ncbi.datasets.v1.RefGeneCatalogDatasetRequest.FileTypeB\x06\xda\xf0\x19\x02\x08\x02R\x05\x66iles\x12V\n\x14\x65lement_flank_config\x18\x03 \x01(\x0b\x32$.ncbi.datasets.v1.ElementFlankConfigR\x12\x65lementFlankConfig\"{\n\x08\x46ileType\x12\x19\n\x15\x46ILE_TYPE_UNSPECIFIED\x10\x00\x12\x18\n\x14\x46ILE_TYPE_NUCLEOTIDE\x10\x01\x12#\n\x1f\x46ILE_TYPE_NUCLEOTIDE_WITH_FLANK\x10\x02\x12\x15\n\x11\x46ILE_TYPE_PROTEIN\x10\x03\x42\x07\n\x05query\"\xf4\x02\n\x18MicroBiggeDatasetRequest\x12,\n\x11opaque_solr_query\x18\x01 \x01(\tH\x00R\x0fopaqueSolrQuery\x12Q\n\x05\x66iles\x18\x02 \x03(\x0e\x32\x33.ncbi.datasets.v1.MicroBiggeDatasetRequest.FileTypeB\x06\xda\xf0\x19\x02\x08\x02R\x05\x66iles\x12V\n\x14\x65lement_flank_config\x18\x03 \x01(\x0b\x32$.ncbi.datasets.v1.ElementFlankConfigR\x12\x65lementFlankConfig\"v\n\x08\x46ileType\x12\x19\n\x15\x46ILE_TYPE_UNSPECIFIED\x10\x00\x12\x11\n\relement_fasta\x10\x01\x12\x17\n\x13\x65lement_flank_fasta\x10\x02\x12\x10\n\x0c\x63ontig_fasta\x10\x03\x12\x11\n\rprotein_fasta\x10\x04\x42\x07\n\x05query\"\x7f\n\x0cVersionReply\x12\x18\n\x07version\x18\x01 \x01(\tR\x07version\x12\x1b\n\tmajor_ver\x18\x02 \x01(\rR\x08majorVer\x12\x1b\n\tminor_ver\x18\x03 \x01(\rR\x08minorVer\x12\x1b\n\tpatch_ver\x18\x04 \x01(\rR\x08patchVer\"\x81\x05\n\x15ProkaryoteGeneRequest\x12\x93\x01\n\naccessions\x18\x01 \x03(\tBs\xc2\xf0\x19o\x12.\n\x0eWP_015878339.1\x12\x1c\x44NA-deoxyinosine glycosylase\x12=\n\x0eWP_001040085.1\x12+type II CRISPR RNA-guided endonuclease Cas9R\naccessions\x12\xf6\x01\n\x17include_annotation_type\x18\x02 \x03(\x0e\x32\x17.ncbi.datasets.v1.FastaB\xa4\x01\xda\xf0\x19\x02\x08\x02\xc2\xf0\x19\x99\x01\x12\x1e\n\nFASTA_GENE\x12\x10Gene range FASTA\x12?\n\tFASTA_RNA\n\rFASTA_PROTEIN\x12#Transcript & protein sequence FASTA\x12\x36\n\nFASTA_GENE\n\tFASTA_RNA\n\rFASTA_PROTEIN\x12\x0e\x41ll annotationR\x15includeAnnotationType\x12\x63\n\x11gene_flank_config\x18\x03 \x01(\x0b\x32\x37.ncbi.datasets.v1.ProkaryoteGeneRequest.GeneFlankConfigR\x0fgeneFlankConfig\x12\x14\n\x05taxon\x18\x04 \x01(\tR\x05taxon\x1a)\n\x0fGeneFlankConfig\x12\x16\n\x06length\x18\x01 \x01(\rR\x06length:3\xc2\xf0\x19/\n-accessions:\n- WP_000000001.1\n- WP_000000002.1\"L\n\x0cSleepRequest\x12\x1d\n\nsleep_msec\x18\x01 \x01(\x05R\tsleepMsec\x12\x1d\n\nerror_rate\x18\x02 \x01(\x02R\terrorRate\"\"\n\nSleepReply\x12\x14\n\x05reply\x18\x01 \x01(\tR\x05reply\"\xea\x04\n\x0cTaxonomyNode\x12\x15\n\x06tax_id\x18\x01 \x01(\rR\x05taxId\x12#\n\rorganism_name\x18\x02 \x01(\tR\x0corganismName\x12\x1f\n\x0b\x63ommon_name\x18\x03 \x01(\tR\ncommonName\x12\x18\n\x07lineage\x18\x04 \x03(\rR\x07lineage\x12\x1a\n\x08\x63hildren\x18\x05 \x03(\rR\x08\x63hildren\x12_\n-descendent_with_described_species_names_count\x18\x06 \x01(\rR(descendentWithDescribedSpeciesNamesCount\x12\x37\n\x04rank\x18\x07 \x01(\x0e\x32#.ncbi.datasets.v1.Organism.RankTypeR\x04rank\x12;\n\x1ahas_described_species_name\x18\x08 \x01(\x08R\x17hasDescribedSpeciesName\x12\x42\n\x06\x63ounts\x18\t \x03(\x0b\x32*.ncbi.datasets.v1.TaxonomyNode.CountByTypeR\x06\x63ounts\x12\x17\n\x07min_ord\x18\x14 \x01(\rR\x06minOrd\x12\x17\n\x07max_ord\x18\x15 \x01(\rR\x06maxOrd\x12$\n\x0e_indexed_names\x18\x65 \x03(\tR\x0cIndexedNames\x1aT\n\x0b\x43ountByType\x12/\n\x04type\x18\x01 \x01(\x0e\x32\x1b.ncbi.datasets.v1.CountTypeR\x04type\x12\x14\n\x05\x63ount\x18\x02 \x01(\rR\x05\x63ount\"\xd6\x02\n\x17TaxonomyMetadataRequest\x12\x91\x01\n\x06taxons\x18\x01 \x03(\tBy\xc2\xf0\x19u\x12\'\n\x04\x39\x36\x30\x36\n\x05\x31\x30\x30\x39\x30\x12\x18NCBI Taxonomy Identifier\x12\x1b\n\x05human\n\x05mouse\x12\x0b\x43ommon Name\x12-\n\x0cHomo sapiens\n\x0cMus musculus\x12\x0fScientific NameR\x06taxons\x12`\n\x10returned_content\x18\x02 \x01(\x0e\x32\x35.ncbi.datasets.v1.TaxonomyMetadataRequest.ContentTypeR\x0freturnedContent\"\'\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\n\n\x06TAXIDS\x10\x01:\x1c\xc2\xf0\x19\x18\n\x16taxons:\n- 9606\n- mouse\"\xc9\x01\n\rTaxonomyMatch\x12\x35\n\x08warnings\x18\x01 \x03(\x0b\x32\x19.ncbi.datasets.v1.WarningR\x08warnings\x12/\n\x06\x65rrors\x18\x02 \x03(\x0b\x32\x17.ncbi.datasets.v1.ErrorR\x06\x65rrors\x12\x14\n\x05query\x18\x03 \x03(\tR\x05query\x12:\n\x08taxonomy\x18\x04 \x01(\x0b\x32\x1e.ncbi.datasets.v1.TaxonomyNodeR\x08taxonomy\"\x99\x01\n\x18TaxonomyMetadataResponse\x12\x35\n\x08messages\x18\x01 \x03(\x0b\x32\x19.ncbi.datasets.v1.MessageR\x08messages\x12\x46\n\x0etaxonomy_nodes\x18\x02 \x03(\x0b\x32\x1f.ncbi.datasets.v1.TaxonomyMatchR\rtaxonomyNodes\"\xa3\x02\n\x1eTaxonomyFilteredSubtreeRequest\x12\x91\x01\n\x06taxons\x18\x01 \x03(\tBy\xc2\xf0\x19u\x12\'\n\x04\x39\x36\x30\x36\n\x05\x31\x30\x30\x39\x30\x12\x18NCBI Taxonomy Identifier\x12\x1b\n\x05human\n\x05mouse\x12\x0b\x43ommon Name\x12-\n\x0cHomo sapiens\n\x0cMus musculus\x12\x0fScientific NameR\x06taxons\x12\'\n\x0fspecified_limit\x18\x02 \x01(\x08R\x0especifiedLimit\x12\x44\n\x0brank_limits\x18\x03 \x03(\x0e\x32#.ncbi.datasets.v1.Organism.RankTypeR\nrankLimits\"\xdf\x04\n\x1fTaxonomyFilteredSubtreeResponse\x12\x1d\n\nroot_nodes\x18\x01 \x03(\rR\trootNodes\x12R\n\x05\x65\x64ges\x18\x02 \x03(\x0b\x32<.ncbi.datasets.v1.TaxonomyFilteredSubtreeResponse.EdgesEntryR\x05\x65\x64ges\x12\x35\n\x08warnings\x18\x03 \x03(\x0b\x32\x19.ncbi.datasets.v1.WarningR\x08warnings\x12/\n\x06\x65rrors\x18\x04 \x03(\x0b\x32\x17.ncbi.datasets.v1.ErrorR\x06\x65rrors\x1a\xee\x01\n\x04\x45\x64ge\x12)\n\x10visible_children\x18\x01 \x03(\rR\x0fvisibleChildren\x12k\n\x0f\x63hildren_status\x18\x02 \x01(\x0e\x32\x42.ncbi.datasets.v1.TaxonomyFilteredSubtreeResponse.Edge.ChildStatusR\x0e\x63hildrenStatus\"N\n\x0b\x43hildStatus\x12\x0f\n\x0bUNSPECIFIED\x10\x00\x12\x15\n\x11HAS_MORE_CHILDREN\x10\x01\x12\x17\n\x13NO_VISIBLE_CHILDREN\x10\x02\x1ap\n\nEdgesEntry\x12\x10\n\x03key\x18\x01 \x01(\rR\x03key\x12L\n\x05value\x18\x02 \x01(\x0b\x32\x36.ncbi.datasets.v1.TaxonomyFilteredSubtreeResponse.EdgeR\x05value:\x02\x38\x01\"\xe9\x01\n\x0e\x44\x61tasetRequest\x12G\n\tgenome_v1\x18\x01 \x01(\x0b\x32(.ncbi.datasets.v1.AssemblyDatasetRequestH\x00R\x08genomeV1\x12?\n\x07gene_v1\x18\x02 \x01(\x0b\x32$.ncbi.datasets.v1.GeneDatasetRequestH\x00R\x06geneV1\x12\x42\n\x08virus_v1\x18\x03 \x01(\x0b\x32%.ncbi.datasets.v1.VirusDatasetRequestH\x00R\x07virusV1B\t\n\x07request*\xf9\x02\n\tCountType\x12\x1a\n\x16\x43OUNT_TYPE_UNSPECIFIED\x10\x00\x12\x17\n\x13\x43OUNT_TYPE_ASSEMBLY\x10\x01\x12\x13\n\x0f\x43OUNT_TYPE_GENE\x10\x02\x12\x13\n\x0f\x43OUNT_TYPE_tRNA\x10\x03\x12\x13\n\x0f\x43OUNT_TYPE_rRNA\x10\x04\x12\x14\n\x10\x43OUNT_TYPE_snRNA\x10\x05\x12\x14\n\x10\x43OUNT_TYPE_scRNA\x10\x06\x12\x15\n\x11\x43OUNT_TYPE_snoRNA\x10\x07\x12\x1d\n\x19\x43OUNT_TYPE_PROTEIN_CODING\x10\x08\x12\x15\n\x11\x43OUNT_TYPE_PSEUDO\x10\t\x12\x19\n\x15\x43OUNT_TYPE_TRANSPOSON\x10\n\x12\x16\n\x12\x43OUNT_TYPE_miscRNA\x10\x0b\x12\x14\n\x10\x43OUNT_TYPE_ncRNA\x10\x0c\x12 \n\x1c\x43OUNT_TYPE_BIOLOGICAL_REGION\x10\r\x12\x14\n\x10\x43OUNT_TYPE_OTHER\x10\x0e*l\n\rSortDirection\x12\x1e\n\x1aSORT_DIRECTION_UNSPECIFIED\x10\x00\x12\x1c\n\x18SORT_DIRECTION_ASCENDING\x10\x01\x12\x1d\n\x19SORT_DIRECTION_DESCENDING\x10\x02*\x99\x01\n\x05\x46\x61sta\x12\x15\n\x11\x46\x41STA_UNSPECIFIED\x10\x00\x12\x0e\n\nFASTA_GENE\x10\x01\x12\r\n\tFASTA_RNA\x10\x02\x12\x11\n\rFASTA_PROTEIN\x10\x03\x12\x14\n\x10\x46\x41STA_GENE_FLANK\x10\x04\x12\r\n\tFASTA_CDS\x10\x05\x12\x10\n\x0c\x46\x41STA_5P_UTR\x10\x06\x12\x10\n\x0c\x46\x41STA_3P_UTR\x10\x07*\xa1\x03\n\x0fVirusTableField\x12\x0f\n\x0bunspecified\x10\x00\x12\x18\n\x14nucleotide_accession\x10\x01\x12\x12\n\x0especies_tax_id\x10\x02\x12\x10\n\x0cspecies_name\x10\x03\x12\t\n\x05genus\x10\x04\x12\n\n\x06\x66\x61mily\x10\x05\x12\x15\n\x11nucleotide_length\x10\x07\x12\x10\n\x0cisolate_name\x10\x08\x12\x11\n\rsequence_type\x10\t\x12\x14\n\x10nuc_completeness\x10\n\x12\x10\n\x0cgeo_location\x10\x0b\x12\x0c\n\x08us_state\x10\x0c\x12\r\n\thost_name\x10\r\x12\x0f\n\x0bhost_tax_id\x10\x0e\x12\x13\n\x0f\x63ollection_date\x10\x0f\x12\x0e\n\nbioproject\x10\x10\x12\r\n\tbiosample\x10\x11\x12\x14\n\x10polyprotein_name\x10\x13\x12\x10\n\x0cprotein_name\x10\x14\x12\x15\n\x11protein_accession\x10\x15\x12\x13\n\x0fprotein_synonym\x10\x16\x12\x0c\n\x08\x63\x64s_span\x10\x17**\n\x0bTableFormat\x12\x07\n\x03tsv\x10\x00\x12\x07\n\x03\x63sv\x10\x01\x12\t\n\x05jsonl\x10\x02\x42\x15Z\x10ncbi/datasets/v1\xf8\x01\x01\x62\x06proto3'
  ,
  dependencies=[google_dot_protobuf_dot_timestamp__pb2.DESCRIPTOR,ncbi_dot_datasets_dot_options_dot_openapi__pb2.DESCRIPTOR,ncbi_dot_datasets_dot_options_dot_request__pb2.DESCRIPTOR,ncbi_dot_datasets_dot_v1_dot_gene__descriptor__pb2.DESCRIPTOR,ncbi_dot_datasets_dot_v1_dot_reports_dot_biosample__pb2.DESCRIPTOR,])

_COUNTTYPE = _descriptor.EnumDescriptor(
  name='CountType',
  full_name='ncbi.datasets.v1.CountType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_UNSPECIFIED', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_ASSEMBLY', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_GENE', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_tRNA', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_rRNA', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_snRNA', index=5, number=5,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_scRNA', index=6, number=6,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_snoRNA', index=7, number=7,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_PROTEIN_CODING', index=8, number=8,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_PSEUDO', index=9, number=9,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_TRANSPOSON', index=10, number=10,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_miscRNA', index=11, number=11,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_ncRNA', index=12, number=12,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_BIOLOGICAL_REGION', index=13, number=13,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNT_TYPE_OTHER', index=14, number=14,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=25083,
  serialized_end=25460,
)
_sym_db.RegisterEnumDescriptor(_COUNTTYPE)

CountType = enum_type_wrapper.EnumTypeWrapper(_COUNTTYPE)
_SORTDIRECTION = _descriptor.EnumDescriptor(
  name='SortDirection',
  full_name='ncbi.datasets.v1.SortDirection',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='SORT_DIRECTION_UNSPECIFIED', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SORT_DIRECTION_ASCENDING', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SORT_DIRECTION_DESCENDING', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=25462,
  serialized_end=25570,
)
_sym_db.RegisterEnumDescriptor(_SORTDIRECTION)

SortDirection = enum_type_wrapper.EnumTypeWrapper(_SORTDIRECTION)
_FASTA = _descriptor.EnumDescriptor(
  name='Fasta',
  full_name='ncbi.datasets.v1.Fasta',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='FASTA_UNSPECIFIED', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FASTA_GENE', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FASTA_RNA', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FASTA_PROTEIN', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FASTA_GENE_FLANK', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FASTA_CDS', index=5, number=5,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FASTA_5P_UTR', index=6, number=6,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FASTA_3P_UTR', index=7, number=7,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=25573,
  serialized_end=25726,
)
_sym_db.RegisterEnumDescriptor(_FASTA)

Fasta = enum_type_wrapper.EnumTypeWrapper(_FASTA)
_VIRUSTABLEFIELD = _descriptor.EnumDescriptor(
  name='VirusTableField',
  full_name='ncbi.datasets.v1.VirusTableField',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='unspecified', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='nucleotide_accession', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='species_tax_id', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='species_name', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='genus', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='family', index=5, number=5,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='nucleotide_length', index=6, number=7,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='isolate_name', index=7, number=8,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='sequence_type', index=8, number=9,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='nuc_completeness', index=9, number=10,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='geo_location', index=10, number=11,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='us_state', index=11, number=12,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='host_name', index=12, number=13,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='host_tax_id', index=13, number=14,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='collection_date', index=14, number=15,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='bioproject', index=15, number=16,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='biosample', index=16, number=17,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='polyprotein_name', index=17, number=19,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='protein_name', index=18, number=20,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='protein_accession', index=19, number=21,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='protein_synonym', index=20, number=22,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='cds_span', index=21, number=23,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=25729,
  serialized_end=26146,
)
_sym_db.RegisterEnumDescriptor(_VIRUSTABLEFIELD)

VirusTableField = enum_type_wrapper.EnumTypeWrapper(_VIRUSTABLEFIELD)
_TABLEFORMAT = _descriptor.EnumDescriptor(
  name='TableFormat',
  full_name='ncbi.datasets.v1.TableFormat',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='tsv', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='csv', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='jsonl', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=26148,
  serialized_end=26190,
)
_sym_db.RegisterEnumDescriptor(_TABLEFORMAT)

TableFormat = enum_type_wrapper.EnumTypeWrapper(_TABLEFORMAT)
COUNT_TYPE_UNSPECIFIED = 0
COUNT_TYPE_ASSEMBLY = 1
COUNT_TYPE_GENE = 2
COUNT_TYPE_tRNA = 3
COUNT_TYPE_rRNA = 4
COUNT_TYPE_snRNA = 5
COUNT_TYPE_scRNA = 6
COUNT_TYPE_snoRNA = 7
COUNT_TYPE_PROTEIN_CODING = 8
COUNT_TYPE_PSEUDO = 9
COUNT_TYPE_TRANSPOSON = 10
COUNT_TYPE_miscRNA = 11
COUNT_TYPE_ncRNA = 12
COUNT_TYPE_BIOLOGICAL_REGION = 13
COUNT_TYPE_OTHER = 14
SORT_DIRECTION_UNSPECIFIED = 0
SORT_DIRECTION_ASCENDING = 1
SORT_DIRECTION_DESCENDING = 2
FASTA_UNSPECIFIED = 0
FASTA_GENE = 1
FASTA_RNA = 2
FASTA_PROTEIN = 3
FASTA_GENE_FLANK = 4
FASTA_CDS = 5
FASTA_5P_UTR = 6
FASTA_3P_UTR = 7
unspecified = 0
nucleotide_accession = 1
species_tax_id = 2
species_name = 3
genus = 4
family = 5
nucleotide_length = 7
isolate_name = 8
sequence_type = 9
nuc_completeness = 10
geo_location = 11
us_state = 12
host_name = 13
host_tax_id = 14
collection_date = 15
bioproject = 16
biosample = 17
polyprotein_name = 19
protein_name = 20
protein_accession = 21
protein_synonym = 22
cds_span = 23
tsv = 0
csv = 1
jsonl = 2


_ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYSOURCE = _descriptor.EnumDescriptor(
  name='AssemblySource',
  full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter.AssemblySource',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='all', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='refseq', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='genbank', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=2080,
  serialized_end=2130,
)
_sym_db.RegisterEnumDescriptor(_ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYSOURCE)

_ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYVERSION = _descriptor.EnumDescriptor(
  name='AssemblyVersion',
  full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter.AssemblyVersion',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='current', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='all_assemblies', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=2132,
  serialized_end=2182,
)
_sym_db.RegisterEnumDescriptor(_ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYVERSION)

_ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYLEVEL = _descriptor.EnumDescriptor(
  name='AssemblyLevel',
  full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter.AssemblyLevel',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='chromosome', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='scaffold', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='contig', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='complete_genome', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=2184,
  serialized_end=2262,
)
_sym_db.RegisterEnumDescriptor(_ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYLEVEL)

_ASSEMBLYMETADATAREQUEST_CONTENTTYPE = _descriptor.EnumDescriptor(
  name='ContentType',
  full_name='ncbi.datasets.v1.AssemblyMetadataRequest.ContentType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='COMPLETE', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='ASSM_ACC', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=3157,
  serialized_end=3198,
)
_sym_db.RegisterEnumDescriptor(_ASSEMBLYMETADATAREQUEST_CONTENTTYPE)

_ASSEMBLYDATASETDESCRIPTORSREQUEST_CONTENTTYPE = _descriptor.EnumDescriptor(
  name='ContentType',
  full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsRequest.ContentType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='COMPLETE', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='ASSM_ACC', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COUNTS', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=3719,
  serialized_end=3772,
)
_sym_db.RegisterEnumDescriptor(_ASSEMBLYDATASETDESCRIPTORSREQUEST_CONTENTTYPE)

_ANNOTATIONFORASSEMBLY_TYPE = _descriptor.EnumDescriptor(
  name='Type',
  full_name='ncbi.datasets.v1.AnnotationForAssembly.Type',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='DEFAULT', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GENOME_GFF', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GENOME_GBFF', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GENOME_GB', index=3, number=2,
      serialized_options=b'\010\001',
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='RNA_FASTA', index=4, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PROT_FASTA', index=5, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GENOME_GTF', index=6, number=5,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='CDS_FASTA', index=7, number=6,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='ASM_STRUCT', index=8, number=7,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GENOME_FASTA', index=9, number=8,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=b'\020\001',
  serialized_start=5946,
  serialized_end=6117,
)
_sym_db.RegisterEnumDescriptor(_ANNOTATIONFORASSEMBLY_TYPE)

_ASSEMBLYDATASETREQUEST_RESOLUTION = _descriptor.EnumDescriptor(
  name='Resolution',
  full_name='ncbi.datasets.v1.AssemblyDatasetRequest.Resolution',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='FULLY_HYDRATED', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='DATA_REPORT_ONLY', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=7164,
  serialized_end=7218,
)
_sym_db.RegisterEnumDescriptor(_ASSEMBLYDATASETREQUEST_RESOLUTION)

_ORGANISM_RANKTYPE = _descriptor.EnumDescriptor(
  name='RankType',
  full_name='ncbi.datasets.v1.Organism.RankType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='NO_RANK', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUPERKINGDOM', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='KINGDOM', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBKINGDOM', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUPERPHYLUM', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBPHYLUM', index=5, number=5,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PHYLUM', index=6, number=6,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='CLADE', index=7, number=31,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUPERCLASS', index=8, number=7,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='CLASS', index=9, number=8,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBCLASS', index=10, number=9,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='INFRACLASS', index=11, number=10,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='COHORT', index=12, number=11,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBCOHORT', index=13, number=12,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUPERORDER', index=14, number=13,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='ORDER', index=15, number=14,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBORDER', index=16, number=15,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='INFRAORDER', index=17, number=16,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PARVORDER', index=18, number=17,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUPERFAMILY', index=19, number=18,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FAMILY', index=20, number=19,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBFAMILY', index=21, number=20,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GENUS', index=22, number=21,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBGENUS', index=23, number=22,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SPECIES_GROUP', index=24, number=23,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SPECIES_SUBGROUP', index=25, number=24,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SPECIES', index=26, number=25,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBSPECIES', index=27, number=26,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='TRIBE', index=28, number=27,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBTRIBE', index=29, number=28,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FORMA', index=30, number=29,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='VARIETAS', index=31, number=30,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='STRAIN', index=32, number=320,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SECTION', index=33, number=330,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBSECTION', index=34, number=340,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PATHOGROUP', index=35, number=350,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SUBVARIETY', index=36, number=360,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GENOTYPE', index=37, number=370,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SEROTYPE', index=38, number=380,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='ISOLATE', index=39, number=390,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='MORPH', index=40, number=400,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SERIES', index=41, number=410,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FORMA_SPECIALIS', index=42, number=420,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SEROGROUP', index=43, number=430,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='BIOTYPE', index=44, number=440,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=8524,
  serialized_end=9193,
)
_sym_db.RegisterEnumDescriptor(_ORGANISM_RANKTYPE)

_ORGANISMQUERYREQUEST_TAXRANKFILTER = _descriptor.EnumDescriptor(
  name='TaxRankFilter',
  full_name='ncbi.datasets.v1.OrganismQueryRequest.TaxRankFilter',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='species', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='higher_taxon', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=9810,
  serialized_end=9856,
)
_sym_db.RegisterEnumDescriptor(_ORGANISMQUERYREQUEST_TAXRANKFILTER)

_ORGANISMQUERYREQUEST_TAXONRESOURCEFILTER = _descriptor.EnumDescriptor(
  name='TaxonResourceFilter',
  full_name='ncbi.datasets.v1.OrganismQueryRequest.TaxonResourceFilter',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='all', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='genome', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='gene', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=9858,
  serialized_end=9910,
)
_sym_db.RegisterEnumDescriptor(_ORGANISMQUERYREQUEST_TAXONRESOURCEFILTER)

_ORTHOLOGREQUEST_CONTENTTYPE = _descriptor.EnumDescriptor(
  name='ContentType',
  full_name='ncbi.datasets.v1.OrthologRequest.ContentType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='COMPLETE', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='IDS_ONLY', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=10488,
  serialized_end=10529,
)
_sym_db.RegisterEnumDescriptor(_ORTHOLOGREQUEST_CONTENTTYPE)

_GENEDATASETREQUEST_CONTENTTYPE = _descriptor.EnumDescriptor(
  name='ContentType',
  full_name='ncbi.datasets.v1.GeneDatasetRequest.ContentType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='COMPLETE', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='IDS_ONLY', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=10488,
  serialized_end=10529,
)
_sym_db.RegisterEnumDescriptor(_GENEDATASETREQUEST_CONTENTTYPE)

_GENEDATASETREQUEST_SORTFIELD = _descriptor.EnumDescriptor(
  name='SortField',
  full_name='ncbi.datasets.v1.GeneDatasetRequest.SortField',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='SORT_FIELD_UNSPECIFIED', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SORT_FIELD_GENE_ID', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SORT_FIELD_GENE_TYPE', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='SORT_FIELD_GENE_SYMBOL', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=11849,
  serialized_end=11966,
)
_sym_db.RegisterEnumDescriptor(_GENEDATASETREQUEST_SORTFIELD)

_ERROR_ASSEMBLYERRORCODE = _descriptor.EnumDescriptor(
  name='AssemblyErrorCode',
  full_name='ncbi.datasets.v1.Error.AssemblyErrorCode',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='UNKNOWN_ASSEMBLY_ERROR_CODE', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='INVALID_BIOPROJECT_IDS', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='NO_ASSEMBLIES_FOR_BIOPROJECTS', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='INVALID_TAXON', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='MISSING_SEARCH_FIELD', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=12418,
  serialized_end=12578,
)
_sym_db.RegisterEnumDescriptor(_ERROR_ASSEMBLYERRORCODE)

_ERROR_GENEERRORCODE = _descriptor.EnumDescriptor(
  name='GeneErrorCode',
  full_name='ncbi.datasets.v1.Error.GeneErrorCode',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='UNKNOWN_GENE_ERROR_CODE', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='INCOMPLETE_LOOKUP_SYMBOL', index=1, number=5,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=12580,
  serialized_end=12654,
)
_sym_db.RegisterEnumDescriptor(_ERROR_GENEERRORCODE)

_ERROR_VIRUSERRORCODE = _descriptor.EnumDescriptor(
  name='VirusErrorCode',
  full_name='ncbi.datasets.v1.Error.VirusErrorCode',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='UNKNOWN_VIRUS_ERROR_CODE', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=12656,
  serialized_end=12702,
)
_sym_db.RegisterEnumDescriptor(_ERROR_VIRUSERRORCODE)

_WARNING_GENEWARNINGCODE = _descriptor.EnumDescriptor(
  name='GeneWarningCode',
  full_name='ncbi.datasets.v1.Warning.GeneWarningCode',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='UNKNOWN_GENE_WARNING_CODE', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='ACCESSION_VERSION_MISMATCH', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='REPLACED_GENE_ID', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='DISCONTINUED_GENE_ID', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='UNRECOGNIZED_GENE_ID', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='UNRECOGNIZED_GENE_SYMBOL', index=5, number=5,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='UNRECOGNIZED_ACCESSION', index=6, number=6,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='UNRECOGNIZED_TAX_TOKEN', index=7, number=7,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=13064,
  serialized_end=13304,
)
_sym_db.RegisterEnumDescriptor(_WARNING_GENEWARNINGCODE)

_ANNOTATIONFORVIRUS_TYPE = _descriptor.EnumDescriptor(
  name='Type',
  full_name='ncbi.datasets.v1.AnnotationForVirus.Type',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='DEFAULT', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='CDS_FASTA', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PROT_FASTA', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GENOME_GBFF', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='GENOME_GPFF', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='PDB_FILES', index=5, number=5,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='BIOSAMPLE', index=6, number=6,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=14356,
  serialized_end=14470,
)
_sym_db.RegisterEnumDescriptor(_ANNOTATIONFORVIRUS_TYPE)

_VIRUSDATAREPORTREQUEST_CONTENTTYPE = _descriptor.EnumDescriptor(
  name='ContentType',
  full_name='ncbi.datasets.v1.VirusDataReportRequest.ContentType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='COMPLETE', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='ACCESSIONS_ONLY', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=17453,
  serialized_end=17501,
)
_sym_db.RegisterEnumDescriptor(_VIRUSDATAREPORTREQUEST_CONTENTTYPE)

_REFGENECATALOGDATASETREQUEST_FILETYPE = _descriptor.EnumDescriptor(
  name='FileType',
  full_name='ncbi.datasets.v1.RefGeneCatalogDatasetRequest.FileType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='FILE_TYPE_UNSPECIFIED', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FILE_TYPE_NUCLEOTIDE', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FILE_TYPE_NUCLEOTIDE_WITH_FLANK', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='FILE_TYPE_PROTEIN', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=21220,
  serialized_end=21343,
)
_sym_db.RegisterEnumDescriptor(_REFGENECATALOGDATASETREQUEST_FILETYPE)

_MICROBIGGEDATASETREQUEST_FILETYPE = _descriptor.EnumDescriptor(
  name='FileType',
  full_name='ncbi.datasets.v1.MicroBiggeDatasetRequest.FileType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='FILE_TYPE_UNSPECIFIED', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='element_fasta', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='element_flank_fasta', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='contig_fasta', index=3, number=3,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='protein_fasta', index=4, number=4,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=21600,
  serialized_end=21718,
)
_sym_db.RegisterEnumDescriptor(_MICROBIGGEDATASETREQUEST_FILETYPE)

_TAXONOMYMETADATAREQUEST_CONTENTTYPE = _descriptor.EnumDescriptor(
  name='ContentType',
  full_name='ncbi.datasets.v1.TaxonomyMetadataRequest.ContentType',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='COMPLETE', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='TAXIDS', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=23511,
  serialized_end=23550,
)
_sym_db.RegisterEnumDescriptor(_TAXONOMYMETADATAREQUEST_CONTENTTYPE)

_TAXONOMYFILTEREDSUBTREERESPONSE_EDGE_CHILDSTATUS = _descriptor.EnumDescriptor(
  name='ChildStatus',
  full_name='ncbi.datasets.v1.TaxonomyFilteredSubtreeResponse.Edge.ChildStatus',
  filename=None,
  file=DESCRIPTOR,
  create_key=_descriptor._internal_create_key,
  values=[
    _descriptor.EnumValueDescriptor(
      name='UNSPECIFIED', index=0, number=0,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='HAS_MORE_CHILDREN', index=1, number=1,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
    _descriptor.EnumValueDescriptor(
      name='NO_VISIBLE_CHILDREN', index=2, number=2,
      serialized_options=None,
      type=None,
      create_key=_descriptor._internal_create_key),
  ],
  containing_type=None,
  serialized_options=None,
  serialized_start=24652,
  serialized_end=24730,
)
_sym_db.RegisterEnumDescriptor(_TAXONOMYFILTEREDSUBTREERESPONSE_EDGE_CHILDSTATUS)


_METHODPAYLOADREQUEST = _descriptor.Descriptor(
  name='MethodPayloadRequest',
  full_name='ncbi.datasets.v1.MethodPayloadRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='method', full_name='ncbi.datasets.v1.MethodPayloadRequest.method', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='method', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='payload', full_name='ncbi.datasets.v1.MethodPayloadRequest.payload', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='payload', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=242,
  serialized_end=314,
)


_ASSEMBLYDATASETDESCRIPTORSFILTER = _descriptor.Descriptor(
  name='AssemblyDatasetDescriptorsFilter',
  full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='reference_only', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter.reference_only', index=0,
      number=1, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031r\n\005false\0223\n\004true\022+Limit to reference & representative genomes\0224\n\005false\022+Include reference and non-reference genomes', json_name='referenceOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='refseq_only', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter.refseq_only', index=1,
      number=2, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='refseqOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_source', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter.assembly_source', index=2,
      number=3, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031a\n\003all\022!\n\006refseq\022\027Limit to RefSeq genomes\022#\n\007genbank\022\030Limit to GenBank genomes\022\022\n\003all\022\013All genomes', json_name='assemblySource', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='has_annotation', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter.has_annotation', index=3,
      number=7, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031N\n\005false\022%\n\004true\022\035Limit to annotated assemblies\022\036\n\005false\022\025Return all assemblies', json_name='hasAnnotation', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='exclude_paired_reports', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter.exclude_paired_reports', index=4,
      number=9, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\\\n\005false\022\036\n\005false\022\025Return all assemblies\0223\n\004true\022+Only return single record for GCA/GCF pairs', json_name='excludePairedReports', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='exclude_atypical', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter.exclude_atypical', index=5,
      number=11, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\226\001\n\005false\022F\n\005false\022=Return all selected genomes, including ones that are atypical\022E\n\004true\022=Only return genomes that have not been classified as atypical', json_name='excludeAtypical', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_version', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter.assembly_version', index=6,
      number=10, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\200\001\n\007current\022)\n\007current\022\036Only return current assemblies\022J\n\016all_assemblies\0228Return all assemblies, including suppressed and replaced', json_name='assemblyVersion', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_level', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter.assembly_level', index=7,
      number=4, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031}\022=\n\nchromosome\n\017complete_genome\022\036Chromosome and complete genome\022<\n\010scaffold\n\nchromosome\n\017complete_genome\022\023Scaffold and higher', json_name='assemblyLevel', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='first_release_date', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter.first_release_date', index=8,
      number=5, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\034\022\032\n\n01/10/2015\022\014Jan 10, 2015', json_name='firstReleaseDate', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='last_release_date', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter.last_release_date', index=9,
      number=6, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\034\022\032\n\n01/10/2021\022\014Jan 10, 2021', json_name='lastReleaseDate', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='search_text', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter.search_text', index=10,
      number=8, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031d\022:\n\033Genome Reference Consortium\022\033Genome Reference Consortium\022\020\n\006GRCh38\022\006GRCh38\022\024\n\010C57BL/6J\022\010C57BL/6J', json_name='searchText', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYSOURCE,
    _ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYVERSION,
    _ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYLEVEL,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=317,
  serialized_end=2262,
)


_ACCESSIONS = _descriptor.Descriptor(
  name='Accessions',
  full_name='ncbi.datasets.v1.Accessions',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v1.Accessions.accessions', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031R\0225\n\020GCF_000001405.40\n\020GCF_000001635.27\022\017Human and Mouse\022\031\n\020GCF_000001405.40\022\005Human', json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=2265,
  serialized_end=2397,
)


_ASSEMBLYMETADATAREQUEST_BIOPROJECTS = _descriptor.Descriptor(
  name='Bioprojects',
  full_name='ncbi.datasets.v1.AssemblyMetadataRequest.Bioprojects',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v1.AssemblyMetadataRequest.Bioprojects.accessions', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031O\022)\n\013PRJNA489243\022\032Vertebrate Genomes Project\022\"\n\nPRJNA31257\022\024Human Genome Project', json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=3025,
  serialized_end=3155,
)

_ASSEMBLYMETADATAREQUEST = _descriptor.Descriptor(
  name='AssemblyMetadataRequest',
  full_name='ncbi.datasets.v1.AssemblyMetadataRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='taxon', full_name='ncbi.datasets.v1.AssemblyMetadataRequest.taxon', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031Y\022 \n\0049606\022\030NCBI Taxonomy Identifier\022\024\n\005human\022\013Common Name\022\037\n\014Homo sapiens\022\017Scientific Name', json_name='taxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v1.AssemblyMetadataRequest.accessions', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='bioprojects', full_name='ncbi.datasets.v1.AssemblyMetadataRequest.bioprojects', index=2,
      number=7, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='bioprojects', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='filters', full_name='ncbi.datasets.v1.AssemblyMetadataRequest.filters', index=3,
      number=4, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='filters', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='tax_exact_match', full_name='ncbi.datasets.v1.AssemblyMetadataRequest.tax_exact_match', index=4,
      number=5, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\007\n\005false', json_name='taxExactMatch', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='returned_content', full_name='ncbi.datasets.v1.AssemblyMetadataRequest.returned_content', index=5,
      number=6, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='returnedContent', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='limit', full_name='ncbi.datasets.v1.AssemblyMetadataRequest.limit', index=6,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='limit', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_size', full_name='ncbi.datasets.v1.AssemblyMetadataRequest.page_size', index=7,
      number=8, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\004\n\00220', json_name='pageSize', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_token', full_name='ncbi.datasets.v1.AssemblyMetadataRequest.page_token', index=8,
      number=9, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='pageToken', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_ASSEMBLYMETADATAREQUEST_BIOPROJECTS, ],
  enum_types=[
    _ASSEMBLYMETADATAREQUEST_CONTENTTYPE,
  ],
  serialized_options=b'\302\360\0310\n.accessions:\n  accessions:\n  - GCF_000001405.40',
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
    _descriptor.OneofDescriptor(
      name='records', full_name='ncbi.datasets.v1.AssemblyMetadataRequest.records',
      index=0, containing_type=None,
      create_key=_descriptor._internal_create_key,
    fields=[]),
  ],
  serialized_start=2400,
  serialized_end=3263,
)


_ASSEMBLYDATASETDESCRIPTORSREQUEST = _descriptor.Descriptor(
  name='AssemblyDatasetDescriptorsRequest',
  full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='tax_id', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsRequest.tax_id', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='taxId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='org_name', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsRequest.org_name', index=1,
      number=5, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='orgName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='tax_name', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsRequest.tax_name', index=2,
      number=7, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='taxName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_accession', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsRequest.assembly_accession', index=3,
      number=4, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='assemblyAccession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='cutoff', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsRequest.cutoff', index=4,
      number=2, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='cutoff', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='limit', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsRequest.limit', index=5,
      number=8, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='limit', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='filters', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsRequest.filters', index=6,
      number=3, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='filters', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='tax_exact_match', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsRequest.tax_exact_match', index=7,
      number=6, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='taxExactMatch', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='returned_content', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsRequest.returned_content', index=8,
      number=9, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='returnedContent', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _ASSEMBLYDATASETDESCRIPTORSREQUEST_CONTENTTYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
    _descriptor.OneofDescriptor(
      name='id', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsRequest.id',
      index=0, containing_type=None,
      create_key=_descriptor._internal_create_key,
    fields=[]),
    _descriptor.OneofDescriptor(
      name='records', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptorsRequest.records',
      index=1, containing_type=None,
      create_key=_descriptor._internal_create_key,
    fields=[]),
  ],
  serialized_start=3266,
  serialized_end=3789,
)


_ASSEMBLYDATASETDESCRIPTORS = _descriptor.Descriptor(
  name='AssemblyDatasetDescriptors',
  full_name='ncbi.datasets.v1.AssemblyDatasetDescriptors',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='datasets', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptors.datasets', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='datasets', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='total_count', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptors.total_count', index=1,
      number=2, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='totalCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=3792,
  serialized_end=3926,
)


_BIOPROJECT = _descriptor.Descriptor(
  name='BioProject',
  full_name='ncbi.datasets.v1.BioProject',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accession', full_name='ncbi.datasets.v1.BioProject.accession', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='accession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='title', full_name='ncbi.datasets.v1.BioProject.title', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='title', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='parent_accession', full_name='ncbi.datasets.v1.BioProject.parent_accession', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='parentAccession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='parent_accessions', full_name='ncbi.datasets.v1.BioProject.parent_accessions', index=3,
      number=4, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='parentAccessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=3929,
  serialized_end=4081,
)


_BIOPROJECTLINEAGE = _descriptor.Descriptor(
  name='BioProjectLineage',
  full_name='ncbi.datasets.v1.BioProjectLineage',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='bioprojects', full_name='ncbi.datasets.v1.BioProjectLineage.bioprojects', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='bioprojects', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=4083,
  serialized_end=4166,
)


_ASSEMBLYDATASETDESCRIPTOR_CHROMOSOME = _descriptor.Descriptor(
  name='Chromosome',
  full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.Chromosome',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='name', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.Chromosome.name', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='name', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='length', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.Chromosome.length', index=1,
      number=2, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='length', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='accession_version', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.Chromosome.accession_version', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='accessionVersion', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='gc_count', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.Chromosome.gc_count', index=3,
      number=4, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='gcCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=5091,
  serialized_end=5219,
)

_ASSEMBLYDATASETDESCRIPTOR = _descriptor.Descriptor(
  name='AssemblyDatasetDescriptor',
  full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='assembly_accession', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.assembly_accession', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='assemblyAccession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='paired_assembly_accession', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.paired_assembly_accession', index=1,
      number=17, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='pairedAssemblyAccession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='bioproject_lineages', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.bioproject_lineages', index=2,
      number=14, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='bioprojectLineages', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='display_name', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.display_name', index=3,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='displayName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='org', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.org', index=4,
      number=10, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='org', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='chromosomes', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.chromosomes', index=5,
      number=4, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='chromosomes', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_category', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.assembly_category', index=6,
      number=5, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='assemblyCategory', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='annotation_metadata', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.annotation_metadata', index=7,
      number=11, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='annotationMetadata', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_level', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.assembly_level', index=8,
      number=7, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='assemblyLevel', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='submitter', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.submitter', index=9,
      number=15, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='submitter', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='submission_date', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.submission_date', index=10,
      number=8, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='submissionDate', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='contig_n50', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.contig_n50', index=11,
      number=9, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='contigN50', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='estimated_size', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.estimated_size', index=12,
      number=12, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='estimatedSize', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='seq_length', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.seq_length', index=13,
      number=13, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='seqLength', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='biosample_accession', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.biosample_accession', index=14,
      number=16, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='biosampleAccession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='biosample', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.biosample', index=15,
      number=20, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='biosample', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='blast_url', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.blast_url', index=16,
      number=18, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='blastUrl', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='gc_count', full_name='ncbi.datasets.v1.AssemblyDatasetDescriptor.gc_count', index=17,
      number=19, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='gcCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_ASSEMBLYDATASETDESCRIPTOR_CHROMOSOME, ],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=4169,
  serialized_end=5219,
)


_BUSCOSTAT = _descriptor.Descriptor(
  name='BuscoStat',
  full_name='ncbi.datasets.v1.BuscoStat',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='busco_lineage', full_name='ncbi.datasets.v1.BuscoStat.busco_lineage', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='buscoLineage', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='busco_ver', full_name='ncbi.datasets.v1.BuscoStat.busco_ver', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='buscoVer', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='complete', full_name='ncbi.datasets.v1.BuscoStat.complete', index=2,
      number=3, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='complete', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='single_copy', full_name='ncbi.datasets.v1.BuscoStat.single_copy', index=3,
      number=4, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='singleCopy', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='duplicated', full_name='ncbi.datasets.v1.BuscoStat.duplicated', index=4,
      number=5, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='duplicated', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='fragmented', full_name='ncbi.datasets.v1.BuscoStat.fragmented', index=5,
      number=6, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='fragmented', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='missing', full_name='ncbi.datasets.v1.BuscoStat.missing', index=6,
      number=7, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='missing', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='total_count', full_name='ncbi.datasets.v1.BuscoStat.total_count', index=7,
      number=8, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='totalCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=5222,
  serialized_end=5483,
)


_ANNOTATIONFORASSEMBLY_FILE = _descriptor.Descriptor(
  name='File',
  full_name='ncbi.datasets.v1.AnnotationForAssembly.File',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='type', full_name='ncbi.datasets.v1.AnnotationForAssembly.File.type', index=0,
      number=1, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='type', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='estimated_size', full_name='ncbi.datasets.v1.AnnotationForAssembly.File.estimated_size', index=1,
      number=2, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='estimatedSize', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=5832,
  serialized_end=5943,
)

_ANNOTATIONFORASSEMBLY = _descriptor.Descriptor(
  name='AnnotationForAssembly',
  full_name='ncbi.datasets.v1.AnnotationForAssembly',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='name', full_name='ncbi.datasets.v1.AnnotationForAssembly.name', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='name', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='source', full_name='ncbi.datasets.v1.AnnotationForAssembly.source', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='source', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='release_number', full_name='ncbi.datasets.v1.AnnotationForAssembly.release_number', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='releaseNumber', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='release_date', full_name='ncbi.datasets.v1.AnnotationForAssembly.release_date', index=3,
      number=4, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='releaseDate', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='report_url', full_name='ncbi.datasets.v1.AnnotationForAssembly.report_url', index=4,
      number=5, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='reportUrl', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='file', full_name='ncbi.datasets.v1.AnnotationForAssembly.file', index=5,
      number=6, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='file', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='stats', full_name='ncbi.datasets.v1.AnnotationForAssembly.stats', index=6,
      number=7, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='stats', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='busco', full_name='ncbi.datasets.v1.AnnotationForAssembly.busco', index=7,
      number=8, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='busco', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_ANNOTATIONFORASSEMBLY_FILE, ],
  enum_types=[
    _ANNOTATIONFORASSEMBLY_TYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=5486,
  serialized_end=6117,
)


_FEATURECOUNTS = _descriptor.Descriptor(
  name='FeatureCounts',
  full_name='ncbi.datasets.v1.FeatureCounts',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='gene_counts', full_name='ncbi.datasets.v1.FeatureCounts.gene_counts', index=0,
      number=1, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='geneCounts', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=6119,
  serialized_end=6197,
)


_GENECOUNTS = _descriptor.Descriptor(
  name='GeneCounts',
  full_name='ncbi.datasets.v1.GeneCounts',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='total', full_name='ncbi.datasets.v1.GeneCounts.total', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='total', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='protein_coding', full_name='ncbi.datasets.v1.GeneCounts.protein_coding', index=1,
      number=2, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='proteinCoding', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=6199,
  serialized_end=6272,
)


_ASSEMBLYDATASETREQUEST = _descriptor.Descriptor(
  name='AssemblyDatasetRequest',
  full_name='ncbi.datasets.v1.AssemblyDatasetRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='assembly_accessions', full_name='ncbi.datasets.v1.AssemblyDatasetRequest.assembly_accessions', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='assemblyAccessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v1.AssemblyDatasetRequest.accessions', index=1,
      number=10, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031R\0225\n\020GCF_000001405.40\n\020GCF_000001635.27\022\017Human and Mouse\022\031\n\020GCF_000001405.40\022\005Human', json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='chromosomes', full_name='ncbi.datasets.v1.AssemblyDatasetRequest.chromosomes', index=2,
      number=2, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031M\022,\n\0011\n\0012\n\0013\n\001X\n\001Y\n\002MT\022\027Selected chromosomes #1\022\035\n\001X\n\001Y\022\025Human sex chromosomes', json_name='chromosomes', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='include_annotation', full_name='ncbi.datasets.v1.AssemblyDatasetRequest.include_annotation', index=3,
      number=3, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001\332\360\031\002\010\002', json_name='includeAnnotation', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='exclude_sequence', full_name='ncbi.datasets.v1.AssemblyDatasetRequest.exclude_sequence', index=4,
      number=7, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\332\360\031\002\010\002\302\360\031L\n\005false\022 \n\004true\022\030Exclude genomic sequence\022!\n\005false\022\030Include genomic sequence', json_name='excludeSequence', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='include_annotation_type', full_name='ncbi.datasets.v1.AssemblyDatasetRequest.include_annotation_type', index=5,
      number=8, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\332\360\031\002\010\002\302\360\031\206\001\0226\n\nPROT_FASTA\n\tRNA_FASTA\022\035Transcript & Protein Sequence\022L\n\nGENOME_GFF\n\013GENOME_GBFF\n\nGENOME_GTF\n\nPROT_FASTA\n\tRNA_FASTA\n\tCDS_FASTA\022\003All', json_name='includeAnnotationType', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='hydrated', full_name='ncbi.datasets.v1.AssemblyDatasetRequest.hydrated', index=6,
      number=9, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='hydrated', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='include_tsv', full_name='ncbi.datasets.v1.AssemblyDatasetRequest.include_tsv', index=7,
      number=11, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\332\360\031\002\010\002\302\360\031\007\n\005false', json_name='includeTsv', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _ASSEMBLYDATASETREQUEST_RESOLUTION,
  ],
  serialized_options=b'\302\360\031 \n\036accessions:\n- GCF_000001405.40',
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=6275,
  serialized_end=7256,
)


_ASSEMBLYDATASETAVAILABILITY = _descriptor.Descriptor(
  name='AssemblyDatasetAvailability',
  full_name='ncbi.datasets.v1.AssemblyDatasetAvailability',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='valid_assemblies', full_name='ncbi.datasets.v1.AssemblyDatasetAvailability.valid_assemblies', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='validAssemblies', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='invalid_assemblies', full_name='ncbi.datasets.v1.AssemblyDatasetAvailability.invalid_assemblies', index=1,
      number=2, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='invalidAssemblies', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='reason', full_name='ncbi.datasets.v1.AssemblyDatasetAvailability.reason', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='reason', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=7259,
  serialized_end=7402,
)


_TAXIDREQUEST = _descriptor.Descriptor(
  name='TaxidRequest',
  full_name='ncbi.datasets.v1.TaxidRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='org_name', full_name='ncbi.datasets.v1.TaxidRequest.org_name', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='orgName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=7404,
  serialized_end=7445,
)


_TAXIDREPLY = _descriptor.Descriptor(
  name='TaxidReply',
  full_name='ncbi.datasets.v1.TaxidReply',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='tax_id', full_name='ncbi.datasets.v1.TaxidReply.tax_id', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='taxId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=7447,
  serialized_end=7482,
)


_ORGANISM_COUNTS = _descriptor.Descriptor(
  name='Counts',
  full_name='ncbi.datasets.v1.Organism.Counts',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='node', full_name='ncbi.datasets.v1.Organism.Counts.node', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='node', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='subtree', full_name='ncbi.datasets.v1.Organism.Counts.subtree', index=1,
      number=2, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='subtree', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=8344,
  serialized_end=8398,
)

_ORGANISM_COUNTBYTYPE = _descriptor.Descriptor(
  name='CountByType',
  full_name='ncbi.datasets.v1.Organism.CountByType',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='type', full_name='ncbi.datasets.v1.Organism.CountByType.type', index=0,
      number=1, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='type', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='counts', full_name='ncbi.datasets.v1.Organism.CountByType.counts', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='counts', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=8400,
  serialized_end=8521,
)

_ORGANISM = _descriptor.Descriptor(
  name='Organism',
  full_name='ncbi.datasets.v1.Organism',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='tax_id', full_name='ncbi.datasets.v1.Organism.tax_id', index=0,
      number=1, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='taxId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='sci_name', full_name='ncbi.datasets.v1.Organism.sci_name', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='sciName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='common_name', full_name='ncbi.datasets.v1.Organism.common_name', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='commonName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='blast_node', full_name='ncbi.datasets.v1.Organism.blast_node', index=3,
      number=16, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='blastNode', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='breed', full_name='ncbi.datasets.v1.Organism.breed', index=4,
      number=10, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='breed', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='cultivar', full_name='ncbi.datasets.v1.Organism.cultivar', index=5,
      number=11, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='cultivar', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='ecotype', full_name='ncbi.datasets.v1.Organism.ecotype', index=6,
      number=12, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='ecotype', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='isolate', full_name='ncbi.datasets.v1.Organism.isolate', index=7,
      number=13, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='isolate', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='sex', full_name='ncbi.datasets.v1.Organism.sex', index=8,
      number=14, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='sex', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='strain', full_name='ncbi.datasets.v1.Organism.strain', index=9,
      number=15, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='strain', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='search_text', full_name='ncbi.datasets.v1.Organism.search_text', index=10,
      number=17, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='searchText', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='rank', full_name='ncbi.datasets.v1.Organism.rank', index=11,
      number=5, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='rank', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='parent_tax_id', full_name='ncbi.datasets.v1.Organism.parent_tax_id', index=12,
      number=6, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='parentTaxId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_count', full_name='ncbi.datasets.v1.Organism.assembly_count', index=13,
      number=7, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='assemblyCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_counts', full_name='ncbi.datasets.v1.Organism.assembly_counts', index=14,
      number=19, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='assemblyCounts', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='counts', full_name='ncbi.datasets.v1.Organism.counts', index=15,
      number=22, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='counts', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='children', full_name='ncbi.datasets.v1.Organism.children', index=16,
      number=8, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='children', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='merged', full_name='ncbi.datasets.v1.Organism.merged', index=17,
      number=9, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='merged', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='merged_tax_ids', full_name='ncbi.datasets.v1.Organism.merged_tax_ids', index=18,
      number=18, type=4, cpp_type=4, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='mergedTaxIds', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='min_ord', full_name='ncbi.datasets.v1.Organism.min_ord', index=19,
      number=20, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='minOrd', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='max_ord', full_name='ncbi.datasets.v1.Organism.max_ord', index=20,
      number=21, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='maxOrd', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='weight', full_name='ncbi.datasets.v1.Organism.weight', index=21,
      number=23, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='weight', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='key', full_name='ncbi.datasets.v1.Organism.key', index=22,
      number=100, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='key', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='title', full_name='ncbi.datasets.v1.Organism.title', index=23,
      number=101, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='title', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='icon', full_name='ncbi.datasets.v1.Organism.icon', index=24,
      number=102, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='icon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_ORGANISM_COUNTS, _ORGANISM_COUNTBYTYPE, ],
  enum_types=[
    _ORGANISM_RANKTYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=7485,
  serialized_end=9193,
)


_TAXTREEREQUEST = _descriptor.Descriptor(
  name='TaxTreeRequest',
  full_name='ncbi.datasets.v1.TaxTreeRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='tax_id', full_name='ncbi.datasets.v1.TaxTreeRequest.tax_id', index=0,
      number=1, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='taxId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='tax_token', full_name='ncbi.datasets.v1.TaxTreeRequest.tax_token', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='taxToken', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxon', full_name='ncbi.datasets.v1.TaxTreeRequest.taxon', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031Y\022 \n\0049606\022\030NCBI Taxonomy Identifier\022\024\n\005human\022\013Common Name\022\037\n\014Homo sapiens\022\017Scientific Name', json_name='taxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='children_only', full_name='ncbi.datasets.v1.TaxTreeRequest.children_only', index=3,
      number=4, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\007\n\005false', json_name='childrenOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
    _descriptor.OneofDescriptor(
      name='taxonomy', full_name='ncbi.datasets.v1.TaxTreeRequest.taxonomy',
      index=0, containing_type=None,
      create_key=_descriptor._internal_create_key,
    fields=[]),
  ],
  serialized_start=9196,
  serialized_end=9457,
)


_ORGANISMQUERYREQUEST = _descriptor.Descriptor(
  name='OrganismQueryRequest',
  full_name='ncbi.datasets.v1.OrganismQueryRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='organism_query', full_name='ncbi.datasets.v1.OrganismQueryRequest.organism_query', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='organismQuery', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxon_query', full_name='ncbi.datasets.v1.OrganismQueryRequest.taxon_query', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\014\022\n\n\003hum\022\003hum', json_name='taxonQuery', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='tax_rank_filter', full_name='ncbi.datasets.v1.OrganismQueryRequest.tax_rank_filter', index=2,
      number=3, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\t\n\007species', json_name='taxRankFilter', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='tax_resource_filter', full_name='ncbi.datasets.v1.OrganismQueryRequest.tax_resource_filter', index=3,
      number=4, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\005\n\003all', json_name='taxResourceFilter', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _ORGANISMQUERYREQUEST_TAXRANKFILTER,
    _ORGANISMQUERYREQUEST_TAXONRESOURCEFILTER,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
    _descriptor.OneofDescriptor(
      name='id', full_name='ncbi.datasets.v1.OrganismQueryRequest.id',
      index=0, containing_type=None,
      create_key=_descriptor._internal_create_key,
    fields=[]),
  ],
  serialized_start=9460,
  serialized_end=9916,
)


_SCINAMEANDIDS_SCINAMEANDID = _descriptor.Descriptor(
  name='SciNameAndId',
  full_name='ncbi.datasets.v1.SciNameAndIds.SciNameAndId',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='sci_name', full_name='ncbi.datasets.v1.SciNameAndIds.SciNameAndId.sci_name', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='sciName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='tax_id', full_name='ncbi.datasets.v1.SciNameAndIds.SciNameAndId.tax_id', index=1,
      number=2, type=4, cpp_type=4, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='taxId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='common_name', full_name='ncbi.datasets.v1.SciNameAndIds.SciNameAndId.common_name', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='commonName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='matched_term', full_name='ncbi.datasets.v1.SciNameAndIds.SciNameAndId.matched_term', index=3,
      number=4, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='matchedTerm', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=10024,
  serialized_end=10156,
)

_SCINAMEANDIDS = _descriptor.Descriptor(
  name='SciNameAndIds',
  full_name='ncbi.datasets.v1.SciNameAndIds',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='sci_name_and_ids', full_name='ncbi.datasets.v1.SciNameAndIds.sci_name_and_ids', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='sciNameAndIds', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_SCINAMEANDIDS_SCINAMEANDID, ],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=9919,
  serialized_end=10156,
)


_ORTHOLOGREQUEST = _descriptor.Descriptor(
  name='OrthologRequest',
  full_name='ncbi.datasets.v1.OrthologRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='gene_id', full_name='ncbi.datasets.v1.OrthologRequest.gene_id', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\030\022\014\n\0042778\022\004GNAS\022\010\n\0012\022\003A2M', json_name='geneId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='returned_content', full_name='ncbi.datasets.v1.OrthologRequest.returned_content', index=1,
      number=2, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='returnedContent', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxon_filter', full_name='ncbi.datasets.v1.OrthologRequest.taxon_filter', index=2,
      number=3, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031{\022\'\n\0049606\n\00510090\022\030NCBI Taxonomy Identifier\022!\n\005human\n\013house mouse\022\013Common Name\022-\n\014homo sapiens\n\014mus musculus\022\017Scientific Name', json_name='taxonFilter', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _ORTHOLOGREQUEST_CONTENTTYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=10159,
  serialized_end=10529,
)


_ORTHOLOGSET = _descriptor.Descriptor(
  name='OrthologSet',
  full_name='ncbi.datasets.v1.OrthologSet',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='ortholog_set_id', full_name='ncbi.datasets.v1.OrthologSet.ortholog_set_id', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='orthologSetId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='method', full_name='ncbi.datasets.v1.OrthologSet.method', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='method', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='genes', full_name='ncbi.datasets.v1.OrthologSet.genes', index=2,
      number=3, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='genes', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=10532,
  serialized_end=10663,
)


_GENEDATASETREQUEST_SYMBOLSFORTAXON = _descriptor.Descriptor(
  name='SymbolsForTaxon',
  full_name='ncbi.datasets.v1.GeneDatasetRequest.SymbolsForTaxon',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='symbols', full_name='ncbi.datasets.v1.GeneDatasetRequest.SymbolsForTaxon.symbols', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\016\022\014\n\004GNAS\022\004GNAS', json_name='symbols', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxon', full_name='ncbi.datasets.v1.GeneDatasetRequest.SymbolsForTaxon.taxon', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031Y\022 \n\0049606\022\030NCBI Taxonomy Identifier\022\024\n\005human\022\013Common Name\022\037\n\014Homo sapiens\022\017Scientific Name', json_name='taxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=11444,
  serialized_end=11624,
)

_GENEDATASETREQUEST_GENEIDS = _descriptor.Descriptor(
  name='GeneIds',
  full_name='ncbi.datasets.v1.GeneDatasetRequest.GeneIds',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='gene_ids', full_name='ncbi.datasets.v1.GeneDatasetRequest.GeneIds.gene_ids', index=0,
      number=1, type=13, cpp_type=3, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='geneIds', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=11626,
  serialized_end=11662,
)

_GENEDATASETREQUEST_SORT = _descriptor.Descriptor(
  name='Sort',
  full_name='ncbi.datasets.v1.GeneDatasetRequest.Sort',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='field', full_name='ncbi.datasets.v1.GeneDatasetRequest.Sort.field', index=0,
      number=1, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='field', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='direction', full_name='ncbi.datasets.v1.GeneDatasetRequest.Sort.direction', index=1,
      number=2, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='direction', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=11665,
  serialized_end=11804,
)

_GENEDATASETREQUEST = _descriptor.Descriptor(
  name='GeneDatasetRequest',
  full_name='ncbi.datasets.v1.GeneDatasetRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='gene_ids', full_name='ncbi.datasets.v1.GeneDatasetRequest.gene_ids', index=0,
      number=1, type=13, cpp_type=3, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031<\022\024\n\00559067\022\013IL21 GeneID\022$\n\00559067\n\00550615\022\024IL21 & IL21R GeneIDs', json_name='geneIds', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v1.GeneDatasetRequest.accessions', index=1,
      number=10, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031]\022.\n\013NM_021803.4\n\013NM_181078.3\022\022RefSeq transcripts\022+\n\013NP_068575.1\n\013NP_851564.1\022\017RefSeq proteins', json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='symbols_for_taxon', full_name='ncbi.datasets.v1.GeneDatasetRequest.symbols_for_taxon', index=2,
      number=9, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='symbolsForTaxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxon', full_name='ncbi.datasets.v1.GeneDatasetRequest.taxon', index=3,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031Y\022 \n\0049606\022\030NCBI Taxonomy Identifier\022\024\n\005human\022\013Common Name\022\037\n\014Homo sapiens\022\017Scientific Name', json_name='taxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='include_annotation_type', full_name='ncbi.datasets.v1.GeneDatasetRequest.include_annotation_type', index=4,
      number=6, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\332\360\031\002\010\002', json_name='includeAnnotationType', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='returned_content', full_name='ncbi.datasets.v1.GeneDatasetRequest.returned_content', index=5,
      number=7, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='returnedContent', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='sort_schema', full_name='ncbi.datasets.v1.GeneDatasetRequest.sort_schema', index=6,
      number=8, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='sortSchema', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='limit', full_name='ncbi.datasets.v1.GeneDatasetRequest.limit', index=7,
      number=11, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='limit', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='fasta_filter', full_name='ncbi.datasets.v1.GeneDatasetRequest.fasta_filter', index=8,
      number=12, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='fastaFilter', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_GENEDATASETREQUEST_SYMBOLSFORTAXON, _GENEDATASETREQUEST_GENEIDS, _GENEDATASETREQUEST_SORT, ],
  enum_types=[
    _GENEDATASETREQUEST_CONTENTTYPE,
    _GENEDATASETREQUEST_SORTFIELD,
  ],
  serialized_options=b'\302\360\031\033\n\031gene_ids:\n- 59067\n- 50615',
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=10666,
  serialized_end=11999,
)


_ERROR = _descriptor.Descriptor(
  name='Error',
  full_name='ncbi.datasets.v1.Error',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='assembly_error_code', full_name='ncbi.datasets.v1.Error.assembly_error_code', index=0,
      number=6, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='assemblyErrorCode', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='gene_error_code', full_name='ncbi.datasets.v1.Error.gene_error_code', index=1,
      number=7, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='geneErrorCode', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='virus_error_code', full_name='ncbi.datasets.v1.Error.virus_error_code', index=2,
      number=8, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='virusErrorCode', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='reason', full_name='ncbi.datasets.v1.Error.reason', index=3,
      number=5, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='reason', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='message', full_name='ncbi.datasets.v1.Error.message', index=4,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='message', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='valid_identifiers', full_name='ncbi.datasets.v1.Error.valid_identifiers', index=5,
      number=3, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='validIdentifiers', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='invalid_identifiers', full_name='ncbi.datasets.v1.Error.invalid_identifiers', index=6,
      number=4, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='invalidIdentifiers', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _ERROR_ASSEMBLYERRORCODE,
    _ERROR_GENEERRORCODE,
    _ERROR_VIRUSERRORCODE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
    _descriptor.OneofDescriptor(
      name='code', full_name='ncbi.datasets.v1.Error.code',
      index=0, containing_type=None,
      create_key=_descriptor._internal_create_key,
    fields=[]),
  ],
  serialized_start=12002,
  serialized_end=12710,
)


_WARNING_REPLACEDID = _descriptor.Descriptor(
  name='ReplacedId',
  full_name='ncbi.datasets.v1.Warning.ReplacedId',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='requested', full_name='ncbi.datasets.v1.Warning.ReplacedId.requested', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='requested', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='returned', full_name='ncbi.datasets.v1.Warning.ReplacedId.returned', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='returned', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=12991,
  serialized_end=13061,
)

_WARNING = _descriptor.Descriptor(
  name='Warning',
  full_name='ncbi.datasets.v1.Warning',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='gene_warning_code', full_name='ncbi.datasets.v1.Warning.gene_warning_code', index=0,
      number=7, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='geneWarningCode', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='reason', full_name='ncbi.datasets.v1.Warning.reason', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='reason', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='message', full_name='ncbi.datasets.v1.Warning.message', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='message', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='replaced_id', full_name='ncbi.datasets.v1.Warning.replaced_id', index=3,
      number=4, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='replacedId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='unrecognized_identifier', full_name='ncbi.datasets.v1.Warning.unrecognized_identifier', index=4,
      number=5, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='unrecognizedIdentifier', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_WARNING_REPLACEDID, ],
  enum_types=[
    _WARNING_GENEWARNINGCODE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
    _descriptor.OneofDescriptor(
      name='code', full_name='ncbi.datasets.v1.Warning.code',
      index=0, containing_type=None,
      create_key=_descriptor._internal_create_key,
    fields=[]),
  ],
  serialized_start=12713,
  serialized_end=13312,
)


_MESSAGE = _descriptor.Descriptor(
  name='Message',
  full_name='ncbi.datasets.v1.Message',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='error', full_name='ncbi.datasets.v1.Message.error', index=0,
      number=1, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='error', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='warning', full_name='ncbi.datasets.v1.Message.warning', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='warning', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
    _descriptor.OneofDescriptor(
      name='message', full_name='ncbi.datasets.v1.Message.message',
      index=0, containing_type=None,
      create_key=_descriptor._internal_create_key,
    fields=[]),
  ],
  serialized_start=13314,
  serialized_end=13438,
)


_GENEMATCH = _descriptor.Descriptor(
  name='GeneMatch',
  full_name='ncbi.datasets.v1.GeneMatch',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='messages', full_name='ncbi.datasets.v1.GeneMatch.messages', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='messages', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='warnings', full_name='ncbi.datasets.v1.GeneMatch.warnings', index=1,
      number=3, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='warnings', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='query', full_name='ncbi.datasets.v1.GeneMatch.query', index=2,
      number=4, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='query', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='gene', full_name='ncbi.datasets.v1.GeneMatch.gene', index=3,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='gene', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='errors', full_name='ncbi.datasets.v1.GeneMatch.errors', index=4,
      number=5, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='errors', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=13441,
  serialized_end=13691,
)


_GENEMETADATA = _descriptor.Descriptor(
  name='GeneMetadata',
  full_name='ncbi.datasets.v1.GeneMetadata',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='messages', full_name='ncbi.datasets.v1.GeneMetadata.messages', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='messages', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='genes', full_name='ncbi.datasets.v1.GeneMetadata.genes', index=1,
      number=2, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='genes', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='total_count', full_name='ncbi.datasets.v1.GeneMetadata.total_count', index=2,
      number=3, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='totalCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=13694,
  serialized_end=13847,
)


_ASSEMBLYMATCH = _descriptor.Descriptor(
  name='AssemblyMatch',
  full_name='ncbi.datasets.v1.AssemblyMatch',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='messages', full_name='ncbi.datasets.v1.AssemblyMatch.messages', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='messages', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly', full_name='ncbi.datasets.v1.AssemblyMatch.assembly', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='assembly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=13850,
  serialized_end=13993,
)


_ASSEMBLYMETADATA = _descriptor.Descriptor(
  name='AssemblyMetadata',
  full_name='ncbi.datasets.v1.AssemblyMetadata',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='messages', full_name='ncbi.datasets.v1.AssemblyMetadata.messages', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='messages', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assemblies', full_name='ncbi.datasets.v1.AssemblyMetadata.assemblies', index=1,
      number=2, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='assemblies', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='total_count', full_name='ncbi.datasets.v1.AssemblyMetadata.total_count', index=2,
      number=3, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='totalCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='next_page_token', full_name='ncbi.datasets.v1.AssemblyMetadata.next_page_token', index=3,
      number=4, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='nextPageToken', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=13996,
  serialized_end=14207,
)


_GENEDESCRIPTORS = _descriptor.Descriptor(
  name='GeneDescriptors',
  full_name='ncbi.datasets.v1.GeneDescriptors',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='genes', full_name='ncbi.datasets.v1.GeneDescriptors.genes', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='genes', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='errors', full_name='ncbi.datasets.v1.GeneDescriptors.errors', index=1,
      number=2, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='errors', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=14209,
  serialized_end=14331,
)


_ANNOTATIONFORVIRUS = _descriptor.Descriptor(
  name='AnnotationForVirus',
  full_name='ncbi.datasets.v1.AnnotationForVirus',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _ANNOTATIONFORVIRUS_TYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=14334,
  serialized_end=14470,
)


_VIRUSDATASETREQUEST = _descriptor.Descriptor(
  name='VirusDatasetRequest',
  full_name='ncbi.datasets.v1.VirusDatasetRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='tax_id', full_name='ncbi.datasets.v1.VirusDatasetRequest.tax_id', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='taxId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='tax_name', full_name='ncbi.datasets.v1.VirusDatasetRequest.tax_name', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='taxName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='accession', full_name='ncbi.datasets.v1.VirusDatasetRequest.accession', index=2,
      number=18, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='accession', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v1.VirusDatasetRequest.accessions', index=3,
      number=16, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031?\022=\n\013NC_045512.2\022.SARS-COV-2 isolate Wuhan-Hu-1, complete genome', json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxon', full_name='ncbi.datasets.v1.VirusDatasetRequest.taxon', index=4,
      number=19, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031@\022\025\n\0072697049\022\nSARS-COV-2\022\026\n\00511118\022\rCoronaviridae\022\017\n\0071335626\022\004MERS', json_name='taxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='refseq_only', full_name='ncbi.datasets.v1.VirusDatasetRequest.refseq_only', index=5,
      number=4, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031P\n\005false\022\037\n\004true\022\027Limit to RefSeq genomes\022&\n\005false\022\035Both GenBank & RefSeq genomes', json_name='refseqOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='annotated_only', full_name='ncbi.datasets.v1.VirusDatasetRequest.annotated_only', index=6,
      number=5, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031A\n\005false\022\"\n\004true\022\032Limit to annotated genomes\022\024\n\005false\022\013All genomes', json_name='annotatedOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='released_since', full_name='ncbi.datasets.v1.VirusDatasetRequest.released_since', index=7,
      number=6, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\036\022\034\n\n2020-08-01\022\016August 1, 2020', json_name='releasedSince', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='updated_since', full_name='ncbi.datasets.v1.VirusDatasetRequest.updated_since', index=8,
      number=17, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\035\022\033\n\n2021-07-18\022\rJuly 18, 2021', json_name='updatedSince', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='host', full_name='ncbi.datasets.v1.VirusDatasetRequest.host', index=9,
      number=7, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\020\022\016\n\005human\022\005human', json_name='host', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='pangolin_classification', full_name='ncbi.datasets.v1.VirusDatasetRequest.pangolin_classification', index=10,
      number=20, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='pangolinClassification', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='geo_location', full_name='ncbi.datasets.v1.VirusDatasetRequest.geo_location', index=11,
      number=8, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\031\022\n\n\003USA\022\003USA\022\013\n\002TX\022\005Texas', json_name='geoLocation', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='complete_only', full_name='ncbi.datasets.v1.VirusDatasetRequest.complete_only', index=12,
      number=9, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031@\n\005false\022!\n\004true\022\031Limit to complete genomes\022\024\n\005false\022\013All genomes', json_name='completeOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_fields', full_name='ncbi.datasets.v1.VirusDatasetRequest.table_fields', index=13,
      number=10, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031N\022L\n\024nucleotide_accession\n\021nucleotide_length\n\020nuc_completeness\022\017Assembly Fields', json_name='tableFields', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='exclude_sequence', full_name='ncbi.datasets.v1.VirusDatasetRequest.exclude_sequence', index=14,
      number=12, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\332\360\031\002\010\002\302\360\031L\n\005false\022 \n\004true\022\030Exclude genomic sequence\022!\n\005false\022\030Include genomic sequence', json_name='excludeSequence', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='include_annotation_type', full_name='ncbi.datasets.v1.VirusDatasetRequest.include_annotation_type', index=15,
      number=13, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\332\360\031\002\010\002\302\360\031K\022 \n\tCDS_FASTA\022\023Transcript sequence\022\'\n\tCDS_FASTA\n\nPROT_FASTA\022\016All annotation', json_name='includeAnnotationType', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='format', full_name='ncbi.datasets.v1.VirusDatasetRequest.format', index=16,
      number=14, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\0312\n\003tsv\022\n\n\003tsv\022\003TSV\022\n\n\003csv\022\003CSV\022\023\n\005jsonl\022\nJSON Lines', json_name='format', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=14473,
  serialized_end=16064,
)


_VIRUSDATASETFILTER = _descriptor.Descriptor(
  name='VirusDatasetFilter',
  full_name='ncbi.datasets.v1.VirusDatasetFilter',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v1.VirusDatasetFilter.accessions', index=0,
      number=2, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\206\001\022\203\001\n\027NC_045512.2,NC_038294.1\022hSARS-COV-2 isolate Wuhan-Hu-1 and Middle East respiratory syndrome-related coronavirus, complete genomes', json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxon', full_name='ncbi.datasets.v1.VirusDatasetFilter.taxon', index=1,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031@\022\025\n\0072697049\022\nSARS-COV-2\022\026\n\00511118\022\rCoronaviridae\022\017\n\0071335626\022\004MERS', json_name='taxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='refseq_only', full_name='ncbi.datasets.v1.VirusDatasetFilter.refseq_only', index=2,
      number=4, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031P\n\005false\022\037\n\004true\022\027Limit to RefSeq genomes\022&\n\005false\022\035Both GenBank & RefSeq genomes', json_name='refseqOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='annotated_only', full_name='ncbi.datasets.v1.VirusDatasetFilter.annotated_only', index=3,
      number=5, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031A\n\005false\022\"\n\004true\022\032Limit to annotated genomes\022\024\n\005false\022\013All genomes', json_name='annotatedOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='released_since', full_name='ncbi.datasets.v1.VirusDatasetFilter.released_since', index=4,
      number=6, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\036\022\034\n\n2020-08-01\022\016August 1, 2020', json_name='releasedSince', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='updated_since', full_name='ncbi.datasets.v1.VirusDatasetFilter.updated_since', index=5,
      number=11, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\035\022\033\n\n2021-07-18\022\rJuly 18, 2021', json_name='updatedSince', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='host', full_name='ncbi.datasets.v1.VirusDatasetFilter.host', index=6,
      number=7, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\020\022\016\n\005human\022\005human', json_name='host', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='pangolin_classification', full_name='ncbi.datasets.v1.VirusDatasetFilter.pangolin_classification', index=7,
      number=8, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='pangolinClassification', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='geo_location', full_name='ncbi.datasets.v1.VirusDatasetFilter.geo_location', index=8,
      number=9, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\031\022\n\n\003USA\022\003USA\022\013\n\002TX\022\005Texas', json_name='geoLocation', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='complete_only', full_name='ncbi.datasets.v1.VirusDatasetFilter.complete_only', index=9,
      number=10, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031@\n\005false\022!\n\004true\022\031Limit to complete genomes\022\024\n\005false\022\013All genomes', json_name='completeOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=16067,
  serialized_end=17058,
)


_VIRUSDATAREPORTREQUEST = _descriptor.Descriptor(
  name='VirusDataReportRequest',
  full_name='ncbi.datasets.v1.VirusDataReportRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='filter', full_name='ncbi.datasets.v1.VirusDataReportRequest.filter', index=0,
      number=1, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='filter', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='returned_content', full_name='ncbi.datasets.v1.VirusDataReportRequest.returned_content', index=1,
      number=2, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='returnedContent', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_fields', full_name='ncbi.datasets.v1.VirusDataReportRequest.table_fields', index=2,
      number=3, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031=\022;\n\taccession\n\013is-complete\n\014is-annotated\022\023Virus Genome Fields', json_name='tableFields', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_format', full_name='ncbi.datasets.v1.VirusDataReportRequest.table_format', index=3,
      number=6, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='tableFormat', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_size', full_name='ncbi.datasets.v1.VirusDataReportRequest.page_size', index=4,
      number=4, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\004\n\00220', json_name='pageSize', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_token', full_name='ncbi.datasets.v1.VirusDataReportRequest.page_token', index=5,
      number=5, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='pageToken', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _VIRUSDATAREPORTREQUEST_CONTENTTYPE,
  ],
  serialized_options=b'\302\360\031\"\n taxon: 2697049\nrefseq_only: true',
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=17061,
  serialized_end=17541,
)


_SARS2PROTEINDATASETREQUEST = _descriptor.Descriptor(
  name='Sars2ProteinDatasetRequest',
  full_name='ncbi.datasets.v1.Sars2ProteinDatasetRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='proteins', full_name='ncbi.datasets.v1.Sars2ProteinDatasetRequest.proteins', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031^\022\036\n\rspike protein\022\rSpike Protein\022<\n\rspike protein\n\020envelope protein\n\004RdRp\022\023SARS-CoV-2 proteins', json_name='proteins', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='refseq_only', full_name='ncbi.datasets.v1.Sars2ProteinDatasetRequest.refseq_only', index=1,
      number=4, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031P\n\005false\022\037\n\004true\022\027Limit to RefSeq genomes\022&\n\005false\022\035Both GenBank & RefSeq genomes', json_name='refseqOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='annotated_only', full_name='ncbi.datasets.v1.Sars2ProteinDatasetRequest.annotated_only', index=2,
      number=5, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031A\n\005false\022\"\n\004true\022\032Limit to annotated genomes\022\024\n\005false\022\013All genomes', json_name='annotatedOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='released_since', full_name='ncbi.datasets.v1.Sars2ProteinDatasetRequest.released_since', index=3,
      number=6, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\036\022\034\n\n2020-08-01\022\016August 1, 2020', json_name='releasedSince', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='updated_since', full_name='ncbi.datasets.v1.Sars2ProteinDatasetRequest.updated_since', index=4,
      number=15, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\035\022\033\n\n2021-07-18\022\rJuly 18, 2021', json_name='updatedSince', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='host', full_name='ncbi.datasets.v1.Sars2ProteinDatasetRequest.host', index=5,
      number=7, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\020\022\016\n\005human\022\005human', json_name='host', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='geo_location', full_name='ncbi.datasets.v1.Sars2ProteinDatasetRequest.geo_location', index=6,
      number=8, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\031\022\n\n\003USA\022\003USA\022\013\n\002TX\022\005Texas', json_name='geoLocation', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='complete_only', full_name='ncbi.datasets.v1.Sars2ProteinDatasetRequest.complete_only', index=7,
      number=9, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031@\n\005false\022!\n\004true\022\031Limit to complete genomes\022\024\n\005false\022\013All genomes', json_name='completeOnly', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='table_fields', full_name='ncbi.datasets.v1.Sars2ProteinDatasetRequest.table_fields', index=8,
      number=10, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031N\022L\n\024nucleotide_accession\n\021nucleotide_length\n\020nuc_completeness\022\017Assembly Fields', json_name='tableFields', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='exclude_sequence', full_name='ncbi.datasets.v1.Sars2ProteinDatasetRequest.exclude_sequence', index=9,
      number=12, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\332\360\031\002\010\002\302\360\031L\n\005false\022 \n\004true\022\030Exclude genomic sequence\022!\n\005false\022\030Include genomic sequence', json_name='excludeSequence', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='include_annotation_type', full_name='ncbi.datasets.v1.Sars2ProteinDatasetRequest.include_annotation_type', index=10,
      number=13, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\332\360\031\002\010\002\302\360\031K\022 \n\tCDS_FASTA\022\023Transcript sequence\022\'\n\tCDS_FASTA\n\nPROT_FASTA\022\016All annotation', json_name='includeAnnotationType', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='format', full_name='ncbi.datasets.v1.Sars2ProteinDatasetRequest.format', index=11,
      number=14, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\0312\n\003tsv\022\n\n\003tsv\022\003TSV\022\n\n\003csv\022\003CSV\022\023\n\005jsonl\022\nJSON Lines', json_name='format', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=17544,
  serialized_end=18932,
)


_VIRUSAVAILABILITYREQUEST = _descriptor.Descriptor(
  name='VirusAvailabilityRequest',
  full_name='ncbi.datasets.v1.VirusAvailabilityRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v1.VirusAvailabilityRequest.accessions', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031\206\001\022\203\001\n\027NC_045512.2,NC_038294.1\022hSARS-COV-2 isolate Wuhan-Hu-1 and Middle East respiratory syndrome-related coronavirus, complete genomes', json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=18935,
  serialized_end=19136,
)


_VIRUSAVAILABILITY = _descriptor.Descriptor(
  name='VirusAvailability',
  full_name='ncbi.datasets.v1.VirusAvailability',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='valid_accessions', full_name='ncbi.datasets.v1.VirusAvailability.valid_accessions', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='validAccessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='invalid_accessions', full_name='ncbi.datasets.v1.VirusAvailability.invalid_accessions', index=1,
      number=2, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='invalidAccessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='message', full_name='ncbi.datasets.v1.VirusAvailability.message', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='message', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=19139,
  serialized_end=19274,
)


_DOWNLOADSUMMARY_HYDRATED = _descriptor.Descriptor(
  name='Hydrated',
  full_name='ncbi.datasets.v1.DownloadSummary.Hydrated',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='estimated_file_size_mb', full_name='ncbi.datasets.v1.DownloadSummary.Hydrated.estimated_file_size_mb', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='estimatedFileSizeMb', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='url', full_name='ncbi.datasets.v1.DownloadSummary.Hydrated.url', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='url', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='cli_download_command_line', full_name='ncbi.datasets.v1.DownloadSummary.Hydrated.cli_download_command_line', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='cliDownloadCommandLine', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=19800,
  serialized_end=19940,
)

_DOWNLOADSUMMARY_DEHYDRATED = _descriptor.Descriptor(
  name='Dehydrated',
  full_name='ncbi.datasets.v1.DownloadSummary.Dehydrated',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='estimated_file_size_mb', full_name='ncbi.datasets.v1.DownloadSummary.Dehydrated.estimated_file_size_mb', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='estimatedFileSizeMb', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='url', full_name='ncbi.datasets.v1.DownloadSummary.Dehydrated.url', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='url', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='cli_download_command_line', full_name='ncbi.datasets.v1.DownloadSummary.Dehydrated.cli_download_command_line', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='cliDownloadCommandLine', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='cli_rehydrate_command_line', full_name='ncbi.datasets.v1.DownloadSummary.Dehydrated.cli_rehydrate_command_line', index=3,
      number=4, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='cliRehydrateCommandLine', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=19943,
  serialized_end=20146,
)

_DOWNLOADSUMMARY_FILESUMMARY = _descriptor.Descriptor(
  name='FileSummary',
  full_name='ncbi.datasets.v1.DownloadSummary.FileSummary',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='file_count', full_name='ncbi.datasets.v1.DownloadSummary.FileSummary.file_count', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='fileCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='size_mb', full_name='ncbi.datasets.v1.DownloadSummary.FileSummary.size_mb', index=1,
      number=2, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='sizeMb', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=20148,
  serialized_end=20217,
)

_DOWNLOADSUMMARY_AVAILABLEFILES = _descriptor.Descriptor(
  name='AvailableFiles',
  full_name='ncbi.datasets.v1.DownloadSummary.AvailableFiles',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='all_genomic_fasta', full_name='ncbi.datasets.v1.DownloadSummary.AvailableFiles.all_genomic_fasta', index=0,
      number=1, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='allGenomicFasta', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='genome_gff', full_name='ncbi.datasets.v1.DownloadSummary.AvailableFiles.genome_gff', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='genomeGff', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='genome_gbff', full_name='ncbi.datasets.v1.DownloadSummary.AvailableFiles.genome_gbff', index=2,
      number=3, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='genomeGbff', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='rna_fasta', full_name='ncbi.datasets.v1.DownloadSummary.AvailableFiles.rna_fasta', index=3,
      number=4, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='rnaFasta', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='prot_fasta', full_name='ncbi.datasets.v1.DownloadSummary.AvailableFiles.prot_fasta', index=4,
      number=5, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='protFasta', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='genome_gtf', full_name='ncbi.datasets.v1.DownloadSummary.AvailableFiles.genome_gtf', index=5,
      number=6, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='genomeGtf', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='cds_fasta', full_name='ncbi.datasets.v1.DownloadSummary.AvailableFiles.cds_fasta', index=6,
      number=7, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='cdsFasta', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='sequence_report', full_name='ncbi.datasets.v1.DownloadSummary.AvailableFiles.sequence_report', index=7,
      number=8, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='sequenceReport', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=20220,
  serialized_end=20881,
)

_DOWNLOADSUMMARY = _descriptor.Descriptor(
  name='DownloadSummary',
  full_name='ncbi.datasets.v1.DownloadSummary',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='record_count', full_name='ncbi.datasets.v1.DownloadSummary.record_count', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='recordCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='assembly_count', full_name='ncbi.datasets.v1.DownloadSummary.assembly_count', index=1,
      number=2, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='assemblyCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='resource_updated_on', full_name='ncbi.datasets.v1.DownloadSummary.resource_updated_on', index=2,
      number=4, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='resourceUpdatedOn', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='hydrated', full_name='ncbi.datasets.v1.DownloadSummary.hydrated', index=3,
      number=5, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='hydrated', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='dehydrated', full_name='ncbi.datasets.v1.DownloadSummary.dehydrated', index=4,
      number=6, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='dehydrated', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='errors', full_name='ncbi.datasets.v1.DownloadSummary.errors', index=5,
      number=7, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\030\001', json_name='errors', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='messages', full_name='ncbi.datasets.v1.DownloadSummary.messages', index=6,
      number=8, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='messages', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='available_files', full_name='ncbi.datasets.v1.DownloadSummary.available_files', index=7,
      number=9, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='availableFiles', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_DOWNLOADSUMMARY_HYDRATED, _DOWNLOADSUMMARY_DEHYDRATED, _DOWNLOADSUMMARY_FILESUMMARY, _DOWNLOADSUMMARY_AVAILABLEFILES, ],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=19277,
  serialized_end=20881,
)


_TABULAROUTPUT = _descriptor.Descriptor(
  name='TabularOutput',
  full_name='ncbi.datasets.v1.TabularOutput',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='data', full_name='ncbi.datasets.v1.TabularOutput.data', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='data', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=20883,
  serialized_end=20918,
)


_ELEMENTFLANKCONFIG = _descriptor.Descriptor(
  name='ElementFlankConfig',
  full_name='ncbi.datasets.v1.ElementFlankConfig',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='length', full_name='ncbi.datasets.v1.ElementFlankConfig.length', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='length', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=20920,
  serialized_end=20964,
)


_REFGENECATALOGDATASETREQUEST = _descriptor.Descriptor(
  name='RefGeneCatalogDatasetRequest',
  full_name='ncbi.datasets.v1.RefGeneCatalogDatasetRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='opaque_solr_query', full_name='ncbi.datasets.v1.RefGeneCatalogDatasetRequest.opaque_solr_query', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='opaqueSolrQuery', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='files', full_name='ncbi.datasets.v1.RefGeneCatalogDatasetRequest.files', index=1,
      number=2, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\332\360\031\002\010\002', json_name='files', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='element_flank_config', full_name='ncbi.datasets.v1.RefGeneCatalogDatasetRequest.element_flank_config', index=2,
      number=3, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='elementFlankConfig', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _REFGENECATALOGDATASETREQUEST_FILETYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
    _descriptor.OneofDescriptor(
      name='query', full_name='ncbi.datasets.v1.RefGeneCatalogDatasetRequest.query',
      index=0, containing_type=None,
      create_key=_descriptor._internal_create_key,
    fields=[]),
  ],
  serialized_start=20967,
  serialized_end=21352,
)


_MICROBIGGEDATASETREQUEST = _descriptor.Descriptor(
  name='MicroBiggeDatasetRequest',
  full_name='ncbi.datasets.v1.MicroBiggeDatasetRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='opaque_solr_query', full_name='ncbi.datasets.v1.MicroBiggeDatasetRequest.opaque_solr_query', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='opaqueSolrQuery', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='files', full_name='ncbi.datasets.v1.MicroBiggeDatasetRequest.files', index=1,
      number=2, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\332\360\031\002\010\002', json_name='files', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='element_flank_config', full_name='ncbi.datasets.v1.MicroBiggeDatasetRequest.element_flank_config', index=2,
      number=3, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='elementFlankConfig', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _MICROBIGGEDATASETREQUEST_FILETYPE,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
    _descriptor.OneofDescriptor(
      name='query', full_name='ncbi.datasets.v1.MicroBiggeDatasetRequest.query',
      index=0, containing_type=None,
      create_key=_descriptor._internal_create_key,
    fields=[]),
  ],
  serialized_start=21355,
  serialized_end=21727,
)


_VERSIONREPLY = _descriptor.Descriptor(
  name='VersionReply',
  full_name='ncbi.datasets.v1.VersionReply',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='version', full_name='ncbi.datasets.v1.VersionReply.version', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='version', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='major_ver', full_name='ncbi.datasets.v1.VersionReply.major_ver', index=1,
      number=2, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='majorVer', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='minor_ver', full_name='ncbi.datasets.v1.VersionReply.minor_ver', index=2,
      number=3, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='minorVer', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='patch_ver', full_name='ncbi.datasets.v1.VersionReply.patch_ver', index=3,
      number=4, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='patchVer', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=21729,
  serialized_end=21856,
)


_PROKARYOTEGENEREQUEST_GENEFLANKCONFIG = _descriptor.Descriptor(
  name='GeneFlankConfig',
  full_name='ncbi.datasets.v1.ProkaryoteGeneRequest.GeneFlankConfig',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='length', full_name='ncbi.datasets.v1.ProkaryoteGeneRequest.GeneFlankConfig.length', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='length', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=22406,
  serialized_end=22447,
)

_PROKARYOTEGENEREQUEST = _descriptor.Descriptor(
  name='ProkaryoteGeneRequest',
  full_name='ncbi.datasets.v1.ProkaryoteGeneRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='accessions', full_name='ncbi.datasets.v1.ProkaryoteGeneRequest.accessions', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031o\022.\n\016WP_015878339.1\022\034DNA-deoxyinosine glycosylase\022=\n\016WP_001040085.1\022+type II CRISPR RNA-guided endonuclease Cas9', json_name='accessions', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='include_annotation_type', full_name='ncbi.datasets.v1.ProkaryoteGeneRequest.include_annotation_type', index=1,
      number=2, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\332\360\031\002\010\002\302\360\031\231\001\022\036\n\nFASTA_GENE\022\020Gene range FASTA\022?\n\tFASTA_RNA\n\rFASTA_PROTEIN\022#Transcript & protein sequence FASTA\0226\n\nFASTA_GENE\n\tFASTA_RNA\n\rFASTA_PROTEIN\022\016All annotation', json_name='includeAnnotationType', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='gene_flank_config', full_name='ncbi.datasets.v1.ProkaryoteGeneRequest.gene_flank_config', index=2,
      number=3, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='geneFlankConfig', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxon', full_name='ncbi.datasets.v1.ProkaryoteGeneRequest.taxon', index=3,
      number=4, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='taxon', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_PROKARYOTEGENEREQUEST_GENEFLANKCONFIG, ],
  enum_types=[
  ],
  serialized_options=b'\302\360\031/\n-accessions:\n- WP_000000001.1\n- WP_000000002.1',
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=21859,
  serialized_end=22500,
)


_SLEEPREQUEST = _descriptor.Descriptor(
  name='SleepRequest',
  full_name='ncbi.datasets.v1.SleepRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='sleep_msec', full_name='ncbi.datasets.v1.SleepRequest.sleep_msec', index=0,
      number=1, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='sleepMsec', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='error_rate', full_name='ncbi.datasets.v1.SleepRequest.error_rate', index=1,
      number=2, type=2, cpp_type=6, label=1,
      has_default_value=False, default_value=float(0),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='errorRate', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=22502,
  serialized_end=22578,
)


_SLEEPREPLY = _descriptor.Descriptor(
  name='SleepReply',
  full_name='ncbi.datasets.v1.SleepReply',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='reply', full_name='ncbi.datasets.v1.SleepReply.reply', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='reply', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=22580,
  serialized_end=22614,
)


_TAXONOMYNODE_COUNTBYTYPE = _descriptor.Descriptor(
  name='CountByType',
  full_name='ncbi.datasets.v1.TaxonomyNode.CountByType',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='type', full_name='ncbi.datasets.v1.TaxonomyNode.CountByType.type', index=0,
      number=1, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='type', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='count', full_name='ncbi.datasets.v1.TaxonomyNode.CountByType.count', index=1,
      number=2, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='count', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=23151,
  serialized_end=23235,
)

_TAXONOMYNODE = _descriptor.Descriptor(
  name='TaxonomyNode',
  full_name='ncbi.datasets.v1.TaxonomyNode',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='tax_id', full_name='ncbi.datasets.v1.TaxonomyNode.tax_id', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='taxId', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='organism_name', full_name='ncbi.datasets.v1.TaxonomyNode.organism_name', index=1,
      number=2, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='organismName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='common_name', full_name='ncbi.datasets.v1.TaxonomyNode.common_name', index=2,
      number=3, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='commonName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='lineage', full_name='ncbi.datasets.v1.TaxonomyNode.lineage', index=3,
      number=4, type=13, cpp_type=3, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='lineage', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='children', full_name='ncbi.datasets.v1.TaxonomyNode.children', index=4,
      number=5, type=13, cpp_type=3, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='children', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='descendent_with_described_species_names_count', full_name='ncbi.datasets.v1.TaxonomyNode.descendent_with_described_species_names_count', index=5,
      number=6, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='descendentWithDescribedSpeciesNamesCount', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='rank', full_name='ncbi.datasets.v1.TaxonomyNode.rank', index=6,
      number=7, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='rank', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='has_described_species_name', full_name='ncbi.datasets.v1.TaxonomyNode.has_described_species_name', index=7,
      number=8, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='hasDescribedSpeciesName', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='counts', full_name='ncbi.datasets.v1.TaxonomyNode.counts', index=8,
      number=9, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='counts', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='min_ord', full_name='ncbi.datasets.v1.TaxonomyNode.min_ord', index=9,
      number=20, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='minOrd', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='max_ord', full_name='ncbi.datasets.v1.TaxonomyNode.max_ord', index=10,
      number=21, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='maxOrd', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='_indexed_names', full_name='ncbi.datasets.v1.TaxonomyNode._indexed_names', index=11,
      number=101, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='IndexedNames', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_TAXONOMYNODE_COUNTBYTYPE, ],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=22617,
  serialized_end=23235,
)


_TAXONOMYMETADATAREQUEST = _descriptor.Descriptor(
  name='TaxonomyMetadataRequest',
  full_name='ncbi.datasets.v1.TaxonomyMetadataRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='taxons', full_name='ncbi.datasets.v1.TaxonomyMetadataRequest.taxons', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031u\022\'\n\0049606\n\00510090\022\030NCBI Taxonomy Identifier\022\033\n\005human\n\005mouse\022\013Common Name\022-\n\014Homo sapiens\n\014Mus musculus\022\017Scientific Name', json_name='taxons', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='returned_content', full_name='ncbi.datasets.v1.TaxonomyMetadataRequest.returned_content', index=1,
      number=2, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='returnedContent', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _TAXONOMYMETADATAREQUEST_CONTENTTYPE,
  ],
  serialized_options=b'\302\360\031\030\n\026taxons:\n- 9606\n- mouse',
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=23238,
  serialized_end=23580,
)


_TAXONOMYMATCH = _descriptor.Descriptor(
  name='TaxonomyMatch',
  full_name='ncbi.datasets.v1.TaxonomyMatch',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='warnings', full_name='ncbi.datasets.v1.TaxonomyMatch.warnings', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='warnings', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='errors', full_name='ncbi.datasets.v1.TaxonomyMatch.errors', index=1,
      number=2, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='errors', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='query', full_name='ncbi.datasets.v1.TaxonomyMatch.query', index=2,
      number=3, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='query', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxonomy', full_name='ncbi.datasets.v1.TaxonomyMatch.taxonomy', index=3,
      number=4, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='taxonomy', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=23583,
  serialized_end=23784,
)


_TAXONOMYMETADATARESPONSE = _descriptor.Descriptor(
  name='TaxonomyMetadataResponse',
  full_name='ncbi.datasets.v1.TaxonomyMetadataResponse',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='messages', full_name='ncbi.datasets.v1.TaxonomyMetadataResponse.messages', index=0,
      number=1, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='messages', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='taxonomy_nodes', full_name='ncbi.datasets.v1.TaxonomyMetadataResponse.taxonomy_nodes', index=1,
      number=2, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='taxonomyNodes', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=23787,
  serialized_end=23940,
)


_TAXONOMYFILTEREDSUBTREEREQUEST = _descriptor.Descriptor(
  name='TaxonomyFilteredSubtreeRequest',
  full_name='ncbi.datasets.v1.TaxonomyFilteredSubtreeRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='taxons', full_name='ncbi.datasets.v1.TaxonomyFilteredSubtreeRequest.taxons', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=b'\302\360\031u\022\'\n\0049606\n\00510090\022\030NCBI Taxonomy Identifier\022\033\n\005human\n\005mouse\022\013Common Name\022-\n\014Homo sapiens\n\014Mus musculus\022\017Scientific Name', json_name='taxons', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='specified_limit', full_name='ncbi.datasets.v1.TaxonomyFilteredSubtreeRequest.specified_limit', index=1,
      number=2, type=8, cpp_type=7, label=1,
      has_default_value=False, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='specifiedLimit', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='rank_limits', full_name='ncbi.datasets.v1.TaxonomyFilteredSubtreeRequest.rank_limits', index=2,
      number=3, type=14, cpp_type=8, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='rankLimits', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=23943,
  serialized_end=24234,
)


_TAXONOMYFILTEREDSUBTREERESPONSE_EDGE = _descriptor.Descriptor(
  name='Edge',
  full_name='ncbi.datasets.v1.TaxonomyFilteredSubtreeResponse.Edge',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='visible_children', full_name='ncbi.datasets.v1.TaxonomyFilteredSubtreeResponse.Edge.visible_children', index=0,
      number=1, type=13, cpp_type=3, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='visibleChildren', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='children_status', full_name='ncbi.datasets.v1.TaxonomyFilteredSubtreeResponse.Edge.children_status', index=1,
      number=2, type=14, cpp_type=8, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='childrenStatus', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
    _TAXONOMYFILTEREDSUBTREERESPONSE_EDGE_CHILDSTATUS,
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=24492,
  serialized_end=24730,
)

_TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY = _descriptor.Descriptor(
  name='EdgesEntry',
  full_name='ncbi.datasets.v1.TaxonomyFilteredSubtreeResponse.EdgesEntry',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='key', full_name='ncbi.datasets.v1.TaxonomyFilteredSubtreeResponse.EdgesEntry.key', index=0,
      number=1, type=13, cpp_type=3, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='key', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='value', full_name='ncbi.datasets.v1.TaxonomyFilteredSubtreeResponse.EdgesEntry.value', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='value', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=b'8\001',
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=24732,
  serialized_end=24844,
)

_TAXONOMYFILTEREDSUBTREERESPONSE = _descriptor.Descriptor(
  name='TaxonomyFilteredSubtreeResponse',
  full_name='ncbi.datasets.v1.TaxonomyFilteredSubtreeResponse',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='root_nodes', full_name='ncbi.datasets.v1.TaxonomyFilteredSubtreeResponse.root_nodes', index=0,
      number=1, type=13, cpp_type=3, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='rootNodes', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='edges', full_name='ncbi.datasets.v1.TaxonomyFilteredSubtreeResponse.edges', index=1,
      number=2, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='edges', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='warnings', full_name='ncbi.datasets.v1.TaxonomyFilteredSubtreeResponse.warnings', index=2,
      number=3, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='warnings', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='errors', full_name='ncbi.datasets.v1.TaxonomyFilteredSubtreeResponse.errors', index=3,
      number=4, type=11, cpp_type=10, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='errors', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[_TAXONOMYFILTEREDSUBTREERESPONSE_EDGE, _TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY, ],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=24237,
  serialized_end=24844,
)


_DATASETREQUEST = _descriptor.Descriptor(
  name='DatasetRequest',
  full_name='ncbi.datasets.v1.DatasetRequest',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='genome_v1', full_name='ncbi.datasets.v1.DatasetRequest.genome_v1', index=0,
      number=1, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='genomeV1', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='gene_v1', full_name='ncbi.datasets.v1.DatasetRequest.gene_v1', index=1,
      number=2, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='geneV1', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='virus_v1', full_name='ncbi.datasets.v1.DatasetRequest.virus_v1', index=2,
      number=3, type=11, cpp_type=10, label=1,
      has_default_value=False, default_value=None,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, json_name='virusV1', file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
    _descriptor.OneofDescriptor(
      name='request', full_name='ncbi.datasets.v1.DatasetRequest.request',
      index=0, containing_type=None,
      create_key=_descriptor._internal_create_key,
    fields=[]),
  ],
  serialized_start=24847,
  serialized_end=25080,
)

_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['assembly_source'].enum_type = _ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYSOURCE
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['assembly_version'].enum_type = _ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYVERSION
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['assembly_level'].enum_type = _ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYLEVEL
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['first_release_date'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['last_release_date'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYSOURCE.containing_type = _ASSEMBLYDATASETDESCRIPTORSFILTER
_ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYVERSION.containing_type = _ASSEMBLYDATASETDESCRIPTORSFILTER
_ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYLEVEL.containing_type = _ASSEMBLYDATASETDESCRIPTORSFILTER
_ASSEMBLYMETADATAREQUEST_BIOPROJECTS.containing_type = _ASSEMBLYMETADATAREQUEST
_ASSEMBLYMETADATAREQUEST.fields_by_name['accessions'].message_type = _ACCESSIONS
_ASSEMBLYMETADATAREQUEST.fields_by_name['bioprojects'].message_type = _ASSEMBLYMETADATAREQUEST_BIOPROJECTS
_ASSEMBLYMETADATAREQUEST.fields_by_name['filters'].message_type = _ASSEMBLYDATASETDESCRIPTORSFILTER
_ASSEMBLYMETADATAREQUEST.fields_by_name['returned_content'].enum_type = _ASSEMBLYMETADATAREQUEST_CONTENTTYPE
_ASSEMBLYMETADATAREQUEST_CONTENTTYPE.containing_type = _ASSEMBLYMETADATAREQUEST
_ASSEMBLYMETADATAREQUEST.oneofs_by_name['records'].fields.append(
  _ASSEMBLYMETADATAREQUEST.fields_by_name['limit'])
_ASSEMBLYMETADATAREQUEST.fields_by_name['limit'].containing_oneof = _ASSEMBLYMETADATAREQUEST.oneofs_by_name['records']
_ASSEMBLYMETADATAREQUEST.oneofs_by_name['records'].fields.append(
  _ASSEMBLYMETADATAREQUEST.fields_by_name['page_size'])
_ASSEMBLYMETADATAREQUEST.fields_by_name['page_size'].containing_oneof = _ASSEMBLYMETADATAREQUEST.oneofs_by_name['records']
_ASSEMBLYDATASETDESCRIPTORSREQUEST.fields_by_name['filters'].message_type = _ASSEMBLYDATASETDESCRIPTORSFILTER
_ASSEMBLYDATASETDESCRIPTORSREQUEST.fields_by_name['returned_content'].enum_type = _ASSEMBLYDATASETDESCRIPTORSREQUEST_CONTENTTYPE
_ASSEMBLYDATASETDESCRIPTORSREQUEST_CONTENTTYPE.containing_type = _ASSEMBLYDATASETDESCRIPTORSREQUEST
_ASSEMBLYDATASETDESCRIPTORSREQUEST.oneofs_by_name['id'].fields.append(
  _ASSEMBLYDATASETDESCRIPTORSREQUEST.fields_by_name['tax_id'])
_ASSEMBLYDATASETDESCRIPTORSREQUEST.fields_by_name['tax_id'].containing_oneof = _ASSEMBLYDATASETDESCRIPTORSREQUEST.oneofs_by_name['id']
_ASSEMBLYDATASETDESCRIPTORSREQUEST.oneofs_by_name['id'].fields.append(
  _ASSEMBLYDATASETDESCRIPTORSREQUEST.fields_by_name['org_name'])
_ASSEMBLYDATASETDESCRIPTORSREQUEST.fields_by_name['org_name'].containing_oneof = _ASSEMBLYDATASETDESCRIPTORSREQUEST.oneofs_by_name['id']
_ASSEMBLYDATASETDESCRIPTORSREQUEST.oneofs_by_name['id'].fields.append(
  _ASSEMBLYDATASETDESCRIPTORSREQUEST.fields_by_name['tax_name'])
_ASSEMBLYDATASETDESCRIPTORSREQUEST.fields_by_name['tax_name'].containing_oneof = _ASSEMBLYDATASETDESCRIPTORSREQUEST.oneofs_by_name['id']
_ASSEMBLYDATASETDESCRIPTORSREQUEST.oneofs_by_name['id'].fields.append(
  _ASSEMBLYDATASETDESCRIPTORSREQUEST.fields_by_name['assembly_accession'])
_ASSEMBLYDATASETDESCRIPTORSREQUEST.fields_by_name['assembly_accession'].containing_oneof = _ASSEMBLYDATASETDESCRIPTORSREQUEST.oneofs_by_name['id']
_ASSEMBLYDATASETDESCRIPTORSREQUEST.oneofs_by_name['records'].fields.append(
  _ASSEMBLYDATASETDESCRIPTORSREQUEST.fields_by_name['cutoff'])
_ASSEMBLYDATASETDESCRIPTORSREQUEST.fields_by_name['cutoff'].containing_oneof = _ASSEMBLYDATASETDESCRIPTORSREQUEST.oneofs_by_name['records']
_ASSEMBLYDATASETDESCRIPTORSREQUEST.oneofs_by_name['records'].fields.append(
  _ASSEMBLYDATASETDESCRIPTORSREQUEST.fields_by_name['limit'])
_ASSEMBLYDATASETDESCRIPTORSREQUEST.fields_by_name['limit'].containing_oneof = _ASSEMBLYDATASETDESCRIPTORSREQUEST.oneofs_by_name['records']
_ASSEMBLYDATASETDESCRIPTORS.fields_by_name['datasets'].message_type = _ASSEMBLYDATASETDESCRIPTOR
_BIOPROJECTLINEAGE.fields_by_name['bioprojects'].message_type = _BIOPROJECT
_ASSEMBLYDATASETDESCRIPTOR_CHROMOSOME.containing_type = _ASSEMBLYDATASETDESCRIPTOR
_ASSEMBLYDATASETDESCRIPTOR.fields_by_name['bioproject_lineages'].message_type = _BIOPROJECTLINEAGE
_ASSEMBLYDATASETDESCRIPTOR.fields_by_name['org'].message_type = _ORGANISM
_ASSEMBLYDATASETDESCRIPTOR.fields_by_name['chromosomes'].message_type = _ASSEMBLYDATASETDESCRIPTOR_CHROMOSOME
_ASSEMBLYDATASETDESCRIPTOR.fields_by_name['annotation_metadata'].message_type = _ANNOTATIONFORASSEMBLY
_ASSEMBLYDATASETDESCRIPTOR.fields_by_name['biosample'].message_type = ncbi_dot_datasets_dot_v1_dot_reports_dot_biosample__pb2._BIOSAMPLEDESCRIPTOR
_ANNOTATIONFORASSEMBLY_FILE.fields_by_name['type'].enum_type = _ANNOTATIONFORASSEMBLY_TYPE
_ANNOTATIONFORASSEMBLY_FILE.containing_type = _ANNOTATIONFORASSEMBLY
_ANNOTATIONFORASSEMBLY.fields_by_name['file'].message_type = _ANNOTATIONFORASSEMBLY_FILE
_ANNOTATIONFORASSEMBLY.fields_by_name['stats'].message_type = _FEATURECOUNTS
_ANNOTATIONFORASSEMBLY.fields_by_name['busco'].message_type = _BUSCOSTAT
_ANNOTATIONFORASSEMBLY_TYPE.containing_type = _ANNOTATIONFORASSEMBLY
_FEATURECOUNTS.fields_by_name['gene_counts'].message_type = _GENECOUNTS
_ASSEMBLYDATASETREQUEST.fields_by_name['include_annotation_type'].enum_type = _ANNOTATIONFORASSEMBLY_TYPE
_ASSEMBLYDATASETREQUEST.fields_by_name['hydrated'].enum_type = _ASSEMBLYDATASETREQUEST_RESOLUTION
_ASSEMBLYDATASETREQUEST_RESOLUTION.containing_type = _ASSEMBLYDATASETREQUEST
_ORGANISM_COUNTS.containing_type = _ORGANISM
_ORGANISM_COUNTBYTYPE.fields_by_name['type'].enum_type = _COUNTTYPE
_ORGANISM_COUNTBYTYPE.fields_by_name['counts'].message_type = _ORGANISM_COUNTS
_ORGANISM_COUNTBYTYPE.containing_type = _ORGANISM
_ORGANISM.fields_by_name['rank'].enum_type = _ORGANISM_RANKTYPE
_ORGANISM.fields_by_name['assembly_counts'].message_type = _ORGANISM_COUNTS
_ORGANISM.fields_by_name['counts'].message_type = _ORGANISM_COUNTBYTYPE
_ORGANISM.fields_by_name['children'].message_type = _ORGANISM
_ORGANISM.fields_by_name['merged'].message_type = _ORGANISM
_ORGANISM_RANKTYPE.containing_type = _ORGANISM
_TAXTREEREQUEST.oneofs_by_name['taxonomy'].fields.append(
  _TAXTREEREQUEST.fields_by_name['tax_id'])
_TAXTREEREQUEST.fields_by_name['tax_id'].containing_oneof = _TAXTREEREQUEST.oneofs_by_name['taxonomy']
_TAXTREEREQUEST.oneofs_by_name['taxonomy'].fields.append(
  _TAXTREEREQUEST.fields_by_name['tax_token'])
_TAXTREEREQUEST.fields_by_name['tax_token'].containing_oneof = _TAXTREEREQUEST.oneofs_by_name['taxonomy']
_TAXTREEREQUEST.oneofs_by_name['taxonomy'].fields.append(
  _TAXTREEREQUEST.fields_by_name['taxon'])
_TAXTREEREQUEST.fields_by_name['taxon'].containing_oneof = _TAXTREEREQUEST.oneofs_by_name['taxonomy']
_ORGANISMQUERYREQUEST.fields_by_name['tax_rank_filter'].enum_type = _ORGANISMQUERYREQUEST_TAXRANKFILTER
_ORGANISMQUERYREQUEST.fields_by_name['tax_resource_filter'].enum_type = _ORGANISMQUERYREQUEST_TAXONRESOURCEFILTER
_ORGANISMQUERYREQUEST_TAXRANKFILTER.containing_type = _ORGANISMQUERYREQUEST
_ORGANISMQUERYREQUEST_TAXONRESOURCEFILTER.containing_type = _ORGANISMQUERYREQUEST
_ORGANISMQUERYREQUEST.oneofs_by_name['id'].fields.append(
  _ORGANISMQUERYREQUEST.fields_by_name['organism_query'])
_ORGANISMQUERYREQUEST.fields_by_name['organism_query'].containing_oneof = _ORGANISMQUERYREQUEST.oneofs_by_name['id']
_ORGANISMQUERYREQUEST.oneofs_by_name['id'].fields.append(
  _ORGANISMQUERYREQUEST.fields_by_name['taxon_query'])
_ORGANISMQUERYREQUEST.fields_by_name['taxon_query'].containing_oneof = _ORGANISMQUERYREQUEST.oneofs_by_name['id']
_SCINAMEANDIDS_SCINAMEANDID.containing_type = _SCINAMEANDIDS
_SCINAMEANDIDS.fields_by_name['sci_name_and_ids'].message_type = _SCINAMEANDIDS_SCINAMEANDID
_ORTHOLOGREQUEST.fields_by_name['returned_content'].enum_type = _ORTHOLOGREQUEST_CONTENTTYPE
_ORTHOLOGREQUEST_CONTENTTYPE.containing_type = _ORTHOLOGREQUEST
_ORTHOLOGSET.fields_by_name['genes'].message_type = _GENEMETADATA
_GENEDATASETREQUEST_SYMBOLSFORTAXON.containing_type = _GENEDATASETREQUEST
_GENEDATASETREQUEST_GENEIDS.containing_type = _GENEDATASETREQUEST
_GENEDATASETREQUEST_SORT.fields_by_name['field'].enum_type = _GENEDATASETREQUEST_SORTFIELD
_GENEDATASETREQUEST_SORT.fields_by_name['direction'].enum_type = _SORTDIRECTION
_GENEDATASETREQUEST_SORT.containing_type = _GENEDATASETREQUEST
_GENEDATASETREQUEST.fields_by_name['symbols_for_taxon'].message_type = _GENEDATASETREQUEST_SYMBOLSFORTAXON
_GENEDATASETREQUEST.fields_by_name['include_annotation_type'].enum_type = _FASTA
_GENEDATASETREQUEST.fields_by_name['returned_content'].enum_type = _GENEDATASETREQUEST_CONTENTTYPE
_GENEDATASETREQUEST.fields_by_name['sort_schema'].message_type = _GENEDATASETREQUEST_SORT
_GENEDATASETREQUEST_CONTENTTYPE.containing_type = _GENEDATASETREQUEST
_GENEDATASETREQUEST_SORTFIELD.containing_type = _GENEDATASETREQUEST
_ERROR.fields_by_name['assembly_error_code'].enum_type = _ERROR_ASSEMBLYERRORCODE
_ERROR.fields_by_name['gene_error_code'].enum_type = _ERROR_GENEERRORCODE
_ERROR.fields_by_name['virus_error_code'].enum_type = _ERROR_VIRUSERRORCODE
_ERROR_ASSEMBLYERRORCODE.containing_type = _ERROR
_ERROR_GENEERRORCODE.containing_type = _ERROR
_ERROR_VIRUSERRORCODE.containing_type = _ERROR
_ERROR.oneofs_by_name['code'].fields.append(
  _ERROR.fields_by_name['assembly_error_code'])
_ERROR.fields_by_name['assembly_error_code'].containing_oneof = _ERROR.oneofs_by_name['code']
_ERROR.oneofs_by_name['code'].fields.append(
  _ERROR.fields_by_name['gene_error_code'])
_ERROR.fields_by_name['gene_error_code'].containing_oneof = _ERROR.oneofs_by_name['code']
_ERROR.oneofs_by_name['code'].fields.append(
  _ERROR.fields_by_name['virus_error_code'])
_ERROR.fields_by_name['virus_error_code'].containing_oneof = _ERROR.oneofs_by_name['code']
_WARNING_REPLACEDID.containing_type = _WARNING
_WARNING.fields_by_name['gene_warning_code'].enum_type = _WARNING_GENEWARNINGCODE
_WARNING.fields_by_name['replaced_id'].message_type = _WARNING_REPLACEDID
_WARNING_GENEWARNINGCODE.containing_type = _WARNING
_WARNING.oneofs_by_name['code'].fields.append(
  _WARNING.fields_by_name['gene_warning_code'])
_WARNING.fields_by_name['gene_warning_code'].containing_oneof = _WARNING.oneofs_by_name['code']
_MESSAGE.fields_by_name['error'].message_type = _ERROR
_MESSAGE.fields_by_name['warning'].message_type = _WARNING
_MESSAGE.oneofs_by_name['message'].fields.append(
  _MESSAGE.fields_by_name['error'])
_MESSAGE.fields_by_name['error'].containing_oneof = _MESSAGE.oneofs_by_name['message']
_MESSAGE.oneofs_by_name['message'].fields.append(
  _MESSAGE.fields_by_name['warning'])
_MESSAGE.fields_by_name['warning'].containing_oneof = _MESSAGE.oneofs_by_name['message']
_GENEMATCH.fields_by_name['messages'].message_type = _MESSAGE
_GENEMATCH.fields_by_name['warnings'].message_type = _WARNING
_GENEMATCH.fields_by_name['gene'].message_type = ncbi_dot_datasets_dot_v1_dot_gene__descriptor__pb2._GENEDESCRIPTOR
_GENEMATCH.fields_by_name['errors'].message_type = _ERROR
_GENEMETADATA.fields_by_name['messages'].message_type = _MESSAGE
_GENEMETADATA.fields_by_name['genes'].message_type = _GENEMATCH
_ASSEMBLYMATCH.fields_by_name['messages'].message_type = _MESSAGE
_ASSEMBLYMATCH.fields_by_name['assembly'].message_type = _ASSEMBLYDATASETDESCRIPTOR
_ASSEMBLYMETADATA.fields_by_name['messages'].message_type = _MESSAGE
_ASSEMBLYMETADATA.fields_by_name['assemblies'].message_type = _ASSEMBLYMATCH
_GENEDESCRIPTORS.fields_by_name['genes'].message_type = ncbi_dot_datasets_dot_v1_dot_gene__descriptor__pb2._GENEDESCRIPTOR
_GENEDESCRIPTORS.fields_by_name['errors'].message_type = _ERROR
_ANNOTATIONFORVIRUS_TYPE.containing_type = _ANNOTATIONFORVIRUS
_VIRUSDATASETREQUEST.fields_by_name['released_since'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_VIRUSDATASETREQUEST.fields_by_name['updated_since'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_VIRUSDATASETREQUEST.fields_by_name['table_fields'].enum_type = _VIRUSTABLEFIELD
_VIRUSDATASETREQUEST.fields_by_name['include_annotation_type'].enum_type = _ANNOTATIONFORVIRUS_TYPE
_VIRUSDATASETREQUEST.fields_by_name['format'].enum_type = _TABLEFORMAT
_VIRUSDATASETFILTER.fields_by_name['released_since'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_VIRUSDATASETFILTER.fields_by_name['updated_since'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_VIRUSDATAREPORTREQUEST.fields_by_name['filter'].message_type = _VIRUSDATASETFILTER
_VIRUSDATAREPORTREQUEST.fields_by_name['returned_content'].enum_type = _VIRUSDATAREPORTREQUEST_CONTENTTYPE
_VIRUSDATAREPORTREQUEST_CONTENTTYPE.containing_type = _VIRUSDATAREPORTREQUEST
_SARS2PROTEINDATASETREQUEST.fields_by_name['released_since'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_SARS2PROTEINDATASETREQUEST.fields_by_name['updated_since'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_SARS2PROTEINDATASETREQUEST.fields_by_name['table_fields'].enum_type = _VIRUSTABLEFIELD
_SARS2PROTEINDATASETREQUEST.fields_by_name['include_annotation_type'].enum_type = _ANNOTATIONFORVIRUS_TYPE
_SARS2PROTEINDATASETREQUEST.fields_by_name['format'].enum_type = _TABLEFORMAT
_DOWNLOADSUMMARY_HYDRATED.containing_type = _DOWNLOADSUMMARY
_DOWNLOADSUMMARY_DEHYDRATED.containing_type = _DOWNLOADSUMMARY
_DOWNLOADSUMMARY_FILESUMMARY.containing_type = _DOWNLOADSUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.fields_by_name['all_genomic_fasta'].message_type = _DOWNLOADSUMMARY_FILESUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.fields_by_name['genome_gff'].message_type = _DOWNLOADSUMMARY_FILESUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.fields_by_name['genome_gbff'].message_type = _DOWNLOADSUMMARY_FILESUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.fields_by_name['rna_fasta'].message_type = _DOWNLOADSUMMARY_FILESUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.fields_by_name['prot_fasta'].message_type = _DOWNLOADSUMMARY_FILESUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.fields_by_name['genome_gtf'].message_type = _DOWNLOADSUMMARY_FILESUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.fields_by_name['cds_fasta'].message_type = _DOWNLOADSUMMARY_FILESUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.fields_by_name['sequence_report'].message_type = _DOWNLOADSUMMARY_FILESUMMARY
_DOWNLOADSUMMARY_AVAILABLEFILES.containing_type = _DOWNLOADSUMMARY
_DOWNLOADSUMMARY.fields_by_name['resource_updated_on'].message_type = google_dot_protobuf_dot_timestamp__pb2._TIMESTAMP
_DOWNLOADSUMMARY.fields_by_name['hydrated'].message_type = _DOWNLOADSUMMARY_HYDRATED
_DOWNLOADSUMMARY.fields_by_name['dehydrated'].message_type = _DOWNLOADSUMMARY_DEHYDRATED
_DOWNLOADSUMMARY.fields_by_name['errors'].message_type = _ERROR
_DOWNLOADSUMMARY.fields_by_name['messages'].message_type = _MESSAGE
_DOWNLOADSUMMARY.fields_by_name['available_files'].message_type = _DOWNLOADSUMMARY_AVAILABLEFILES
_REFGENECATALOGDATASETREQUEST.fields_by_name['files'].enum_type = _REFGENECATALOGDATASETREQUEST_FILETYPE
_REFGENECATALOGDATASETREQUEST.fields_by_name['element_flank_config'].message_type = _ELEMENTFLANKCONFIG
_REFGENECATALOGDATASETREQUEST_FILETYPE.containing_type = _REFGENECATALOGDATASETREQUEST
_REFGENECATALOGDATASETREQUEST.oneofs_by_name['query'].fields.append(
  _REFGENECATALOGDATASETREQUEST.fields_by_name['opaque_solr_query'])
_REFGENECATALOGDATASETREQUEST.fields_by_name['opaque_solr_query'].containing_oneof = _REFGENECATALOGDATASETREQUEST.oneofs_by_name['query']
_MICROBIGGEDATASETREQUEST.fields_by_name['files'].enum_type = _MICROBIGGEDATASETREQUEST_FILETYPE
_MICROBIGGEDATASETREQUEST.fields_by_name['element_flank_config'].message_type = _ELEMENTFLANKCONFIG
_MICROBIGGEDATASETREQUEST_FILETYPE.containing_type = _MICROBIGGEDATASETREQUEST
_MICROBIGGEDATASETREQUEST.oneofs_by_name['query'].fields.append(
  _MICROBIGGEDATASETREQUEST.fields_by_name['opaque_solr_query'])
_MICROBIGGEDATASETREQUEST.fields_by_name['opaque_solr_query'].containing_oneof = _MICROBIGGEDATASETREQUEST.oneofs_by_name['query']
_PROKARYOTEGENEREQUEST_GENEFLANKCONFIG.containing_type = _PROKARYOTEGENEREQUEST
_PROKARYOTEGENEREQUEST.fields_by_name['include_annotation_type'].enum_type = _FASTA
_PROKARYOTEGENEREQUEST.fields_by_name['gene_flank_config'].message_type = _PROKARYOTEGENEREQUEST_GENEFLANKCONFIG
_TAXONOMYNODE_COUNTBYTYPE.fields_by_name['type'].enum_type = _COUNTTYPE
_TAXONOMYNODE_COUNTBYTYPE.containing_type = _TAXONOMYNODE
_TAXONOMYNODE.fields_by_name['rank'].enum_type = _ORGANISM_RANKTYPE
_TAXONOMYNODE.fields_by_name['counts'].message_type = _TAXONOMYNODE_COUNTBYTYPE
_TAXONOMYMETADATAREQUEST.fields_by_name['returned_content'].enum_type = _TAXONOMYMETADATAREQUEST_CONTENTTYPE
_TAXONOMYMETADATAREQUEST_CONTENTTYPE.containing_type = _TAXONOMYMETADATAREQUEST
_TAXONOMYMATCH.fields_by_name['warnings'].message_type = _WARNING
_TAXONOMYMATCH.fields_by_name['errors'].message_type = _ERROR
_TAXONOMYMATCH.fields_by_name['taxonomy'].message_type = _TAXONOMYNODE
_TAXONOMYMETADATARESPONSE.fields_by_name['messages'].message_type = _MESSAGE
_TAXONOMYMETADATARESPONSE.fields_by_name['taxonomy_nodes'].message_type = _TAXONOMYMATCH
_TAXONOMYFILTEREDSUBTREEREQUEST.fields_by_name['rank_limits'].enum_type = _ORGANISM_RANKTYPE
_TAXONOMYFILTEREDSUBTREERESPONSE_EDGE.fields_by_name['children_status'].enum_type = _TAXONOMYFILTEREDSUBTREERESPONSE_EDGE_CHILDSTATUS
_TAXONOMYFILTEREDSUBTREERESPONSE_EDGE.containing_type = _TAXONOMYFILTEREDSUBTREERESPONSE
_TAXONOMYFILTEREDSUBTREERESPONSE_EDGE_CHILDSTATUS.containing_type = _TAXONOMYFILTEREDSUBTREERESPONSE_EDGE
_TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY.fields_by_name['value'].message_type = _TAXONOMYFILTEREDSUBTREERESPONSE_EDGE
_TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY.containing_type = _TAXONOMYFILTEREDSUBTREERESPONSE
_TAXONOMYFILTEREDSUBTREERESPONSE.fields_by_name['edges'].message_type = _TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY
_TAXONOMYFILTEREDSUBTREERESPONSE.fields_by_name['warnings'].message_type = _WARNING
_TAXONOMYFILTEREDSUBTREERESPONSE.fields_by_name['errors'].message_type = _ERROR
_DATASETREQUEST.fields_by_name['genome_v1'].message_type = _ASSEMBLYDATASETREQUEST
_DATASETREQUEST.fields_by_name['gene_v1'].message_type = _GENEDATASETREQUEST
_DATASETREQUEST.fields_by_name['virus_v1'].message_type = _VIRUSDATASETREQUEST
_DATASETREQUEST.oneofs_by_name['request'].fields.append(
  _DATASETREQUEST.fields_by_name['genome_v1'])
_DATASETREQUEST.fields_by_name['genome_v1'].containing_oneof = _DATASETREQUEST.oneofs_by_name['request']
_DATASETREQUEST.oneofs_by_name['request'].fields.append(
  _DATASETREQUEST.fields_by_name['gene_v1'])
_DATASETREQUEST.fields_by_name['gene_v1'].containing_oneof = _DATASETREQUEST.oneofs_by_name['request']
_DATASETREQUEST.oneofs_by_name['request'].fields.append(
  _DATASETREQUEST.fields_by_name['virus_v1'])
_DATASETREQUEST.fields_by_name['virus_v1'].containing_oneof = _DATASETREQUEST.oneofs_by_name['request']
DESCRIPTOR.message_types_by_name['MethodPayloadRequest'] = _METHODPAYLOADREQUEST
DESCRIPTOR.message_types_by_name['AssemblyDatasetDescriptorsFilter'] = _ASSEMBLYDATASETDESCRIPTORSFILTER
DESCRIPTOR.message_types_by_name['Accessions'] = _ACCESSIONS
DESCRIPTOR.message_types_by_name['AssemblyMetadataRequest'] = _ASSEMBLYMETADATAREQUEST
DESCRIPTOR.message_types_by_name['AssemblyDatasetDescriptorsRequest'] = _ASSEMBLYDATASETDESCRIPTORSREQUEST
DESCRIPTOR.message_types_by_name['AssemblyDatasetDescriptors'] = _ASSEMBLYDATASETDESCRIPTORS
DESCRIPTOR.message_types_by_name['BioProject'] = _BIOPROJECT
DESCRIPTOR.message_types_by_name['BioProjectLineage'] = _BIOPROJECTLINEAGE
DESCRIPTOR.message_types_by_name['AssemblyDatasetDescriptor'] = _ASSEMBLYDATASETDESCRIPTOR
DESCRIPTOR.message_types_by_name['BuscoStat'] = _BUSCOSTAT
DESCRIPTOR.message_types_by_name['AnnotationForAssembly'] = _ANNOTATIONFORASSEMBLY
DESCRIPTOR.message_types_by_name['FeatureCounts'] = _FEATURECOUNTS
DESCRIPTOR.message_types_by_name['GeneCounts'] = _GENECOUNTS
DESCRIPTOR.message_types_by_name['AssemblyDatasetRequest'] = _ASSEMBLYDATASETREQUEST
DESCRIPTOR.message_types_by_name['AssemblyDatasetAvailability'] = _ASSEMBLYDATASETAVAILABILITY
DESCRIPTOR.message_types_by_name['TaxidRequest'] = _TAXIDREQUEST
DESCRIPTOR.message_types_by_name['TaxidReply'] = _TAXIDREPLY
DESCRIPTOR.message_types_by_name['Organism'] = _ORGANISM
DESCRIPTOR.message_types_by_name['TaxTreeRequest'] = _TAXTREEREQUEST
DESCRIPTOR.message_types_by_name['OrganismQueryRequest'] = _ORGANISMQUERYREQUEST
DESCRIPTOR.message_types_by_name['SciNameAndIds'] = _SCINAMEANDIDS
DESCRIPTOR.message_types_by_name['OrthologRequest'] = _ORTHOLOGREQUEST
DESCRIPTOR.message_types_by_name['OrthologSet'] = _ORTHOLOGSET
DESCRIPTOR.message_types_by_name['GeneDatasetRequest'] = _GENEDATASETREQUEST
DESCRIPTOR.message_types_by_name['Error'] = _ERROR
DESCRIPTOR.message_types_by_name['Warning'] = _WARNING
DESCRIPTOR.message_types_by_name['Message'] = _MESSAGE
DESCRIPTOR.message_types_by_name['GeneMatch'] = _GENEMATCH
DESCRIPTOR.message_types_by_name['GeneMetadata'] = _GENEMETADATA
DESCRIPTOR.message_types_by_name['AssemblyMatch'] = _ASSEMBLYMATCH
DESCRIPTOR.message_types_by_name['AssemblyMetadata'] = _ASSEMBLYMETADATA
DESCRIPTOR.message_types_by_name['GeneDescriptors'] = _GENEDESCRIPTORS
DESCRIPTOR.message_types_by_name['AnnotationForVirus'] = _ANNOTATIONFORVIRUS
DESCRIPTOR.message_types_by_name['VirusDatasetRequest'] = _VIRUSDATASETREQUEST
DESCRIPTOR.message_types_by_name['VirusDatasetFilter'] = _VIRUSDATASETFILTER
DESCRIPTOR.message_types_by_name['VirusDataReportRequest'] = _VIRUSDATAREPORTREQUEST
DESCRIPTOR.message_types_by_name['Sars2ProteinDatasetRequest'] = _SARS2PROTEINDATASETREQUEST
DESCRIPTOR.message_types_by_name['VirusAvailabilityRequest'] = _VIRUSAVAILABILITYREQUEST
DESCRIPTOR.message_types_by_name['VirusAvailability'] = _VIRUSAVAILABILITY
DESCRIPTOR.message_types_by_name['DownloadSummary'] = _DOWNLOADSUMMARY
DESCRIPTOR.message_types_by_name['TabularOutput'] = _TABULAROUTPUT
DESCRIPTOR.message_types_by_name['ElementFlankConfig'] = _ELEMENTFLANKCONFIG
DESCRIPTOR.message_types_by_name['RefGeneCatalogDatasetRequest'] = _REFGENECATALOGDATASETREQUEST
DESCRIPTOR.message_types_by_name['MicroBiggeDatasetRequest'] = _MICROBIGGEDATASETREQUEST
DESCRIPTOR.message_types_by_name['VersionReply'] = _VERSIONREPLY
DESCRIPTOR.message_types_by_name['ProkaryoteGeneRequest'] = _PROKARYOTEGENEREQUEST
DESCRIPTOR.message_types_by_name['SleepRequest'] = _SLEEPREQUEST
DESCRIPTOR.message_types_by_name['SleepReply'] = _SLEEPREPLY
DESCRIPTOR.message_types_by_name['TaxonomyNode'] = _TAXONOMYNODE
DESCRIPTOR.message_types_by_name['TaxonomyMetadataRequest'] = _TAXONOMYMETADATAREQUEST
DESCRIPTOR.message_types_by_name['TaxonomyMatch'] = _TAXONOMYMATCH
DESCRIPTOR.message_types_by_name['TaxonomyMetadataResponse'] = _TAXONOMYMETADATARESPONSE
DESCRIPTOR.message_types_by_name['TaxonomyFilteredSubtreeRequest'] = _TAXONOMYFILTEREDSUBTREEREQUEST
DESCRIPTOR.message_types_by_name['TaxonomyFilteredSubtreeResponse'] = _TAXONOMYFILTEREDSUBTREERESPONSE
DESCRIPTOR.message_types_by_name['DatasetRequest'] = _DATASETREQUEST
DESCRIPTOR.enum_types_by_name['CountType'] = _COUNTTYPE
DESCRIPTOR.enum_types_by_name['SortDirection'] = _SORTDIRECTION
DESCRIPTOR.enum_types_by_name['Fasta'] = _FASTA
DESCRIPTOR.enum_types_by_name['VirusTableField'] = _VIRUSTABLEFIELD
DESCRIPTOR.enum_types_by_name['TableFormat'] = _TABLEFORMAT
_sym_db.RegisterFileDescriptor(DESCRIPTOR)

MethodPayloadRequest = _reflection.GeneratedProtocolMessageType('MethodPayloadRequest', (_message.Message,), {
  'DESCRIPTOR' : _METHODPAYLOADREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.MethodPayloadRequest)
  })
_sym_db.RegisterMessage(MethodPayloadRequest)

AssemblyDatasetDescriptorsFilter = _reflection.GeneratedProtocolMessageType('AssemblyDatasetDescriptorsFilter', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYDATASETDESCRIPTORSFILTER,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.AssemblyDatasetDescriptorsFilter)
  })
_sym_db.RegisterMessage(AssemblyDatasetDescriptorsFilter)

Accessions = _reflection.GeneratedProtocolMessageType('Accessions', (_message.Message,), {
  'DESCRIPTOR' : _ACCESSIONS,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.Accessions)
  })
_sym_db.RegisterMessage(Accessions)

AssemblyMetadataRequest = _reflection.GeneratedProtocolMessageType('AssemblyMetadataRequest', (_message.Message,), {

  'Bioprojects' : _reflection.GeneratedProtocolMessageType('Bioprojects', (_message.Message,), {
    'DESCRIPTOR' : _ASSEMBLYMETADATAREQUEST_BIOPROJECTS,
    '__module__' : 'ncbi.datasets.v1.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.AssemblyMetadataRequest.Bioprojects)
    })
  ,
  'DESCRIPTOR' : _ASSEMBLYMETADATAREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.AssemblyMetadataRequest)
  })
_sym_db.RegisterMessage(AssemblyMetadataRequest)
_sym_db.RegisterMessage(AssemblyMetadataRequest.Bioprojects)

AssemblyDatasetDescriptorsRequest = _reflection.GeneratedProtocolMessageType('AssemblyDatasetDescriptorsRequest', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYDATASETDESCRIPTORSREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.AssemblyDatasetDescriptorsRequest)
  })
_sym_db.RegisterMessage(AssemblyDatasetDescriptorsRequest)

AssemblyDatasetDescriptors = _reflection.GeneratedProtocolMessageType('AssemblyDatasetDescriptors', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYDATASETDESCRIPTORS,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.AssemblyDatasetDescriptors)
  })
_sym_db.RegisterMessage(AssemblyDatasetDescriptors)

BioProject = _reflection.GeneratedProtocolMessageType('BioProject', (_message.Message,), {
  'DESCRIPTOR' : _BIOPROJECT,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.BioProject)
  })
_sym_db.RegisterMessage(BioProject)

BioProjectLineage = _reflection.GeneratedProtocolMessageType('BioProjectLineage', (_message.Message,), {
  'DESCRIPTOR' : _BIOPROJECTLINEAGE,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.BioProjectLineage)
  })
_sym_db.RegisterMessage(BioProjectLineage)

AssemblyDatasetDescriptor = _reflection.GeneratedProtocolMessageType('AssemblyDatasetDescriptor', (_message.Message,), {

  'Chromosome' : _reflection.GeneratedProtocolMessageType('Chromosome', (_message.Message,), {
    'DESCRIPTOR' : _ASSEMBLYDATASETDESCRIPTOR_CHROMOSOME,
    '__module__' : 'ncbi.datasets.v1.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.AssemblyDatasetDescriptor.Chromosome)
    })
  ,
  'DESCRIPTOR' : _ASSEMBLYDATASETDESCRIPTOR,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.AssemblyDatasetDescriptor)
  })
_sym_db.RegisterMessage(AssemblyDatasetDescriptor)
_sym_db.RegisterMessage(AssemblyDatasetDescriptor.Chromosome)

BuscoStat = _reflection.GeneratedProtocolMessageType('BuscoStat', (_message.Message,), {
  'DESCRIPTOR' : _BUSCOSTAT,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.BuscoStat)
  })
_sym_db.RegisterMessage(BuscoStat)

AnnotationForAssembly = _reflection.GeneratedProtocolMessageType('AnnotationForAssembly', (_message.Message,), {

  'File' : _reflection.GeneratedProtocolMessageType('File', (_message.Message,), {
    'DESCRIPTOR' : _ANNOTATIONFORASSEMBLY_FILE,
    '__module__' : 'ncbi.datasets.v1.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.AnnotationForAssembly.File)
    })
  ,
  'DESCRIPTOR' : _ANNOTATIONFORASSEMBLY,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.AnnotationForAssembly)
  })
_sym_db.RegisterMessage(AnnotationForAssembly)
_sym_db.RegisterMessage(AnnotationForAssembly.File)

FeatureCounts = _reflection.GeneratedProtocolMessageType('FeatureCounts', (_message.Message,), {
  'DESCRIPTOR' : _FEATURECOUNTS,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.FeatureCounts)
  })
_sym_db.RegisterMessage(FeatureCounts)

GeneCounts = _reflection.GeneratedProtocolMessageType('GeneCounts', (_message.Message,), {
  'DESCRIPTOR' : _GENECOUNTS,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.GeneCounts)
  })
_sym_db.RegisterMessage(GeneCounts)

AssemblyDatasetRequest = _reflection.GeneratedProtocolMessageType('AssemblyDatasetRequest', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYDATASETREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.AssemblyDatasetRequest)
  })
_sym_db.RegisterMessage(AssemblyDatasetRequest)

AssemblyDatasetAvailability = _reflection.GeneratedProtocolMessageType('AssemblyDatasetAvailability', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYDATASETAVAILABILITY,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.AssemblyDatasetAvailability)
  })
_sym_db.RegisterMessage(AssemblyDatasetAvailability)

TaxidRequest = _reflection.GeneratedProtocolMessageType('TaxidRequest', (_message.Message,), {
  'DESCRIPTOR' : _TAXIDREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.TaxidRequest)
  })
_sym_db.RegisterMessage(TaxidRequest)

TaxidReply = _reflection.GeneratedProtocolMessageType('TaxidReply', (_message.Message,), {
  'DESCRIPTOR' : _TAXIDREPLY,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.TaxidReply)
  })
_sym_db.RegisterMessage(TaxidReply)

Organism = _reflection.GeneratedProtocolMessageType('Organism', (_message.Message,), {

  'Counts' : _reflection.GeneratedProtocolMessageType('Counts', (_message.Message,), {
    'DESCRIPTOR' : _ORGANISM_COUNTS,
    '__module__' : 'ncbi.datasets.v1.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.Organism.Counts)
    })
  ,

  'CountByType' : _reflection.GeneratedProtocolMessageType('CountByType', (_message.Message,), {
    'DESCRIPTOR' : _ORGANISM_COUNTBYTYPE,
    '__module__' : 'ncbi.datasets.v1.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.Organism.CountByType)
    })
  ,
  'DESCRIPTOR' : _ORGANISM,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.Organism)
  })
_sym_db.RegisterMessage(Organism)
_sym_db.RegisterMessage(Organism.Counts)
_sym_db.RegisterMessage(Organism.CountByType)

TaxTreeRequest = _reflection.GeneratedProtocolMessageType('TaxTreeRequest', (_message.Message,), {
  'DESCRIPTOR' : _TAXTREEREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.TaxTreeRequest)
  })
_sym_db.RegisterMessage(TaxTreeRequest)

OrganismQueryRequest = _reflection.GeneratedProtocolMessageType('OrganismQueryRequest', (_message.Message,), {
  'DESCRIPTOR' : _ORGANISMQUERYREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.OrganismQueryRequest)
  })
_sym_db.RegisterMessage(OrganismQueryRequest)

SciNameAndIds = _reflection.GeneratedProtocolMessageType('SciNameAndIds', (_message.Message,), {

  'SciNameAndId' : _reflection.GeneratedProtocolMessageType('SciNameAndId', (_message.Message,), {
    'DESCRIPTOR' : _SCINAMEANDIDS_SCINAMEANDID,
    '__module__' : 'ncbi.datasets.v1.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.SciNameAndIds.SciNameAndId)
    })
  ,
  'DESCRIPTOR' : _SCINAMEANDIDS,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.SciNameAndIds)
  })
_sym_db.RegisterMessage(SciNameAndIds)
_sym_db.RegisterMessage(SciNameAndIds.SciNameAndId)

OrthologRequest = _reflection.GeneratedProtocolMessageType('OrthologRequest', (_message.Message,), {
  'DESCRIPTOR' : _ORTHOLOGREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.OrthologRequest)
  })
_sym_db.RegisterMessage(OrthologRequest)

OrthologSet = _reflection.GeneratedProtocolMessageType('OrthologSet', (_message.Message,), {
  'DESCRIPTOR' : _ORTHOLOGSET,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.OrthologSet)
  })
_sym_db.RegisterMessage(OrthologSet)

GeneDatasetRequest = _reflection.GeneratedProtocolMessageType('GeneDatasetRequest', (_message.Message,), {

  'SymbolsForTaxon' : _reflection.GeneratedProtocolMessageType('SymbolsForTaxon', (_message.Message,), {
    'DESCRIPTOR' : _GENEDATASETREQUEST_SYMBOLSFORTAXON,
    '__module__' : 'ncbi.datasets.v1.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.GeneDatasetRequest.SymbolsForTaxon)
    })
  ,

  'GeneIds' : _reflection.GeneratedProtocolMessageType('GeneIds', (_message.Message,), {
    'DESCRIPTOR' : _GENEDATASETREQUEST_GENEIDS,
    '__module__' : 'ncbi.datasets.v1.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.GeneDatasetRequest.GeneIds)
    })
  ,

  'Sort' : _reflection.GeneratedProtocolMessageType('Sort', (_message.Message,), {
    'DESCRIPTOR' : _GENEDATASETREQUEST_SORT,
    '__module__' : 'ncbi.datasets.v1.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.GeneDatasetRequest.Sort)
    })
  ,
  'DESCRIPTOR' : _GENEDATASETREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.GeneDatasetRequest)
  })
_sym_db.RegisterMessage(GeneDatasetRequest)
_sym_db.RegisterMessage(GeneDatasetRequest.SymbolsForTaxon)
_sym_db.RegisterMessage(GeneDatasetRequest.GeneIds)
_sym_db.RegisterMessage(GeneDatasetRequest.Sort)

Error = _reflection.GeneratedProtocolMessageType('Error', (_message.Message,), {
  'DESCRIPTOR' : _ERROR,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.Error)
  })
_sym_db.RegisterMessage(Error)

Warning = _reflection.GeneratedProtocolMessageType('Warning', (_message.Message,), {

  'ReplacedId' : _reflection.GeneratedProtocolMessageType('ReplacedId', (_message.Message,), {
    'DESCRIPTOR' : _WARNING_REPLACEDID,
    '__module__' : 'ncbi.datasets.v1.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.Warning.ReplacedId)
    })
  ,
  'DESCRIPTOR' : _WARNING,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.Warning)
  })
_sym_db.RegisterMessage(Warning)
_sym_db.RegisterMessage(Warning.ReplacedId)

Message = _reflection.GeneratedProtocolMessageType('Message', (_message.Message,), {
  'DESCRIPTOR' : _MESSAGE,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.Message)
  })
_sym_db.RegisterMessage(Message)

GeneMatch = _reflection.GeneratedProtocolMessageType('GeneMatch', (_message.Message,), {
  'DESCRIPTOR' : _GENEMATCH,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.GeneMatch)
  })
_sym_db.RegisterMessage(GeneMatch)

GeneMetadata = _reflection.GeneratedProtocolMessageType('GeneMetadata', (_message.Message,), {
  'DESCRIPTOR' : _GENEMETADATA,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.GeneMetadata)
  })
_sym_db.RegisterMessage(GeneMetadata)

AssemblyMatch = _reflection.GeneratedProtocolMessageType('AssemblyMatch', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYMATCH,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.AssemblyMatch)
  })
_sym_db.RegisterMessage(AssemblyMatch)

AssemblyMetadata = _reflection.GeneratedProtocolMessageType('AssemblyMetadata', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYMETADATA,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.AssemblyMetadata)
  })
_sym_db.RegisterMessage(AssemblyMetadata)

GeneDescriptors = _reflection.GeneratedProtocolMessageType('GeneDescriptors', (_message.Message,), {
  'DESCRIPTOR' : _GENEDESCRIPTORS,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.GeneDescriptors)
  })
_sym_db.RegisterMessage(GeneDescriptors)

AnnotationForVirus = _reflection.GeneratedProtocolMessageType('AnnotationForVirus', (_message.Message,), {
  'DESCRIPTOR' : _ANNOTATIONFORVIRUS,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.AnnotationForVirus)
  })
_sym_db.RegisterMessage(AnnotationForVirus)

VirusDatasetRequest = _reflection.GeneratedProtocolMessageType('VirusDatasetRequest', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSDATASETREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.VirusDatasetRequest)
  })
_sym_db.RegisterMessage(VirusDatasetRequest)

VirusDatasetFilter = _reflection.GeneratedProtocolMessageType('VirusDatasetFilter', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSDATASETFILTER,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.VirusDatasetFilter)
  })
_sym_db.RegisterMessage(VirusDatasetFilter)

VirusDataReportRequest = _reflection.GeneratedProtocolMessageType('VirusDataReportRequest', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSDATAREPORTREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.VirusDataReportRequest)
  })
_sym_db.RegisterMessage(VirusDataReportRequest)

Sars2ProteinDatasetRequest = _reflection.GeneratedProtocolMessageType('Sars2ProteinDatasetRequest', (_message.Message,), {
  'DESCRIPTOR' : _SARS2PROTEINDATASETREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.Sars2ProteinDatasetRequest)
  })
_sym_db.RegisterMessage(Sars2ProteinDatasetRequest)

VirusAvailabilityRequest = _reflection.GeneratedProtocolMessageType('VirusAvailabilityRequest', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSAVAILABILITYREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.VirusAvailabilityRequest)
  })
_sym_db.RegisterMessage(VirusAvailabilityRequest)

VirusAvailability = _reflection.GeneratedProtocolMessageType('VirusAvailability', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSAVAILABILITY,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.VirusAvailability)
  })
_sym_db.RegisterMessage(VirusAvailability)

DownloadSummary = _reflection.GeneratedProtocolMessageType('DownloadSummary', (_message.Message,), {

  'Hydrated' : _reflection.GeneratedProtocolMessageType('Hydrated', (_message.Message,), {
    'DESCRIPTOR' : _DOWNLOADSUMMARY_HYDRATED,
    '__module__' : 'ncbi.datasets.v1.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.DownloadSummary.Hydrated)
    })
  ,

  'Dehydrated' : _reflection.GeneratedProtocolMessageType('Dehydrated', (_message.Message,), {
    'DESCRIPTOR' : _DOWNLOADSUMMARY_DEHYDRATED,
    '__module__' : 'ncbi.datasets.v1.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.DownloadSummary.Dehydrated)
    })
  ,

  'FileSummary' : _reflection.GeneratedProtocolMessageType('FileSummary', (_message.Message,), {
    'DESCRIPTOR' : _DOWNLOADSUMMARY_FILESUMMARY,
    '__module__' : 'ncbi.datasets.v1.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.DownloadSummary.FileSummary)
    })
  ,

  'AvailableFiles' : _reflection.GeneratedProtocolMessageType('AvailableFiles', (_message.Message,), {
    'DESCRIPTOR' : _DOWNLOADSUMMARY_AVAILABLEFILES,
    '__module__' : 'ncbi.datasets.v1.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.DownloadSummary.AvailableFiles)
    })
  ,
  'DESCRIPTOR' : _DOWNLOADSUMMARY,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.DownloadSummary)
  })
_sym_db.RegisterMessage(DownloadSummary)
_sym_db.RegisterMessage(DownloadSummary.Hydrated)
_sym_db.RegisterMessage(DownloadSummary.Dehydrated)
_sym_db.RegisterMessage(DownloadSummary.FileSummary)
_sym_db.RegisterMessage(DownloadSummary.AvailableFiles)

TabularOutput = _reflection.GeneratedProtocolMessageType('TabularOutput', (_message.Message,), {
  'DESCRIPTOR' : _TABULAROUTPUT,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.TabularOutput)
  })
_sym_db.RegisterMessage(TabularOutput)

ElementFlankConfig = _reflection.GeneratedProtocolMessageType('ElementFlankConfig', (_message.Message,), {
  'DESCRIPTOR' : _ELEMENTFLANKCONFIG,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.ElementFlankConfig)
  })
_sym_db.RegisterMessage(ElementFlankConfig)

RefGeneCatalogDatasetRequest = _reflection.GeneratedProtocolMessageType('RefGeneCatalogDatasetRequest', (_message.Message,), {
  'DESCRIPTOR' : _REFGENECATALOGDATASETREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.RefGeneCatalogDatasetRequest)
  })
_sym_db.RegisterMessage(RefGeneCatalogDatasetRequest)

MicroBiggeDatasetRequest = _reflection.GeneratedProtocolMessageType('MicroBiggeDatasetRequest', (_message.Message,), {
  'DESCRIPTOR' : _MICROBIGGEDATASETREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.MicroBiggeDatasetRequest)
  })
_sym_db.RegisterMessage(MicroBiggeDatasetRequest)

VersionReply = _reflection.GeneratedProtocolMessageType('VersionReply', (_message.Message,), {
  'DESCRIPTOR' : _VERSIONREPLY,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.VersionReply)
  })
_sym_db.RegisterMessage(VersionReply)

ProkaryoteGeneRequest = _reflection.GeneratedProtocolMessageType('ProkaryoteGeneRequest', (_message.Message,), {

  'GeneFlankConfig' : _reflection.GeneratedProtocolMessageType('GeneFlankConfig', (_message.Message,), {
    'DESCRIPTOR' : _PROKARYOTEGENEREQUEST_GENEFLANKCONFIG,
    '__module__' : 'ncbi.datasets.v1.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.ProkaryoteGeneRequest.GeneFlankConfig)
    })
  ,
  'DESCRIPTOR' : _PROKARYOTEGENEREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.ProkaryoteGeneRequest)
  })
_sym_db.RegisterMessage(ProkaryoteGeneRequest)
_sym_db.RegisterMessage(ProkaryoteGeneRequest.GeneFlankConfig)

SleepRequest = _reflection.GeneratedProtocolMessageType('SleepRequest', (_message.Message,), {
  'DESCRIPTOR' : _SLEEPREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.SleepRequest)
  })
_sym_db.RegisterMessage(SleepRequest)

SleepReply = _reflection.GeneratedProtocolMessageType('SleepReply', (_message.Message,), {
  'DESCRIPTOR' : _SLEEPREPLY,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.SleepReply)
  })
_sym_db.RegisterMessage(SleepReply)

TaxonomyNode = _reflection.GeneratedProtocolMessageType('TaxonomyNode', (_message.Message,), {

  'CountByType' : _reflection.GeneratedProtocolMessageType('CountByType', (_message.Message,), {
    'DESCRIPTOR' : _TAXONOMYNODE_COUNTBYTYPE,
    '__module__' : 'ncbi.datasets.v1.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.TaxonomyNode.CountByType)
    })
  ,
  'DESCRIPTOR' : _TAXONOMYNODE,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.TaxonomyNode)
  })
_sym_db.RegisterMessage(TaxonomyNode)
_sym_db.RegisterMessage(TaxonomyNode.CountByType)

TaxonomyMetadataRequest = _reflection.GeneratedProtocolMessageType('TaxonomyMetadataRequest', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYMETADATAREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.TaxonomyMetadataRequest)
  })
_sym_db.RegisterMessage(TaxonomyMetadataRequest)

TaxonomyMatch = _reflection.GeneratedProtocolMessageType('TaxonomyMatch', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYMATCH,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.TaxonomyMatch)
  })
_sym_db.RegisterMessage(TaxonomyMatch)

TaxonomyMetadataResponse = _reflection.GeneratedProtocolMessageType('TaxonomyMetadataResponse', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYMETADATARESPONSE,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.TaxonomyMetadataResponse)
  })
_sym_db.RegisterMessage(TaxonomyMetadataResponse)

TaxonomyFilteredSubtreeRequest = _reflection.GeneratedProtocolMessageType('TaxonomyFilteredSubtreeRequest', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYFILTEREDSUBTREEREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.TaxonomyFilteredSubtreeRequest)
  })
_sym_db.RegisterMessage(TaxonomyFilteredSubtreeRequest)

TaxonomyFilteredSubtreeResponse = _reflection.GeneratedProtocolMessageType('TaxonomyFilteredSubtreeResponse', (_message.Message,), {

  'Edge' : _reflection.GeneratedProtocolMessageType('Edge', (_message.Message,), {
    'DESCRIPTOR' : _TAXONOMYFILTEREDSUBTREERESPONSE_EDGE,
    '__module__' : 'ncbi.datasets.v1.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.TaxonomyFilteredSubtreeResponse.Edge)
    })
  ,

  'EdgesEntry' : _reflection.GeneratedProtocolMessageType('EdgesEntry', (_message.Message,), {
    'DESCRIPTOR' : _TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY,
    '__module__' : 'ncbi.datasets.v1.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.TaxonomyFilteredSubtreeResponse.EdgesEntry)
    })
  ,
  'DESCRIPTOR' : _TAXONOMYFILTEREDSUBTREERESPONSE,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.TaxonomyFilteredSubtreeResponse)
  })
_sym_db.RegisterMessage(TaxonomyFilteredSubtreeResponse)
_sym_db.RegisterMessage(TaxonomyFilteredSubtreeResponse.Edge)
_sym_db.RegisterMessage(TaxonomyFilteredSubtreeResponse.EdgesEntry)

DatasetRequest = _reflection.GeneratedProtocolMessageType('DatasetRequest', (_message.Message,), {
  'DESCRIPTOR' : _DATASETREQUEST,
  '__module__' : 'ncbi.datasets.v1.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v1.DatasetRequest)
  })
_sym_db.RegisterMessage(DatasetRequest)


DESCRIPTOR._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['reference_only']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['refseq_only']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['assembly_source']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['has_annotation']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['exclude_paired_reports']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['exclude_atypical']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['assembly_version']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['assembly_level']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['first_release_date']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['last_release_date']._options = None
_ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['search_text']._options = None
_ACCESSIONS.fields_by_name['accessions']._options = None
_ASSEMBLYMETADATAREQUEST_BIOPROJECTS.fields_by_name['accessions']._options = None
_ASSEMBLYMETADATAREQUEST.fields_by_name['taxon']._options = None
_ASSEMBLYMETADATAREQUEST.fields_by_name['tax_exact_match']._options = None
_ASSEMBLYMETADATAREQUEST.fields_by_name['limit']._options = None
_ASSEMBLYMETADATAREQUEST.fields_by_name['page_size']._options = None
_ASSEMBLYMETADATAREQUEST._options = None
_ASSEMBLYDATASETDESCRIPTORSREQUEST.fields_by_name['org_name']._options = None
_ASSEMBLYDATASETDESCRIPTORSREQUEST.fields_by_name['cutoff']._options = None
_ANNOTATIONFORASSEMBLY_TYPE._options = None
_ANNOTATIONFORASSEMBLY_TYPE.values_by_name["GENOME_GB"]._options = None
_ASSEMBLYDATASETREQUEST.fields_by_name['assembly_accessions']._options = None
_ASSEMBLYDATASETREQUEST.fields_by_name['accessions']._options = None
_ASSEMBLYDATASETREQUEST.fields_by_name['chromosomes']._options = None
_ASSEMBLYDATASETREQUEST.fields_by_name['include_annotation']._options = None
_ASSEMBLYDATASETREQUEST.fields_by_name['exclude_sequence']._options = None
_ASSEMBLYDATASETREQUEST.fields_by_name['include_annotation_type']._options = None
_ASSEMBLYDATASETREQUEST.fields_by_name['include_tsv']._options = None
_ASSEMBLYDATASETREQUEST._options = None
_ORGANISM.fields_by_name['assembly_count']._options = None
_TAXTREEREQUEST.fields_by_name['tax_id']._options = None
_TAXTREEREQUEST.fields_by_name['tax_token']._options = None
_TAXTREEREQUEST.fields_by_name['taxon']._options = None
_TAXTREEREQUEST.fields_by_name['children_only']._options = None
_ORGANISMQUERYREQUEST.fields_by_name['organism_query']._options = None
_ORGANISMQUERYREQUEST.fields_by_name['taxon_query']._options = None
_ORGANISMQUERYREQUEST.fields_by_name['tax_rank_filter']._options = None
_ORGANISMQUERYREQUEST.fields_by_name['tax_resource_filter']._options = None
_ORTHOLOGREQUEST.fields_by_name['gene_id']._options = None
_ORTHOLOGREQUEST.fields_by_name['taxon_filter']._options = None
_GENEDATASETREQUEST_SYMBOLSFORTAXON.fields_by_name['symbols']._options = None
_GENEDATASETREQUEST_SYMBOLSFORTAXON.fields_by_name['taxon']._options = None
_GENEDATASETREQUEST.fields_by_name['gene_ids']._options = None
_GENEDATASETREQUEST.fields_by_name['accessions']._options = None
_GENEDATASETREQUEST.fields_by_name['taxon']._options = None
_GENEDATASETREQUEST.fields_by_name['include_annotation_type']._options = None
_GENEDATASETREQUEST._options = None
_ERROR.fields_by_name['valid_identifiers']._options = None
_GENEMATCH.fields_by_name['messages']._options = None
_VIRUSDATASETREQUEST.fields_by_name['tax_id']._options = None
_VIRUSDATASETREQUEST.fields_by_name['tax_name']._options = None
_VIRUSDATASETREQUEST.fields_by_name['accessions']._options = None
_VIRUSDATASETREQUEST.fields_by_name['taxon']._options = None
_VIRUSDATASETREQUEST.fields_by_name['refseq_only']._options = None
_VIRUSDATASETREQUEST.fields_by_name['annotated_only']._options = None
_VIRUSDATASETREQUEST.fields_by_name['released_since']._options = None
_VIRUSDATASETREQUEST.fields_by_name['updated_since']._options = None
_VIRUSDATASETREQUEST.fields_by_name['host']._options = None
_VIRUSDATASETREQUEST.fields_by_name['geo_location']._options = None
_VIRUSDATASETREQUEST.fields_by_name['complete_only']._options = None
_VIRUSDATASETREQUEST.fields_by_name['table_fields']._options = None
_VIRUSDATASETREQUEST.fields_by_name['exclude_sequence']._options = None
_VIRUSDATASETREQUEST.fields_by_name['include_annotation_type']._options = None
_VIRUSDATASETREQUEST.fields_by_name['format']._options = None
_VIRUSDATASETFILTER.fields_by_name['accessions']._options = None
_VIRUSDATASETFILTER.fields_by_name['taxon']._options = None
_VIRUSDATASETFILTER.fields_by_name['refseq_only']._options = None
_VIRUSDATASETFILTER.fields_by_name['annotated_only']._options = None
_VIRUSDATASETFILTER.fields_by_name['released_since']._options = None
_VIRUSDATASETFILTER.fields_by_name['updated_since']._options = None
_VIRUSDATASETFILTER.fields_by_name['host']._options = None
_VIRUSDATASETFILTER.fields_by_name['geo_location']._options = None
_VIRUSDATASETFILTER.fields_by_name['complete_only']._options = None
_VIRUSDATAREPORTREQUEST.fields_by_name['table_fields']._options = None
_VIRUSDATAREPORTREQUEST.fields_by_name['page_size']._options = None
_VIRUSDATAREPORTREQUEST._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['proteins']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['refseq_only']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['annotated_only']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['released_since']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['updated_since']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['host']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['geo_location']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['complete_only']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['table_fields']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['exclude_sequence']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['include_annotation_type']._options = None
_SARS2PROTEINDATASETREQUEST.fields_by_name['format']._options = None
_VIRUSAVAILABILITYREQUEST.fields_by_name['accessions']._options = None
_DOWNLOADSUMMARY.fields_by_name['assembly_count']._options = None
_DOWNLOADSUMMARY.fields_by_name['errors']._options = None
_REFGENECATALOGDATASETREQUEST.fields_by_name['files']._options = None
_MICROBIGGEDATASETREQUEST.fields_by_name['files']._options = None
_PROKARYOTEGENEREQUEST.fields_by_name['accessions']._options = None
_PROKARYOTEGENEREQUEST.fields_by_name['include_annotation_type']._options = None
_PROKARYOTEGENEREQUEST._options = None
_TAXONOMYMETADATAREQUEST.fields_by_name['taxons']._options = None
_TAXONOMYMETADATAREQUEST._options = None
_TAXONOMYFILTEREDSUBTREEREQUEST.fields_by_name['taxons']._options = None
_TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY._options = None
# @@protoc_insertion_point(module_scope)
